// @hubspot/ui-extensions/dist/experimental/index.js
import { createRemoteReactComponent } from '@remote-ui/react';
export { useCrmProperties } from './hooks/useCrmProperties';
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const Iframe = createRemoteReactComponent('Iframe');
/** @experimental */
export const Tab = createRemoteReactComponent('Tab');
/** @experimental */
export const Tabs = createRemoteReactComponent('Tabs');
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const MediaObject = createRemoteReactComponent('MediaObject', {
    fragmentProps: ['itemRight', 'itemLeft'],
});
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const Inline = createRemoteReactComponent('Inline');
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const Stack2 = createRemoteReactComponent('Stack2');
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const Center = createRemoteReactComponent('Center');
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const SimpleGrid = createRemoteReactComponent('SimpleGrid');
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const Grid = createRemoteReactComponent('Grid');
/** @experimental This component is experimental. Avoid using it in production due to potential breaking changes. Your feedback is valuable for improvements. Stay tuned for updates. */
const GridItem = createRemoteReactComponent('GridItem');
const SettingsView = createRemoteReactComponent('SettingsView');
export { Iframe, MediaObject, Inline, Stack2, Center, SimpleGrid, GridItem, Grid, SettingsView, };
