// @hubspot/api-client/lib/codegen/webhooks/models/BatchResponseSubscriptionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSubscriptionResponseStatusEnum = exports.BatchResponseSubscriptionResponse = void 0;
class BatchResponseSubscriptionResponse {
    static getAttributeTypeMap() {
        return BatchResponseSubscriptionResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseSubscriptionResponse = BatchResponseSubscriptionResponse;
BatchResponseSubscriptionResponse.discriminator = undefined;
BatchResponseSubscriptionResponse.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SubscriptionResponse>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSubscriptionResponseStatusEnum",
        "format": ""
    }
];
var BatchResponseSubscriptionResponseStatusEnum;
(function (BatchResponseSubscriptionResponseStatusEnum) {
    BatchResponseSubscriptionResponseStatusEnum["Pending"] = "PENDING";
    BatchResponseSubscriptionResponseStatusEnum["Processing"] = "PROCESSING";
    BatchResponseSubscriptionResponseStatusEnum["Canceled"] = "CANCELED";
    BatchResponseSubscriptionResponseStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseSubscriptionResponseStatusEnum = exports.BatchResponseSubscriptionResponseStatusEnum || (exports.BatchResponseSubscriptionResponseStatusEnum = {}));
