// @hubspot/api-client/lib/codegen/webhooks/models/BatchResponseSubscriptionResponseWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSubscriptionResponseWithErrorsStatusEnum = exports.BatchResponseSubscriptionResponseWithErrors = void 0;
class BatchResponseSubscriptionResponseWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseSubscriptionResponseWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseSubscriptionResponseWithErrors = BatchResponseSubscriptionResponseWithErrors;
BatchResponseSubscriptionResponseWithErrors.discriminator = undefined;
BatchResponseSubscriptionResponseWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SubscriptionResponse>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSubscriptionResponseWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseSubscriptionResponseWithErrorsStatusEnum;
(function (BatchResponseSubscriptionResponseWithErrorsStatusEnum) {
    BatchResponseSubscriptionResponseWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseSubscriptionResponseWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseSubscriptionResponseWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseSubscriptionResponseWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseSubscriptionResponseWithErrorsStatusEnum = exports.BatchResponseSubscriptionResponseWithErrorsStatusEnum || (exports.BatchResponseSubscriptionResponseWithErrorsStatusEnum = {}));
