// @hubspot/api-client/lib/codegen/webhooks/models/SubscriptionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionResponseEventTypeEnum = exports.SubscriptionResponse = void 0;
class SubscriptionResponse {
    static getAttributeTypeMap() {
        return SubscriptionResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SubscriptionResponse = SubscriptionResponse;
SubscriptionResponse.discriminator = undefined;
SubscriptionResponse.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "objectTypeId",
        "baseName": "objectTypeId",
        "type": "string",
        "format": ""
    },
    {
        "name": "propertyName",
        "baseName": "propertyName",
        "type": "string",
        "format": ""
    },
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "eventType",
        "baseName": "eventType",
        "type": "SubscriptionResponseEventTypeEnum",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    }
];
var SubscriptionResponseEventTypeEnum;
(function (SubscriptionResponseEventTypeEnum) {
    SubscriptionResponseEventTypeEnum["ContactPropertyChange"] = "contact.propertyChange";
    SubscriptionResponseEventTypeEnum["CompanyPropertyChange"] = "company.propertyChange";
    SubscriptionResponseEventTypeEnum["DealPropertyChange"] = "deal.propertyChange";
    SubscriptionResponseEventTypeEnum["TicketPropertyChange"] = "ticket.propertyChange";
    SubscriptionResponseEventTypeEnum["ProductPropertyChange"] = "product.propertyChange";
    SubscriptionResponseEventTypeEnum["LineItemPropertyChange"] = "line_item.propertyChange";
    SubscriptionResponseEventTypeEnum["ContactCreation"] = "contact.creation";
    SubscriptionResponseEventTypeEnum["ContactDeletion"] = "contact.deletion";
    SubscriptionResponseEventTypeEnum["ContactPrivacyDeletion"] = "contact.privacyDeletion";
    SubscriptionResponseEventTypeEnum["CompanyCreation"] = "company.creation";
    SubscriptionResponseEventTypeEnum["CompanyDeletion"] = "company.deletion";
    SubscriptionResponseEventTypeEnum["DealCreation"] = "deal.creation";
    SubscriptionResponseEventTypeEnum["DealDeletion"] = "deal.deletion";
    SubscriptionResponseEventTypeEnum["TicketCreation"] = "ticket.creation";
    SubscriptionResponseEventTypeEnum["TicketDeletion"] = "ticket.deletion";
    SubscriptionResponseEventTypeEnum["ProductCreation"] = "product.creation";
    SubscriptionResponseEventTypeEnum["ProductDeletion"] = "product.deletion";
    SubscriptionResponseEventTypeEnum["LineItemCreation"] = "line_item.creation";
    SubscriptionResponseEventTypeEnum["LineItemDeletion"] = "line_item.deletion";
    SubscriptionResponseEventTypeEnum["ConversationCreation"] = "conversation.creation";
    SubscriptionResponseEventTypeEnum["ConversationDeletion"] = "conversation.deletion";
    SubscriptionResponseEventTypeEnum["ConversationNewMessage"] = "conversation.newMessage";
    SubscriptionResponseEventTypeEnum["ConversationPrivacyDeletion"] = "conversation.privacyDeletion";
    SubscriptionResponseEventTypeEnum["ConversationPropertyChange"] = "conversation.propertyChange";
    SubscriptionResponseEventTypeEnum["ContactMerge"] = "contact.merge";
    SubscriptionResponseEventTypeEnum["CompanyMerge"] = "company.merge";
    SubscriptionResponseEventTypeEnum["DealMerge"] = "deal.merge";
    SubscriptionResponseEventTypeEnum["TicketMerge"] = "ticket.merge";
    SubscriptionResponseEventTypeEnum["ProductMerge"] = "product.merge";
    SubscriptionResponseEventTypeEnum["LineItemMerge"] = "line_item.merge";
    SubscriptionResponseEventTypeEnum["ContactRestore"] = "contact.restore";
    SubscriptionResponseEventTypeEnum["CompanyRestore"] = "company.restore";
    SubscriptionResponseEventTypeEnum["DealRestore"] = "deal.restore";
    SubscriptionResponseEventTypeEnum["TicketRestore"] = "ticket.restore";
    SubscriptionResponseEventTypeEnum["ProductRestore"] = "product.restore";
    SubscriptionResponseEventTypeEnum["LineItemRestore"] = "line_item.restore";
    SubscriptionResponseEventTypeEnum["ContactAssociationChange"] = "contact.associationChange";
    SubscriptionResponseEventTypeEnum["CompanyAssociationChange"] = "company.associationChange";
    SubscriptionResponseEventTypeEnum["DealAssociationChange"] = "deal.associationChange";
    SubscriptionResponseEventTypeEnum["TicketAssociationChange"] = "ticket.associationChange";
    SubscriptionResponseEventTypeEnum["LineItemAssociationChange"] = "line_item.associationChange";
    SubscriptionResponseEventTypeEnum["ObjectPropertyChange"] = "object.propertyChange";
    SubscriptionResponseEventTypeEnum["ObjectCreation"] = "object.creation";
    SubscriptionResponseEventTypeEnum["ObjectDeletion"] = "object.deletion";
    SubscriptionResponseEventTypeEnum["ObjectMerge"] = "object.merge";
    SubscriptionResponseEventTypeEnum["ObjectRestore"] = "object.restore";
    SubscriptionResponseEventTypeEnum["ObjectAssociationChange"] = "object.associationChange";
})(SubscriptionResponseEventTypeEnum = exports.SubscriptionResponseEventTypeEnum || (exports.SubscriptionResponseEventTypeEnum = {}));
