// @hubspot/api-client/lib/codegen/webhooks/models/SubscriptionCreateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionCreateRequestEventTypeEnum = exports.SubscriptionCreateRequest = void 0;
class SubscriptionCreateRequest {
    static getAttributeTypeMap() {
        return SubscriptionCreateRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SubscriptionCreateRequest = SubscriptionCreateRequest;
SubscriptionCreateRequest.discriminator = undefined;
SubscriptionCreateRequest.attributeTypeMap = [
    {
        "name": "objectTypeId",
        "baseName": "objectTypeId",
        "type": "string",
        "format": ""
    },
    {
        "name": "propertyName",
        "baseName": "propertyName",
        "type": "string",
        "format": ""
    },
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "eventType",
        "baseName": "eventType",
        "type": "SubscriptionCreateRequestEventTypeEnum",
        "format": ""
    }
];
var SubscriptionCreateRequestEventTypeEnum;
(function (SubscriptionCreateRequestEventTypeEnum) {
    SubscriptionCreateRequestEventTypeEnum["ContactPropertyChange"] = "contact.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["CompanyPropertyChange"] = "company.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["DealPropertyChange"] = "deal.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["TicketPropertyChange"] = "ticket.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["ProductPropertyChange"] = "product.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["LineItemPropertyChange"] = "line_item.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["ContactCreation"] = "contact.creation";
    SubscriptionCreateRequestEventTypeEnum["ContactDeletion"] = "contact.deletion";
    SubscriptionCreateRequestEventTypeEnum["ContactPrivacyDeletion"] = "contact.privacyDeletion";
    SubscriptionCreateRequestEventTypeEnum["CompanyCreation"] = "company.creation";
    SubscriptionCreateRequestEventTypeEnum["CompanyDeletion"] = "company.deletion";
    SubscriptionCreateRequestEventTypeEnum["DealCreation"] = "deal.creation";
    SubscriptionCreateRequestEventTypeEnum["DealDeletion"] = "deal.deletion";
    SubscriptionCreateRequestEventTypeEnum["TicketCreation"] = "ticket.creation";
    SubscriptionCreateRequestEventTypeEnum["TicketDeletion"] = "ticket.deletion";
    SubscriptionCreateRequestEventTypeEnum["ProductCreation"] = "product.creation";
    SubscriptionCreateRequestEventTypeEnum["ProductDeletion"] = "product.deletion";
    SubscriptionCreateRequestEventTypeEnum["LineItemCreation"] = "line_item.creation";
    SubscriptionCreateRequestEventTypeEnum["LineItemDeletion"] = "line_item.deletion";
    SubscriptionCreateRequestEventTypeEnum["ConversationCreation"] = "conversation.creation";
    SubscriptionCreateRequestEventTypeEnum["ConversationDeletion"] = "conversation.deletion";
    SubscriptionCreateRequestEventTypeEnum["ConversationNewMessage"] = "conversation.newMessage";
    SubscriptionCreateRequestEventTypeEnum["ConversationPrivacyDeletion"] = "conversation.privacyDeletion";
    SubscriptionCreateRequestEventTypeEnum["ConversationPropertyChange"] = "conversation.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["ContactMerge"] = "contact.merge";
    SubscriptionCreateRequestEventTypeEnum["CompanyMerge"] = "company.merge";
    SubscriptionCreateRequestEventTypeEnum["DealMerge"] = "deal.merge";
    SubscriptionCreateRequestEventTypeEnum["TicketMerge"] = "ticket.merge";
    SubscriptionCreateRequestEventTypeEnum["ProductMerge"] = "product.merge";
    SubscriptionCreateRequestEventTypeEnum["LineItemMerge"] = "line_item.merge";
    SubscriptionCreateRequestEventTypeEnum["ContactRestore"] = "contact.restore";
    SubscriptionCreateRequestEventTypeEnum["CompanyRestore"] = "company.restore";
    SubscriptionCreateRequestEventTypeEnum["DealRestore"] = "deal.restore";
    SubscriptionCreateRequestEventTypeEnum["TicketRestore"] = "ticket.restore";
    SubscriptionCreateRequestEventTypeEnum["ProductRestore"] = "product.restore";
    SubscriptionCreateRequestEventTypeEnum["LineItemRestore"] = "line_item.restore";
    SubscriptionCreateRequestEventTypeEnum["ContactAssociationChange"] = "contact.associationChange";
    SubscriptionCreateRequestEventTypeEnum["CompanyAssociationChange"] = "company.associationChange";
    SubscriptionCreateRequestEventTypeEnum["DealAssociationChange"] = "deal.associationChange";
    SubscriptionCreateRequestEventTypeEnum["TicketAssociationChange"] = "ticket.associationChange";
    SubscriptionCreateRequestEventTypeEnum["LineItemAssociationChange"] = "line_item.associationChange";
    SubscriptionCreateRequestEventTypeEnum["ObjectPropertyChange"] = "object.propertyChange";
    SubscriptionCreateRequestEventTypeEnum["ObjectCreation"] = "object.creation";
    SubscriptionCreateRequestEventTypeEnum["ObjectDeletion"] = "object.deletion";
    SubscriptionCreateRequestEventTypeEnum["ObjectMerge"] = "object.merge";
    SubscriptionCreateRequestEventTypeEnum["ObjectRestore"] = "object.restore";
    SubscriptionCreateRequestEventTypeEnum["ObjectAssociationChange"] = "object.associationChange";
})(SubscriptionCreateRequestEventTypeEnum = exports.SubscriptionCreateRequestEventTypeEnum || (exports.SubscriptionCreateRequestEventTypeEnum = {}));
