// @hubspot/api-client/lib/codegen/marketing/transactional/types/ObservableAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservableSingleSendApi = exports.ObservablePublicSMTPTokensApi = void 0;
const rxjsStub_1 = require("../rxjsStub");
const rxjsStub_2 = require("../rxjsStub");
const PublicSMTPTokensApi_1 = require("../apis/PublicSMTPTokensApi");
class ObservablePublicSMTPTokensApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.configuration = configuration;
        this.requestFactory = requestFactory || new PublicSMTPTokensApi_1.PublicSMTPTokensApiRequestFactory(configuration);
        this.responseProcessor = responseProcessor || new PublicSMTPTokensApi_1.PublicSMTPTokensApiResponseProcessor();
    }
    archiveTokenWithHttpInfo(tokenId, _options) {
        const requestContextPromise = this.requestFactory.archiveToken(tokenId, _options);
        let middlewarePreObservable = (0, rxjsStub_1.from)(requestContextPromise);
        for (let middleware of this.configuration.middleware) {
            middlewarePreObservable = middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => middleware.pre(ctx)));
        }
        return middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => this.configuration.httpApi.send(ctx))).
            pipe((0, rxjsStub_2.mergeMap)((response) => {
            let middlewarePostObservable = (0, rxjsStub_1.of)(response);
            for (let middleware of this.configuration.middleware) {
                middlewarePostObservable = middlewarePostObservable.pipe((0, rxjsStub_2.mergeMap)((rsp) => middleware.post(rsp)));
            }
            return middlewarePostObservable.pipe((0, rxjsStub_2.map)((rsp) => this.responseProcessor.archiveTokenWithHttpInfo(rsp)));
        }));
    }
    archiveToken(tokenId, _options) {
        return this.archiveTokenWithHttpInfo(tokenId, _options).pipe((0, rxjsStub_2.map)((apiResponse) => apiResponse.data));
    }
    createTokenWithHttpInfo(smtpApiTokenRequestEgg, _options) {
        const requestContextPromise = this.requestFactory.createToken(smtpApiTokenRequestEgg, _options);
        let middlewarePreObservable = (0, rxjsStub_1.from)(requestContextPromise);
        for (let middleware of this.configuration.middleware) {
            middlewarePreObservable = middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => middleware.pre(ctx)));
        }
        return middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => this.configuration.httpApi.send(ctx))).
            pipe((0, rxjsStub_2.mergeMap)((response) => {
            let middlewarePostObservable = (0, rxjsStub_1.of)(response);
            for (let middleware of this.configuration.middleware) {
                middlewarePostObservable = middlewarePostObservable.pipe((0, rxjsStub_2.mergeMap)((rsp) => middleware.post(rsp)));
            }
            return middlewarePostObservable.pipe((0, rxjsStub_2.map)((rsp) => this.responseProcessor.createTokenWithHttpInfo(rsp)));
        }));
    }
    createToken(smtpApiTokenRequestEgg, _options) {
        return this.createTokenWithHttpInfo(smtpApiTokenRequestEgg, _options).pipe((0, rxjsStub_2.map)((apiResponse) => apiResponse.data));
    }
    getTokenByIdWithHttpInfo(tokenId, _options) {
        const requestContextPromise = this.requestFactory.getTokenById(tokenId, _options);
        let middlewarePreObservable = (0, rxjsStub_1.from)(requestContextPromise);
        for (let middleware of this.configuration.middleware) {
            middlewarePreObservable = middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => middleware.pre(ctx)));
        }
        return middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => this.configuration.httpApi.send(ctx))).
            pipe((0, rxjsStub_2.mergeMap)((response) => {
            let middlewarePostObservable = (0, rxjsStub_1.of)(response);
            for (let middleware of this.configuration.middleware) {
                middlewarePostObservable = middlewarePostObservable.pipe((0, rxjsStub_2.mergeMap)((rsp) => middleware.post(rsp)));
            }
            return middlewarePostObservable.pipe((0, rxjsStub_2.map)((rsp) => this.responseProcessor.getTokenByIdWithHttpInfo(rsp)));
        }));
    }
    getTokenById(tokenId, _options) {
        return this.getTokenByIdWithHttpInfo(tokenId, _options).pipe((0, rxjsStub_2.map)((apiResponse) => apiResponse.data));
    }
    getTokensPageWithHttpInfo(campaignName, emailCampaignId, after, limit, _options) {
        const requestContextPromise = this.requestFactory.getTokensPage(campaignName, emailCampaignId, after, limit, _options);
        let middlewarePreObservable = (0, rxjsStub_1.from)(requestContextPromise);
        for (let middleware of this.configuration.middleware) {
            middlewarePreObservable = middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => middleware.pre(ctx)));
        }
        return middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => this.configuration.httpApi.send(ctx))).
            pipe((0, rxjsStub_2.mergeMap)((response) => {
            let middlewarePostObservable = (0, rxjsStub_1.of)(response);
            for (let middleware of this.configuration.middleware) {
                middlewarePostObservable = middlewarePostObservable.pipe((0, rxjsStub_2.mergeMap)((rsp) => middleware.post(rsp)));
            }
            return middlewarePostObservable.pipe((0, rxjsStub_2.map)((rsp) => this.responseProcessor.getTokensPageWithHttpInfo(rsp)));
        }));
    }
    getTokensPage(campaignName, emailCampaignId, after, limit, _options) {
        return this.getTokensPageWithHttpInfo(campaignName, emailCampaignId, after, limit, _options).pipe((0, rxjsStub_2.map)((apiResponse) => apiResponse.data));
    }
    resetPasswordWithHttpInfo(tokenId, _options) {
        const requestContextPromise = this.requestFactory.resetPassword(tokenId, _options);
        let middlewarePreObservable = (0, rxjsStub_1.from)(requestContextPromise);
        for (let middleware of this.configuration.middleware) {
            middlewarePreObservable = middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => middleware.pre(ctx)));
        }
        return middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => this.configuration.httpApi.send(ctx))).
            pipe((0, rxjsStub_2.mergeMap)((response) => {
            let middlewarePostObservable = (0, rxjsStub_1.of)(response);
            for (let middleware of this.configuration.middleware) {
                middlewarePostObservable = middlewarePostObservable.pipe((0, rxjsStub_2.mergeMap)((rsp) => middleware.post(rsp)));
            }
            return middlewarePostObservable.pipe((0, rxjsStub_2.map)((rsp) => this.responseProcessor.resetPasswordWithHttpInfo(rsp)));
        }));
    }
    resetPassword(tokenId, _options) {
        return this.resetPasswordWithHttpInfo(tokenId, _options).pipe((0, rxjsStub_2.map)((apiResponse) => apiResponse.data));
    }
}
exports.ObservablePublicSMTPTokensApi = ObservablePublicSMTPTokensApi;
const SingleSendApi_1 = require("../apis/SingleSendApi");
class ObservableSingleSendApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.configuration = configuration;
        this.requestFactory = requestFactory || new SingleSendApi_1.SingleSendApiRequestFactory(configuration);
        this.responseProcessor = responseProcessor || new SingleSendApi_1.SingleSendApiResponseProcessor();
    }
    sendEmailWithHttpInfo(publicSingleSendRequestEgg, _options) {
        const requestContextPromise = this.requestFactory.sendEmail(publicSingleSendRequestEgg, _options);
        let middlewarePreObservable = (0, rxjsStub_1.from)(requestContextPromise);
        for (let middleware of this.configuration.middleware) {
            middlewarePreObservable = middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => middleware.pre(ctx)));
        }
        return middlewarePreObservable.pipe((0, rxjsStub_2.mergeMap)((ctx) => this.configuration.httpApi.send(ctx))).
            pipe((0, rxjsStub_2.mergeMap)((response) => {
            let middlewarePostObservable = (0, rxjsStub_1.of)(response);
            for (let middleware of this.configuration.middleware) {
                middlewarePostObservable = middlewarePostObservable.pipe((0, rxjsStub_2.mergeMap)((rsp) => middleware.post(rsp)));
            }
            return middlewarePostObservable.pipe((0, rxjsStub_2.map)((rsp) => this.responseProcessor.sendEmailWithHttpInfo(rsp)));
        }));
    }
    sendEmail(publicSingleSendRequestEgg, _options) {
        return this.sendEmailWithHttpInfo(publicSingleSendRequestEgg, _options).pipe((0, rxjsStub_2.map)((apiResponse) => apiResponse.data));
    }
}
exports.ObservableSingleSendApi = ObservableSingleSendApi;
