// @hubspot/api-client/lib/codegen/crm/imports/models/PublicImportResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicImportResponseStateEnum = exports.PublicImportResponseImportSourceEnum = exports.PublicImportResponse = void 0;
class PublicImportResponse {
    static getAttributeTypeMap() {
        return PublicImportResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicImportResponse = PublicImportResponse;
PublicImportResponse.discriminator = undefined;
PublicImportResponse.attributeTypeMap = [
    {
        "name": "importTemplate",
        "baseName": "importTemplate",
        "type": "ImportTemplate",
        "format": ""
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "PublicImportMetadata",
        "format": ""
    },
    {
        "name": "importRequestJson",
        "baseName": "importRequestJson",
        "type": "any",
        "format": ""
    },
    {
        "name": "importSource",
        "baseName": "importSource",
        "type": "PublicImportResponseImportSourceEnum",
        "format": ""
    },
    {
        "name": "importName",
        "baseName": "importName",
        "type": "string",
        "format": ""
    },
    {
        "name": "state",
        "baseName": "state",
        "type": "PublicImportResponseStateEnum",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "optOutImport",
        "baseName": "optOutImport",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    }
];
var PublicImportResponseImportSourceEnum;
(function (PublicImportResponseImportSourceEnum) {
    PublicImportResponseImportSourceEnum["Api"] = "API";
    PublicImportResponseImportSourceEnum["CrmUi"] = "CRM_UI";
    PublicImportResponseImportSourceEnum["Import"] = "IMPORT";
    PublicImportResponseImportSourceEnum["MobileAndroid"] = "MOBILE_ANDROID";
    PublicImportResponseImportSourceEnum["MobileIos"] = "MOBILE_IOS";
    PublicImportResponseImportSourceEnum["Salesforce"] = "SALESFORCE";
})(PublicImportResponseImportSourceEnum = exports.PublicImportResponseImportSourceEnum || (exports.PublicImportResponseImportSourceEnum = {}));
var PublicImportResponseStateEnum;
(function (PublicImportResponseStateEnum) {
    PublicImportResponseStateEnum["Started"] = "STARTED";
    PublicImportResponseStateEnum["Processing"] = "PROCESSING";
    PublicImportResponseStateEnum["Done"] = "DONE";
    PublicImportResponseStateEnum["Failed"] = "FAILED";
    PublicImportResponseStateEnum["Canceled"] = "CANCELED";
    PublicImportResponseStateEnum["Deferred"] = "DEFERRED";
    PublicImportResponseStateEnum["Reverted"] = "REVERTED";
})(PublicImportResponseStateEnum = exports.PublicImportResponseStateEnum || (exports.PublicImportResponseStateEnum = {}));
