// @hubspot/api-client/lib/codegen/crm/imports/models/PublicImportError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicImportErrorObjectTypeEnum = exports.PublicImportErrorErrorTypeEnum = exports.PublicImportError = void 0;
class PublicImportError {
    static getAttributeTypeMap() {
        return PublicImportError.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicImportError = PublicImportError;
PublicImportError.discriminator = undefined;
PublicImportError.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "extraContext",
        "baseName": "extraContext",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectTypeId",
        "baseName": "objectTypeId",
        "type": "string",
        "format": ""
    },
    {
        "name": "errorType",
        "baseName": "errorType",
        "type": "PublicImportErrorErrorTypeEnum",
        "format": ""
    },
    {
        "name": "knownColumnNumber",
        "baseName": "knownColumnNumber",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "sourceData",
        "baseName": "sourceData",
        "type": "ImportRowCore",
        "format": ""
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "PublicImportErrorObjectTypeEnum",
        "format": ""
    },
    {
        "name": "invalidValue",
        "baseName": "invalidValue",
        "type": "string",
        "format": ""
    }
];
var PublicImportErrorErrorTypeEnum;
(function (PublicImportErrorErrorTypeEnum) {
    PublicImportErrorErrorTypeEnum["IncorrectNumberOfColumns"] = "INCORRECT_NUMBER_OF_COLUMNS";
    PublicImportErrorErrorTypeEnum["InvalidObjectId"] = "INVALID_OBJECT_ID";
    PublicImportErrorErrorTypeEnum["InvalidAssociationIdentifier"] = "INVALID_ASSOCIATION_IDENTIFIER";
    PublicImportErrorErrorTypeEnum["NoObjectIdFromAssociationIdentifier"] = "NO_OBJECT_ID_FROM_ASSOCIATION_IDENTIFIER";
    PublicImportErrorErrorTypeEnum["MultipleCompaniesWithThisDomain"] = "MULTIPLE_COMPANIES_WITH_THIS_DOMAIN";
    PublicImportErrorErrorTypeEnum["PropertyDefinitionNotFound"] = "PROPERTY_DEFINITION_NOT_FOUND";
    PublicImportErrorErrorTypeEnum["PropertyValueNotFound"] = "PROPERTY_VALUE_NOT_FOUND";
    PublicImportErrorErrorTypeEnum["CouldNotFindOwner"] = "COULD_NOT_FIND_OWNER";
    PublicImportErrorErrorTypeEnum["MultipleOwnersFound"] = "MULTIPLE_OWNERS_FOUND";
    PublicImportErrorErrorTypeEnum["CouldNotFindBusinessUnit"] = "COULD_NOT_FIND_BUSINESS_UNIT";
    PublicImportErrorErrorTypeEnum["CouldNotParseNumber"] = "COULD_NOT_PARSE_NUMBER";
    PublicImportErrorErrorTypeEnum["CouldNotParseDate"] = "COULD_NOT_PARSE_DATE";
    PublicImportErrorErrorTypeEnum["CouldNotParseTerm"] = "COULD_NOT_PARSE_TERM";
    PublicImportErrorErrorTypeEnum["OutsideValidTimeRange"] = "OUTSIDE_VALID_TIME_RANGE";
    PublicImportErrorErrorTypeEnum["OutsideValidTermRange"] = "OUTSIDE_VALID_TERM_RANGE";
    PublicImportErrorErrorTypeEnum["CouldNotParseRow"] = "COULD_NOT_PARSE_ROW";
    PublicImportErrorErrorTypeEnum["InvalidEnumerationOption"] = "INVALID_ENUMERATION_OPTION";
    PublicImportErrorErrorTypeEnum["AmbiguousEnumerationOption"] = "AMBIGUOUS_ENUMERATION_OPTION";
    PublicImportErrorErrorTypeEnum["FailedValidation"] = "FAILED_VALIDATION";
    PublicImportErrorErrorTypeEnum["FailedToCreateAssociation"] = "FAILED_TO_CREATE_ASSOCIATION";
    PublicImportErrorErrorTypeEnum["FileNotFound"] = "FILE_NOT_FOUND";
    PublicImportErrorErrorTypeEnum["InvalidColumnConfiguration"] = "INVALID_COLUMN_CONFIGURATION";
    PublicImportErrorErrorTypeEnum["InvalidFileType"] = "INVALID_FILE_TYPE";
    PublicImportErrorErrorTypeEnum["InvalidSpreadsheet"] = "INVALID_SPREADSHEET";
    PublicImportErrorErrorTypeEnum["InvalidSheetCount"] = "INVALID_SHEET_COUNT";
    PublicImportErrorErrorTypeEnum["FailedToProcessObjectWithEmptyPropertyValues"] = "FAILED_TO_PROCESS_OBJECT_WITH_EMPTY_PROPERTY_VALUES";
    PublicImportErrorErrorTypeEnum["UnknownBadRequest"] = "UNKNOWN_BAD_REQUEST";
    PublicImportErrorErrorTypeEnum["GdprBlacklistedEmail"] = "GDPR_BLACKLISTED_EMAIL";
    PublicImportErrorErrorTypeEnum["DuplicateAssociationId"] = "DUPLICATE_ASSOCIATION_ID";
    PublicImportErrorErrorTypeEnum["LimitExceeded"] = "LIMIT_EXCEEDED";
    PublicImportErrorErrorTypeEnum["PortalWideCustomObjectLimitExceeded"] = "PORTAL_WIDE_CUSTOM_OBJECT_LIMIT_EXCEEDED";
    PublicImportErrorErrorTypeEnum["InvalidAlternateId"] = "INVALID_ALTERNATE_ID";
    PublicImportErrorErrorTypeEnum["InvalidEmail"] = "INVALID_EMAIL";
    PublicImportErrorErrorTypeEnum["InvalidDomain"] = "INVALID_DOMAIN";
    PublicImportErrorErrorTypeEnum["DuplicateRowContent"] = "DUPLICATE_ROW_CONTENT";
    PublicImportErrorErrorTypeEnum["InvalidNumberSize"] = "INVALID_NUMBER_SIZE";
    PublicImportErrorErrorTypeEnum["UnknownError"] = "UNKNOWN_ERROR";
    PublicImportErrorErrorTypeEnum["FailedToOptOutContact"] = "FAILED_TO_OPT_OUT_CONTACT";
    PublicImportErrorErrorTypeEnum["InvalidRequiredProperty"] = "INVALID_REQUIRED_PROPERTY";
    PublicImportErrorErrorTypeEnum["MissingRequiredProperty"] = "MISSING_REQUIRED_PROPERTY";
    PublicImportErrorErrorTypeEnum["DuplicateAlternateId"] = "DUPLICATE_ALTERNATE_ID";
    PublicImportErrorErrorTypeEnum["DuplicateObjectId"] = "DUPLICATE_OBJECT_ID";
    PublicImportErrorErrorTypeEnum["DuplicateUniquePropertyValue"] = "DUPLICATE_UNIQUE_PROPERTY_VALUE";
    PublicImportErrorErrorTypeEnum["BlankValueProvided"] = "BLANK_VALUE_PROVIDED";
    PublicImportErrorErrorTypeEnum["UnknownAssociationRecordId"] = "UNKNOWN_ASSOCIATION_RECORD_ID";
    PublicImportErrorErrorTypeEnum["InvalidRecordId"] = "INVALID_RECORD_ID";
    PublicImportErrorErrorTypeEnum["DuplicateRecordId"] = "DUPLICATE_RECORD_ID";
    PublicImportErrorErrorTypeEnum["InvalidCustomPropertyValidation"] = "INVALID_CUSTOM_PROPERTY_VALIDATION";
    PublicImportErrorErrorTypeEnum["CreateOnlyImport"] = "CREATE_ONLY_IMPORT";
    PublicImportErrorErrorTypeEnum["UpdateOnlyImport"] = "UPDATE_ONLY_IMPORT";
    PublicImportErrorErrorTypeEnum["ColumnTooLarge"] = "COLUMN_TOO_LARGE";
    PublicImportErrorErrorTypeEnum["RowDataTooLarge"] = "ROW_DATA_TOO_LARGE";
    PublicImportErrorErrorTypeEnum["InvalidEventTimestamp"] = "INVALID_EVENT_TIMESTAMP";
    PublicImportErrorErrorTypeEnum["InvalidEvent"] = "INVALID_EVENT";
    PublicImportErrorErrorTypeEnum["DuplicateEvent"] = "DUPLICATE_EVENT";
    PublicImportErrorErrorTypeEnum["MissingEventDefinition"] = "MISSING_EVENT_DEFINITION";
})(PublicImportErrorErrorTypeEnum = exports.PublicImportErrorErrorTypeEnum || (exports.PublicImportErrorErrorTypeEnum = {}));
var PublicImportErrorObjectTypeEnum;
(function (PublicImportErrorObjectTypeEnum) {
    PublicImportErrorObjectTypeEnum["Contact"] = "CONTACT";
    PublicImportErrorObjectTypeEnum["Company"] = "COMPANY";
    PublicImportErrorObjectTypeEnum["Deal"] = "DEAL";
    PublicImportErrorObjectTypeEnum["Engagement"] = "ENGAGEMENT";
    PublicImportErrorObjectTypeEnum["Ticket"] = "TICKET";
    PublicImportErrorObjectTypeEnum["Owner"] = "OWNER";
    PublicImportErrorObjectTypeEnum["Product"] = "PRODUCT";
    PublicImportErrorObjectTypeEnum["LineItem"] = "LINE_ITEM";
    PublicImportErrorObjectTypeEnum["BetDeliverableService"] = "BET_DELIVERABLE_SERVICE";
    PublicImportErrorObjectTypeEnum["Content"] = "CONTENT";
    PublicImportErrorObjectTypeEnum["Conversation"] = "CONVERSATION";
    PublicImportErrorObjectTypeEnum["BetAlert"] = "BET_ALERT";
    PublicImportErrorObjectTypeEnum["Portal"] = "PORTAL";
    PublicImportErrorObjectTypeEnum["Quote"] = "QUOTE";
    PublicImportErrorObjectTypeEnum["FormSubmissionInbounddb"] = "FORM_SUBMISSION_INBOUNDDB";
    PublicImportErrorObjectTypeEnum["Quota"] = "QUOTA";
    PublicImportErrorObjectTypeEnum["Unsubscribe"] = "UNSUBSCRIBE";
    PublicImportErrorObjectTypeEnum["Communication"] = "COMMUNICATION";
    PublicImportErrorObjectTypeEnum["FeedbackSubmission"] = "FEEDBACK_SUBMISSION";
    PublicImportErrorObjectTypeEnum["Attribution"] = "ATTRIBUTION";
    PublicImportErrorObjectTypeEnum["SalesforceSyncError"] = "SALESFORCE_SYNC_ERROR";
    PublicImportErrorObjectTypeEnum["RestorableCrmObject"] = "RESTORABLE_CRM_OBJECT";
    PublicImportErrorObjectTypeEnum["Hub"] = "HUB";
    PublicImportErrorObjectTypeEnum["LandingPage"] = "LANDING_PAGE";
    PublicImportErrorObjectTypeEnum["ProductOrFolder"] = "PRODUCT_OR_FOLDER";
    PublicImportErrorObjectTypeEnum["Task"] = "TASK";
    PublicImportErrorObjectTypeEnum["Form"] = "FORM";
    PublicImportErrorObjectTypeEnum["MarketingEmail"] = "MARKETING_EMAIL";
    PublicImportErrorObjectTypeEnum["AdAccount"] = "AD_ACCOUNT";
    PublicImportErrorObjectTypeEnum["AdCampaign"] = "AD_CAMPAIGN";
    PublicImportErrorObjectTypeEnum["AdGroup"] = "AD_GROUP";
    PublicImportErrorObjectTypeEnum["Ad"] = "AD";
    PublicImportErrorObjectTypeEnum["Keyword"] = "KEYWORD";
    PublicImportErrorObjectTypeEnum["Campaign"] = "CAMPAIGN";
    PublicImportErrorObjectTypeEnum["SocialChannel"] = "SOCIAL_CHANNEL";
    PublicImportErrorObjectTypeEnum["SocialPost"] = "SOCIAL_POST";
    PublicImportErrorObjectTypeEnum["SitePage"] = "SITE_PAGE";
    PublicImportErrorObjectTypeEnum["BlogPost"] = "BLOG_POST";
    PublicImportErrorObjectTypeEnum["Import"] = "IMPORT";
    PublicImportErrorObjectTypeEnum["Export"] = "EXPORT";
    PublicImportErrorObjectTypeEnum["Cta"] = "CTA";
    PublicImportErrorObjectTypeEnum["TaskTemplate"] = "TASK_TEMPLATE";
    PublicImportErrorObjectTypeEnum["AutomationPlatformFlow"] = "AUTOMATION_PLATFORM_FLOW";
    PublicImportErrorObjectTypeEnum["ObjectList"] = "OBJECT_LIST";
    PublicImportErrorObjectTypeEnum["Note"] = "NOTE";
    PublicImportErrorObjectTypeEnum["MeetingEvent"] = "MEETING_EVENT";
    PublicImportErrorObjectTypeEnum["Call"] = "CALL";
    PublicImportErrorObjectTypeEnum["Email"] = "EMAIL";
    PublicImportErrorObjectTypeEnum["PublishingTask"] = "PUBLISHING_TASK";
    PublicImportErrorObjectTypeEnum["ConversationSession"] = "CONVERSATION_SESSION";
    PublicImportErrorObjectTypeEnum["ContactCreateAttribution"] = "CONTACT_CREATE_ATTRIBUTION";
    PublicImportErrorObjectTypeEnum["Invoice"] = "INVOICE";
    PublicImportErrorObjectTypeEnum["MarketingEvent"] = "MARKETING_EVENT";
    PublicImportErrorObjectTypeEnum["ConversationInbox"] = "CONVERSATION_INBOX";
    PublicImportErrorObjectTypeEnum["Chatflow"] = "CHATFLOW";
    PublicImportErrorObjectTypeEnum["MediaBridge"] = "MEDIA_BRIDGE";
    PublicImportErrorObjectTypeEnum["Sequence"] = "SEQUENCE";
    PublicImportErrorObjectTypeEnum["SequenceStep"] = "SEQUENCE_STEP";
    PublicImportErrorObjectTypeEnum["Forecast"] = "FORECAST";
    PublicImportErrorObjectTypeEnum["Snippet"] = "SNIPPET";
    PublicImportErrorObjectTypeEnum["Template"] = "TEMPLATE";
    PublicImportErrorObjectTypeEnum["DealCreateAttribution"] = "DEAL_CREATE_ATTRIBUTION";
    PublicImportErrorObjectTypeEnum["QuoteTemplate"] = "QUOTE_TEMPLATE";
    PublicImportErrorObjectTypeEnum["QuoteModule"] = "QUOTE_MODULE";
    PublicImportErrorObjectTypeEnum["QuoteModuleField"] = "QUOTE_MODULE_FIELD";
    PublicImportErrorObjectTypeEnum["QuoteField"] = "QUOTE_FIELD";
    PublicImportErrorObjectTypeEnum["SequenceEnrollment"] = "SEQUENCE_ENROLLMENT";
    PublicImportErrorObjectTypeEnum["Subscription"] = "SUBSCRIPTION";
    PublicImportErrorObjectTypeEnum["AcceptanceTest"] = "ACCEPTANCE_TEST";
    PublicImportErrorObjectTypeEnum["SocialBroadcast"] = "SOCIAL_BROADCAST";
    PublicImportErrorObjectTypeEnum["DealSplit"] = "DEAL_SPLIT";
    PublicImportErrorObjectTypeEnum["DealRegistration"] = "DEAL_REGISTRATION";
    PublicImportErrorObjectTypeEnum["GoalTarget"] = "GOAL_TARGET";
    PublicImportErrorObjectTypeEnum["GoalTargetGroup"] = "GOAL_TARGET_GROUP";
    PublicImportErrorObjectTypeEnum["PortalObjectSyncMessage"] = "PORTAL_OBJECT_SYNC_MESSAGE";
    PublicImportErrorObjectTypeEnum["FileManagerFile"] = "FILE_MANAGER_FILE";
    PublicImportErrorObjectTypeEnum["FileManagerFolder"] = "FILE_MANAGER_FOLDER";
    PublicImportErrorObjectTypeEnum["SequenceStepEnrollment"] = "SEQUENCE_STEP_ENROLLMENT";
    PublicImportErrorObjectTypeEnum["Approval"] = "APPROVAL";
    PublicImportErrorObjectTypeEnum["ApprovalStep"] = "APPROVAL_STEP";
    PublicImportErrorObjectTypeEnum["CtaVariant"] = "CTA_VARIANT";
    PublicImportErrorObjectTypeEnum["SalesDocument"] = "SALES_DOCUMENT";
    PublicImportErrorObjectTypeEnum["Discount"] = "DISCOUNT";
    PublicImportErrorObjectTypeEnum["Fee"] = "FEE";
    PublicImportErrorObjectTypeEnum["Tax"] = "TAX";
    PublicImportErrorObjectTypeEnum["MarketingCalendar"] = "MARKETING_CALENDAR";
    PublicImportErrorObjectTypeEnum["PermissionsTesting"] = "PERMISSIONS_TESTING";
    PublicImportErrorObjectTypeEnum["PrivacyScannerCookie"] = "PRIVACY_SCANNER_COOKIE";
    PublicImportErrorObjectTypeEnum["DataSyncState"] = "DATA_SYNC_STATE";
    PublicImportErrorObjectTypeEnum["WebInteractive"] = "WEB_INTERACTIVE";
    PublicImportErrorObjectTypeEnum["Playbook"] = "PLAYBOOK";
    PublicImportErrorObjectTypeEnum["Folder"] = "FOLDER";
    PublicImportErrorObjectTypeEnum["PlaybookQuestion"] = "PLAYBOOK_QUESTION";
    PublicImportErrorObjectTypeEnum["PlaybookSubmission"] = "PLAYBOOK_SUBMISSION";
    PublicImportErrorObjectTypeEnum["PlaybookSubmissionAnswer"] = "PLAYBOOK_SUBMISSION_ANSWER";
    PublicImportErrorObjectTypeEnum["CommercePayment"] = "COMMERCE_PAYMENT";
    PublicImportErrorObjectTypeEnum["GscProperty"] = "GSC_PROPERTY";
    PublicImportErrorObjectTypeEnum["SoxProtectedDummyType"] = "SOX_PROTECTED_DUMMY_TYPE";
    PublicImportErrorObjectTypeEnum["BlogListingPage"] = "BLOG_LISTING_PAGE";
    PublicImportErrorObjectTypeEnum["QuarantinedSubmission"] = "QUARANTINED_SUBMISSION";
    PublicImportErrorObjectTypeEnum["PaymentSchedule"] = "PAYMENT_SCHEDULE";
    PublicImportErrorObjectTypeEnum["PaymentScheduleInstallment"] = "PAYMENT_SCHEDULE_INSTALLMENT";
    PublicImportErrorObjectTypeEnum["MarketingCampaignUtm"] = "MARKETING_CAMPAIGN_UTM";
    PublicImportErrorObjectTypeEnum["DiscountTemplate"] = "DISCOUNT_TEMPLATE";
    PublicImportErrorObjectTypeEnum["DiscountCode"] = "DISCOUNT_CODE";
    PublicImportErrorObjectTypeEnum["FeedbackSurvey"] = "FEEDBACK_SURVEY";
    PublicImportErrorObjectTypeEnum["CmsUrl"] = "CMS_URL";
    PublicImportErrorObjectTypeEnum["SalesTask"] = "SALES_TASK";
    PublicImportErrorObjectTypeEnum["SalesWorkload"] = "SALES_WORKLOAD";
    PublicImportErrorObjectTypeEnum["User"] = "USER";
    PublicImportErrorObjectTypeEnum["PostalMail"] = "POSTAL_MAIL";
    PublicImportErrorObjectTypeEnum["SchemasBackendTest"] = "SCHEMAS_BACKEND_TEST";
    PublicImportErrorObjectTypeEnum["PaymentLink"] = "PAYMENT_LINK";
    PublicImportErrorObjectTypeEnum["SubmissionTag"] = "SUBMISSION_TAG";
    PublicImportErrorObjectTypeEnum["CampaignStep"] = "CAMPAIGN_STEP";
    PublicImportErrorObjectTypeEnum["SchedulingPage"] = "SCHEDULING_PAGE";
    PublicImportErrorObjectTypeEnum["SoxProtectedTestType"] = "SOX_PROTECTED_TEST_TYPE";
    PublicImportErrorObjectTypeEnum["Order"] = "ORDER";
    PublicImportErrorObjectTypeEnum["MarketingSms"] = "MARKETING_SMS";
    PublicImportErrorObjectTypeEnum["PartnerAccount"] = "PARTNER_ACCOUNT";
    PublicImportErrorObjectTypeEnum["CampaignTemplate"] = "CAMPAIGN_TEMPLATE";
    PublicImportErrorObjectTypeEnum["CampaignTemplateStep"] = "CAMPAIGN_TEMPLATE_STEP";
    PublicImportErrorObjectTypeEnum["Playlist"] = "PLAYLIST";
    PublicImportErrorObjectTypeEnum["Clip"] = "CLIP";
    PublicImportErrorObjectTypeEnum["CampaignBudgetItem"] = "CAMPAIGN_BUDGET_ITEM";
    PublicImportErrorObjectTypeEnum["CampaignSpendItem"] = "CAMPAIGN_SPEND_ITEM";
    PublicImportErrorObjectTypeEnum["Mic"] = "MIC";
    PublicImportErrorObjectTypeEnum["ContentAudit"] = "CONTENT_AUDIT";
    PublicImportErrorObjectTypeEnum["ContentAuditPage"] = "CONTENT_AUDIT_PAGE";
    PublicImportErrorObjectTypeEnum["PlaylistFolder"] = "PLAYLIST_FOLDER";
    PublicImportErrorObjectTypeEnum["Lead"] = "LEAD";
    PublicImportErrorObjectTypeEnum["AbandonedCart"] = "ABANDONED_CART";
    PublicImportErrorObjectTypeEnum["ExternalWebUrl"] = "EXTERNAL_WEB_URL";
    PublicImportErrorObjectTypeEnum["View"] = "VIEW";
    PublicImportErrorObjectTypeEnum["ViewBlock"] = "VIEW_BLOCK";
    PublicImportErrorObjectTypeEnum["Roster"] = "ROSTER";
    PublicImportErrorObjectTypeEnum["Cart"] = "CART";
    PublicImportErrorObjectTypeEnum["AutomationPlatformFlowAction"] = "AUTOMATION_PLATFORM_FLOW_ACTION";
    PublicImportErrorObjectTypeEnum["SocialProfile"] = "SOCIAL_PROFILE";
    PublicImportErrorObjectTypeEnum["PartnerClient"] = "PARTNER_CLIENT";
    PublicImportErrorObjectTypeEnum["RosterMember"] = "ROSTER_MEMBER";
    PublicImportErrorObjectTypeEnum["MarketingEventAttendance"] = "MARKETING_EVENT_ATTENDANCE";
    PublicImportErrorObjectTypeEnum["AllPages"] = "ALL_PAGES";
    PublicImportErrorObjectTypeEnum["AiForecast"] = "AI_FORECAST";
    PublicImportErrorObjectTypeEnum["CrmPipelinesDummyType"] = "CRM_PIPELINES_DUMMY_TYPE";
    PublicImportErrorObjectTypeEnum["Unknown"] = "UNKNOWN";
})(PublicImportErrorObjectTypeEnum = exports.PublicImportErrorObjectTypeEnum || (exports.PublicImportErrorObjectTypeEnum = {}));
