// @hubspot/api-client/lib/codegen/files/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectFoldersApi = exports.ObjectFilesApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectFilesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableFilesApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(param, options) {
        return this.api.archiveWithHttpInfo(param.fileId, options).toPromise();
    }
    archive(param, options) {
        return this.api.archive(param.fileId, options).toPromise();
    }
    archiveGDPRWithHttpInfo(param, options) {
        return this.api.archiveGDPRWithHttpInfo(param.fileId, options).toPromise();
    }
    archiveGDPR(param, options) {
        return this.api.archiveGDPR(param.fileId, options).toPromise();
    }
    checkImportWithHttpInfo(param, options) {
        return this.api.checkImportWithHttpInfo(param.taskId, options).toPromise();
    }
    checkImport(param, options) {
        return this.api.checkImport(param.taskId, options).toPromise();
    }
    doSearchWithHttpInfo(param = {}, options) {
        return this.api.doSearchWithHttpInfo(param.properties, param.after, param.before, param.limit, param.sort, param.id, param.createdAt, param.createdAtLte, param.createdAtGte, param.updatedAt, param.updatedAtLte, param.updatedAtGte, param.name, param.path, param.parentFolderId, param.size, param.height, param.width, param.encoding, param.type, param.extension, param.url, param.isUsableInContent, param.allowsAnonymousAccess, options).toPromise();
    }
    doSearch(param = {}, options) {
        return this.api.doSearch(param.properties, param.after, param.before, param.limit, param.sort, param.id, param.createdAt, param.createdAtLte, param.createdAtGte, param.updatedAt, param.updatedAtLte, param.updatedAtGte, param.name, param.path, param.parentFolderId, param.size, param.height, param.width, param.encoding, param.type, param.extension, param.url, param.isUsableInContent, param.allowsAnonymousAccess, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.fileId, param.properties, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.fileId, param.properties, options).toPromise();
    }
    getMetadataWithHttpInfo(param, options) {
        return this.api.getMetadataWithHttpInfo(param.path, param.properties, options).toPromise();
    }
    getMetadata(param, options) {
        return this.api.getMetadata(param.path, param.properties, options).toPromise();
    }
    getSignedUrlWithHttpInfo(param, options) {
        return this.api.getSignedUrlWithHttpInfo(param.fileId, param.size, param.expirationSeconds, param.upscale, options).toPromise();
    }
    getSignedUrl(param, options) {
        return this.api.getSignedUrl(param.fileId, param.size, param.expirationSeconds, param.upscale, options).toPromise();
    }
    importFromUrlWithHttpInfo(param, options) {
        return this.api.importFromUrlWithHttpInfo(param.importFromUrlInput, options).toPromise();
    }
    importFromUrl(param, options) {
        return this.api.importFromUrl(param.importFromUrlInput, options).toPromise();
    }
    replaceWithHttpInfo(param, options) {
        return this.api.replaceWithHttpInfo(param.fileId, param.file, param.charsetHunch, param.options, options).toPromise();
    }
    replace(param, options) {
        return this.api.replace(param.fileId, param.file, param.charsetHunch, param.options, options).toPromise();
    }
    updatePropertiesWithHttpInfo(param, options) {
        return this.api.updatePropertiesWithHttpInfo(param.fileId, param.fileUpdateInput, options).toPromise();
    }
    updateProperties(param, options) {
        return this.api.updateProperties(param.fileId, param.fileUpdateInput, options).toPromise();
    }
    uploadWithHttpInfo(param = {}, options) {
        return this.api.uploadWithHttpInfo(param.file, param.folderId, param.folderPath, param.fileName, param.charsetHunch, param.options, options).toPromise();
    }
    upload(param = {}, options) {
        return this.api.upload(param.file, param.folderId, param.folderPath, param.fileName, param.charsetHunch, param.options, options).toPromise();
    }
}
exports.ObjectFilesApi = ObjectFilesApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectFoldersApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableFoldersApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(param, options) {
        return this.api.archiveWithHttpInfo(param.folderId, options).toPromise();
    }
    archive(param, options) {
        return this.api.archive(param.folderId, options).toPromise();
    }
    archiveByPathWithHttpInfo(param, options) {
        return this.api.archiveByPathWithHttpInfo(param.folderPath, options).toPromise();
    }
    archiveByPath(param, options) {
        return this.api.archiveByPath(param.folderPath, options).toPromise();
    }
    checkUpdateStatusWithHttpInfo(param, options) {
        return this.api.checkUpdateStatusWithHttpInfo(param.taskId, options).toPromise();
    }
    checkUpdateStatus(param, options) {
        return this.api.checkUpdateStatus(param.taskId, options).toPromise();
    }
    createWithHttpInfo(param, options) {
        return this.api.createWithHttpInfo(param.folderInput, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.folderInput, options).toPromise();
    }
    doSearchWithHttpInfo(param = {}, options) {
        return this.api.doSearchWithHttpInfo(param.properties, param.after, param.before, param.limit, param.sort, param.id, param.createdAt, param.createdAtLte, param.createdAtGte, param.updatedAt, param.updatedAtLte, param.updatedAtGte, param.name, param.path, param.parentFolderId, options).toPromise();
    }
    doSearch(param = {}, options) {
        return this.api.doSearch(param.properties, param.after, param.before, param.limit, param.sort, param.id, param.createdAt, param.createdAtLte, param.createdAtGte, param.updatedAt, param.updatedAtLte, param.updatedAtGte, param.name, param.path, param.parentFolderId, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.folderId, param.properties, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.folderId, param.properties, options).toPromise();
    }
    getByPathWithHttpInfo(param, options) {
        return this.api.getByPathWithHttpInfo(param.folderPath, param.properties, options).toPromise();
    }
    getByPath(param, options) {
        return this.api.getByPath(param.folderPath, param.properties, options).toPromise();
    }
    updatePropertiesWithHttpInfo(param, options) {
        return this.api.updatePropertiesWithHttpInfo(param.folderUpdateInput, options).toPromise();
    }
    updateProperties(param, options) {
        return this.api.updateProperties(param.folderUpdateInput, options).toPromise();
    }
}
exports.ObjectFoldersApi = ObjectFoldersApi;
