// @hubspot/api-client/lib/codegen/crm/associations/v4/models/AssociationSpecWithLabel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssociationSpecWithLabelCategoryEnum = exports.AssociationSpecWithLabel = void 0;
class AssociationSpecWithLabel {
    static getAttributeTypeMap() {
        return AssociationSpecWithLabel.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AssociationSpecWithLabel = AssociationSpecWithLabel;
AssociationSpecWithLabel.discriminator = undefined;
AssociationSpecWithLabel.attributeTypeMap = [
    {
        "name": "typeId",
        "baseName": "typeId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "AssociationSpecWithLabelCategoryEnum",
        "format": ""
    }
];
var AssociationSpecWithLabelCategoryEnum;
(function (AssociationSpecWithLabelCategoryEnum) {
    AssociationSpecWithLabelCategoryEnum["HubspotDefined"] = "HUBSPOT_DEFINED";
    AssociationSpecWithLabelCategoryEnum["UserDefined"] = "USER_DEFINED";
    AssociationSpecWithLabelCategoryEnum["IntegratorDefined"] = "INTEGRATOR_DEFINED";
})(AssociationSpecWithLabelCategoryEnum = exports.AssociationSpecWithLabelCategoryEnum || (exports.AssociationSpecWithLabelCategoryEnum = {}));
