// @hubspot/api-client/lib/codegen/crm/associations/v4/models/BatchResponseLabelsBetweenObjectPairWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum = exports.BatchResponseLabelsBetweenObjectPairWithErrors = void 0;
class BatchResponseLabelsBetweenObjectPairWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseLabelsBetweenObjectPairWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseLabelsBetweenObjectPairWithErrors = BatchResponseLabelsBetweenObjectPairWithErrors;
BatchResponseLabelsBetweenObjectPairWithErrors.discriminator = undefined;
BatchResponseLabelsBetweenObjectPairWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<LabelsBetweenObjectPair>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum;
(function (BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum) {
    BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum = exports.BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum || (exports.BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum = {}));
