// @hubspot/api-client/lib/codegen/crm/associations/v4/models/BatchResponsePublicDefaultAssociation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicDefaultAssociationStatusEnum = exports.BatchResponsePublicDefaultAssociation = void 0;
class BatchResponsePublicDefaultAssociation {
    static getAttributeTypeMap() {
        return BatchResponsePublicDefaultAssociation.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicDefaultAssociation = BatchResponsePublicDefaultAssociation;
BatchResponsePublicDefaultAssociation.discriminator = undefined;
BatchResponsePublicDefaultAssociation.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicDefaultAssociation>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicDefaultAssociationStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicDefaultAssociationStatusEnum;
(function (BatchResponsePublicDefaultAssociationStatusEnum) {
    BatchResponsePublicDefaultAssociationStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicDefaultAssociationStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicDefaultAssociationStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicDefaultAssociationStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicDefaultAssociationStatusEnum = exports.BatchResponsePublicDefaultAssociationStatusEnum || (exports.BatchResponsePublicDefaultAssociationStatusEnum = {}));
