// @hubspot/api-client/lib/codegen/crm/associations/v4/models/BatchResponsePublicAssociationMultiWithLabel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationMultiWithLabelStatusEnum = exports.BatchResponsePublicAssociationMultiWithLabel = void 0;
class BatchResponsePublicAssociationMultiWithLabel {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationMultiWithLabel.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationMultiWithLabel = BatchResponsePublicAssociationMultiWithLabel;
BatchResponsePublicAssociationMultiWithLabel.discriminator = undefined;
BatchResponsePublicAssociationMultiWithLabel.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationMultiWithLabel>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationMultiWithLabelStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationMultiWithLabelStatusEnum;
(function (BatchResponsePublicAssociationMultiWithLabelStatusEnum) {
    BatchResponsePublicAssociationMultiWithLabelStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationMultiWithLabelStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationMultiWithLabelStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationMultiWithLabelStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationMultiWithLabelStatusEnum = exports.BatchResponsePublicAssociationMultiWithLabelStatusEnum || (exports.BatchResponsePublicAssociationMultiWithLabelStatusEnum = {}));
