// @hubspot/api-client/lib/codegen/crm/extensions/cards/models/ActionHookActionBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionHookActionBodyHttpMethodEnum = exports.ActionHookActionBodyTypeEnum = exports.ActionHookActionBody = void 0;
class ActionHookActionBody {
    static getAttributeTypeMap() {
        return ActionHookActionBody.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ActionHookActionBody = ActionHookActionBody;
ActionHookActionBody.discriminator = undefined;
ActionHookActionBody.attributeTypeMap = [
    {
        "name": "propertyNamesIncluded",
        "baseName": "propertyNamesIncluded",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "confirmation",
        "baseName": "confirmation",
        "type": "ActionConfirmationBody",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ActionHookActionBodyTypeEnum",
        "format": ""
    },
    {
        "name": "httpMethod",
        "baseName": "httpMethod",
        "type": "ActionHookActionBodyHttpMethodEnum",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    }
];
var ActionHookActionBodyTypeEnum;
(function (ActionHookActionBodyTypeEnum) {
    ActionHookActionBodyTypeEnum["ActionHook"] = "ACTION_HOOK";
})(ActionHookActionBodyTypeEnum = exports.ActionHookActionBodyTypeEnum || (exports.ActionHookActionBodyTypeEnum = {}));
var ActionHookActionBodyHttpMethodEnum;
(function (ActionHookActionBodyHttpMethodEnum) {
    ActionHookActionBodyHttpMethodEnum["Connect"] = "CONNECT";
    ActionHookActionBodyHttpMethodEnum["Delete"] = "DELETE";
    ActionHookActionBodyHttpMethodEnum["Get"] = "GET";
    ActionHookActionBodyHttpMethodEnum["Head"] = "HEAD";
    ActionHookActionBodyHttpMethodEnum["Options"] = "OPTIONS";
    ActionHookActionBodyHttpMethodEnum["Patch"] = "PATCH";
    ActionHookActionBodyHttpMethodEnum["Post"] = "POST";
    ActionHookActionBodyHttpMethodEnum["Put"] = "PUT";
    ActionHookActionBodyHttpMethodEnum["Trace"] = "TRACE";
})(ActionHookActionBodyHttpMethodEnum = exports.ActionHookActionBodyHttpMethodEnum || (exports.ActionHookActionBodyHttpMethodEnum = {}));
