// @hubspot/api-client/lib/codegen/crm/extensions/cards/models/CardDisplayProperty.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardDisplayPropertyDataTypeEnum = exports.CardDisplayProperty = void 0;
class CardDisplayProperty {
    static getAttributeTypeMap() {
        return CardDisplayProperty.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CardDisplayProperty = CardDisplayProperty;
CardDisplayProperty.discriminator = undefined;
CardDisplayProperty.attributeTypeMap = [
    {
        "name": "dataType",
        "baseName": "dataType",
        "type": "CardDisplayPropertyDataTypeEnum",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<DisplayOption>",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    }
];
var CardDisplayPropertyDataTypeEnum;
(function (CardDisplayPropertyDataTypeEnum) {
    CardDisplayPropertyDataTypeEnum["Boolean"] = "BOOLEAN";
    CardDisplayPropertyDataTypeEnum["Currency"] = "CURRENCY";
    CardDisplayPropertyDataTypeEnum["Date"] = "DATE";
    CardDisplayPropertyDataTypeEnum["Datetime"] = "DATETIME";
    CardDisplayPropertyDataTypeEnum["Email"] = "EMAIL";
    CardDisplayPropertyDataTypeEnum["Link"] = "LINK";
    CardDisplayPropertyDataTypeEnum["Numeric"] = "NUMERIC";
    CardDisplayPropertyDataTypeEnum["String"] = "STRING";
    CardDisplayPropertyDataTypeEnum["Status"] = "STATUS";
})(CardDisplayPropertyDataTypeEnum = exports.CardDisplayPropertyDataTypeEnum || (exports.CardDisplayPropertyDataTypeEnum = {}));
