// @hubspot/api-client/lib/codegen/crm/extensions/cards/models/CardObjectTypeBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardObjectTypeBodyNameEnum = exports.CardObjectTypeBody = void 0;
class CardObjectTypeBody {
    static getAttributeTypeMap() {
        return CardObjectTypeBody.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CardObjectTypeBody = CardObjectTypeBody;
CardObjectTypeBody.discriminator = undefined;
CardObjectTypeBody.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "CardObjectTypeBodyNameEnum",
        "format": ""
    },
    {
        "name": "propertiesToSend",
        "baseName": "propertiesToSend",
        "type": "Array<string>",
        "format": ""
    }
];
var CardObjectTypeBodyNameEnum;
(function (CardObjectTypeBodyNameEnum) {
    CardObjectTypeBodyNameEnum["Contacts"] = "contacts";
    CardObjectTypeBodyNameEnum["Deals"] = "deals";
    CardObjectTypeBodyNameEnum["Companies"] = "companies";
    CardObjectTypeBodyNameEnum["Tickets"] = "tickets";
    CardObjectTypeBodyNameEnum["MarketingEvents"] = "marketing_events";
})(CardObjectTypeBodyNameEnum = exports.CardObjectTypeBodyNameEnum || (exports.CardObjectTypeBodyNameEnum = {}));
