// @hubspot/api-client/lib/codegen/crm/extensions/cards/models/DisplayOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisplayOptionTypeEnum = exports.DisplayOption = void 0;
class DisplayOption {
    static getAttributeTypeMap() {
        return DisplayOption.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DisplayOption = DisplayOption;
DisplayOption.discriminator = undefined;
DisplayOption.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DisplayOptionTypeEnum",
        "format": ""
    }
];
var DisplayOptionTypeEnum;
(function (DisplayOptionTypeEnum) {
    DisplayOptionTypeEnum["Default"] = "DEFAULT";
    DisplayOptionTypeEnum["Success"] = "SUCCESS";
    DisplayOptionTypeEnum["Warning"] = "WARNING";
    DisplayOptionTypeEnum["Danger"] = "DANGER";
    DisplayOptionTypeEnum["Info"] = "INFO";
})(DisplayOptionTypeEnum = exports.DisplayOptionTypeEnum || (exports.DisplayOptionTypeEnum = {}));
