// @hubspot/api-client/lib/codegen/crm/extensions/cards/models/IntegratorObjectResultActionsInner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegratorObjectResultActionsInnerHttpMethodEnum = exports.IntegratorObjectResultActionsInnerTypeEnum = exports.IntegratorObjectResultActionsInner = void 0;
class IntegratorObjectResultActionsInner {
    static getAttributeTypeMap() {
        return IntegratorObjectResultActionsInner.attributeTypeMap;
    }
    constructor() {
    }
}
exports.IntegratorObjectResultActionsInner = IntegratorObjectResultActionsInner;
IntegratorObjectResultActionsInner.discriminator = undefined;
IntegratorObjectResultActionsInner.attributeTypeMap = [
    {
        "name": "propertyNamesIncluded",
        "baseName": "propertyNamesIncluded",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "confirmation",
        "baseName": "confirmation",
        "type": "ActionConfirmationBody",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "IntegratorObjectResultActionsInnerTypeEnum",
        "format": ""
    },
    {
        "name": "httpMethod",
        "baseName": "httpMethod",
        "type": "IntegratorObjectResultActionsInnerHttpMethodEnum",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    },
    {
        "name": "width",
        "baseName": "width",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "height",
        "baseName": "height",
        "type": "number",
        "format": "int32"
    }
];
var IntegratorObjectResultActionsInnerTypeEnum;
(function (IntegratorObjectResultActionsInnerTypeEnum) {
    IntegratorObjectResultActionsInnerTypeEnum["ActionHook"] = "ACTION_HOOK";
    IntegratorObjectResultActionsInnerTypeEnum["Iframe"] = "IFRAME";
})(IntegratorObjectResultActionsInnerTypeEnum = exports.IntegratorObjectResultActionsInnerTypeEnum || (exports.IntegratorObjectResultActionsInnerTypeEnum = {}));
var IntegratorObjectResultActionsInnerHttpMethodEnum;
(function (IntegratorObjectResultActionsInnerHttpMethodEnum) {
    IntegratorObjectResultActionsInnerHttpMethodEnum["Connect"] = "CONNECT";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Delete"] = "DELETE";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Get"] = "GET";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Head"] = "HEAD";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Options"] = "OPTIONS";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Patch"] = "PATCH";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Post"] = "POST";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Put"] = "PUT";
    IntegratorObjectResultActionsInnerHttpMethodEnum["Trace"] = "TRACE";
})(IntegratorObjectResultActionsInnerHttpMethodEnum = exports.IntegratorObjectResultActionsInnerHttpMethodEnum || (exports.IntegratorObjectResultActionsInnerHttpMethodEnum = {}));
