// @hubspot/api-client/lib/codegen/crm/extensions/cards/models/ObjectToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectTokenDataTypeEnum = exports.ObjectToken = void 0;
class ObjectToken {
    static getAttributeTypeMap() {
        return ObjectToken.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ObjectToken = ObjectToken;
ObjectToken.discriminator = undefined;
ObjectToken.attributeTypeMap = [
    {
        "name": "dataType",
        "baseName": "dataType",
        "type": "ObjectTokenDataTypeEnum",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string",
        "format": ""
    }
];
var ObjectTokenDataTypeEnum;
(function (ObjectTokenDataTypeEnum) {
    ObjectTokenDataTypeEnum["Boolean"] = "BOOLEAN";
    ObjectTokenDataTypeEnum["Currency"] = "CURRENCY";
    ObjectTokenDataTypeEnum["Date"] = "DATE";
    ObjectTokenDataTypeEnum["Datetime"] = "DATETIME";
    ObjectTokenDataTypeEnum["Email"] = "EMAIL";
    ObjectTokenDataTypeEnum["Link"] = "LINK";
    ObjectTokenDataTypeEnum["Numeric"] = "NUMERIC";
    ObjectTokenDataTypeEnum["String"] = "STRING";
    ObjectTokenDataTypeEnum["Status"] = "STATUS";
})(ObjectTokenDataTypeEnum = exports.ObjectTokenDataTypeEnum || (exports.ObjectTokenDataTypeEnum = {}));
