// @hubspot/api-client/lib/codegen/crm/pipelines/models/PipelineStage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipelineStageWritePermissionsEnum = exports.PipelineStage = void 0;
class PipelineStage {
    static getAttributeTypeMap() {
        return PipelineStage.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PipelineStage = PipelineStage;
PipelineStage.discriminator = undefined;
PipelineStage.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "archivedAt",
        "baseName": "archivedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "displayOrder",
        "baseName": "displayOrder",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "writePermissions",
        "baseName": "writePermissions",
        "type": "PipelineStageWritePermissionsEnum",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    }
];
var PipelineStageWritePermissionsEnum;
(function (PipelineStageWritePermissionsEnum) {
    PipelineStageWritePermissionsEnum["CrmPermissionsEnforcement"] = "CRM_PERMISSIONS_ENFORCEMENT";
    PipelineStageWritePermissionsEnum["ReadOnly"] = "READ_ONLY";
    PipelineStageWritePermissionsEnum["InternalOnly"] = "INTERNAL_ONLY";
})(PipelineStageWritePermissionsEnum = exports.PipelineStageWritePermissionsEnum || (exports.PipelineStageWritePermissionsEnum = {}));
