// @hubspot/api-client/lib/codegen/crm/objects/taxes/models/AssociationSpec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssociationSpecAssociationCategoryEnum = exports.AssociationSpec = void 0;
class AssociationSpec {
    static getAttributeTypeMap() {
        return AssociationSpec.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AssociationSpec = AssociationSpec;
AssociationSpec.discriminator = undefined;
AssociationSpec.attributeTypeMap = [
    {
        "name": "associationCategory",
        "baseName": "associationCategory",
        "type": "AssociationSpecAssociationCategoryEnum",
        "format": ""
    },
    {
        "name": "associationTypeId",
        "baseName": "associationTypeId",
        "type": "number",
        "format": "int32"
    }
];
var AssociationSpecAssociationCategoryEnum;
(function (AssociationSpecAssociationCategoryEnum) {
    AssociationSpecAssociationCategoryEnum["HubspotDefined"] = "HUBSPOT_DEFINED";
    AssociationSpecAssociationCategoryEnum["UserDefined"] = "USER_DEFINED";
    AssociationSpecAssociationCategoryEnum["IntegratorDefined"] = "INTEGRATOR_DEFINED";
})(AssociationSpecAssociationCategoryEnum = exports.AssociationSpecAssociationCategoryEnum || (exports.AssociationSpecAssociationCategoryEnum = {}));
