// @hubspot/api-client/lib/codegen/crm/objects/taxes/models/Filter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterOperatorEnum = exports.Filter = void 0;
class Filter {
    static getAttributeTypeMap() {
        return Filter.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Filter = Filter;
Filter.discriminator = undefined;
Filter.attributeTypeMap = [
    {
        "name": "highValue",
        "baseName": "highValue",
        "type": "string",
        "format": ""
    },
    {
        "name": "propertyName",
        "baseName": "propertyName",
        "type": "string",
        "format": ""
    },
    {
        "name": "values",
        "baseName": "values",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "FilterOperatorEnum",
        "format": ""
    }
];
var FilterOperatorEnum;
(function (FilterOperatorEnum) {
    FilterOperatorEnum["Eq"] = "EQ";
    FilterOperatorEnum["Neq"] = "NEQ";
    FilterOperatorEnum["Lt"] = "LT";
    FilterOperatorEnum["Lte"] = "LTE";
    FilterOperatorEnum["Gt"] = "GT";
    FilterOperatorEnum["Gte"] = "GTE";
    FilterOperatorEnum["Between"] = "BETWEEN";
    FilterOperatorEnum["In"] = "IN";
    FilterOperatorEnum["NotIn"] = "NOT_IN";
    FilterOperatorEnum["HasProperty"] = "HAS_PROPERTY";
    FilterOperatorEnum["NotHasProperty"] = "NOT_HAS_PROPERTY";
    FilterOperatorEnum["ContainsToken"] = "CONTAINS_TOKEN";
    FilterOperatorEnum["NotContainsToken"] = "NOT_CONTAINS_TOKEN";
})(FilterOperatorEnum = exports.FilterOperatorEnum || (exports.FilterOperatorEnum = {}));
