// @hubspot/api-client/lib/codegen/cms/blogs/tags/models/BatchResponseTag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseTagStatusEnum = exports.BatchResponseTag = void 0;
class BatchResponseTag {
    static getAttributeTypeMap() {
        return BatchResponseTag.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseTag = BatchResponseTag;
BatchResponseTag.discriminator = undefined;
BatchResponseTag.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<Tag>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseTagStatusEnum",
        "format": ""
    }
];
var BatchResponseTagStatusEnum;
(function (BatchResponseTagStatusEnum) {
    BatchResponseTagStatusEnum["Pending"] = "PENDING";
    BatchResponseTagStatusEnum["Processing"] = "PROCESSING";
    BatchResponseTagStatusEnum["Canceled"] = "CANCELED";
    BatchResponseTagStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseTagStatusEnum = exports.BatchResponseTagStatusEnum || (exports.BatchResponseTagStatusEnum = {}));
