// @hubspot/api-client/lib/codegen/cms/blogs/tags/models/BatchResponseTagWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseTagWithErrorsStatusEnum = exports.BatchResponseTagWithErrors = void 0;
class BatchResponseTagWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseTagWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseTagWithErrors = BatchResponseTagWithErrors;
BatchResponseTagWithErrors.discriminator = undefined;
BatchResponseTagWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<Tag>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseTagWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseTagWithErrorsStatusEnum;
(function (BatchResponseTagWithErrorsStatusEnum) {
    BatchResponseTagWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseTagWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseTagWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseTagWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseTagWithErrorsStatusEnum = exports.BatchResponseTagWithErrorsStatusEnum || (exports.BatchResponseTagWithErrorsStatusEnum = {}));
