// @hubspot/api-client/lib/codegen/cms/blogs/tags/models/Tag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagLanguageEnum = exports.Tag = void 0;
class Tag {
    static getAttributeTypeMap() {
        return Tag.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Tag = Tag;
Tag.discriminator = undefined;
Tag.attributeTypeMap = [
    {
        "name": "deletedAt",
        "baseName": "deletedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "TagLanguageEnum",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "translatedFromId",
        "baseName": "translatedFromId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date",
        "format": "date-time"
    }
];
var TagLanguageEnum;
(function (TagLanguageEnum) {
    TagLanguageEnum["Af"] = "af";
    TagLanguageEnum["AfNa"] = "af-na";
    TagLanguageEnum["AfZa"] = "af-za";
    TagLanguageEnum["Agq"] = "agq";
    TagLanguageEnum["AgqCm"] = "agq-cm";
    TagLanguageEnum["Ak"] = "ak";
    TagLanguageEnum["AkGh"] = "ak-gh";
    TagLanguageEnum["Am"] = "am";
    TagLanguageEnum["AmEt"] = "am-et";
    TagLanguageEnum["Ar"] = "ar";
    TagLanguageEnum["Ar001"] = "ar-001";
    TagLanguageEnum["ArAe"] = "ar-ae";
    TagLanguageEnum["ArBh"] = "ar-bh";
    TagLanguageEnum["ArDj"] = "ar-dj";
    TagLanguageEnum["ArDz"] = "ar-dz";
    TagLanguageEnum["ArEg"] = "ar-eg";
    TagLanguageEnum["ArEh"] = "ar-eh";
    TagLanguageEnum["ArEr"] = "ar-er";
    TagLanguageEnum["ArIl"] = "ar-il";
    TagLanguageEnum["ArIq"] = "ar-iq";
    TagLanguageEnum["ArJo"] = "ar-jo";
    TagLanguageEnum["ArKm"] = "ar-km";
    TagLanguageEnum["ArKw"] = "ar-kw";
    TagLanguageEnum["ArLb"] = "ar-lb";
    TagLanguageEnum["ArLy"] = "ar-ly";
    TagLanguageEnum["ArMa"] = "ar-ma";
    TagLanguageEnum["ArMr"] = "ar-mr";
    TagLanguageEnum["ArOm"] = "ar-om";
    TagLanguageEnum["ArPs"] = "ar-ps";
    TagLanguageEnum["ArQa"] = "ar-qa";
    TagLanguageEnum["ArSa"] = "ar-sa";
    TagLanguageEnum["ArSd"] = "ar-sd";
    TagLanguageEnum["ArSo"] = "ar-so";
    TagLanguageEnum["ArSs"] = "ar-ss";
    TagLanguageEnum["ArSy"] = "ar-sy";
    TagLanguageEnum["ArTd"] = "ar-td";
    TagLanguageEnum["ArTn"] = "ar-tn";
    TagLanguageEnum["ArYe"] = "ar-ye";
    TagLanguageEnum["As"] = "as";
    TagLanguageEnum["AsIn"] = "as-in";
    TagLanguageEnum["Asa"] = "asa";
    TagLanguageEnum["AsaTz"] = "asa-tz";
    TagLanguageEnum["Ast"] = "ast";
    TagLanguageEnum["AstEs"] = "ast-es";
    TagLanguageEnum["Az"] = "az";
    TagLanguageEnum["AzAz"] = "az-az";
    TagLanguageEnum["Bas"] = "bas";
    TagLanguageEnum["BasCm"] = "bas-cm";
    TagLanguageEnum["Be"] = "be";
    TagLanguageEnum["BeBy"] = "be-by";
    TagLanguageEnum["Bem"] = "bem";
    TagLanguageEnum["BemZm"] = "bem-zm";
    TagLanguageEnum["Bez"] = "bez";
    TagLanguageEnum["BezTz"] = "bez-tz";
    TagLanguageEnum["Bg"] = "bg";
    TagLanguageEnum["BgBg"] = "bg-bg";
    TagLanguageEnum["Bm"] = "bm";
    TagLanguageEnum["BmMl"] = "bm-ml";
    TagLanguageEnum["Bn"] = "bn";
    TagLanguageEnum["BnBd"] = "bn-bd";
    TagLanguageEnum["BnIn"] = "bn-in";
    TagLanguageEnum["Bo"] = "bo";
    TagLanguageEnum["BoCn"] = "bo-cn";
    TagLanguageEnum["BoIn"] = "bo-in";
    TagLanguageEnum["Br"] = "br";
    TagLanguageEnum["BrFr"] = "br-fr";
    TagLanguageEnum["Brx"] = "brx";
    TagLanguageEnum["BrxIn"] = "brx-in";
    TagLanguageEnum["Bs"] = "bs";
    TagLanguageEnum["BsBa"] = "bs-ba";
    TagLanguageEnum["Ca"] = "ca";
    TagLanguageEnum["CaAd"] = "ca-ad";
    TagLanguageEnum["CaEs"] = "ca-es";
    TagLanguageEnum["CaFr"] = "ca-fr";
    TagLanguageEnum["CaIt"] = "ca-it";
    TagLanguageEnum["Ccp"] = "ccp";
    TagLanguageEnum["CcpBd"] = "ccp-bd";
    TagLanguageEnum["CcpIn"] = "ccp-in";
    TagLanguageEnum["Ce"] = "ce";
    TagLanguageEnum["CeRu"] = "ce-ru";
    TagLanguageEnum["Ceb"] = "ceb";
    TagLanguageEnum["CebPh"] = "ceb-ph";
    TagLanguageEnum["Cgg"] = "cgg";
    TagLanguageEnum["CggUg"] = "cgg-ug";
    TagLanguageEnum["Chr"] = "chr";
    TagLanguageEnum["ChrUs"] = "chr-us";
    TagLanguageEnum["Ckb"] = "ckb";
    TagLanguageEnum["CkbIq"] = "ckb-iq";
    TagLanguageEnum["CkbIr"] = "ckb-ir";
    TagLanguageEnum["Cs"] = "cs";
    TagLanguageEnum["CsCz"] = "cs-cz";
    TagLanguageEnum["Cu"] = "cu";
    TagLanguageEnum["CuRu"] = "cu-ru";
    TagLanguageEnum["Cy"] = "cy";
    TagLanguageEnum["CyGb"] = "cy-gb";
    TagLanguageEnum["Da"] = "da";
    TagLanguageEnum["DaDk"] = "da-dk";
    TagLanguageEnum["DaGl"] = "da-gl";
    TagLanguageEnum["Dav"] = "dav";
    TagLanguageEnum["DavKe"] = "dav-ke";
    TagLanguageEnum["De"] = "de";
    TagLanguageEnum["DeAt"] = "de-at";
    TagLanguageEnum["DeBe"] = "de-be";
    TagLanguageEnum["DeCh"] = "de-ch";
    TagLanguageEnum["DeDe"] = "de-de";
    TagLanguageEnum["DeGr"] = "de-gr";
    TagLanguageEnum["DeIt"] = "de-it";
    TagLanguageEnum["DeLi"] = "de-li";
    TagLanguageEnum["DeLu"] = "de-lu";
    TagLanguageEnum["Dje"] = "dje";
    TagLanguageEnum["DjeNe"] = "dje-ne";
    TagLanguageEnum["Doi"] = "doi";
    TagLanguageEnum["DoiIn"] = "doi-in";
    TagLanguageEnum["Dsb"] = "dsb";
    TagLanguageEnum["DsbDe"] = "dsb-de";
    TagLanguageEnum["Dua"] = "dua";
    TagLanguageEnum["DuaCm"] = "dua-cm";
    TagLanguageEnum["Dyo"] = "dyo";
    TagLanguageEnum["DyoSn"] = "dyo-sn";
    TagLanguageEnum["Dz"] = "dz";
    TagLanguageEnum["DzBt"] = "dz-bt";
    TagLanguageEnum["Ebu"] = "ebu";
    TagLanguageEnum["EbuKe"] = "ebu-ke";
    TagLanguageEnum["Ee"] = "ee";
    TagLanguageEnum["EeGh"] = "ee-gh";
    TagLanguageEnum["EeTg"] = "ee-tg";
    TagLanguageEnum["El"] = "el";
    TagLanguageEnum["ElCy"] = "el-cy";
    TagLanguageEnum["ElGr"] = "el-gr";
    TagLanguageEnum["En"] = "en";
    TagLanguageEnum["En001"] = "en-001";
    TagLanguageEnum["En150"] = "en-150";
    TagLanguageEnum["EnAe"] = "en-ae";
    TagLanguageEnum["EnAg"] = "en-ag";
    TagLanguageEnum["EnAi"] = "en-ai";
    TagLanguageEnum["EnAs"] = "en-as";
    TagLanguageEnum["EnAt"] = "en-at";
    TagLanguageEnum["EnAu"] = "en-au";
    TagLanguageEnum["EnBb"] = "en-bb";
    TagLanguageEnum["EnBe"] = "en-be";
    TagLanguageEnum["EnBi"] = "en-bi";
    TagLanguageEnum["EnBm"] = "en-bm";
    TagLanguageEnum["EnBs"] = "en-bs";
    TagLanguageEnum["EnBw"] = "en-bw";
    TagLanguageEnum["EnBz"] = "en-bz";
    TagLanguageEnum["EnCa"] = "en-ca";
    TagLanguageEnum["EnCc"] = "en-cc";
    TagLanguageEnum["EnCh"] = "en-ch";
    TagLanguageEnum["EnCk"] = "en-ck";
    TagLanguageEnum["EnCm"] = "en-cm";
    TagLanguageEnum["EnCn"] = "en-cn";
    TagLanguageEnum["EnCx"] = "en-cx";
    TagLanguageEnum["EnCy"] = "en-cy";
    TagLanguageEnum["EnDe"] = "en-de";
    TagLanguageEnum["EnDg"] = "en-dg";
    TagLanguageEnum["EnDk"] = "en-dk";
    TagLanguageEnum["EnDm"] = "en-dm";
    TagLanguageEnum["EnEr"] = "en-er";
    TagLanguageEnum["EnFi"] = "en-fi";
    TagLanguageEnum["EnFj"] = "en-fj";
    TagLanguageEnum["EnFk"] = "en-fk";
    TagLanguageEnum["EnFm"] = "en-fm";
    TagLanguageEnum["EnGb"] = "en-gb";
    TagLanguageEnum["EnGd"] = "en-gd";
    TagLanguageEnum["EnGg"] = "en-gg";
    TagLanguageEnum["EnGh"] = "en-gh";
    TagLanguageEnum["EnGi"] = "en-gi";
    TagLanguageEnum["EnGm"] = "en-gm";
    TagLanguageEnum["EnGu"] = "en-gu";
    TagLanguageEnum["EnGy"] = "en-gy";
    TagLanguageEnum["EnHk"] = "en-hk";
    TagLanguageEnum["EnIe"] = "en-ie";
    TagLanguageEnum["EnIl"] = "en-il";
    TagLanguageEnum["EnIm"] = "en-im";
    TagLanguageEnum["EnIn"] = "en-in";
    TagLanguageEnum["EnIo"] = "en-io";
    TagLanguageEnum["EnJe"] = "en-je";
    TagLanguageEnum["EnJm"] = "en-jm";
    TagLanguageEnum["EnKe"] = "en-ke";
    TagLanguageEnum["EnKi"] = "en-ki";
    TagLanguageEnum["EnKn"] = "en-kn";
    TagLanguageEnum["EnKy"] = "en-ky";
    TagLanguageEnum["EnLc"] = "en-lc";
    TagLanguageEnum["EnLr"] = "en-lr";
    TagLanguageEnum["EnLs"] = "en-ls";
    TagLanguageEnum["EnLu"] = "en-lu";
    TagLanguageEnum["EnMg"] = "en-mg";
    TagLanguageEnum["EnMh"] = "en-mh";
    TagLanguageEnum["EnMo"] = "en-mo";
    TagLanguageEnum["EnMp"] = "en-mp";
    TagLanguageEnum["EnMs"] = "en-ms";
    TagLanguageEnum["EnMt"] = "en-mt";
    TagLanguageEnum["EnMu"] = "en-mu";
    TagLanguageEnum["EnMw"] = "en-mw";
    TagLanguageEnum["EnMx"] = "en-mx";
    TagLanguageEnum["EnMy"] = "en-my";
    TagLanguageEnum["EnNa"] = "en-na";
    TagLanguageEnum["EnNf"] = "en-nf";
    TagLanguageEnum["EnNg"] = "en-ng";
    TagLanguageEnum["EnNl"] = "en-nl";
    TagLanguageEnum["EnNr"] = "en-nr";
    TagLanguageEnum["EnNu"] = "en-nu";
    TagLanguageEnum["EnNz"] = "en-nz";
    TagLanguageEnum["EnPg"] = "en-pg";
    TagLanguageEnum["EnPh"] = "en-ph";
    TagLanguageEnum["EnPk"] = "en-pk";
    TagLanguageEnum["EnPn"] = "en-pn";
    TagLanguageEnum["EnPr"] = "en-pr";
    TagLanguageEnum["EnPw"] = "en-pw";
    TagLanguageEnum["EnRw"] = "en-rw";
    TagLanguageEnum["EnSb"] = "en-sb";
    TagLanguageEnum["EnSc"] = "en-sc";
    TagLanguageEnum["EnSd"] = "en-sd";
    TagLanguageEnum["EnSe"] = "en-se";
    TagLanguageEnum["EnSg"] = "en-sg";
    TagLanguageEnum["EnSh"] = "en-sh";
    TagLanguageEnum["EnSi"] = "en-si";
    TagLanguageEnum["EnSl"] = "en-sl";
    TagLanguageEnum["EnSs"] = "en-ss";
    TagLanguageEnum["EnSx"] = "en-sx";
    TagLanguageEnum["EnSz"] = "en-sz";
    TagLanguageEnum["EnTc"] = "en-tc";
    TagLanguageEnum["EnTk"] = "en-tk";
    TagLanguageEnum["EnTo"] = "en-to";
    TagLanguageEnum["EnTt"] = "en-tt";
    TagLanguageEnum["EnTv"] = "en-tv";
    TagLanguageEnum["EnTz"] = "en-tz";
    TagLanguageEnum["EnUg"] = "en-ug";
    TagLanguageEnum["EnUm"] = "en-um";
    TagLanguageEnum["EnUs"] = "en-us";
    TagLanguageEnum["EnVc"] = "en-vc";
    TagLanguageEnum["EnVg"] = "en-vg";
    TagLanguageEnum["EnVi"] = "en-vi";
    TagLanguageEnum["EnVu"] = "en-vu";
    TagLanguageEnum["EnWs"] = "en-ws";
    TagLanguageEnum["EnZa"] = "en-za";
    TagLanguageEnum["EnZm"] = "en-zm";
    TagLanguageEnum["EnZw"] = "en-zw";
    TagLanguageEnum["Eo"] = "eo";
    TagLanguageEnum["Eo001"] = "eo-001";
    TagLanguageEnum["Es"] = "es";
    TagLanguageEnum["Es419"] = "es-419";
    TagLanguageEnum["EsAr"] = "es-ar";
    TagLanguageEnum["EsBo"] = "es-bo";
    TagLanguageEnum["EsBr"] = "es-br";
    TagLanguageEnum["EsBz"] = "es-bz";
    TagLanguageEnum["EsCl"] = "es-cl";
    TagLanguageEnum["EsCo"] = "es-co";
    TagLanguageEnum["EsCr"] = "es-cr";
    TagLanguageEnum["EsCu"] = "es-cu";
    TagLanguageEnum["EsDo"] = "es-do";
    TagLanguageEnum["EsEa"] = "es-ea";
    TagLanguageEnum["EsEc"] = "es-ec";
    TagLanguageEnum["EsEs"] = "es-es";
    TagLanguageEnum["EsGq"] = "es-gq";
    TagLanguageEnum["EsGt"] = "es-gt";
    TagLanguageEnum["EsHn"] = "es-hn";
    TagLanguageEnum["EsIc"] = "es-ic";
    TagLanguageEnum["EsMx"] = "es-mx";
    TagLanguageEnum["EsNi"] = "es-ni";
    TagLanguageEnum["EsPa"] = "es-pa";
    TagLanguageEnum["EsPe"] = "es-pe";
    TagLanguageEnum["EsPh"] = "es-ph";
    TagLanguageEnum["EsPr"] = "es-pr";
    TagLanguageEnum["EsPy"] = "es-py";
    TagLanguageEnum["EsSv"] = "es-sv";
    TagLanguageEnum["EsUs"] = "es-us";
    TagLanguageEnum["EsUy"] = "es-uy";
    TagLanguageEnum["EsVe"] = "es-ve";
    TagLanguageEnum["Et"] = "et";
    TagLanguageEnum["EtEe"] = "et-ee";
    TagLanguageEnum["Eu"] = "eu";
    TagLanguageEnum["EuEs"] = "eu-es";
    TagLanguageEnum["Ewo"] = "ewo";
    TagLanguageEnum["EwoCm"] = "ewo-cm";
    TagLanguageEnum["Fa"] = "fa";
    TagLanguageEnum["FaAf"] = "fa-af";
    TagLanguageEnum["FaIr"] = "fa-ir";
    TagLanguageEnum["Ff"] = "ff";
    TagLanguageEnum["FfBf"] = "ff-bf";
    TagLanguageEnum["FfCm"] = "ff-cm";
    TagLanguageEnum["FfGh"] = "ff-gh";
    TagLanguageEnum["FfGm"] = "ff-gm";
    TagLanguageEnum["FfGn"] = "ff-gn";
    TagLanguageEnum["FfGw"] = "ff-gw";
    TagLanguageEnum["FfLr"] = "ff-lr";
    TagLanguageEnum["FfMr"] = "ff-mr";
    TagLanguageEnum["FfNe"] = "ff-ne";
    TagLanguageEnum["FfNg"] = "ff-ng";
    TagLanguageEnum["FfSl"] = "ff-sl";
    TagLanguageEnum["FfSn"] = "ff-sn";
    TagLanguageEnum["Fi"] = "fi";
    TagLanguageEnum["FiFi"] = "fi-fi";
    TagLanguageEnum["Fil"] = "fil";
    TagLanguageEnum["FilPh"] = "fil-ph";
    TagLanguageEnum["Fo"] = "fo";
    TagLanguageEnum["FoDk"] = "fo-dk";
    TagLanguageEnum["FoFo"] = "fo-fo";
    TagLanguageEnum["Fr"] = "fr";
    TagLanguageEnum["FrBe"] = "fr-be";
    TagLanguageEnum["FrBf"] = "fr-bf";
    TagLanguageEnum["FrBi"] = "fr-bi";
    TagLanguageEnum["FrBj"] = "fr-bj";
    TagLanguageEnum["FrBl"] = "fr-bl";
    TagLanguageEnum["FrCa"] = "fr-ca";
    TagLanguageEnum["FrCd"] = "fr-cd";
    TagLanguageEnum["FrCf"] = "fr-cf";
    TagLanguageEnum["FrCg"] = "fr-cg";
    TagLanguageEnum["FrCh"] = "fr-ch";
    TagLanguageEnum["FrCi"] = "fr-ci";
    TagLanguageEnum["FrCm"] = "fr-cm";
    TagLanguageEnum["FrDj"] = "fr-dj";
    TagLanguageEnum["FrDz"] = "fr-dz";
    TagLanguageEnum["FrFr"] = "fr-fr";
    TagLanguageEnum["FrGa"] = "fr-ga";
    TagLanguageEnum["FrGf"] = "fr-gf";
    TagLanguageEnum["FrGn"] = "fr-gn";
    TagLanguageEnum["FrGp"] = "fr-gp";
    TagLanguageEnum["FrGq"] = "fr-gq";
    TagLanguageEnum["FrHt"] = "fr-ht";
    TagLanguageEnum["FrKm"] = "fr-km";
    TagLanguageEnum["FrLu"] = "fr-lu";
    TagLanguageEnum["FrMa"] = "fr-ma";
    TagLanguageEnum["FrMc"] = "fr-mc";
    TagLanguageEnum["FrMf"] = "fr-mf";
    TagLanguageEnum["FrMg"] = "fr-mg";
    TagLanguageEnum["FrMl"] = "fr-ml";
    TagLanguageEnum["FrMq"] = "fr-mq";
    TagLanguageEnum["FrMr"] = "fr-mr";
    TagLanguageEnum["FrMu"] = "fr-mu";
    TagLanguageEnum["FrNc"] = "fr-nc";
    TagLanguageEnum["FrNe"] = "fr-ne";
    TagLanguageEnum["FrPf"] = "fr-pf";
    TagLanguageEnum["FrPm"] = "fr-pm";
    TagLanguageEnum["FrRe"] = "fr-re";
    TagLanguageEnum["FrRw"] = "fr-rw";
    TagLanguageEnum["FrSc"] = "fr-sc";
    TagLanguageEnum["FrSn"] = "fr-sn";
    TagLanguageEnum["FrSy"] = "fr-sy";
    TagLanguageEnum["FrTd"] = "fr-td";
    TagLanguageEnum["FrTg"] = "fr-tg";
    TagLanguageEnum["FrTn"] = "fr-tn";
    TagLanguageEnum["FrVu"] = "fr-vu";
    TagLanguageEnum["FrWf"] = "fr-wf";
    TagLanguageEnum["FrYt"] = "fr-yt";
    TagLanguageEnum["Fur"] = "fur";
    TagLanguageEnum["FurIt"] = "fur-it";
    TagLanguageEnum["Fy"] = "fy";
    TagLanguageEnum["FyNl"] = "fy-nl";
    TagLanguageEnum["Ga"] = "ga";
    TagLanguageEnum["GaGb"] = "ga-gb";
    TagLanguageEnum["GaIe"] = "ga-ie";
    TagLanguageEnum["Gd"] = "gd";
    TagLanguageEnum["GdGb"] = "gd-gb";
    TagLanguageEnum["Gl"] = "gl";
    TagLanguageEnum["GlEs"] = "gl-es";
    TagLanguageEnum["Gsw"] = "gsw";
    TagLanguageEnum["GswCh"] = "gsw-ch";
    TagLanguageEnum["GswFr"] = "gsw-fr";
    TagLanguageEnum["GswLi"] = "gsw-li";
    TagLanguageEnum["Gu"] = "gu";
    TagLanguageEnum["GuIn"] = "gu-in";
    TagLanguageEnum["Guz"] = "guz";
    TagLanguageEnum["GuzKe"] = "guz-ke";
    TagLanguageEnum["Gv"] = "gv";
    TagLanguageEnum["GvIm"] = "gv-im";
    TagLanguageEnum["Ha"] = "ha";
    TagLanguageEnum["HaGh"] = "ha-gh";
    TagLanguageEnum["HaNe"] = "ha-ne";
    TagLanguageEnum["HaNg"] = "ha-ng";
    TagLanguageEnum["Haw"] = "haw";
    TagLanguageEnum["HawUs"] = "haw-us";
    TagLanguageEnum["He"] = "he";
    TagLanguageEnum["Hi"] = "hi";
    TagLanguageEnum["HiIn"] = "hi-in";
    TagLanguageEnum["Hr"] = "hr";
    TagLanguageEnum["HrBa"] = "hr-ba";
    TagLanguageEnum["HrHr"] = "hr-hr";
    TagLanguageEnum["Hsb"] = "hsb";
    TagLanguageEnum["HsbDe"] = "hsb-de";
    TagLanguageEnum["Hu"] = "hu";
    TagLanguageEnum["HuHu"] = "hu-hu";
    TagLanguageEnum["Hy"] = "hy";
    TagLanguageEnum["HyAm"] = "hy-am";
    TagLanguageEnum["Ia"] = "ia";
    TagLanguageEnum["Ia001"] = "ia-001";
    TagLanguageEnum["Id"] = "id";
    TagLanguageEnum["Ig"] = "ig";
    TagLanguageEnum["IgNg"] = "ig-ng";
    TagLanguageEnum["Ii"] = "ii";
    TagLanguageEnum["IiCn"] = "ii-cn";
    TagLanguageEnum["IdId"] = "id-id";
    TagLanguageEnum["Is"] = "is";
    TagLanguageEnum["IsIs"] = "is-is";
    TagLanguageEnum["It"] = "it";
    TagLanguageEnum["ItCh"] = "it-ch";
    TagLanguageEnum["ItIt"] = "it-it";
    TagLanguageEnum["ItSm"] = "it-sm";
    TagLanguageEnum["ItVa"] = "it-va";
    TagLanguageEnum["HeIl"] = "he-il";
    TagLanguageEnum["Ja"] = "ja";
    TagLanguageEnum["JaJp"] = "ja-jp";
    TagLanguageEnum["Jgo"] = "jgo";
    TagLanguageEnum["JgoCm"] = "jgo-cm";
    TagLanguageEnum["Yi"] = "yi";
    TagLanguageEnum["Yi001"] = "yi-001";
    TagLanguageEnum["Jmc"] = "jmc";
    TagLanguageEnum["JmcTz"] = "jmc-tz";
    TagLanguageEnum["Jv"] = "jv";
    TagLanguageEnum["JvId"] = "jv-id";
    TagLanguageEnum["Ka"] = "ka";
    TagLanguageEnum["KaGe"] = "ka-ge";
    TagLanguageEnum["Kab"] = "kab";
    TagLanguageEnum["KabDz"] = "kab-dz";
    TagLanguageEnum["Kam"] = "kam";
    TagLanguageEnum["KamKe"] = "kam-ke";
    TagLanguageEnum["Kde"] = "kde";
    TagLanguageEnum["KdeTz"] = "kde-tz";
    TagLanguageEnum["Kea"] = "kea";
    TagLanguageEnum["KeaCv"] = "kea-cv";
    TagLanguageEnum["Khq"] = "khq";
    TagLanguageEnum["KhqMl"] = "khq-ml";
    TagLanguageEnum["Ki"] = "ki";
    TagLanguageEnum["KiKe"] = "ki-ke";
    TagLanguageEnum["Kk"] = "kk";
    TagLanguageEnum["KkKz"] = "kk-kz";
    TagLanguageEnum["Kkj"] = "kkj";
    TagLanguageEnum["KkjCm"] = "kkj-cm";
    TagLanguageEnum["Kl"] = "kl";
    TagLanguageEnum["KlGl"] = "kl-gl";
    TagLanguageEnum["Kln"] = "kln";
    TagLanguageEnum["KlnKe"] = "kln-ke";
    TagLanguageEnum["Km"] = "km";
    TagLanguageEnum["KmKh"] = "km-kh";
    TagLanguageEnum["Kn"] = "kn";
    TagLanguageEnum["KnIn"] = "kn-in";
    TagLanguageEnum["Ko"] = "ko";
    TagLanguageEnum["KoKp"] = "ko-kp";
    TagLanguageEnum["KoKr"] = "ko-kr";
    TagLanguageEnum["Kok"] = "kok";
    TagLanguageEnum["KokIn"] = "kok-in";
    TagLanguageEnum["Ks"] = "ks";
    TagLanguageEnum["KsIn"] = "ks-in";
    TagLanguageEnum["Ksb"] = "ksb";
    TagLanguageEnum["KsbTz"] = "ksb-tz";
    TagLanguageEnum["Ksf"] = "ksf";
    TagLanguageEnum["KsfCm"] = "ksf-cm";
    TagLanguageEnum["Ksh"] = "ksh";
    TagLanguageEnum["KshDe"] = "ksh-de";
    TagLanguageEnum["Kw"] = "kw";
    TagLanguageEnum["KwGb"] = "kw-gb";
    TagLanguageEnum["Ku"] = "ku";
    TagLanguageEnum["KuTr"] = "ku-tr";
    TagLanguageEnum["Ky"] = "ky";
    TagLanguageEnum["KyKg"] = "ky-kg";
    TagLanguageEnum["Lag"] = "lag";
    TagLanguageEnum["LagTz"] = "lag-tz";
    TagLanguageEnum["Lb"] = "lb";
    TagLanguageEnum["LbLu"] = "lb-lu";
    TagLanguageEnum["Lg"] = "lg";
    TagLanguageEnum["LgUg"] = "lg-ug";
    TagLanguageEnum["Lkt"] = "lkt";
    TagLanguageEnum["LktUs"] = "lkt-us";
    TagLanguageEnum["Ln"] = "ln";
    TagLanguageEnum["LnAo"] = "ln-ao";
    TagLanguageEnum["LnCd"] = "ln-cd";
    TagLanguageEnum["LnCf"] = "ln-cf";
    TagLanguageEnum["LnCg"] = "ln-cg";
    TagLanguageEnum["Lo"] = "lo";
    TagLanguageEnum["LoLa"] = "lo-la";
    TagLanguageEnum["Lrc"] = "lrc";
    TagLanguageEnum["LrcIq"] = "lrc-iq";
    TagLanguageEnum["LrcIr"] = "lrc-ir";
    TagLanguageEnum["Lt"] = "lt";
    TagLanguageEnum["LtLt"] = "lt-lt";
    TagLanguageEnum["Lu"] = "lu";
    TagLanguageEnum["LuCd"] = "lu-cd";
    TagLanguageEnum["Luo"] = "luo";
    TagLanguageEnum["LuoKe"] = "luo-ke";
    TagLanguageEnum["Luy"] = "luy";
    TagLanguageEnum["LuyKe"] = "luy-ke";
    TagLanguageEnum["Lv"] = "lv";
    TagLanguageEnum["LvLv"] = "lv-lv";
    TagLanguageEnum["Mai"] = "mai";
    TagLanguageEnum["MaiIn"] = "mai-in";
    TagLanguageEnum["Mas"] = "mas";
    TagLanguageEnum["MasKe"] = "mas-ke";
    TagLanguageEnum["MasTz"] = "mas-tz";
    TagLanguageEnum["Mer"] = "mer";
    TagLanguageEnum["MerKe"] = "mer-ke";
    TagLanguageEnum["Mfe"] = "mfe";
    TagLanguageEnum["MfeMu"] = "mfe-mu";
    TagLanguageEnum["Mg"] = "mg";
    TagLanguageEnum["MgMg"] = "mg-mg";
    TagLanguageEnum["Mgh"] = "mgh";
    TagLanguageEnum["MghMz"] = "mgh-mz";
    TagLanguageEnum["Mgo"] = "mgo";
    TagLanguageEnum["MgoCm"] = "mgo-cm";
    TagLanguageEnum["Mi"] = "mi";
    TagLanguageEnum["MiNz"] = "mi-nz";
    TagLanguageEnum["Mk"] = "mk";
    TagLanguageEnum["MkMk"] = "mk-mk";
    TagLanguageEnum["Ml"] = "ml";
    TagLanguageEnum["MlIn"] = "ml-in";
    TagLanguageEnum["Mn"] = "mn";
    TagLanguageEnum["MnMn"] = "mn-mn";
    TagLanguageEnum["Mni"] = "mni";
    TagLanguageEnum["MniIn"] = "mni-in";
    TagLanguageEnum["Mr"] = "mr";
    TagLanguageEnum["MrIn"] = "mr-in";
    TagLanguageEnum["Ms"] = "ms";
    TagLanguageEnum["MsBn"] = "ms-bn";
    TagLanguageEnum["MsId"] = "ms-id";
    TagLanguageEnum["MsMy"] = "ms-my";
    TagLanguageEnum["MsSg"] = "ms-sg";
    TagLanguageEnum["Mt"] = "mt";
    TagLanguageEnum["MtMt"] = "mt-mt";
    TagLanguageEnum["Mua"] = "mua";
    TagLanguageEnum["MuaCm"] = "mua-cm";
    TagLanguageEnum["My"] = "my";
    TagLanguageEnum["MyMm"] = "my-mm";
    TagLanguageEnum["Mzn"] = "mzn";
    TagLanguageEnum["MznIr"] = "mzn-ir";
    TagLanguageEnum["Naq"] = "naq";
    TagLanguageEnum["NaqNa"] = "naq-na";
    TagLanguageEnum["Nb"] = "nb";
    TagLanguageEnum["NbNo"] = "nb-no";
    TagLanguageEnum["NbSj"] = "nb-sj";
    TagLanguageEnum["Nd"] = "nd";
    TagLanguageEnum["NdZw"] = "nd-zw";
    TagLanguageEnum["Nds"] = "nds";
    TagLanguageEnum["NdsDe"] = "nds-de";
    TagLanguageEnum["NdsNl"] = "nds-nl";
    TagLanguageEnum["Ne"] = "ne";
    TagLanguageEnum["NeIn"] = "ne-in";
    TagLanguageEnum["NeNp"] = "ne-np";
    TagLanguageEnum["Nl"] = "nl";
    TagLanguageEnum["NlAw"] = "nl-aw";
    TagLanguageEnum["NlBe"] = "nl-be";
    TagLanguageEnum["NlCh"] = "nl-ch";
    TagLanguageEnum["NlBq"] = "nl-bq";
    TagLanguageEnum["NlCw"] = "nl-cw";
    TagLanguageEnum["NlLu"] = "nl-lu";
    TagLanguageEnum["NlNl"] = "nl-nl";
    TagLanguageEnum["NlSr"] = "nl-sr";
    TagLanguageEnum["NlSx"] = "nl-sx";
    TagLanguageEnum["Nmg"] = "nmg";
    TagLanguageEnum["NmgCm"] = "nmg-cm";
    TagLanguageEnum["Nn"] = "nn";
    TagLanguageEnum["NnNo"] = "nn-no";
    TagLanguageEnum["Nnh"] = "nnh";
    TagLanguageEnum["NnhCm"] = "nnh-cm";
    TagLanguageEnum["No"] = "no";
    TagLanguageEnum["NoNo"] = "no-no";
    TagLanguageEnum["Nus"] = "nus";
    TagLanguageEnum["NusSs"] = "nus-ss";
    TagLanguageEnum["Nyn"] = "nyn";
    TagLanguageEnum["NynUg"] = "nyn-ug";
    TagLanguageEnum["Om"] = "om";
    TagLanguageEnum["OmEt"] = "om-et";
    TagLanguageEnum["OmKe"] = "om-ke";
    TagLanguageEnum["Or"] = "or";
    TagLanguageEnum["OrIn"] = "or-in";
    TagLanguageEnum["Os"] = "os";
    TagLanguageEnum["OsGe"] = "os-ge";
    TagLanguageEnum["OsRu"] = "os-ru";
    TagLanguageEnum["Pa"] = "pa";
    TagLanguageEnum["PaIn"] = "pa-in";
    TagLanguageEnum["PaPk"] = "pa-pk";
    TagLanguageEnum["Pcm"] = "pcm";
    TagLanguageEnum["PcmNg"] = "pcm-ng";
    TagLanguageEnum["Pl"] = "pl";
    TagLanguageEnum["PlPl"] = "pl-pl";
    TagLanguageEnum["Prg"] = "prg";
    TagLanguageEnum["Prg001"] = "prg-001";
    TagLanguageEnum["Ps"] = "ps";
    TagLanguageEnum["PsAf"] = "ps-af";
    TagLanguageEnum["PsPk"] = "ps-pk";
    TagLanguageEnum["Pt"] = "pt";
    TagLanguageEnum["PtAo"] = "pt-ao";
    TagLanguageEnum["PtBr"] = "pt-br";
    TagLanguageEnum["PtCh"] = "pt-ch";
    TagLanguageEnum["PtCv"] = "pt-cv";
    TagLanguageEnum["PtGq"] = "pt-gq";
    TagLanguageEnum["PtGw"] = "pt-gw";
    TagLanguageEnum["PtLu"] = "pt-lu";
    TagLanguageEnum["PtMo"] = "pt-mo";
    TagLanguageEnum["PtMz"] = "pt-mz";
    TagLanguageEnum["PtPt"] = "pt-pt";
    TagLanguageEnum["PtSt"] = "pt-st";
    TagLanguageEnum["PtTl"] = "pt-tl";
    TagLanguageEnum["Qu"] = "qu";
    TagLanguageEnum["QuBo"] = "qu-bo";
    TagLanguageEnum["QuEc"] = "qu-ec";
    TagLanguageEnum["QuPe"] = "qu-pe";
    TagLanguageEnum["Rm"] = "rm";
    TagLanguageEnum["RmCh"] = "rm-ch";
    TagLanguageEnum["Rn"] = "rn";
    TagLanguageEnum["RnBi"] = "rn-bi";
    TagLanguageEnum["Ro"] = "ro";
    TagLanguageEnum["RoMd"] = "ro-md";
    TagLanguageEnum["RoRo"] = "ro-ro";
    TagLanguageEnum["Rof"] = "rof";
    TagLanguageEnum["RofTz"] = "rof-tz";
    TagLanguageEnum["Ru"] = "ru";
    TagLanguageEnum["RuBy"] = "ru-by";
    TagLanguageEnum["RuKg"] = "ru-kg";
    TagLanguageEnum["RuKz"] = "ru-kz";
    TagLanguageEnum["RuMd"] = "ru-md";
    TagLanguageEnum["RuRu"] = "ru-ru";
    TagLanguageEnum["RuUa"] = "ru-ua";
    TagLanguageEnum["Rw"] = "rw";
    TagLanguageEnum["RwRw"] = "rw-rw";
    TagLanguageEnum["Rwk"] = "rwk";
    TagLanguageEnum["RwkTz"] = "rwk-tz";
    TagLanguageEnum["Sa"] = "sa";
    TagLanguageEnum["SaIn"] = "sa-in";
    TagLanguageEnum["Sah"] = "sah";
    TagLanguageEnum["SahRu"] = "sah-ru";
    TagLanguageEnum["Saq"] = "saq";
    TagLanguageEnum["SaqKe"] = "saq-ke";
    TagLanguageEnum["Sat"] = "sat";
    TagLanguageEnum["SatIn"] = "sat-in";
    TagLanguageEnum["Sbp"] = "sbp";
    TagLanguageEnum["SbpTz"] = "sbp-tz";
    TagLanguageEnum["Sd"] = "sd";
    TagLanguageEnum["SdIn"] = "sd-in";
    TagLanguageEnum["SdPk"] = "sd-pk";
    TagLanguageEnum["Se"] = "se";
    TagLanguageEnum["SeFi"] = "se-fi";
    TagLanguageEnum["SeNo"] = "se-no";
    TagLanguageEnum["SeSe"] = "se-se";
    TagLanguageEnum["Seh"] = "seh";
    TagLanguageEnum["SehMz"] = "seh-mz";
    TagLanguageEnum["Ses"] = "ses";
    TagLanguageEnum["SesMl"] = "ses-ml";
    TagLanguageEnum["Sg"] = "sg";
    TagLanguageEnum["SgCf"] = "sg-cf";
    TagLanguageEnum["Shi"] = "shi";
    TagLanguageEnum["ShiMa"] = "shi-ma";
    TagLanguageEnum["Si"] = "si";
    TagLanguageEnum["SiLk"] = "si-lk";
    TagLanguageEnum["Sk"] = "sk";
    TagLanguageEnum["SkSk"] = "sk-sk";
    TagLanguageEnum["Sl"] = "sl";
    TagLanguageEnum["SlSi"] = "sl-si";
    TagLanguageEnum["Smn"] = "smn";
    TagLanguageEnum["SmnFi"] = "smn-fi";
    TagLanguageEnum["Sn"] = "sn";
    TagLanguageEnum["SnZw"] = "sn-zw";
    TagLanguageEnum["So"] = "so";
    TagLanguageEnum["SoDj"] = "so-dj";
    TagLanguageEnum["SoEt"] = "so-et";
    TagLanguageEnum["SoKe"] = "so-ke";
    TagLanguageEnum["SoSo"] = "so-so";
    TagLanguageEnum["Sq"] = "sq";
    TagLanguageEnum["SqAl"] = "sq-al";
    TagLanguageEnum["SqMk"] = "sq-mk";
    TagLanguageEnum["SqXk"] = "sq-xk";
    TagLanguageEnum["Sr"] = "sr";
    TagLanguageEnum["SrBa"] = "sr-ba";
    TagLanguageEnum["SrCs"] = "sr-cs";
    TagLanguageEnum["SrMe"] = "sr-me";
    TagLanguageEnum["SrRs"] = "sr-rs";
    TagLanguageEnum["SrXk"] = "sr-xk";
    TagLanguageEnum["Su"] = "su";
    TagLanguageEnum["SuId"] = "su-id";
    TagLanguageEnum["Sv"] = "sv";
    TagLanguageEnum["SvAx"] = "sv-ax";
    TagLanguageEnum["SvFi"] = "sv-fi";
    TagLanguageEnum["SvSe"] = "sv-se";
    TagLanguageEnum["Sw"] = "sw";
    TagLanguageEnum["SwCd"] = "sw-cd";
    TagLanguageEnum["SwKe"] = "sw-ke";
    TagLanguageEnum["SwTz"] = "sw-tz";
    TagLanguageEnum["SwUg"] = "sw-ug";
    TagLanguageEnum["Sy"] = "sy";
    TagLanguageEnum["Ta"] = "ta";
    TagLanguageEnum["TaIn"] = "ta-in";
    TagLanguageEnum["TaLk"] = "ta-lk";
    TagLanguageEnum["TaMy"] = "ta-my";
    TagLanguageEnum["TaSg"] = "ta-sg";
    TagLanguageEnum["Te"] = "te";
    TagLanguageEnum["TeIn"] = "te-in";
    TagLanguageEnum["Teo"] = "teo";
    TagLanguageEnum["TeoKe"] = "teo-ke";
    TagLanguageEnum["TeoUg"] = "teo-ug";
    TagLanguageEnum["Tg"] = "tg";
    TagLanguageEnum["TgTj"] = "tg-tj";
    TagLanguageEnum["Th"] = "th";
    TagLanguageEnum["ThTh"] = "th-th";
    TagLanguageEnum["Ti"] = "ti";
    TagLanguageEnum["TiEr"] = "ti-er";
    TagLanguageEnum["TiEt"] = "ti-et";
    TagLanguageEnum["Tk"] = "tk";
    TagLanguageEnum["TkTm"] = "tk-tm";
    TagLanguageEnum["Tl"] = "tl";
    TagLanguageEnum["To"] = "to";
    TagLanguageEnum["ToTo"] = "to-to";
    TagLanguageEnum["Tr"] = "tr";
    TagLanguageEnum["TrCy"] = "tr-cy";
    TagLanguageEnum["TrTr"] = "tr-tr";
    TagLanguageEnum["Tt"] = "tt";
    TagLanguageEnum["TtRu"] = "tt-ru";
    TagLanguageEnum["Twq"] = "twq";
    TagLanguageEnum["TwqNe"] = "twq-ne";
    TagLanguageEnum["Tzm"] = "tzm";
    TagLanguageEnum["TzmMa"] = "tzm-ma";
    TagLanguageEnum["Ug"] = "ug";
    TagLanguageEnum["UgCn"] = "ug-cn";
    TagLanguageEnum["Uk"] = "uk";
    TagLanguageEnum["UkUa"] = "uk-ua";
    TagLanguageEnum["Ur"] = "ur";
    TagLanguageEnum["UrIn"] = "ur-in";
    TagLanguageEnum["UrPk"] = "ur-pk";
    TagLanguageEnum["Uz"] = "uz";
    TagLanguageEnum["UzAf"] = "uz-af";
    TagLanguageEnum["UzUz"] = "uz-uz";
    TagLanguageEnum["Vai"] = "vai";
    TagLanguageEnum["VaiLr"] = "vai-lr";
    TagLanguageEnum["Vi"] = "vi";
    TagLanguageEnum["ViVn"] = "vi-vn";
    TagLanguageEnum["Vo"] = "vo";
    TagLanguageEnum["Vo001"] = "vo-001";
    TagLanguageEnum["Vun"] = "vun";
    TagLanguageEnum["VunTz"] = "vun-tz";
    TagLanguageEnum["Wae"] = "wae";
    TagLanguageEnum["WaeCh"] = "wae-ch";
    TagLanguageEnum["Wo"] = "wo";
    TagLanguageEnum["WoSn"] = "wo-sn";
    TagLanguageEnum["Xh"] = "xh";
    TagLanguageEnum["XhZa"] = "xh-za";
    TagLanguageEnum["Xog"] = "xog";
    TagLanguageEnum["XogUg"] = "xog-ug";
    TagLanguageEnum["Yav"] = "yav";
    TagLanguageEnum["YavCm"] = "yav-cm";
    TagLanguageEnum["Yo"] = "yo";
    TagLanguageEnum["YoBj"] = "yo-bj";
    TagLanguageEnum["YoNg"] = "yo-ng";
    TagLanguageEnum["Yue"] = "yue";
    TagLanguageEnum["YueCn"] = "yue-cn";
    TagLanguageEnum["YueHk"] = "yue-hk";
    TagLanguageEnum["Zgh"] = "zgh";
    TagLanguageEnum["ZghMa"] = "zgh-ma";
    TagLanguageEnum["Zh"] = "zh";
    TagLanguageEnum["ZhCn"] = "zh-cn";
    TagLanguageEnum["ZhHk"] = "zh-hk";
    TagLanguageEnum["ZhMo"] = "zh-mo";
    TagLanguageEnum["ZhSg"] = "zh-sg";
    TagLanguageEnum["ZhTw"] = "zh-tw";
    TagLanguageEnum["ZhHans"] = "zh-hans";
    TagLanguageEnum["ZhHant"] = "zh-hant";
    TagLanguageEnum["Zu"] = "zu";
    TagLanguageEnum["ZuZa"] = "zu-za";
})(TagLanguageEnum = exports.TagLanguageEnum || (exports.TagLanguageEnum = {}));
