// @hubspot/api-client/lib/codegen/marketing/events/models/BatchResponseMarketingEventPublicDefaultResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseMarketingEventPublicDefaultResponseStatusEnum = exports.BatchResponseMarketingEventPublicDefaultResponse = void 0;
class BatchResponseMarketingEventPublicDefaultResponse {
    static getAttributeTypeMap() {
        return BatchResponseMarketingEventPublicDefaultResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseMarketingEventPublicDefaultResponse = BatchResponseMarketingEventPublicDefaultResponse;
BatchResponseMarketingEventPublicDefaultResponse.discriminator = undefined;
BatchResponseMarketingEventPublicDefaultResponse.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<MarketingEventPublicDefaultResponse>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseMarketingEventPublicDefaultResponseStatusEnum",
        "format": ""
    }
];
var BatchResponseMarketingEventPublicDefaultResponseStatusEnum;
(function (BatchResponseMarketingEventPublicDefaultResponseStatusEnum) {
    BatchResponseMarketingEventPublicDefaultResponseStatusEnum["Pending"] = "PENDING";
    BatchResponseMarketingEventPublicDefaultResponseStatusEnum["Processing"] = "PROCESSING";
    BatchResponseMarketingEventPublicDefaultResponseStatusEnum["Canceled"] = "CANCELED";
    BatchResponseMarketingEventPublicDefaultResponseStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseMarketingEventPublicDefaultResponseStatusEnum = exports.BatchResponseMarketingEventPublicDefaultResponseStatusEnum || (exports.BatchResponseMarketingEventPublicDefaultResponseStatusEnum = {}));
