// @hubspot/api-client/lib/codegen/marketing/events/models/BatchResponseSubscriberVidResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSubscriberVidResponseStatusEnum = exports.BatchResponseSubscriberVidResponse = void 0;
class BatchResponseSubscriberVidResponse {
    static getAttributeTypeMap() {
        return BatchResponseSubscriberVidResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseSubscriberVidResponse = BatchResponseSubscriberVidResponse;
BatchResponseSubscriberVidResponse.discriminator = undefined;
BatchResponseSubscriberVidResponse.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SubscriberVidResponse>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSubscriberVidResponseStatusEnum",
        "format": ""
    }
];
var BatchResponseSubscriberVidResponseStatusEnum;
(function (BatchResponseSubscriberVidResponseStatusEnum) {
    BatchResponseSubscriberVidResponseStatusEnum["Pending"] = "PENDING";
    BatchResponseSubscriberVidResponseStatusEnum["Processing"] = "PROCESSING";
    BatchResponseSubscriberVidResponseStatusEnum["Canceled"] = "CANCELED";
    BatchResponseSubscriberVidResponseStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseSubscriberVidResponseStatusEnum = exports.BatchResponseSubscriberVidResponseStatusEnum || (exports.BatchResponseSubscriberVidResponseStatusEnum = {}));
