// @hubspot/api-client/lib/codegen/marketing/events/models/ParticipationProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParticipationPropertiesAttendanceStateEnum = exports.ParticipationProperties = void 0;
class ParticipationProperties {
    static getAttributeTypeMap() {
        return ParticipationProperties.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ParticipationProperties = ParticipationProperties;
ParticipationProperties.discriminator = undefined;
ParticipationProperties.attributeTypeMap = [
    {
        "name": "occurredAt",
        "baseName": "occurredAt",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "attendancePercentage",
        "baseName": "attendancePercentage",
        "type": "string",
        "format": ""
    },
    {
        "name": "attendanceState",
        "baseName": "attendanceState",
        "type": "ParticipationPropertiesAttendanceStateEnum",
        "format": ""
    },
    {
        "name": "attendanceDurationSeconds",
        "baseName": "attendanceDurationSeconds",
        "type": "number",
        "format": "int32"
    }
];
var ParticipationPropertiesAttendanceStateEnum;
(function (ParticipationPropertiesAttendanceStateEnum) {
    ParticipationPropertiesAttendanceStateEnum["Registered"] = "REGISTERED";
    ParticipationPropertiesAttendanceStateEnum["Attended"] = "ATTENDED";
    ParticipationPropertiesAttendanceStateEnum["Cancelled"] = "CANCELLED";
    ParticipationPropertiesAttendanceStateEnum["Empty"] = "EMPTY";
    ParticipationPropertiesAttendanceStateEnum["NoShow"] = "NO_SHOW";
})(ParticipationPropertiesAttendanceStateEnum = exports.ParticipationPropertiesAttendanceStateEnum || (exports.ParticipationPropertiesAttendanceStateEnum = {}));
