// @hubspot/api-client/lib/codegen/marketing/events/models/PropertyValue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyValueDataSensitivityEnum = exports.PropertyValueSourceEnum = exports.PropertyValue = void 0;
class PropertyValue {
    static getAttributeTypeMap() {
        return PropertyValue.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PropertyValue = PropertyValue;
PropertyValue.discriminator = undefined;
PropertyValue.attributeTypeMap = [
    {
        "name": "sourceId",
        "baseName": "sourceId",
        "type": "string",
        "format": ""
    },
    {
        "name": "selectedByUser",
        "baseName": "selectedByUser",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "sourceLabel",
        "baseName": "sourceLabel",
        "type": "string",
        "format": ""
    },
    {
        "name": "source",
        "baseName": "source",
        "type": "PropertyValueSourceEnum",
        "format": ""
    },
    {
        "name": "updatedByUserId",
        "baseName": "updatedByUserId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "persistenceTimestamp",
        "baseName": "persistenceTimestamp",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "sourceMetadata",
        "baseName": "sourceMetadata",
        "type": "string",
        "format": ""
    },
    {
        "name": "dataSensitivity",
        "baseName": "dataSensitivity",
        "type": "PropertyValueDataSensitivityEnum",
        "format": ""
    },
    {
        "name": "sourceVid",
        "baseName": "sourceVid",
        "type": "Array<number>",
        "format": "int64"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "string",
        "format": ""
    },
    {
        "name": "requestId",
        "baseName": "requestId",
        "type": "string",
        "format": ""
    },
    {
        "name": "isEncrypted",
        "baseName": "isEncrypted",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "useTimestampAsPersistenceTimestamp",
        "baseName": "useTimestampAsPersistenceTimestamp",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string",
        "format": ""
    },
    {
        "name": "selectedByUserTimestamp",
        "baseName": "selectedByUserTimestamp",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "isLargeValue",
        "baseName": "isLargeValue",
        "type": "boolean",
        "format": ""
    }
];
var PropertyValueSourceEnum;
(function (PropertyValueSourceEnum) {
    PropertyValueSourceEnum["Unknown"] = "UNKNOWN";
    PropertyValueSourceEnum["Import"] = "IMPORT";
    PropertyValueSourceEnum["Api"] = "API";
    PropertyValueSourceEnum["Form"] = "FORM";
    PropertyValueSourceEnum["Analytics"] = "ANALYTICS";
    PropertyValueSourceEnum["Migration"] = "MIGRATION";
    PropertyValueSourceEnum["Salesforce"] = "SALESFORCE";
    PropertyValueSourceEnum["Integration"] = "INTEGRATION";
    PropertyValueSourceEnum["ContactsWeb"] = "CONTACTS_WEB";
    PropertyValueSourceEnum["WalIncremental"] = "WAL_INCREMENTAL";
    PropertyValueSourceEnum["Task"] = "TASK";
    PropertyValueSourceEnum["Email"] = "EMAIL";
    PropertyValueSourceEnum["Workflows"] = "WORKFLOWS";
    PropertyValueSourceEnum["Calculated"] = "CALCULATED";
    PropertyValueSourceEnum["Social"] = "SOCIAL";
    PropertyValueSourceEnum["BatchUpdate"] = "BATCH_UPDATE";
    PropertyValueSourceEnum["Signals"] = "SIGNALS";
    PropertyValueSourceEnum["Biden"] = "BIDEN";
    PropertyValueSourceEnum["Default"] = "DEFAULT";
    PropertyValueSourceEnum["Companies"] = "COMPANIES";
    PropertyValueSourceEnum["Deals"] = "DEALS";
    PropertyValueSourceEnum["Assists"] = "ASSISTS";
    PropertyValueSourceEnum["Presentations"] = "PRESENTATIONS";
    PropertyValueSourceEnum["Tally"] = "TALLY";
    PropertyValueSourceEnum["Sidekick"] = "SIDEKICK";
    PropertyValueSourceEnum["CrmUi"] = "CRM_UI";
    PropertyValueSourceEnum["MergeContacts"] = "MERGE_CONTACTS";
    PropertyValueSourceEnum["PortalUserAssociator"] = "PORTAL_USER_ASSOCIATOR";
    PropertyValueSourceEnum["IntegrationsPlatform"] = "INTEGRATIONS_PLATFORM";
    PropertyValueSourceEnum["BccToCrm"] = "BCC_TO_CRM";
    PropertyValueSourceEnum["ForwardToCrm"] = "FORWARD_TO_CRM";
    PropertyValueSourceEnum["Engagements"] = "ENGAGEMENTS";
    PropertyValueSourceEnum["Sales"] = "SALES";
    PropertyValueSourceEnum["Heisenberg"] = "HEISENBERG";
    PropertyValueSourceEnum["Leadin"] = "LEADIN";
    PropertyValueSourceEnum["GmailIntegration"] = "GMAIL_INTEGRATION";
    PropertyValueSourceEnum["Academy"] = "ACADEMY";
    PropertyValueSourceEnum["SalesMessages"] = "SALES_MESSAGES";
    PropertyValueSourceEnum["AvatarsService"] = "AVATARS_SERVICE";
    PropertyValueSourceEnum["MergeCompanies"] = "MERGE_COMPANIES";
    PropertyValueSourceEnum["Sequences"] = "SEQUENCES";
    PropertyValueSourceEnum["CompanyFamilies"] = "COMPANY_FAMILIES";
    PropertyValueSourceEnum["MobileIos"] = "MOBILE_IOS";
    PropertyValueSourceEnum["MobileAndroid"] = "MOBILE_ANDROID";
    PropertyValueSourceEnum["Contacts"] = "CONTACTS";
    PropertyValueSourceEnum["Associations"] = "ASSOCIATIONS";
    PropertyValueSourceEnum["Extension"] = "EXTENSION";
    PropertyValueSourceEnum["Success"] = "SUCCESS";
    PropertyValueSourceEnum["Bot"] = "BOT";
    PropertyValueSourceEnum["IntegrationsSync"] = "INTEGRATIONS_SYNC";
    PropertyValueSourceEnum["AutomationPlatform"] = "AUTOMATION_PLATFORM";
    PropertyValueSourceEnum["Conversations"] = "CONVERSATIONS";
    PropertyValueSourceEnum["EmailIntegration"] = "EMAIL_INTEGRATION";
    PropertyValueSourceEnum["ContentMembership"] = "CONTENT_MEMBERSHIP";
    PropertyValueSourceEnum["Quotes"] = "QUOTES";
    PropertyValueSourceEnum["BetAssignment"] = "BET_ASSIGNMENT";
    PropertyValueSourceEnum["Quotas"] = "QUOTAS";
    PropertyValueSourceEnum["BetCrmConnector"] = "BET_CRM_CONNECTOR";
    PropertyValueSourceEnum["Meetings"] = "MEETINGS";
    PropertyValueSourceEnum["MergeObjects"] = "MERGE_OBJECTS";
    PropertyValueSourceEnum["RecyclingBin"] = "RECYCLING_BIN";
    PropertyValueSourceEnum["Ads"] = "ADS";
    PropertyValueSourceEnum["AiGroup"] = "AI_GROUP";
    PropertyValueSourceEnum["Communicator"] = "COMMUNICATOR";
    PropertyValueSourceEnum["Settings"] = "SETTINGS";
    PropertyValueSourceEnum["PropertySettings"] = "PROPERTY_SETTINGS";
    PropertyValueSourceEnum["PipelineSettings"] = "PIPELINE_SETTINGS";
    PropertyValueSourceEnum["CompanyInsights"] = "COMPANY_INSIGHTS";
    PropertyValueSourceEnum["BehavioralEvents"] = "BEHAVIORAL_EVENTS";
    PropertyValueSourceEnum["Payments"] = "PAYMENTS";
    PropertyValueSourceEnum["Goals"] = "GOALS";
    PropertyValueSourceEnum["PortalObjectSync"] = "PORTAL_OBJECT_SYNC";
    PropertyValueSourceEnum["Approvals"] = "APPROVALS";
    PropertyValueSourceEnum["FileManager"] = "FILE_MANAGER";
    PropertyValueSourceEnum["Marketplace"] = "MARKETPLACE";
    PropertyValueSourceEnum["InternalProcessing"] = "INTERNAL_PROCESSING";
    PropertyValueSourceEnum["Forecasting"] = "FORECASTING";
    PropertyValueSourceEnum["SlackIntegration"] = "SLACK_INTEGRATION";
    PropertyValueSourceEnum["CrmUiBulkAction"] = "CRM_UI_BULK_ACTION";
    PropertyValueSourceEnum["WorkflowContactDeleteAction"] = "WORKFLOW_CONTACT_DELETE_ACTION";
    PropertyValueSourceEnum["AcceptanceTest"] = "ACCEPTANCE_TEST";
    PropertyValueSourceEnum["Playbooks"] = "PLAYBOOKS";
    PropertyValueSourceEnum["Chatspot"] = "CHATSPOT";
    PropertyValueSourceEnum["FlywheelProductDataSync"] = "FLYWHEEL_PRODUCT_DATA_SYNC";
    PropertyValueSourceEnum["HelpDesk"] = "HELP_DESK";
    PropertyValueSourceEnum["Billing"] = "BILLING";
    PropertyValueSourceEnum["DataEnrichment"] = "DATA_ENRICHMENT";
    PropertyValueSourceEnum["AutomationJourney"] = "AUTOMATION_JOURNEY";
    PropertyValueSourceEnum["Microapps"] = "MICROAPPS";
    PropertyValueSourceEnum["Intent"] = "INTENT";
    PropertyValueSourceEnum["ProspectingAgent"] = "PROSPECTING_AGENT";
})(PropertyValueSourceEnum = exports.PropertyValueSourceEnum || (exports.PropertyValueSourceEnum = {}));
var PropertyValueDataSensitivityEnum;
(function (PropertyValueDataSensitivityEnum) {
    PropertyValueDataSensitivityEnum["None"] = "none";
    PropertyValueDataSensitivityEnum["Standard"] = "standard";
    PropertyValueDataSensitivityEnum["High"] = "high";
})(PropertyValueDataSensitivityEnum = exports.PropertyValueDataSensitivityEnum || (exports.PropertyValueDataSensitivityEnum = {}));
