// @hubspot/api-client/lib/codegen/cms/audit_logs/models/PublicAuditLog.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAuditLogObjectTypeEnum = exports.PublicAuditLogEventEnum = exports.PublicAuditLog = void 0;
class PublicAuditLog {
    static getAttributeTypeMap() {
        return PublicAuditLog.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicAuditLog = PublicAuditLog;
PublicAuditLog.discriminator = undefined;
PublicAuditLog.attributeTypeMap = [
    {
        "name": "meta",
        "baseName": "meta",
        "type": "any",
        "format": ""
    },
    {
        "name": "objectName",
        "baseName": "objectName",
        "type": "string",
        "format": ""
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string",
        "format": ""
    },
    {
        "name": "event",
        "baseName": "event",
        "type": "PublicAuditLogEventEnum",
        "format": ""
    },
    {
        "name": "userId",
        "baseName": "userId",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectId",
        "baseName": "objectId",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "PublicAuditLogObjectTypeEnum",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    }
];
var PublicAuditLogEventEnum;
(function (PublicAuditLogEventEnum) {
    PublicAuditLogEventEnum["Created"] = "CREATED";
    PublicAuditLogEventEnum["Updated"] = "UPDATED";
    PublicAuditLogEventEnum["Published"] = "PUBLISHED";
    PublicAuditLogEventEnum["Deleted"] = "DELETED";
    PublicAuditLogEventEnum["Unpublished"] = "UNPUBLISHED";
    PublicAuditLogEventEnum["Restore"] = "RESTORE";
})(PublicAuditLogEventEnum = exports.PublicAuditLogEventEnum || (exports.PublicAuditLogEventEnum = {}));
var PublicAuditLogObjectTypeEnum;
(function (PublicAuditLogObjectTypeEnum) {
    PublicAuditLogObjectTypeEnum["Blog"] = "BLOG";
    PublicAuditLogObjectTypeEnum["BlogPost"] = "BLOG_POST";
    PublicAuditLogObjectTypeEnum["LandingPage"] = "LANDING_PAGE";
    PublicAuditLogObjectTypeEnum["WebsitePage"] = "WEBSITE_PAGE";
    PublicAuditLogObjectTypeEnum["Template"] = "TEMPLATE";
    PublicAuditLogObjectTypeEnum["Module"] = "MODULE";
    PublicAuditLogObjectTypeEnum["GlobalModule"] = "GLOBAL_MODULE";
    PublicAuditLogObjectTypeEnum["ServerlessFunction"] = "SERVERLESS_FUNCTION";
    PublicAuditLogObjectTypeEnum["Domain"] = "DOMAIN";
    PublicAuditLogObjectTypeEnum["UrlMapping"] = "URL_MAPPING";
    PublicAuditLogObjectTypeEnum["Email"] = "EMAIL";
    PublicAuditLogObjectTypeEnum["ContentSettings"] = "CONTENT_SETTINGS";
    PublicAuditLogObjectTypeEnum["HubdbTable"] = "HUBDB_TABLE";
    PublicAuditLogObjectTypeEnum["KnowledgeBaseArticle"] = "KNOWLEDGE_BASE_ARTICLE";
    PublicAuditLogObjectTypeEnum["KnowledgeBase"] = "KNOWLEDGE_BASE";
    PublicAuditLogObjectTypeEnum["Theme"] = "THEME";
    PublicAuditLogObjectTypeEnum["Css"] = "CSS";
    PublicAuditLogObjectTypeEnum["Js"] = "JS";
    PublicAuditLogObjectTypeEnum["Cta"] = "CTA";
    PublicAuditLogObjectTypeEnum["File"] = "FILE";
})(PublicAuditLogObjectTypeEnum = exports.PublicAuditLogObjectTypeEnum || (exports.PublicAuditLogObjectTypeEnum = {}));
