// @hubspot/api-client/lib/codegen/cms/blogs/authors/models/BatchResponseBlogAuthorWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseBlogAuthorWithErrorsStatusEnum = exports.BatchResponseBlogAuthorWithErrors = void 0;
class BatchResponseBlogAuthorWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseBlogAuthorWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseBlogAuthorWithErrors = BatchResponseBlogAuthorWithErrors;
BatchResponseBlogAuthorWithErrors.discriminator = undefined;
BatchResponseBlogAuthorWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<BlogAuthor>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseBlogAuthorWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseBlogAuthorWithErrorsStatusEnum;
(function (BatchResponseBlogAuthorWithErrorsStatusEnum) {
    BatchResponseBlogAuthorWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseBlogAuthorWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseBlogAuthorWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseBlogAuthorWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseBlogAuthorWithErrorsStatusEnum = exports.BatchResponseBlogAuthorWithErrorsStatusEnum || (exports.BatchResponseBlogAuthorWithErrorsStatusEnum = {}));
