// @hubspot/api-client/lib/codegen/cms/blogs/authors/models/BatchResponseBlogAuthor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseBlogAuthorStatusEnum = exports.BatchResponseBlogAuthor = void 0;
class BatchResponseBlogAuthor {
    static getAttributeTypeMap() {
        return BatchResponseBlogAuthor.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseBlogAuthor = BatchResponseBlogAuthor;
BatchResponseBlogAuthor.discriminator = undefined;
BatchResponseBlogAuthor.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<BlogAuthor>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseBlogAuthorStatusEnum",
        "format": ""
    }
];
var BatchResponseBlogAuthorStatusEnum;
(function (BatchResponseBlogAuthorStatusEnum) {
    BatchResponseBlogAuthorStatusEnum["Pending"] = "PENDING";
    BatchResponseBlogAuthorStatusEnum["Processing"] = "PROCESSING";
    BatchResponseBlogAuthorStatusEnum["Canceled"] = "CANCELED";
    BatchResponseBlogAuthorStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseBlogAuthorStatusEnum = exports.BatchResponseBlogAuthorStatusEnum || (exports.BatchResponseBlogAuthorStatusEnum = {}));
