// @hubspot/api-client/lib/codegen/cms/blogs/authors/models/BlogAuthor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlogAuthorLanguageEnum = exports.BlogAuthor = void 0;
class BlogAuthor {
    static getAttributeTypeMap() {
        return BlogAuthor.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BlogAuthor = BlogAuthor;
BlogAuthor.discriminator = undefined;
BlogAuthor.attributeTypeMap = [
    {
        "name": "website",
        "baseName": "website",
        "type": "string",
        "format": ""
    },
    {
        "name": "displayName",
        "baseName": "displayName",
        "type": "string",
        "format": ""
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "facebook",
        "baseName": "facebook",
        "type": "string",
        "format": ""
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string",
        "format": ""
    },
    {
        "name": "bio",
        "baseName": "bio",
        "type": "string",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "BlogAuthorLanguageEnum",
        "format": ""
    },
    {
        "name": "linkedin",
        "baseName": "linkedin",
        "type": "string",
        "format": ""
    },
    {
        "name": "avatar",
        "baseName": "avatar",
        "type": "string",
        "format": ""
    },
    {
        "name": "translatedFromId",
        "baseName": "translatedFromId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "twitter",
        "baseName": "twitter",
        "type": "string",
        "format": ""
    },
    {
        "name": "deletedAt",
        "baseName": "deletedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "email",
        "baseName": "email",
        "type": "string",
        "format": ""
    },
    {
        "name": "slug",
        "baseName": "slug",
        "type": "string",
        "format": ""
    }
];
var BlogAuthorLanguageEnum;
(function (BlogAuthorLanguageEnum) {
    BlogAuthorLanguageEnum["Af"] = "af";
    BlogAuthorLanguageEnum["AfNa"] = "af-na";
    BlogAuthorLanguageEnum["AfZa"] = "af-za";
    BlogAuthorLanguageEnum["Agq"] = "agq";
    BlogAuthorLanguageEnum["AgqCm"] = "agq-cm";
    BlogAuthorLanguageEnum["Ak"] = "ak";
    BlogAuthorLanguageEnum["AkGh"] = "ak-gh";
    BlogAuthorLanguageEnum["Am"] = "am";
    BlogAuthorLanguageEnum["AmEt"] = "am-et";
    BlogAuthorLanguageEnum["Ar"] = "ar";
    BlogAuthorLanguageEnum["Ar001"] = "ar-001";
    BlogAuthorLanguageEnum["ArAe"] = "ar-ae";
    BlogAuthorLanguageEnum["ArBh"] = "ar-bh";
    BlogAuthorLanguageEnum["ArDj"] = "ar-dj";
    BlogAuthorLanguageEnum["ArDz"] = "ar-dz";
    BlogAuthorLanguageEnum["ArEg"] = "ar-eg";
    BlogAuthorLanguageEnum["ArEh"] = "ar-eh";
    BlogAuthorLanguageEnum["ArEr"] = "ar-er";
    BlogAuthorLanguageEnum["ArIl"] = "ar-il";
    BlogAuthorLanguageEnum["ArIq"] = "ar-iq";
    BlogAuthorLanguageEnum["ArJo"] = "ar-jo";
    BlogAuthorLanguageEnum["ArKm"] = "ar-km";
    BlogAuthorLanguageEnum["ArKw"] = "ar-kw";
    BlogAuthorLanguageEnum["ArLb"] = "ar-lb";
    BlogAuthorLanguageEnum["ArLy"] = "ar-ly";
    BlogAuthorLanguageEnum["ArMa"] = "ar-ma";
    BlogAuthorLanguageEnum["ArMr"] = "ar-mr";
    BlogAuthorLanguageEnum["ArOm"] = "ar-om";
    BlogAuthorLanguageEnum["ArPs"] = "ar-ps";
    BlogAuthorLanguageEnum["ArQa"] = "ar-qa";
    BlogAuthorLanguageEnum["ArSa"] = "ar-sa";
    BlogAuthorLanguageEnum["ArSd"] = "ar-sd";
    BlogAuthorLanguageEnum["ArSo"] = "ar-so";
    BlogAuthorLanguageEnum["ArSs"] = "ar-ss";
    BlogAuthorLanguageEnum["ArSy"] = "ar-sy";
    BlogAuthorLanguageEnum["ArTd"] = "ar-td";
    BlogAuthorLanguageEnum["ArTn"] = "ar-tn";
    BlogAuthorLanguageEnum["ArYe"] = "ar-ye";
    BlogAuthorLanguageEnum["As"] = "as";
    BlogAuthorLanguageEnum["AsIn"] = "as-in";
    BlogAuthorLanguageEnum["Asa"] = "asa";
    BlogAuthorLanguageEnum["AsaTz"] = "asa-tz";
    BlogAuthorLanguageEnum["Ast"] = "ast";
    BlogAuthorLanguageEnum["AstEs"] = "ast-es";
    BlogAuthorLanguageEnum["Az"] = "az";
    BlogAuthorLanguageEnum["AzAz"] = "az-az";
    BlogAuthorLanguageEnum["Bas"] = "bas";
    BlogAuthorLanguageEnum["BasCm"] = "bas-cm";
    BlogAuthorLanguageEnum["Be"] = "be";
    BlogAuthorLanguageEnum["BeBy"] = "be-by";
    BlogAuthorLanguageEnum["Bem"] = "bem";
    BlogAuthorLanguageEnum["BemZm"] = "bem-zm";
    BlogAuthorLanguageEnum["Bez"] = "bez";
    BlogAuthorLanguageEnum["BezTz"] = "bez-tz";
    BlogAuthorLanguageEnum["Bg"] = "bg";
    BlogAuthorLanguageEnum["BgBg"] = "bg-bg";
    BlogAuthorLanguageEnum["Bm"] = "bm";
    BlogAuthorLanguageEnum["BmMl"] = "bm-ml";
    BlogAuthorLanguageEnum["Bn"] = "bn";
    BlogAuthorLanguageEnum["BnBd"] = "bn-bd";
    BlogAuthorLanguageEnum["BnIn"] = "bn-in";
    BlogAuthorLanguageEnum["Bo"] = "bo";
    BlogAuthorLanguageEnum["BoCn"] = "bo-cn";
    BlogAuthorLanguageEnum["BoIn"] = "bo-in";
    BlogAuthorLanguageEnum["Br"] = "br";
    BlogAuthorLanguageEnum["BrFr"] = "br-fr";
    BlogAuthorLanguageEnum["Brx"] = "brx";
    BlogAuthorLanguageEnum["BrxIn"] = "brx-in";
    BlogAuthorLanguageEnum["Bs"] = "bs";
    BlogAuthorLanguageEnum["BsBa"] = "bs-ba";
    BlogAuthorLanguageEnum["Ca"] = "ca";
    BlogAuthorLanguageEnum["CaAd"] = "ca-ad";
    BlogAuthorLanguageEnum["CaEs"] = "ca-es";
    BlogAuthorLanguageEnum["CaFr"] = "ca-fr";
    BlogAuthorLanguageEnum["CaIt"] = "ca-it";
    BlogAuthorLanguageEnum["Ccp"] = "ccp";
    BlogAuthorLanguageEnum["CcpBd"] = "ccp-bd";
    BlogAuthorLanguageEnum["CcpIn"] = "ccp-in";
    BlogAuthorLanguageEnum["Ce"] = "ce";
    BlogAuthorLanguageEnum["CeRu"] = "ce-ru";
    BlogAuthorLanguageEnum["Ceb"] = "ceb";
    BlogAuthorLanguageEnum["CebPh"] = "ceb-ph";
    BlogAuthorLanguageEnum["Cgg"] = "cgg";
    BlogAuthorLanguageEnum["CggUg"] = "cgg-ug";
    BlogAuthorLanguageEnum["Chr"] = "chr";
    BlogAuthorLanguageEnum["ChrUs"] = "chr-us";
    BlogAuthorLanguageEnum["Ckb"] = "ckb";
    BlogAuthorLanguageEnum["CkbIq"] = "ckb-iq";
    BlogAuthorLanguageEnum["CkbIr"] = "ckb-ir";
    BlogAuthorLanguageEnum["Cs"] = "cs";
    BlogAuthorLanguageEnum["CsCz"] = "cs-cz";
    BlogAuthorLanguageEnum["Cu"] = "cu";
    BlogAuthorLanguageEnum["CuRu"] = "cu-ru";
    BlogAuthorLanguageEnum["Cy"] = "cy";
    BlogAuthorLanguageEnum["CyGb"] = "cy-gb";
    BlogAuthorLanguageEnum["Da"] = "da";
    BlogAuthorLanguageEnum["DaDk"] = "da-dk";
    BlogAuthorLanguageEnum["DaGl"] = "da-gl";
    BlogAuthorLanguageEnum["Dav"] = "dav";
    BlogAuthorLanguageEnum["DavKe"] = "dav-ke";
    BlogAuthorLanguageEnum["De"] = "de";
    BlogAuthorLanguageEnum["DeAt"] = "de-at";
    BlogAuthorLanguageEnum["DeBe"] = "de-be";
    BlogAuthorLanguageEnum["DeCh"] = "de-ch";
    BlogAuthorLanguageEnum["DeDe"] = "de-de";
    BlogAuthorLanguageEnum["DeGr"] = "de-gr";
    BlogAuthorLanguageEnum["DeIt"] = "de-it";
    BlogAuthorLanguageEnum["DeLi"] = "de-li";
    BlogAuthorLanguageEnum["DeLu"] = "de-lu";
    BlogAuthorLanguageEnum["Dje"] = "dje";
    BlogAuthorLanguageEnum["DjeNe"] = "dje-ne";
    BlogAuthorLanguageEnum["Doi"] = "doi";
    BlogAuthorLanguageEnum["DoiIn"] = "doi-in";
    BlogAuthorLanguageEnum["Dsb"] = "dsb";
    BlogAuthorLanguageEnum["DsbDe"] = "dsb-de";
    BlogAuthorLanguageEnum["Dua"] = "dua";
    BlogAuthorLanguageEnum["DuaCm"] = "dua-cm";
    BlogAuthorLanguageEnum["Dyo"] = "dyo";
    BlogAuthorLanguageEnum["DyoSn"] = "dyo-sn";
    BlogAuthorLanguageEnum["Dz"] = "dz";
    BlogAuthorLanguageEnum["DzBt"] = "dz-bt";
    BlogAuthorLanguageEnum["Ebu"] = "ebu";
    BlogAuthorLanguageEnum["EbuKe"] = "ebu-ke";
    BlogAuthorLanguageEnum["Ee"] = "ee";
    BlogAuthorLanguageEnum["EeGh"] = "ee-gh";
    BlogAuthorLanguageEnum["EeTg"] = "ee-tg";
    BlogAuthorLanguageEnum["El"] = "el";
    BlogAuthorLanguageEnum["ElCy"] = "el-cy";
    BlogAuthorLanguageEnum["ElGr"] = "el-gr";
    BlogAuthorLanguageEnum["En"] = "en";
    BlogAuthorLanguageEnum["En001"] = "en-001";
    BlogAuthorLanguageEnum["En150"] = "en-150";
    BlogAuthorLanguageEnum["EnAe"] = "en-ae";
    BlogAuthorLanguageEnum["EnAg"] = "en-ag";
    BlogAuthorLanguageEnum["EnAi"] = "en-ai";
    BlogAuthorLanguageEnum["EnAs"] = "en-as";
    BlogAuthorLanguageEnum["EnAt"] = "en-at";
    BlogAuthorLanguageEnum["EnAu"] = "en-au";
    BlogAuthorLanguageEnum["EnBb"] = "en-bb";
    BlogAuthorLanguageEnum["EnBe"] = "en-be";
    BlogAuthorLanguageEnum["EnBi"] = "en-bi";
    BlogAuthorLanguageEnum["EnBm"] = "en-bm";
    BlogAuthorLanguageEnum["EnBs"] = "en-bs";
    BlogAuthorLanguageEnum["EnBw"] = "en-bw";
    BlogAuthorLanguageEnum["EnBz"] = "en-bz";
    BlogAuthorLanguageEnum["EnCa"] = "en-ca";
    BlogAuthorLanguageEnum["EnCc"] = "en-cc";
    BlogAuthorLanguageEnum["EnCh"] = "en-ch";
    BlogAuthorLanguageEnum["EnCk"] = "en-ck";
    BlogAuthorLanguageEnum["EnCm"] = "en-cm";
    BlogAuthorLanguageEnum["EnCn"] = "en-cn";
    BlogAuthorLanguageEnum["EnCx"] = "en-cx";
    BlogAuthorLanguageEnum["EnCy"] = "en-cy";
    BlogAuthorLanguageEnum["EnDe"] = "en-de";
    BlogAuthorLanguageEnum["EnDg"] = "en-dg";
    BlogAuthorLanguageEnum["EnDk"] = "en-dk";
    BlogAuthorLanguageEnum["EnDm"] = "en-dm";
    BlogAuthorLanguageEnum["EnEr"] = "en-er";
    BlogAuthorLanguageEnum["EnFi"] = "en-fi";
    BlogAuthorLanguageEnum["EnFj"] = "en-fj";
    BlogAuthorLanguageEnum["EnFk"] = "en-fk";
    BlogAuthorLanguageEnum["EnFm"] = "en-fm";
    BlogAuthorLanguageEnum["EnGb"] = "en-gb";
    BlogAuthorLanguageEnum["EnGd"] = "en-gd";
    BlogAuthorLanguageEnum["EnGg"] = "en-gg";
    BlogAuthorLanguageEnum["EnGh"] = "en-gh";
    BlogAuthorLanguageEnum["EnGi"] = "en-gi";
    BlogAuthorLanguageEnum["EnGm"] = "en-gm";
    BlogAuthorLanguageEnum["EnGu"] = "en-gu";
    BlogAuthorLanguageEnum["EnGy"] = "en-gy";
    BlogAuthorLanguageEnum["EnHk"] = "en-hk";
    BlogAuthorLanguageEnum["EnIe"] = "en-ie";
    BlogAuthorLanguageEnum["EnIl"] = "en-il";
    BlogAuthorLanguageEnum["EnIm"] = "en-im";
    BlogAuthorLanguageEnum["EnIn"] = "en-in";
    BlogAuthorLanguageEnum["EnIo"] = "en-io";
    BlogAuthorLanguageEnum["EnJe"] = "en-je";
    BlogAuthorLanguageEnum["EnJm"] = "en-jm";
    BlogAuthorLanguageEnum["EnKe"] = "en-ke";
    BlogAuthorLanguageEnum["EnKi"] = "en-ki";
    BlogAuthorLanguageEnum["EnKn"] = "en-kn";
    BlogAuthorLanguageEnum["EnKy"] = "en-ky";
    BlogAuthorLanguageEnum["EnLc"] = "en-lc";
    BlogAuthorLanguageEnum["EnLr"] = "en-lr";
    BlogAuthorLanguageEnum["EnLs"] = "en-ls";
    BlogAuthorLanguageEnum["EnLu"] = "en-lu";
    BlogAuthorLanguageEnum["EnMg"] = "en-mg";
    BlogAuthorLanguageEnum["EnMh"] = "en-mh";
    BlogAuthorLanguageEnum["EnMo"] = "en-mo";
    BlogAuthorLanguageEnum["EnMp"] = "en-mp";
    BlogAuthorLanguageEnum["EnMs"] = "en-ms";
    BlogAuthorLanguageEnum["EnMt"] = "en-mt";
    BlogAuthorLanguageEnum["EnMu"] = "en-mu";
    BlogAuthorLanguageEnum["EnMw"] = "en-mw";
    BlogAuthorLanguageEnum["EnMx"] = "en-mx";
    BlogAuthorLanguageEnum["EnMy"] = "en-my";
    BlogAuthorLanguageEnum["EnNa"] = "en-na";
    BlogAuthorLanguageEnum["EnNf"] = "en-nf";
    BlogAuthorLanguageEnum["EnNg"] = "en-ng";
    BlogAuthorLanguageEnum["EnNl"] = "en-nl";
    BlogAuthorLanguageEnum["EnNr"] = "en-nr";
    BlogAuthorLanguageEnum["EnNu"] = "en-nu";
    BlogAuthorLanguageEnum["EnNz"] = "en-nz";
    BlogAuthorLanguageEnum["EnPg"] = "en-pg";
    BlogAuthorLanguageEnum["EnPh"] = "en-ph";
    BlogAuthorLanguageEnum["EnPk"] = "en-pk";
    BlogAuthorLanguageEnum["EnPn"] = "en-pn";
    BlogAuthorLanguageEnum["EnPr"] = "en-pr";
    BlogAuthorLanguageEnum["EnPw"] = "en-pw";
    BlogAuthorLanguageEnum["EnRw"] = "en-rw";
    BlogAuthorLanguageEnum["EnSb"] = "en-sb";
    BlogAuthorLanguageEnum["EnSc"] = "en-sc";
    BlogAuthorLanguageEnum["EnSd"] = "en-sd";
    BlogAuthorLanguageEnum["EnSe"] = "en-se";
    BlogAuthorLanguageEnum["EnSg"] = "en-sg";
    BlogAuthorLanguageEnum["EnSh"] = "en-sh";
    BlogAuthorLanguageEnum["EnSi"] = "en-si";
    BlogAuthorLanguageEnum["EnSl"] = "en-sl";
    BlogAuthorLanguageEnum["EnSs"] = "en-ss";
    BlogAuthorLanguageEnum["EnSx"] = "en-sx";
    BlogAuthorLanguageEnum["EnSz"] = "en-sz";
    BlogAuthorLanguageEnum["EnTc"] = "en-tc";
    BlogAuthorLanguageEnum["EnTk"] = "en-tk";
    BlogAuthorLanguageEnum["EnTo"] = "en-to";
    BlogAuthorLanguageEnum["EnTt"] = "en-tt";
    BlogAuthorLanguageEnum["EnTv"] = "en-tv";
    BlogAuthorLanguageEnum["EnTz"] = "en-tz";
    BlogAuthorLanguageEnum["EnUg"] = "en-ug";
    BlogAuthorLanguageEnum["EnUm"] = "en-um";
    BlogAuthorLanguageEnum["EnUs"] = "en-us";
    BlogAuthorLanguageEnum["EnVc"] = "en-vc";
    BlogAuthorLanguageEnum["EnVg"] = "en-vg";
    BlogAuthorLanguageEnum["EnVi"] = "en-vi";
    BlogAuthorLanguageEnum["EnVu"] = "en-vu";
    BlogAuthorLanguageEnum["EnWs"] = "en-ws";
    BlogAuthorLanguageEnum["EnZa"] = "en-za";
    BlogAuthorLanguageEnum["EnZm"] = "en-zm";
    BlogAuthorLanguageEnum["EnZw"] = "en-zw";
    BlogAuthorLanguageEnum["Eo"] = "eo";
    BlogAuthorLanguageEnum["Eo001"] = "eo-001";
    BlogAuthorLanguageEnum["Es"] = "es";
    BlogAuthorLanguageEnum["Es419"] = "es-419";
    BlogAuthorLanguageEnum["EsAr"] = "es-ar";
    BlogAuthorLanguageEnum["EsBo"] = "es-bo";
    BlogAuthorLanguageEnum["EsBr"] = "es-br";
    BlogAuthorLanguageEnum["EsBz"] = "es-bz";
    BlogAuthorLanguageEnum["EsCl"] = "es-cl";
    BlogAuthorLanguageEnum["EsCo"] = "es-co";
    BlogAuthorLanguageEnum["EsCr"] = "es-cr";
    BlogAuthorLanguageEnum["EsCu"] = "es-cu";
    BlogAuthorLanguageEnum["EsDo"] = "es-do";
    BlogAuthorLanguageEnum["EsEa"] = "es-ea";
    BlogAuthorLanguageEnum["EsEc"] = "es-ec";
    BlogAuthorLanguageEnum["EsEs"] = "es-es";
    BlogAuthorLanguageEnum["EsGq"] = "es-gq";
    BlogAuthorLanguageEnum["EsGt"] = "es-gt";
    BlogAuthorLanguageEnum["EsHn"] = "es-hn";
    BlogAuthorLanguageEnum["EsIc"] = "es-ic";
    BlogAuthorLanguageEnum["EsMx"] = "es-mx";
    BlogAuthorLanguageEnum["EsNi"] = "es-ni";
    BlogAuthorLanguageEnum["EsPa"] = "es-pa";
    BlogAuthorLanguageEnum["EsPe"] = "es-pe";
    BlogAuthorLanguageEnum["EsPh"] = "es-ph";
    BlogAuthorLanguageEnum["EsPr"] = "es-pr";
    BlogAuthorLanguageEnum["EsPy"] = "es-py";
    BlogAuthorLanguageEnum["EsSv"] = "es-sv";
    BlogAuthorLanguageEnum["EsUs"] = "es-us";
    BlogAuthorLanguageEnum["EsUy"] = "es-uy";
    BlogAuthorLanguageEnum["EsVe"] = "es-ve";
    BlogAuthorLanguageEnum["Et"] = "et";
    BlogAuthorLanguageEnum["EtEe"] = "et-ee";
    BlogAuthorLanguageEnum["Eu"] = "eu";
    BlogAuthorLanguageEnum["EuEs"] = "eu-es";
    BlogAuthorLanguageEnum["Ewo"] = "ewo";
    BlogAuthorLanguageEnum["EwoCm"] = "ewo-cm";
    BlogAuthorLanguageEnum["Fa"] = "fa";
    BlogAuthorLanguageEnum["FaAf"] = "fa-af";
    BlogAuthorLanguageEnum["FaIr"] = "fa-ir";
    BlogAuthorLanguageEnum["Ff"] = "ff";
    BlogAuthorLanguageEnum["FfBf"] = "ff-bf";
    BlogAuthorLanguageEnum["FfCm"] = "ff-cm";
    BlogAuthorLanguageEnum["FfGh"] = "ff-gh";
    BlogAuthorLanguageEnum["FfGm"] = "ff-gm";
    BlogAuthorLanguageEnum["FfGn"] = "ff-gn";
    BlogAuthorLanguageEnum["FfGw"] = "ff-gw";
    BlogAuthorLanguageEnum["FfLr"] = "ff-lr";
    BlogAuthorLanguageEnum["FfMr"] = "ff-mr";
    BlogAuthorLanguageEnum["FfNe"] = "ff-ne";
    BlogAuthorLanguageEnum["FfNg"] = "ff-ng";
    BlogAuthorLanguageEnum["FfSl"] = "ff-sl";
    BlogAuthorLanguageEnum["FfSn"] = "ff-sn";
    BlogAuthorLanguageEnum["Fi"] = "fi";
    BlogAuthorLanguageEnum["FiFi"] = "fi-fi";
    BlogAuthorLanguageEnum["Fil"] = "fil";
    BlogAuthorLanguageEnum["FilPh"] = "fil-ph";
    BlogAuthorLanguageEnum["Fo"] = "fo";
    BlogAuthorLanguageEnum["FoDk"] = "fo-dk";
    BlogAuthorLanguageEnum["FoFo"] = "fo-fo";
    BlogAuthorLanguageEnum["Fr"] = "fr";
    BlogAuthorLanguageEnum["FrBe"] = "fr-be";
    BlogAuthorLanguageEnum["FrBf"] = "fr-bf";
    BlogAuthorLanguageEnum["FrBi"] = "fr-bi";
    BlogAuthorLanguageEnum["FrBj"] = "fr-bj";
    BlogAuthorLanguageEnum["FrBl"] = "fr-bl";
    BlogAuthorLanguageEnum["FrCa"] = "fr-ca";
    BlogAuthorLanguageEnum["FrCd"] = "fr-cd";
    BlogAuthorLanguageEnum["FrCf"] = "fr-cf";
    BlogAuthorLanguageEnum["FrCg"] = "fr-cg";
    BlogAuthorLanguageEnum["FrCh"] = "fr-ch";
    BlogAuthorLanguageEnum["FrCi"] = "fr-ci";
    BlogAuthorLanguageEnum["FrCm"] = "fr-cm";
    BlogAuthorLanguageEnum["FrDj"] = "fr-dj";
    BlogAuthorLanguageEnum["FrDz"] = "fr-dz";
    BlogAuthorLanguageEnum["FrFr"] = "fr-fr";
    BlogAuthorLanguageEnum["FrGa"] = "fr-ga";
    BlogAuthorLanguageEnum["FrGf"] = "fr-gf";
    BlogAuthorLanguageEnum["FrGn"] = "fr-gn";
    BlogAuthorLanguageEnum["FrGp"] = "fr-gp";
    BlogAuthorLanguageEnum["FrGq"] = "fr-gq";
    BlogAuthorLanguageEnum["FrHt"] = "fr-ht";
    BlogAuthorLanguageEnum["FrKm"] = "fr-km";
    BlogAuthorLanguageEnum["FrLu"] = "fr-lu";
    BlogAuthorLanguageEnum["FrMa"] = "fr-ma";
    BlogAuthorLanguageEnum["FrMc"] = "fr-mc";
    BlogAuthorLanguageEnum["FrMf"] = "fr-mf";
    BlogAuthorLanguageEnum["FrMg"] = "fr-mg";
    BlogAuthorLanguageEnum["FrMl"] = "fr-ml";
    BlogAuthorLanguageEnum["FrMq"] = "fr-mq";
    BlogAuthorLanguageEnum["FrMr"] = "fr-mr";
    BlogAuthorLanguageEnum["FrMu"] = "fr-mu";
    BlogAuthorLanguageEnum["FrNc"] = "fr-nc";
    BlogAuthorLanguageEnum["FrNe"] = "fr-ne";
    BlogAuthorLanguageEnum["FrPf"] = "fr-pf";
    BlogAuthorLanguageEnum["FrPm"] = "fr-pm";
    BlogAuthorLanguageEnum["FrRe"] = "fr-re";
    BlogAuthorLanguageEnum["FrRw"] = "fr-rw";
    BlogAuthorLanguageEnum["FrSc"] = "fr-sc";
    BlogAuthorLanguageEnum["FrSn"] = "fr-sn";
    BlogAuthorLanguageEnum["FrSy"] = "fr-sy";
    BlogAuthorLanguageEnum["FrTd"] = "fr-td";
    BlogAuthorLanguageEnum["FrTg"] = "fr-tg";
    BlogAuthorLanguageEnum["FrTn"] = "fr-tn";
    BlogAuthorLanguageEnum["FrVu"] = "fr-vu";
    BlogAuthorLanguageEnum["FrWf"] = "fr-wf";
    BlogAuthorLanguageEnum["FrYt"] = "fr-yt";
    BlogAuthorLanguageEnum["Fur"] = "fur";
    BlogAuthorLanguageEnum["FurIt"] = "fur-it";
    BlogAuthorLanguageEnum["Fy"] = "fy";
    BlogAuthorLanguageEnum["FyNl"] = "fy-nl";
    BlogAuthorLanguageEnum["Ga"] = "ga";
    BlogAuthorLanguageEnum["GaGb"] = "ga-gb";
    BlogAuthorLanguageEnum["GaIe"] = "ga-ie";
    BlogAuthorLanguageEnum["Gd"] = "gd";
    BlogAuthorLanguageEnum["GdGb"] = "gd-gb";
    BlogAuthorLanguageEnum["Gl"] = "gl";
    BlogAuthorLanguageEnum["GlEs"] = "gl-es";
    BlogAuthorLanguageEnum["Gsw"] = "gsw";
    BlogAuthorLanguageEnum["GswCh"] = "gsw-ch";
    BlogAuthorLanguageEnum["GswFr"] = "gsw-fr";
    BlogAuthorLanguageEnum["GswLi"] = "gsw-li";
    BlogAuthorLanguageEnum["Gu"] = "gu";
    BlogAuthorLanguageEnum["GuIn"] = "gu-in";
    BlogAuthorLanguageEnum["Guz"] = "guz";
    BlogAuthorLanguageEnum["GuzKe"] = "guz-ke";
    BlogAuthorLanguageEnum["Gv"] = "gv";
    BlogAuthorLanguageEnum["GvIm"] = "gv-im";
    BlogAuthorLanguageEnum["Ha"] = "ha";
    BlogAuthorLanguageEnum["HaGh"] = "ha-gh";
    BlogAuthorLanguageEnum["HaNe"] = "ha-ne";
    BlogAuthorLanguageEnum["HaNg"] = "ha-ng";
    BlogAuthorLanguageEnum["Haw"] = "haw";
    BlogAuthorLanguageEnum["HawUs"] = "haw-us";
    BlogAuthorLanguageEnum["He"] = "he";
    BlogAuthorLanguageEnum["Hi"] = "hi";
    BlogAuthorLanguageEnum["HiIn"] = "hi-in";
    BlogAuthorLanguageEnum["Hr"] = "hr";
    BlogAuthorLanguageEnum["HrBa"] = "hr-ba";
    BlogAuthorLanguageEnum["HrHr"] = "hr-hr";
    BlogAuthorLanguageEnum["Hsb"] = "hsb";
    BlogAuthorLanguageEnum["HsbDe"] = "hsb-de";
    BlogAuthorLanguageEnum["Hu"] = "hu";
    BlogAuthorLanguageEnum["HuHu"] = "hu-hu";
    BlogAuthorLanguageEnum["Hy"] = "hy";
    BlogAuthorLanguageEnum["HyAm"] = "hy-am";
    BlogAuthorLanguageEnum["Ia"] = "ia";
    BlogAuthorLanguageEnum["Ia001"] = "ia-001";
    BlogAuthorLanguageEnum["Id"] = "id";
    BlogAuthorLanguageEnum["Ig"] = "ig";
    BlogAuthorLanguageEnum["IgNg"] = "ig-ng";
    BlogAuthorLanguageEnum["Ii"] = "ii";
    BlogAuthorLanguageEnum["IiCn"] = "ii-cn";
    BlogAuthorLanguageEnum["IdId"] = "id-id";
    BlogAuthorLanguageEnum["Is"] = "is";
    BlogAuthorLanguageEnum["IsIs"] = "is-is";
    BlogAuthorLanguageEnum["It"] = "it";
    BlogAuthorLanguageEnum["ItCh"] = "it-ch";
    BlogAuthorLanguageEnum["ItIt"] = "it-it";
    BlogAuthorLanguageEnum["ItSm"] = "it-sm";
    BlogAuthorLanguageEnum["ItVa"] = "it-va";
    BlogAuthorLanguageEnum["HeIl"] = "he-il";
    BlogAuthorLanguageEnum["Ja"] = "ja";
    BlogAuthorLanguageEnum["JaJp"] = "ja-jp";
    BlogAuthorLanguageEnum["Jgo"] = "jgo";
    BlogAuthorLanguageEnum["JgoCm"] = "jgo-cm";
    BlogAuthorLanguageEnum["Yi"] = "yi";
    BlogAuthorLanguageEnum["Yi001"] = "yi-001";
    BlogAuthorLanguageEnum["Jmc"] = "jmc";
    BlogAuthorLanguageEnum["JmcTz"] = "jmc-tz";
    BlogAuthorLanguageEnum["Jv"] = "jv";
    BlogAuthorLanguageEnum["JvId"] = "jv-id";
    BlogAuthorLanguageEnum["Ka"] = "ka";
    BlogAuthorLanguageEnum["KaGe"] = "ka-ge";
    BlogAuthorLanguageEnum["Kab"] = "kab";
    BlogAuthorLanguageEnum["KabDz"] = "kab-dz";
    BlogAuthorLanguageEnum["Kam"] = "kam";
    BlogAuthorLanguageEnum["KamKe"] = "kam-ke";
    BlogAuthorLanguageEnum["Kde"] = "kde";
    BlogAuthorLanguageEnum["KdeTz"] = "kde-tz";
    BlogAuthorLanguageEnum["Kea"] = "kea";
    BlogAuthorLanguageEnum["KeaCv"] = "kea-cv";
    BlogAuthorLanguageEnum["Khq"] = "khq";
    BlogAuthorLanguageEnum["KhqMl"] = "khq-ml";
    BlogAuthorLanguageEnum["Ki"] = "ki";
    BlogAuthorLanguageEnum["KiKe"] = "ki-ke";
    BlogAuthorLanguageEnum["Kk"] = "kk";
    BlogAuthorLanguageEnum["KkKz"] = "kk-kz";
    BlogAuthorLanguageEnum["Kkj"] = "kkj";
    BlogAuthorLanguageEnum["KkjCm"] = "kkj-cm";
    BlogAuthorLanguageEnum["Kl"] = "kl";
    BlogAuthorLanguageEnum["KlGl"] = "kl-gl";
    BlogAuthorLanguageEnum["Kln"] = "kln";
    BlogAuthorLanguageEnum["KlnKe"] = "kln-ke";
    BlogAuthorLanguageEnum["Km"] = "km";
    BlogAuthorLanguageEnum["KmKh"] = "km-kh";
    BlogAuthorLanguageEnum["Kn"] = "kn";
    BlogAuthorLanguageEnum["KnIn"] = "kn-in";
    BlogAuthorLanguageEnum["Ko"] = "ko";
    BlogAuthorLanguageEnum["KoKp"] = "ko-kp";
    BlogAuthorLanguageEnum["KoKr"] = "ko-kr";
    BlogAuthorLanguageEnum["Kok"] = "kok";
    BlogAuthorLanguageEnum["KokIn"] = "kok-in";
    BlogAuthorLanguageEnum["Ks"] = "ks";
    BlogAuthorLanguageEnum["KsIn"] = "ks-in";
    BlogAuthorLanguageEnum["Ksb"] = "ksb";
    BlogAuthorLanguageEnum["KsbTz"] = "ksb-tz";
    BlogAuthorLanguageEnum["Ksf"] = "ksf";
    BlogAuthorLanguageEnum["KsfCm"] = "ksf-cm";
    BlogAuthorLanguageEnum["Ksh"] = "ksh";
    BlogAuthorLanguageEnum["KshDe"] = "ksh-de";
    BlogAuthorLanguageEnum["Kw"] = "kw";
    BlogAuthorLanguageEnum["KwGb"] = "kw-gb";
    BlogAuthorLanguageEnum["Ku"] = "ku";
    BlogAuthorLanguageEnum["KuTr"] = "ku-tr";
    BlogAuthorLanguageEnum["Ky"] = "ky";
    BlogAuthorLanguageEnum["KyKg"] = "ky-kg";
    BlogAuthorLanguageEnum["Lag"] = "lag";
    BlogAuthorLanguageEnum["LagTz"] = "lag-tz";
    BlogAuthorLanguageEnum["Lb"] = "lb";
    BlogAuthorLanguageEnum["LbLu"] = "lb-lu";
    BlogAuthorLanguageEnum["Lg"] = "lg";
    BlogAuthorLanguageEnum["LgUg"] = "lg-ug";
    BlogAuthorLanguageEnum["Lkt"] = "lkt";
    BlogAuthorLanguageEnum["LktUs"] = "lkt-us";
    BlogAuthorLanguageEnum["Ln"] = "ln";
    BlogAuthorLanguageEnum["LnAo"] = "ln-ao";
    BlogAuthorLanguageEnum["LnCd"] = "ln-cd";
    BlogAuthorLanguageEnum["LnCf"] = "ln-cf";
    BlogAuthorLanguageEnum["LnCg"] = "ln-cg";
    BlogAuthorLanguageEnum["Lo"] = "lo";
    BlogAuthorLanguageEnum["LoLa"] = "lo-la";
    BlogAuthorLanguageEnum["Lrc"] = "lrc";
    BlogAuthorLanguageEnum["LrcIq"] = "lrc-iq";
    BlogAuthorLanguageEnum["LrcIr"] = "lrc-ir";
    BlogAuthorLanguageEnum["Lt"] = "lt";
    BlogAuthorLanguageEnum["LtLt"] = "lt-lt";
    BlogAuthorLanguageEnum["Lu"] = "lu";
    BlogAuthorLanguageEnum["LuCd"] = "lu-cd";
    BlogAuthorLanguageEnum["Luo"] = "luo";
    BlogAuthorLanguageEnum["LuoKe"] = "luo-ke";
    BlogAuthorLanguageEnum["Luy"] = "luy";
    BlogAuthorLanguageEnum["LuyKe"] = "luy-ke";
    BlogAuthorLanguageEnum["Lv"] = "lv";
    BlogAuthorLanguageEnum["LvLv"] = "lv-lv";
    BlogAuthorLanguageEnum["Mai"] = "mai";
    BlogAuthorLanguageEnum["MaiIn"] = "mai-in";
    BlogAuthorLanguageEnum["Mas"] = "mas";
    BlogAuthorLanguageEnum["MasKe"] = "mas-ke";
    BlogAuthorLanguageEnum["MasTz"] = "mas-tz";
    BlogAuthorLanguageEnum["Mer"] = "mer";
    BlogAuthorLanguageEnum["MerKe"] = "mer-ke";
    BlogAuthorLanguageEnum["Mfe"] = "mfe";
    BlogAuthorLanguageEnum["MfeMu"] = "mfe-mu";
    BlogAuthorLanguageEnum["Mg"] = "mg";
    BlogAuthorLanguageEnum["MgMg"] = "mg-mg";
    BlogAuthorLanguageEnum["Mgh"] = "mgh";
    BlogAuthorLanguageEnum["MghMz"] = "mgh-mz";
    BlogAuthorLanguageEnum["Mgo"] = "mgo";
    BlogAuthorLanguageEnum["MgoCm"] = "mgo-cm";
    BlogAuthorLanguageEnum["Mi"] = "mi";
    BlogAuthorLanguageEnum["MiNz"] = "mi-nz";
    BlogAuthorLanguageEnum["Mk"] = "mk";
    BlogAuthorLanguageEnum["MkMk"] = "mk-mk";
    BlogAuthorLanguageEnum["Ml"] = "ml";
    BlogAuthorLanguageEnum["MlIn"] = "ml-in";
    BlogAuthorLanguageEnum["Mn"] = "mn";
    BlogAuthorLanguageEnum["MnMn"] = "mn-mn";
    BlogAuthorLanguageEnum["Mni"] = "mni";
    BlogAuthorLanguageEnum["MniIn"] = "mni-in";
    BlogAuthorLanguageEnum["Mr"] = "mr";
    BlogAuthorLanguageEnum["MrIn"] = "mr-in";
    BlogAuthorLanguageEnum["Ms"] = "ms";
    BlogAuthorLanguageEnum["MsBn"] = "ms-bn";
    BlogAuthorLanguageEnum["MsId"] = "ms-id";
    BlogAuthorLanguageEnum["MsMy"] = "ms-my";
    BlogAuthorLanguageEnum["MsSg"] = "ms-sg";
    BlogAuthorLanguageEnum["Mt"] = "mt";
    BlogAuthorLanguageEnum["MtMt"] = "mt-mt";
    BlogAuthorLanguageEnum["Mua"] = "mua";
    BlogAuthorLanguageEnum["MuaCm"] = "mua-cm";
    BlogAuthorLanguageEnum["My"] = "my";
    BlogAuthorLanguageEnum["MyMm"] = "my-mm";
    BlogAuthorLanguageEnum["Mzn"] = "mzn";
    BlogAuthorLanguageEnum["MznIr"] = "mzn-ir";
    BlogAuthorLanguageEnum["Naq"] = "naq";
    BlogAuthorLanguageEnum["NaqNa"] = "naq-na";
    BlogAuthorLanguageEnum["Nb"] = "nb";
    BlogAuthorLanguageEnum["NbNo"] = "nb-no";
    BlogAuthorLanguageEnum["NbSj"] = "nb-sj";
    BlogAuthorLanguageEnum["Nd"] = "nd";
    BlogAuthorLanguageEnum["NdZw"] = "nd-zw";
    BlogAuthorLanguageEnum["Nds"] = "nds";
    BlogAuthorLanguageEnum["NdsDe"] = "nds-de";
    BlogAuthorLanguageEnum["NdsNl"] = "nds-nl";
    BlogAuthorLanguageEnum["Ne"] = "ne";
    BlogAuthorLanguageEnum["NeIn"] = "ne-in";
    BlogAuthorLanguageEnum["NeNp"] = "ne-np";
    BlogAuthorLanguageEnum["Nl"] = "nl";
    BlogAuthorLanguageEnum["NlAw"] = "nl-aw";
    BlogAuthorLanguageEnum["NlBe"] = "nl-be";
    BlogAuthorLanguageEnum["NlCh"] = "nl-ch";
    BlogAuthorLanguageEnum["NlBq"] = "nl-bq";
    BlogAuthorLanguageEnum["NlCw"] = "nl-cw";
    BlogAuthorLanguageEnum["NlLu"] = "nl-lu";
    BlogAuthorLanguageEnum["NlNl"] = "nl-nl";
    BlogAuthorLanguageEnum["NlSr"] = "nl-sr";
    BlogAuthorLanguageEnum["NlSx"] = "nl-sx";
    BlogAuthorLanguageEnum["Nmg"] = "nmg";
    BlogAuthorLanguageEnum["NmgCm"] = "nmg-cm";
    BlogAuthorLanguageEnum["Nn"] = "nn";
    BlogAuthorLanguageEnum["NnNo"] = "nn-no";
    BlogAuthorLanguageEnum["Nnh"] = "nnh";
    BlogAuthorLanguageEnum["NnhCm"] = "nnh-cm";
    BlogAuthorLanguageEnum["No"] = "no";
    BlogAuthorLanguageEnum["NoNo"] = "no-no";
    BlogAuthorLanguageEnum["Nus"] = "nus";
    BlogAuthorLanguageEnum["NusSs"] = "nus-ss";
    BlogAuthorLanguageEnum["Nyn"] = "nyn";
    BlogAuthorLanguageEnum["NynUg"] = "nyn-ug";
    BlogAuthorLanguageEnum["Om"] = "om";
    BlogAuthorLanguageEnum["OmEt"] = "om-et";
    BlogAuthorLanguageEnum["OmKe"] = "om-ke";
    BlogAuthorLanguageEnum["Or"] = "or";
    BlogAuthorLanguageEnum["OrIn"] = "or-in";
    BlogAuthorLanguageEnum["Os"] = "os";
    BlogAuthorLanguageEnum["OsGe"] = "os-ge";
    BlogAuthorLanguageEnum["OsRu"] = "os-ru";
    BlogAuthorLanguageEnum["Pa"] = "pa";
    BlogAuthorLanguageEnum["PaIn"] = "pa-in";
    BlogAuthorLanguageEnum["PaPk"] = "pa-pk";
    BlogAuthorLanguageEnum["Pcm"] = "pcm";
    BlogAuthorLanguageEnum["PcmNg"] = "pcm-ng";
    BlogAuthorLanguageEnum["Pl"] = "pl";
    BlogAuthorLanguageEnum["PlPl"] = "pl-pl";
    BlogAuthorLanguageEnum["Prg"] = "prg";
    BlogAuthorLanguageEnum["Prg001"] = "prg-001";
    BlogAuthorLanguageEnum["Ps"] = "ps";
    BlogAuthorLanguageEnum["PsAf"] = "ps-af";
    BlogAuthorLanguageEnum["PsPk"] = "ps-pk";
    BlogAuthorLanguageEnum["Pt"] = "pt";
    BlogAuthorLanguageEnum["PtAo"] = "pt-ao";
    BlogAuthorLanguageEnum["PtBr"] = "pt-br";
    BlogAuthorLanguageEnum["PtCh"] = "pt-ch";
    BlogAuthorLanguageEnum["PtCv"] = "pt-cv";
    BlogAuthorLanguageEnum["PtGq"] = "pt-gq";
    BlogAuthorLanguageEnum["PtGw"] = "pt-gw";
    BlogAuthorLanguageEnum["PtLu"] = "pt-lu";
    BlogAuthorLanguageEnum["PtMo"] = "pt-mo";
    BlogAuthorLanguageEnum["PtMz"] = "pt-mz";
    BlogAuthorLanguageEnum["PtPt"] = "pt-pt";
    BlogAuthorLanguageEnum["PtSt"] = "pt-st";
    BlogAuthorLanguageEnum["PtTl"] = "pt-tl";
    BlogAuthorLanguageEnum["Qu"] = "qu";
    BlogAuthorLanguageEnum["QuBo"] = "qu-bo";
    BlogAuthorLanguageEnum["QuEc"] = "qu-ec";
    BlogAuthorLanguageEnum["QuPe"] = "qu-pe";
    BlogAuthorLanguageEnum["Rm"] = "rm";
    BlogAuthorLanguageEnum["RmCh"] = "rm-ch";
    BlogAuthorLanguageEnum["Rn"] = "rn";
    BlogAuthorLanguageEnum["RnBi"] = "rn-bi";
    BlogAuthorLanguageEnum["Ro"] = "ro";
    BlogAuthorLanguageEnum["RoMd"] = "ro-md";
    BlogAuthorLanguageEnum["RoRo"] = "ro-ro";
    BlogAuthorLanguageEnum["Rof"] = "rof";
    BlogAuthorLanguageEnum["RofTz"] = "rof-tz";
    BlogAuthorLanguageEnum["Ru"] = "ru";
    BlogAuthorLanguageEnum["RuBy"] = "ru-by";
    BlogAuthorLanguageEnum["RuKg"] = "ru-kg";
    BlogAuthorLanguageEnum["RuKz"] = "ru-kz";
    BlogAuthorLanguageEnum["RuMd"] = "ru-md";
    BlogAuthorLanguageEnum["RuRu"] = "ru-ru";
    BlogAuthorLanguageEnum["RuUa"] = "ru-ua";
    BlogAuthorLanguageEnum["Rw"] = "rw";
    BlogAuthorLanguageEnum["RwRw"] = "rw-rw";
    BlogAuthorLanguageEnum["Rwk"] = "rwk";
    BlogAuthorLanguageEnum["RwkTz"] = "rwk-tz";
    BlogAuthorLanguageEnum["Sa"] = "sa";
    BlogAuthorLanguageEnum["SaIn"] = "sa-in";
    BlogAuthorLanguageEnum["Sah"] = "sah";
    BlogAuthorLanguageEnum["SahRu"] = "sah-ru";
    BlogAuthorLanguageEnum["Saq"] = "saq";
    BlogAuthorLanguageEnum["SaqKe"] = "saq-ke";
    BlogAuthorLanguageEnum["Sat"] = "sat";
    BlogAuthorLanguageEnum["SatIn"] = "sat-in";
    BlogAuthorLanguageEnum["Sbp"] = "sbp";
    BlogAuthorLanguageEnum["SbpTz"] = "sbp-tz";
    BlogAuthorLanguageEnum["Sd"] = "sd";
    BlogAuthorLanguageEnum["SdIn"] = "sd-in";
    BlogAuthorLanguageEnum["SdPk"] = "sd-pk";
    BlogAuthorLanguageEnum["Se"] = "se";
    BlogAuthorLanguageEnum["SeFi"] = "se-fi";
    BlogAuthorLanguageEnum["SeNo"] = "se-no";
    BlogAuthorLanguageEnum["SeSe"] = "se-se";
    BlogAuthorLanguageEnum["Seh"] = "seh";
    BlogAuthorLanguageEnum["SehMz"] = "seh-mz";
    BlogAuthorLanguageEnum["Ses"] = "ses";
    BlogAuthorLanguageEnum["SesMl"] = "ses-ml";
    BlogAuthorLanguageEnum["Sg"] = "sg";
    BlogAuthorLanguageEnum["SgCf"] = "sg-cf";
    BlogAuthorLanguageEnum["Shi"] = "shi";
    BlogAuthorLanguageEnum["ShiMa"] = "shi-ma";
    BlogAuthorLanguageEnum["Si"] = "si";
    BlogAuthorLanguageEnum["SiLk"] = "si-lk";
    BlogAuthorLanguageEnum["Sk"] = "sk";
    BlogAuthorLanguageEnum["SkSk"] = "sk-sk";
    BlogAuthorLanguageEnum["Sl"] = "sl";
    BlogAuthorLanguageEnum["SlSi"] = "sl-si";
    BlogAuthorLanguageEnum["Smn"] = "smn";
    BlogAuthorLanguageEnum["SmnFi"] = "smn-fi";
    BlogAuthorLanguageEnum["Sn"] = "sn";
    BlogAuthorLanguageEnum["SnZw"] = "sn-zw";
    BlogAuthorLanguageEnum["So"] = "so";
    BlogAuthorLanguageEnum["SoDj"] = "so-dj";
    BlogAuthorLanguageEnum["SoEt"] = "so-et";
    BlogAuthorLanguageEnum["SoKe"] = "so-ke";
    BlogAuthorLanguageEnum["SoSo"] = "so-so";
    BlogAuthorLanguageEnum["Sq"] = "sq";
    BlogAuthorLanguageEnum["SqAl"] = "sq-al";
    BlogAuthorLanguageEnum["SqMk"] = "sq-mk";
    BlogAuthorLanguageEnum["SqXk"] = "sq-xk";
    BlogAuthorLanguageEnum["Sr"] = "sr";
    BlogAuthorLanguageEnum["SrBa"] = "sr-ba";
    BlogAuthorLanguageEnum["SrCs"] = "sr-cs";
    BlogAuthorLanguageEnum["SrMe"] = "sr-me";
    BlogAuthorLanguageEnum["SrRs"] = "sr-rs";
    BlogAuthorLanguageEnum["SrXk"] = "sr-xk";
    BlogAuthorLanguageEnum["Su"] = "su";
    BlogAuthorLanguageEnum["SuId"] = "su-id";
    BlogAuthorLanguageEnum["Sv"] = "sv";
    BlogAuthorLanguageEnum["SvAx"] = "sv-ax";
    BlogAuthorLanguageEnum["SvFi"] = "sv-fi";
    BlogAuthorLanguageEnum["SvSe"] = "sv-se";
    BlogAuthorLanguageEnum["Sw"] = "sw";
    BlogAuthorLanguageEnum["SwCd"] = "sw-cd";
    BlogAuthorLanguageEnum["SwKe"] = "sw-ke";
    BlogAuthorLanguageEnum["SwTz"] = "sw-tz";
    BlogAuthorLanguageEnum["SwUg"] = "sw-ug";
    BlogAuthorLanguageEnum["Sy"] = "sy";
    BlogAuthorLanguageEnum["Ta"] = "ta";
    BlogAuthorLanguageEnum["TaIn"] = "ta-in";
    BlogAuthorLanguageEnum["TaLk"] = "ta-lk";
    BlogAuthorLanguageEnum["TaMy"] = "ta-my";
    BlogAuthorLanguageEnum["TaSg"] = "ta-sg";
    BlogAuthorLanguageEnum["Te"] = "te";
    BlogAuthorLanguageEnum["TeIn"] = "te-in";
    BlogAuthorLanguageEnum["Teo"] = "teo";
    BlogAuthorLanguageEnum["TeoKe"] = "teo-ke";
    BlogAuthorLanguageEnum["TeoUg"] = "teo-ug";
    BlogAuthorLanguageEnum["Tg"] = "tg";
    BlogAuthorLanguageEnum["TgTj"] = "tg-tj";
    BlogAuthorLanguageEnum["Th"] = "th";
    BlogAuthorLanguageEnum["ThTh"] = "th-th";
    BlogAuthorLanguageEnum["Ti"] = "ti";
    BlogAuthorLanguageEnum["TiEr"] = "ti-er";
    BlogAuthorLanguageEnum["TiEt"] = "ti-et";
    BlogAuthorLanguageEnum["Tk"] = "tk";
    BlogAuthorLanguageEnum["TkTm"] = "tk-tm";
    BlogAuthorLanguageEnum["Tl"] = "tl";
    BlogAuthorLanguageEnum["To"] = "to";
    BlogAuthorLanguageEnum["ToTo"] = "to-to";
    BlogAuthorLanguageEnum["Tr"] = "tr";
    BlogAuthorLanguageEnum["TrCy"] = "tr-cy";
    BlogAuthorLanguageEnum["TrTr"] = "tr-tr";
    BlogAuthorLanguageEnum["Tt"] = "tt";
    BlogAuthorLanguageEnum["TtRu"] = "tt-ru";
    BlogAuthorLanguageEnum["Twq"] = "twq";
    BlogAuthorLanguageEnum["TwqNe"] = "twq-ne";
    BlogAuthorLanguageEnum["Tzm"] = "tzm";
    BlogAuthorLanguageEnum["TzmMa"] = "tzm-ma";
    BlogAuthorLanguageEnum["Ug"] = "ug";
    BlogAuthorLanguageEnum["UgCn"] = "ug-cn";
    BlogAuthorLanguageEnum["Uk"] = "uk";
    BlogAuthorLanguageEnum["UkUa"] = "uk-ua";
    BlogAuthorLanguageEnum["Ur"] = "ur";
    BlogAuthorLanguageEnum["UrIn"] = "ur-in";
    BlogAuthorLanguageEnum["UrPk"] = "ur-pk";
    BlogAuthorLanguageEnum["Uz"] = "uz";
    BlogAuthorLanguageEnum["UzAf"] = "uz-af";
    BlogAuthorLanguageEnum["UzUz"] = "uz-uz";
    BlogAuthorLanguageEnum["Vai"] = "vai";
    BlogAuthorLanguageEnum["VaiLr"] = "vai-lr";
    BlogAuthorLanguageEnum["Vi"] = "vi";
    BlogAuthorLanguageEnum["ViVn"] = "vi-vn";
    BlogAuthorLanguageEnum["Vo"] = "vo";
    BlogAuthorLanguageEnum["Vo001"] = "vo-001";
    BlogAuthorLanguageEnum["Vun"] = "vun";
    BlogAuthorLanguageEnum["VunTz"] = "vun-tz";
    BlogAuthorLanguageEnum["Wae"] = "wae";
    BlogAuthorLanguageEnum["WaeCh"] = "wae-ch";
    BlogAuthorLanguageEnum["Wo"] = "wo";
    BlogAuthorLanguageEnum["WoSn"] = "wo-sn";
    BlogAuthorLanguageEnum["Xh"] = "xh";
    BlogAuthorLanguageEnum["XhZa"] = "xh-za";
    BlogAuthorLanguageEnum["Xog"] = "xog";
    BlogAuthorLanguageEnum["XogUg"] = "xog-ug";
    BlogAuthorLanguageEnum["Yav"] = "yav";
    BlogAuthorLanguageEnum["YavCm"] = "yav-cm";
    BlogAuthorLanguageEnum["Yo"] = "yo";
    BlogAuthorLanguageEnum["YoBj"] = "yo-bj";
    BlogAuthorLanguageEnum["YoNg"] = "yo-ng";
    BlogAuthorLanguageEnum["Yue"] = "yue";
    BlogAuthorLanguageEnum["YueCn"] = "yue-cn";
    BlogAuthorLanguageEnum["YueHk"] = "yue-hk";
    BlogAuthorLanguageEnum["Zgh"] = "zgh";
    BlogAuthorLanguageEnum["ZghMa"] = "zgh-ma";
    BlogAuthorLanguageEnum["Zh"] = "zh";
    BlogAuthorLanguageEnum["ZhCn"] = "zh-cn";
    BlogAuthorLanguageEnum["ZhHk"] = "zh-hk";
    BlogAuthorLanguageEnum["ZhMo"] = "zh-mo";
    BlogAuthorLanguageEnum["ZhSg"] = "zh-sg";
    BlogAuthorLanguageEnum["ZhTw"] = "zh-tw";
    BlogAuthorLanguageEnum["ZhHans"] = "zh-hans";
    BlogAuthorLanguageEnum["ZhHant"] = "zh-hant";
    BlogAuthorLanguageEnum["Zu"] = "zu";
    BlogAuthorLanguageEnum["ZuZa"] = "zu-za";
})(BlogAuthorLanguageEnum = exports.BlogAuthorLanguageEnum || (exports.BlogAuthorLanguageEnum = {}));
