// @hubspot/api-client/lib/codegen/crm/objects/tasks/models/BatchResponseSimplePublicUpsertObject.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSimplePublicUpsertObjectStatusEnum = exports.BatchResponseSimplePublicUpsertObject = void 0;
class BatchResponseSimplePublicUpsertObject {
    static getAttributeTypeMap() {
        return BatchResponseSimplePublicUpsertObject.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseSimplePublicUpsertObject = BatchResponseSimplePublicUpsertObject;
BatchResponseSimplePublicUpsertObject.discriminator = undefined;
BatchResponseSimplePublicUpsertObject.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SimplePublicUpsertObject>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSimplePublicUpsertObjectStatusEnum",
        "format": ""
    }
];
var BatchResponseSimplePublicUpsertObjectStatusEnum;
(function (BatchResponseSimplePublicUpsertObjectStatusEnum) {
    BatchResponseSimplePublicUpsertObjectStatusEnum["Pending"] = "PENDING";
    BatchResponseSimplePublicUpsertObjectStatusEnum["Processing"] = "PROCESSING";
    BatchResponseSimplePublicUpsertObjectStatusEnum["Canceled"] = "CANCELED";
    BatchResponseSimplePublicUpsertObjectStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseSimplePublicUpsertObjectStatusEnum = exports.BatchResponseSimplePublicUpsertObjectStatusEnum || (exports.BatchResponseSimplePublicUpsertObjectStatusEnum = {}));
