// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/models/AttachToLangPrimaryRequestVNext.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachToLangPrimaryRequestVNextLanguageEnum = exports.AttachToLangPrimaryRequestVNext = void 0;
class AttachToLangPrimaryRequestVNext {
    static getAttributeTypeMap() {
        return AttachToLangPrimaryRequestVNext.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AttachToLangPrimaryRequestVNext = AttachToLangPrimaryRequestVNext;
AttachToLangPrimaryRequestVNext.discriminator = undefined;
AttachToLangPrimaryRequestVNext.attributeTypeMap = [
    {
        "name": "language",
        "baseName": "language",
        "type": "AttachToLangPrimaryRequestVNextLanguageEnum",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "primaryId",
        "baseName": "primaryId",
        "type": "string",
        "format": ""
    },
    {
        "name": "primaryLanguage",
        "baseName": "primaryLanguage",
        "type": "string",
        "format": ""
    }
];
var AttachToLangPrimaryRequestVNextLanguageEnum;
(function (AttachToLangPrimaryRequestVNextLanguageEnum) {
    AttachToLangPrimaryRequestVNextLanguageEnum["Af"] = "af";
    AttachToLangPrimaryRequestVNextLanguageEnum["AfNa"] = "af-na";
    AttachToLangPrimaryRequestVNextLanguageEnum["AfZa"] = "af-za";
    AttachToLangPrimaryRequestVNextLanguageEnum["Agq"] = "agq";
    AttachToLangPrimaryRequestVNextLanguageEnum["AgqCm"] = "agq-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ak"] = "ak";
    AttachToLangPrimaryRequestVNextLanguageEnum["AkGh"] = "ak-gh";
    AttachToLangPrimaryRequestVNextLanguageEnum["Am"] = "am";
    AttachToLangPrimaryRequestVNextLanguageEnum["AmEt"] = "am-et";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ar"] = "ar";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ar001"] = "ar-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArAe"] = "ar-ae";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArBh"] = "ar-bh";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArDj"] = "ar-dj";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArDz"] = "ar-dz";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArEg"] = "ar-eg";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArEh"] = "ar-eh";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArEr"] = "ar-er";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArIl"] = "ar-il";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArIq"] = "ar-iq";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArJo"] = "ar-jo";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArKm"] = "ar-km";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArKw"] = "ar-kw";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArLb"] = "ar-lb";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArLy"] = "ar-ly";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArMa"] = "ar-ma";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArMr"] = "ar-mr";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArOm"] = "ar-om";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArPs"] = "ar-ps";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArQa"] = "ar-qa";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArSa"] = "ar-sa";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArSd"] = "ar-sd";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArSo"] = "ar-so";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArSs"] = "ar-ss";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArSy"] = "ar-sy";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArTd"] = "ar-td";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArTn"] = "ar-tn";
    AttachToLangPrimaryRequestVNextLanguageEnum["ArYe"] = "ar-ye";
    AttachToLangPrimaryRequestVNextLanguageEnum["As"] = "as";
    AttachToLangPrimaryRequestVNextLanguageEnum["AsIn"] = "as-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Asa"] = "asa";
    AttachToLangPrimaryRequestVNextLanguageEnum["AsaTz"] = "asa-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ast"] = "ast";
    AttachToLangPrimaryRequestVNextLanguageEnum["AstEs"] = "ast-es";
    AttachToLangPrimaryRequestVNextLanguageEnum["Az"] = "az";
    AttachToLangPrimaryRequestVNextLanguageEnum["AzAz"] = "az-az";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bas"] = "bas";
    AttachToLangPrimaryRequestVNextLanguageEnum["BasCm"] = "bas-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Be"] = "be";
    AttachToLangPrimaryRequestVNextLanguageEnum["BeBy"] = "be-by";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bem"] = "bem";
    AttachToLangPrimaryRequestVNextLanguageEnum["BemZm"] = "bem-zm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bez"] = "bez";
    AttachToLangPrimaryRequestVNextLanguageEnum["BezTz"] = "bez-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bg"] = "bg";
    AttachToLangPrimaryRequestVNextLanguageEnum["BgBg"] = "bg-bg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bm"] = "bm";
    AttachToLangPrimaryRequestVNextLanguageEnum["BmMl"] = "bm-ml";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bn"] = "bn";
    AttachToLangPrimaryRequestVNextLanguageEnum["BnBd"] = "bn-bd";
    AttachToLangPrimaryRequestVNextLanguageEnum["BnIn"] = "bn-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bo"] = "bo";
    AttachToLangPrimaryRequestVNextLanguageEnum["BoCn"] = "bo-cn";
    AttachToLangPrimaryRequestVNextLanguageEnum["BoIn"] = "bo-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Br"] = "br";
    AttachToLangPrimaryRequestVNextLanguageEnum["BrFr"] = "br-fr";
    AttachToLangPrimaryRequestVNextLanguageEnum["Brx"] = "brx";
    AttachToLangPrimaryRequestVNextLanguageEnum["BrxIn"] = "brx-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Bs"] = "bs";
    AttachToLangPrimaryRequestVNextLanguageEnum["BsBa"] = "bs-ba";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ca"] = "ca";
    AttachToLangPrimaryRequestVNextLanguageEnum["CaAd"] = "ca-ad";
    AttachToLangPrimaryRequestVNextLanguageEnum["CaEs"] = "ca-es";
    AttachToLangPrimaryRequestVNextLanguageEnum["CaFr"] = "ca-fr";
    AttachToLangPrimaryRequestVNextLanguageEnum["CaIt"] = "ca-it";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ccp"] = "ccp";
    AttachToLangPrimaryRequestVNextLanguageEnum["CcpBd"] = "ccp-bd";
    AttachToLangPrimaryRequestVNextLanguageEnum["CcpIn"] = "ccp-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ce"] = "ce";
    AttachToLangPrimaryRequestVNextLanguageEnum["CeRu"] = "ce-ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ceb"] = "ceb";
    AttachToLangPrimaryRequestVNextLanguageEnum["CebPh"] = "ceb-ph";
    AttachToLangPrimaryRequestVNextLanguageEnum["Cgg"] = "cgg";
    AttachToLangPrimaryRequestVNextLanguageEnum["CggUg"] = "cgg-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["Chr"] = "chr";
    AttachToLangPrimaryRequestVNextLanguageEnum["ChrUs"] = "chr-us";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ckb"] = "ckb";
    AttachToLangPrimaryRequestVNextLanguageEnum["CkbIq"] = "ckb-iq";
    AttachToLangPrimaryRequestVNextLanguageEnum["CkbIr"] = "ckb-ir";
    AttachToLangPrimaryRequestVNextLanguageEnum["Cs"] = "cs";
    AttachToLangPrimaryRequestVNextLanguageEnum["CsCz"] = "cs-cz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Cu"] = "cu";
    AttachToLangPrimaryRequestVNextLanguageEnum["CuRu"] = "cu-ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["Cy"] = "cy";
    AttachToLangPrimaryRequestVNextLanguageEnum["CyGb"] = "cy-gb";
    AttachToLangPrimaryRequestVNextLanguageEnum["Da"] = "da";
    AttachToLangPrimaryRequestVNextLanguageEnum["DaDk"] = "da-dk";
    AttachToLangPrimaryRequestVNextLanguageEnum["DaGl"] = "da-gl";
    AttachToLangPrimaryRequestVNextLanguageEnum["Dav"] = "dav";
    AttachToLangPrimaryRequestVNextLanguageEnum["DavKe"] = "dav-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["De"] = "de";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeAt"] = "de-at";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeBe"] = "de-be";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeCh"] = "de-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeDe"] = "de-de";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeGr"] = "de-gr";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeIt"] = "de-it";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeLi"] = "de-li";
    AttachToLangPrimaryRequestVNextLanguageEnum["DeLu"] = "de-lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["Dje"] = "dje";
    AttachToLangPrimaryRequestVNextLanguageEnum["DjeNe"] = "dje-ne";
    AttachToLangPrimaryRequestVNextLanguageEnum["Doi"] = "doi";
    AttachToLangPrimaryRequestVNextLanguageEnum["DoiIn"] = "doi-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Dsb"] = "dsb";
    AttachToLangPrimaryRequestVNextLanguageEnum["DsbDe"] = "dsb-de";
    AttachToLangPrimaryRequestVNextLanguageEnum["Dua"] = "dua";
    AttachToLangPrimaryRequestVNextLanguageEnum["DuaCm"] = "dua-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Dyo"] = "dyo";
    AttachToLangPrimaryRequestVNextLanguageEnum["DyoSn"] = "dyo-sn";
    AttachToLangPrimaryRequestVNextLanguageEnum["Dz"] = "dz";
    AttachToLangPrimaryRequestVNextLanguageEnum["DzBt"] = "dz-bt";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ebu"] = "ebu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EbuKe"] = "ebu-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ee"] = "ee";
    AttachToLangPrimaryRequestVNextLanguageEnum["EeGh"] = "ee-gh";
    AttachToLangPrimaryRequestVNextLanguageEnum["EeTg"] = "ee-tg";
    AttachToLangPrimaryRequestVNextLanguageEnum["El"] = "el";
    AttachToLangPrimaryRequestVNextLanguageEnum["ElCy"] = "el-cy";
    AttachToLangPrimaryRequestVNextLanguageEnum["ElGr"] = "el-gr";
    AttachToLangPrimaryRequestVNextLanguageEnum["En"] = "en";
    AttachToLangPrimaryRequestVNextLanguageEnum["En001"] = "en-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["En150"] = "en-150";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnAe"] = "en-ae";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnAg"] = "en-ag";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnAi"] = "en-ai";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnAs"] = "en-as";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnAt"] = "en-at";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnAu"] = "en-au";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBb"] = "en-bb";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBe"] = "en-be";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBi"] = "en-bi";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBm"] = "en-bm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBs"] = "en-bs";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBw"] = "en-bw";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnBz"] = "en-bz";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCa"] = "en-ca";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCc"] = "en-cc";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCh"] = "en-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCk"] = "en-ck";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCm"] = "en-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCn"] = "en-cn";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCx"] = "en-cx";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnCy"] = "en-cy";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnDe"] = "en-de";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnDg"] = "en-dg";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnDk"] = "en-dk";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnDm"] = "en-dm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnEe"] = "en-ee";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnEr"] = "en-er";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnFr"] = "en-fr";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnFi"] = "en-fi";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnFj"] = "en-fj";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnFk"] = "en-fk";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnFm"] = "en-fm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGb"] = "en-gb";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGd"] = "en-gd";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGg"] = "en-gg";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGh"] = "en-gh";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGi"] = "en-gi";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGm"] = "en-gm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGu"] = "en-gu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnGy"] = "en-gy";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnHk"] = "en-hk";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnIe"] = "en-ie";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnIl"] = "en-il";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnIm"] = "en-im";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnIn"] = "en-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnIo"] = "en-io";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnJe"] = "en-je";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnJm"] = "en-jm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnKe"] = "en-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnKi"] = "en-ki";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnKn"] = "en-kn";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnKy"] = "en-ky";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnLc"] = "en-lc";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnLr"] = "en-lr";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnLs"] = "en-ls";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnLu"] = "en-lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMg"] = "en-mg";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMh"] = "en-mh";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMo"] = "en-mo";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMp"] = "en-mp";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMs"] = "en-ms";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMt"] = "en-mt";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMu"] = "en-mu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMw"] = "en-mw";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMx"] = "en-mx";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnMy"] = "en-my";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNa"] = "en-na";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNf"] = "en-nf";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNg"] = "en-ng";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNl"] = "en-nl";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNr"] = "en-nr";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNu"] = "en-nu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnNz"] = "en-nz";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnPg"] = "en-pg";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnPh"] = "en-ph";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnPk"] = "en-pk";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnPn"] = "en-pn";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnPr"] = "en-pr";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnPw"] = "en-pw";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnRw"] = "en-rw";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSb"] = "en-sb";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSc"] = "en-sc";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSd"] = "en-sd";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSe"] = "en-se";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSg"] = "en-sg";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSh"] = "en-sh";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSi"] = "en-si";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSl"] = "en-sl";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSs"] = "en-ss";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSx"] = "en-sx";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnSz"] = "en-sz";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnTc"] = "en-tc";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnTk"] = "en-tk";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnTo"] = "en-to";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnTt"] = "en-tt";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnTv"] = "en-tv";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnTz"] = "en-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnUg"] = "en-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnUm"] = "en-um";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnUs"] = "en-us";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnVc"] = "en-vc";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnVg"] = "en-vg";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnVi"] = "en-vi";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnVu"] = "en-vu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnWs"] = "en-ws";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnZa"] = "en-za";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnZm"] = "en-zm";
    AttachToLangPrimaryRequestVNextLanguageEnum["EnZw"] = "en-zw";
    AttachToLangPrimaryRequestVNextLanguageEnum["Eo"] = "eo";
    AttachToLangPrimaryRequestVNextLanguageEnum["Eo001"] = "eo-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["Es"] = "es";
    AttachToLangPrimaryRequestVNextLanguageEnum["Es419"] = "es-419";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsAr"] = "es-ar";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsBo"] = "es-bo";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsBr"] = "es-br";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsBz"] = "es-bz";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsCl"] = "es-cl";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsCo"] = "es-co";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsCr"] = "es-cr";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsCu"] = "es-cu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsDo"] = "es-do";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsEa"] = "es-ea";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsEc"] = "es-ec";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsEs"] = "es-es";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsGq"] = "es-gq";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsGt"] = "es-gt";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsHn"] = "es-hn";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsIc"] = "es-ic";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsMx"] = "es-mx";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsNi"] = "es-ni";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsPa"] = "es-pa";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsPe"] = "es-pe";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsPh"] = "es-ph";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsPr"] = "es-pr";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsPy"] = "es-py";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsSv"] = "es-sv";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsUs"] = "es-us";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsUy"] = "es-uy";
    AttachToLangPrimaryRequestVNextLanguageEnum["EsVe"] = "es-ve";
    AttachToLangPrimaryRequestVNextLanguageEnum["Et"] = "et";
    AttachToLangPrimaryRequestVNextLanguageEnum["EtEe"] = "et-ee";
    AttachToLangPrimaryRequestVNextLanguageEnum["Eu"] = "eu";
    AttachToLangPrimaryRequestVNextLanguageEnum["EuEs"] = "eu-es";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ewo"] = "ewo";
    AttachToLangPrimaryRequestVNextLanguageEnum["EwoCm"] = "ewo-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fa"] = "fa";
    AttachToLangPrimaryRequestVNextLanguageEnum["FaAf"] = "fa-af";
    AttachToLangPrimaryRequestVNextLanguageEnum["FaIr"] = "fa-ir";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ff"] = "ff";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfBf"] = "ff-bf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfCm"] = "ff-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfGh"] = "ff-gh";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfGm"] = "ff-gm";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfGn"] = "ff-gn";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfGw"] = "ff-gw";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfLr"] = "ff-lr";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfMr"] = "ff-mr";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfNe"] = "ff-ne";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfNg"] = "ff-ng";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfSl"] = "ff-sl";
    AttachToLangPrimaryRequestVNextLanguageEnum["FfSn"] = "ff-sn";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fi"] = "fi";
    AttachToLangPrimaryRequestVNextLanguageEnum["FiFi"] = "fi-fi";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fil"] = "fil";
    AttachToLangPrimaryRequestVNextLanguageEnum["FilPh"] = "fil-ph";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fo"] = "fo";
    AttachToLangPrimaryRequestVNextLanguageEnum["FoDk"] = "fo-dk";
    AttachToLangPrimaryRequestVNextLanguageEnum["FoFo"] = "fo-fo";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fr"] = "fr";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrBe"] = "fr-be";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrBf"] = "fr-bf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrBi"] = "fr-bi";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrBj"] = "fr-bj";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrBl"] = "fr-bl";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCa"] = "fr-ca";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCd"] = "fr-cd";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCf"] = "fr-cf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCg"] = "fr-cg";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCh"] = "fr-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCi"] = "fr-ci";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrCm"] = "fr-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrDj"] = "fr-dj";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrDz"] = "fr-dz";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrFr"] = "fr-fr";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrGa"] = "fr-ga";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrGf"] = "fr-gf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrGn"] = "fr-gn";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrGp"] = "fr-gp";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrGq"] = "fr-gq";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrHt"] = "fr-ht";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrKm"] = "fr-km";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrLu"] = "fr-lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMa"] = "fr-ma";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMc"] = "fr-mc";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMf"] = "fr-mf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMg"] = "fr-mg";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMl"] = "fr-ml";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMq"] = "fr-mq";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMr"] = "fr-mr";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrMu"] = "fr-mu";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrNc"] = "fr-nc";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrNe"] = "fr-ne";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrPf"] = "fr-pf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrPm"] = "fr-pm";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrRe"] = "fr-re";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrRw"] = "fr-rw";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrSc"] = "fr-sc";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrSn"] = "fr-sn";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrSy"] = "fr-sy";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrTd"] = "fr-td";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrTg"] = "fr-tg";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrTn"] = "fr-tn";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrVu"] = "fr-vu";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrWf"] = "fr-wf";
    AttachToLangPrimaryRequestVNextLanguageEnum["FrYt"] = "fr-yt";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fur"] = "fur";
    AttachToLangPrimaryRequestVNextLanguageEnum["FurIt"] = "fur-it";
    AttachToLangPrimaryRequestVNextLanguageEnum["Fy"] = "fy";
    AttachToLangPrimaryRequestVNextLanguageEnum["FyNl"] = "fy-nl";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ga"] = "ga";
    AttachToLangPrimaryRequestVNextLanguageEnum["GaGb"] = "ga-gb";
    AttachToLangPrimaryRequestVNextLanguageEnum["GaIe"] = "ga-ie";
    AttachToLangPrimaryRequestVNextLanguageEnum["Gd"] = "gd";
    AttachToLangPrimaryRequestVNextLanguageEnum["GdGb"] = "gd-gb";
    AttachToLangPrimaryRequestVNextLanguageEnum["Gl"] = "gl";
    AttachToLangPrimaryRequestVNextLanguageEnum["GlEs"] = "gl-es";
    AttachToLangPrimaryRequestVNextLanguageEnum["Gsw"] = "gsw";
    AttachToLangPrimaryRequestVNextLanguageEnum["GswCh"] = "gsw-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["GswFr"] = "gsw-fr";
    AttachToLangPrimaryRequestVNextLanguageEnum["GswLi"] = "gsw-li";
    AttachToLangPrimaryRequestVNextLanguageEnum["Gu"] = "gu";
    AttachToLangPrimaryRequestVNextLanguageEnum["GuIn"] = "gu-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Guz"] = "guz";
    AttachToLangPrimaryRequestVNextLanguageEnum["GuzKe"] = "guz-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Gv"] = "gv";
    AttachToLangPrimaryRequestVNextLanguageEnum["GvIm"] = "gv-im";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ha"] = "ha";
    AttachToLangPrimaryRequestVNextLanguageEnum["HaGh"] = "ha-gh";
    AttachToLangPrimaryRequestVNextLanguageEnum["HaNe"] = "ha-ne";
    AttachToLangPrimaryRequestVNextLanguageEnum["HaNg"] = "ha-ng";
    AttachToLangPrimaryRequestVNextLanguageEnum["Haw"] = "haw";
    AttachToLangPrimaryRequestVNextLanguageEnum["HawUs"] = "haw-us";
    AttachToLangPrimaryRequestVNextLanguageEnum["He"] = "he";
    AttachToLangPrimaryRequestVNextLanguageEnum["Hi"] = "hi";
    AttachToLangPrimaryRequestVNextLanguageEnum["HiIn"] = "hi-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Hr"] = "hr";
    AttachToLangPrimaryRequestVNextLanguageEnum["HrBa"] = "hr-ba";
    AttachToLangPrimaryRequestVNextLanguageEnum["HrHr"] = "hr-hr";
    AttachToLangPrimaryRequestVNextLanguageEnum["Hsb"] = "hsb";
    AttachToLangPrimaryRequestVNextLanguageEnum["HsbDe"] = "hsb-de";
    AttachToLangPrimaryRequestVNextLanguageEnum["Hu"] = "hu";
    AttachToLangPrimaryRequestVNextLanguageEnum["HuHu"] = "hu-hu";
    AttachToLangPrimaryRequestVNextLanguageEnum["Hy"] = "hy";
    AttachToLangPrimaryRequestVNextLanguageEnum["HyAm"] = "hy-am";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ia"] = "ia";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ia001"] = "ia-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["Id"] = "id";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ig"] = "ig";
    AttachToLangPrimaryRequestVNextLanguageEnum["IgNg"] = "ig-ng";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ii"] = "ii";
    AttachToLangPrimaryRequestVNextLanguageEnum["IiCn"] = "ii-cn";
    AttachToLangPrimaryRequestVNextLanguageEnum["IdId"] = "id-id";
    AttachToLangPrimaryRequestVNextLanguageEnum["Is"] = "is";
    AttachToLangPrimaryRequestVNextLanguageEnum["IsIs"] = "is-is";
    AttachToLangPrimaryRequestVNextLanguageEnum["It"] = "it";
    AttachToLangPrimaryRequestVNextLanguageEnum["ItCh"] = "it-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["ItIt"] = "it-it";
    AttachToLangPrimaryRequestVNextLanguageEnum["ItSm"] = "it-sm";
    AttachToLangPrimaryRequestVNextLanguageEnum["ItVa"] = "it-va";
    AttachToLangPrimaryRequestVNextLanguageEnum["HeIl"] = "he-il";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ja"] = "ja";
    AttachToLangPrimaryRequestVNextLanguageEnum["JaJp"] = "ja-jp";
    AttachToLangPrimaryRequestVNextLanguageEnum["Jgo"] = "jgo";
    AttachToLangPrimaryRequestVNextLanguageEnum["JgoCm"] = "jgo-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Yi"] = "yi";
    AttachToLangPrimaryRequestVNextLanguageEnum["Yi001"] = "yi-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["Jmc"] = "jmc";
    AttachToLangPrimaryRequestVNextLanguageEnum["JmcTz"] = "jmc-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Jv"] = "jv";
    AttachToLangPrimaryRequestVNextLanguageEnum["JvId"] = "jv-id";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ka"] = "ka";
    AttachToLangPrimaryRequestVNextLanguageEnum["KaGe"] = "ka-ge";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kab"] = "kab";
    AttachToLangPrimaryRequestVNextLanguageEnum["KabDz"] = "kab-dz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kam"] = "kam";
    AttachToLangPrimaryRequestVNextLanguageEnum["KamKe"] = "kam-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kde"] = "kde";
    AttachToLangPrimaryRequestVNextLanguageEnum["KdeTz"] = "kde-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kea"] = "kea";
    AttachToLangPrimaryRequestVNextLanguageEnum["KeaCv"] = "kea-cv";
    AttachToLangPrimaryRequestVNextLanguageEnum["Khq"] = "khq";
    AttachToLangPrimaryRequestVNextLanguageEnum["KhqMl"] = "khq-ml";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ki"] = "ki";
    AttachToLangPrimaryRequestVNextLanguageEnum["KiKe"] = "ki-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kk"] = "kk";
    AttachToLangPrimaryRequestVNextLanguageEnum["KkKz"] = "kk-kz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kkj"] = "kkj";
    AttachToLangPrimaryRequestVNextLanguageEnum["KkjCm"] = "kkj-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kl"] = "kl";
    AttachToLangPrimaryRequestVNextLanguageEnum["KlGl"] = "kl-gl";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kln"] = "kln";
    AttachToLangPrimaryRequestVNextLanguageEnum["KlnKe"] = "kln-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Km"] = "km";
    AttachToLangPrimaryRequestVNextLanguageEnum["KmKh"] = "km-kh";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kn"] = "kn";
    AttachToLangPrimaryRequestVNextLanguageEnum["KnIn"] = "kn-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ko"] = "ko";
    AttachToLangPrimaryRequestVNextLanguageEnum["KoKp"] = "ko-kp";
    AttachToLangPrimaryRequestVNextLanguageEnum["KoKr"] = "ko-kr";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kok"] = "kok";
    AttachToLangPrimaryRequestVNextLanguageEnum["KokIn"] = "kok-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ks"] = "ks";
    AttachToLangPrimaryRequestVNextLanguageEnum["KsIn"] = "ks-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ksb"] = "ksb";
    AttachToLangPrimaryRequestVNextLanguageEnum["KsbTz"] = "ksb-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ksf"] = "ksf";
    AttachToLangPrimaryRequestVNextLanguageEnum["KsfCm"] = "ksf-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ksh"] = "ksh";
    AttachToLangPrimaryRequestVNextLanguageEnum["KshDe"] = "ksh-de";
    AttachToLangPrimaryRequestVNextLanguageEnum["Kw"] = "kw";
    AttachToLangPrimaryRequestVNextLanguageEnum["KwGb"] = "kw-gb";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ku"] = "ku";
    AttachToLangPrimaryRequestVNextLanguageEnum["KuTr"] = "ku-tr";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ky"] = "ky";
    AttachToLangPrimaryRequestVNextLanguageEnum["KyKg"] = "ky-kg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lag"] = "lag";
    AttachToLangPrimaryRequestVNextLanguageEnum["LagTz"] = "lag-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lb"] = "lb";
    AttachToLangPrimaryRequestVNextLanguageEnum["LbLu"] = "lb-lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lg"] = "lg";
    AttachToLangPrimaryRequestVNextLanguageEnum["LgUg"] = "lg-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lkt"] = "lkt";
    AttachToLangPrimaryRequestVNextLanguageEnum["LktUs"] = "lkt-us";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ln"] = "ln";
    AttachToLangPrimaryRequestVNextLanguageEnum["LnAo"] = "ln-ao";
    AttachToLangPrimaryRequestVNextLanguageEnum["LnCd"] = "ln-cd";
    AttachToLangPrimaryRequestVNextLanguageEnum["LnCf"] = "ln-cf";
    AttachToLangPrimaryRequestVNextLanguageEnum["LnCg"] = "ln-cg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lo"] = "lo";
    AttachToLangPrimaryRequestVNextLanguageEnum["LoLa"] = "lo-la";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lrc"] = "lrc";
    AttachToLangPrimaryRequestVNextLanguageEnum["LrcIq"] = "lrc-iq";
    AttachToLangPrimaryRequestVNextLanguageEnum["LrcIr"] = "lrc-ir";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lt"] = "lt";
    AttachToLangPrimaryRequestVNextLanguageEnum["LtLt"] = "lt-lt";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lu"] = "lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["LuCd"] = "lu-cd";
    AttachToLangPrimaryRequestVNextLanguageEnum["Luo"] = "luo";
    AttachToLangPrimaryRequestVNextLanguageEnum["LuoKe"] = "luo-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Luy"] = "luy";
    AttachToLangPrimaryRequestVNextLanguageEnum["LuyKe"] = "luy-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Lv"] = "lv";
    AttachToLangPrimaryRequestVNextLanguageEnum["LvLv"] = "lv-lv";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mai"] = "mai";
    AttachToLangPrimaryRequestVNextLanguageEnum["MaiIn"] = "mai-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mas"] = "mas";
    AttachToLangPrimaryRequestVNextLanguageEnum["MasKe"] = "mas-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["MasTz"] = "mas-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mer"] = "mer";
    AttachToLangPrimaryRequestVNextLanguageEnum["MerKe"] = "mer-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mfe"] = "mfe";
    AttachToLangPrimaryRequestVNextLanguageEnum["MfeMu"] = "mfe-mu";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mg"] = "mg";
    AttachToLangPrimaryRequestVNextLanguageEnum["MgMg"] = "mg-mg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mgh"] = "mgh";
    AttachToLangPrimaryRequestVNextLanguageEnum["MghMz"] = "mgh-mz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mgo"] = "mgo";
    AttachToLangPrimaryRequestVNextLanguageEnum["MgoCm"] = "mgo-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mi"] = "mi";
    AttachToLangPrimaryRequestVNextLanguageEnum["MiNz"] = "mi-nz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mk"] = "mk";
    AttachToLangPrimaryRequestVNextLanguageEnum["MkMk"] = "mk-mk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ml"] = "ml";
    AttachToLangPrimaryRequestVNextLanguageEnum["MlIn"] = "ml-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mn"] = "mn";
    AttachToLangPrimaryRequestVNextLanguageEnum["MnMn"] = "mn-mn";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mni"] = "mni";
    AttachToLangPrimaryRequestVNextLanguageEnum["MniIn"] = "mni-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mr"] = "mr";
    AttachToLangPrimaryRequestVNextLanguageEnum["MrIn"] = "mr-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ms"] = "ms";
    AttachToLangPrimaryRequestVNextLanguageEnum["MsBn"] = "ms-bn";
    AttachToLangPrimaryRequestVNextLanguageEnum["MsId"] = "ms-id";
    AttachToLangPrimaryRequestVNextLanguageEnum["MsMy"] = "ms-my";
    AttachToLangPrimaryRequestVNextLanguageEnum["MsSg"] = "ms-sg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mt"] = "mt";
    AttachToLangPrimaryRequestVNextLanguageEnum["MtMt"] = "mt-mt";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mua"] = "mua";
    AttachToLangPrimaryRequestVNextLanguageEnum["MuaCm"] = "mua-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["My"] = "my";
    AttachToLangPrimaryRequestVNextLanguageEnum["MyMm"] = "my-mm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Mzn"] = "mzn";
    AttachToLangPrimaryRequestVNextLanguageEnum["MznIr"] = "mzn-ir";
    AttachToLangPrimaryRequestVNextLanguageEnum["Naq"] = "naq";
    AttachToLangPrimaryRequestVNextLanguageEnum["NaqNa"] = "naq-na";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nb"] = "nb";
    AttachToLangPrimaryRequestVNextLanguageEnum["NbNo"] = "nb-no";
    AttachToLangPrimaryRequestVNextLanguageEnum["NbSj"] = "nb-sj";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nd"] = "nd";
    AttachToLangPrimaryRequestVNextLanguageEnum["NdZw"] = "nd-zw";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nds"] = "nds";
    AttachToLangPrimaryRequestVNextLanguageEnum["NdsDe"] = "nds-de";
    AttachToLangPrimaryRequestVNextLanguageEnum["NdsNl"] = "nds-nl";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ne"] = "ne";
    AttachToLangPrimaryRequestVNextLanguageEnum["NeIn"] = "ne-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["NeNp"] = "ne-np";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nl"] = "nl";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlAw"] = "nl-aw";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlBe"] = "nl-be";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlCh"] = "nl-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlBq"] = "nl-bq";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlCw"] = "nl-cw";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlLu"] = "nl-lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlNl"] = "nl-nl";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlSr"] = "nl-sr";
    AttachToLangPrimaryRequestVNextLanguageEnum["NlSx"] = "nl-sx";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nmg"] = "nmg";
    AttachToLangPrimaryRequestVNextLanguageEnum["NmgCm"] = "nmg-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nn"] = "nn";
    AttachToLangPrimaryRequestVNextLanguageEnum["NnNo"] = "nn-no";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nnh"] = "nnh";
    AttachToLangPrimaryRequestVNextLanguageEnum["NnhCm"] = "nnh-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["No"] = "no";
    AttachToLangPrimaryRequestVNextLanguageEnum["NoNo"] = "no-no";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nus"] = "nus";
    AttachToLangPrimaryRequestVNextLanguageEnum["NusSs"] = "nus-ss";
    AttachToLangPrimaryRequestVNextLanguageEnum["Nyn"] = "nyn";
    AttachToLangPrimaryRequestVNextLanguageEnum["NynUg"] = "nyn-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["Om"] = "om";
    AttachToLangPrimaryRequestVNextLanguageEnum["OmEt"] = "om-et";
    AttachToLangPrimaryRequestVNextLanguageEnum["OmKe"] = "om-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Or"] = "or";
    AttachToLangPrimaryRequestVNextLanguageEnum["OrIn"] = "or-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Os"] = "os";
    AttachToLangPrimaryRequestVNextLanguageEnum["OsGe"] = "os-ge";
    AttachToLangPrimaryRequestVNextLanguageEnum["OsRu"] = "os-ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["Pa"] = "pa";
    AttachToLangPrimaryRequestVNextLanguageEnum["PaIn"] = "pa-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["PaPk"] = "pa-pk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Pcm"] = "pcm";
    AttachToLangPrimaryRequestVNextLanguageEnum["PcmNg"] = "pcm-ng";
    AttachToLangPrimaryRequestVNextLanguageEnum["Pl"] = "pl";
    AttachToLangPrimaryRequestVNextLanguageEnum["PlPl"] = "pl-pl";
    AttachToLangPrimaryRequestVNextLanguageEnum["Prg"] = "prg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Prg001"] = "prg-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ps"] = "ps";
    AttachToLangPrimaryRequestVNextLanguageEnum["PsAf"] = "ps-af";
    AttachToLangPrimaryRequestVNextLanguageEnum["PsPk"] = "ps-pk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Pt"] = "pt";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtAo"] = "pt-ao";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtBr"] = "pt-br";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtCh"] = "pt-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtCv"] = "pt-cv";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtGq"] = "pt-gq";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtGw"] = "pt-gw";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtLu"] = "pt-lu";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtMo"] = "pt-mo";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtMz"] = "pt-mz";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtPt"] = "pt-pt";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtSt"] = "pt-st";
    AttachToLangPrimaryRequestVNextLanguageEnum["PtTl"] = "pt-tl";
    AttachToLangPrimaryRequestVNextLanguageEnum["Qu"] = "qu";
    AttachToLangPrimaryRequestVNextLanguageEnum["QuBo"] = "qu-bo";
    AttachToLangPrimaryRequestVNextLanguageEnum["QuEc"] = "qu-ec";
    AttachToLangPrimaryRequestVNextLanguageEnum["QuPe"] = "qu-pe";
    AttachToLangPrimaryRequestVNextLanguageEnum["Rm"] = "rm";
    AttachToLangPrimaryRequestVNextLanguageEnum["RmCh"] = "rm-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["Rn"] = "rn";
    AttachToLangPrimaryRequestVNextLanguageEnum["RnBi"] = "rn-bi";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ro"] = "ro";
    AttachToLangPrimaryRequestVNextLanguageEnum["RoMd"] = "ro-md";
    AttachToLangPrimaryRequestVNextLanguageEnum["RoRo"] = "ro-ro";
    AttachToLangPrimaryRequestVNextLanguageEnum["Rof"] = "rof";
    AttachToLangPrimaryRequestVNextLanguageEnum["RofTz"] = "rof-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ru"] = "ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["RuBy"] = "ru-by";
    AttachToLangPrimaryRequestVNextLanguageEnum["RuKg"] = "ru-kg";
    AttachToLangPrimaryRequestVNextLanguageEnum["RuKz"] = "ru-kz";
    AttachToLangPrimaryRequestVNextLanguageEnum["RuMd"] = "ru-md";
    AttachToLangPrimaryRequestVNextLanguageEnum["RuRu"] = "ru-ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["RuUa"] = "ru-ua";
    AttachToLangPrimaryRequestVNextLanguageEnum["Rw"] = "rw";
    AttachToLangPrimaryRequestVNextLanguageEnum["RwRw"] = "rw-rw";
    AttachToLangPrimaryRequestVNextLanguageEnum["Rwk"] = "rwk";
    AttachToLangPrimaryRequestVNextLanguageEnum["RwkTz"] = "rwk-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sa"] = "sa";
    AttachToLangPrimaryRequestVNextLanguageEnum["SaIn"] = "sa-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sah"] = "sah";
    AttachToLangPrimaryRequestVNextLanguageEnum["SahRu"] = "sah-ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["Saq"] = "saq";
    AttachToLangPrimaryRequestVNextLanguageEnum["SaqKe"] = "saq-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sat"] = "sat";
    AttachToLangPrimaryRequestVNextLanguageEnum["SatIn"] = "sat-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sbp"] = "sbp";
    AttachToLangPrimaryRequestVNextLanguageEnum["SbpTz"] = "sbp-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sd"] = "sd";
    AttachToLangPrimaryRequestVNextLanguageEnum["SdIn"] = "sd-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["SdPk"] = "sd-pk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Se"] = "se";
    AttachToLangPrimaryRequestVNextLanguageEnum["SeFi"] = "se-fi";
    AttachToLangPrimaryRequestVNextLanguageEnum["SeNo"] = "se-no";
    AttachToLangPrimaryRequestVNextLanguageEnum["SeSe"] = "se-se";
    AttachToLangPrimaryRequestVNextLanguageEnum["Seh"] = "seh";
    AttachToLangPrimaryRequestVNextLanguageEnum["SehMz"] = "seh-mz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ses"] = "ses";
    AttachToLangPrimaryRequestVNextLanguageEnum["SesMl"] = "ses-ml";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sg"] = "sg";
    AttachToLangPrimaryRequestVNextLanguageEnum["SgCf"] = "sg-cf";
    AttachToLangPrimaryRequestVNextLanguageEnum["Shi"] = "shi";
    AttachToLangPrimaryRequestVNextLanguageEnum["ShiMa"] = "shi-ma";
    AttachToLangPrimaryRequestVNextLanguageEnum["Si"] = "si";
    AttachToLangPrimaryRequestVNextLanguageEnum["SiLk"] = "si-lk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sk"] = "sk";
    AttachToLangPrimaryRequestVNextLanguageEnum["SkSk"] = "sk-sk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sl"] = "sl";
    AttachToLangPrimaryRequestVNextLanguageEnum["SlSi"] = "sl-si";
    AttachToLangPrimaryRequestVNextLanguageEnum["Smn"] = "smn";
    AttachToLangPrimaryRequestVNextLanguageEnum["SmnFi"] = "smn-fi";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sn"] = "sn";
    AttachToLangPrimaryRequestVNextLanguageEnum["SnZw"] = "sn-zw";
    AttachToLangPrimaryRequestVNextLanguageEnum["So"] = "so";
    AttachToLangPrimaryRequestVNextLanguageEnum["SoDj"] = "so-dj";
    AttachToLangPrimaryRequestVNextLanguageEnum["SoEt"] = "so-et";
    AttachToLangPrimaryRequestVNextLanguageEnum["SoKe"] = "so-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["SoSo"] = "so-so";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sq"] = "sq";
    AttachToLangPrimaryRequestVNextLanguageEnum["SqAl"] = "sq-al";
    AttachToLangPrimaryRequestVNextLanguageEnum["SqMk"] = "sq-mk";
    AttachToLangPrimaryRequestVNextLanguageEnum["SqXk"] = "sq-xk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sr"] = "sr";
    AttachToLangPrimaryRequestVNextLanguageEnum["SrBa"] = "sr-ba";
    AttachToLangPrimaryRequestVNextLanguageEnum["SrCs"] = "sr-cs";
    AttachToLangPrimaryRequestVNextLanguageEnum["SrMe"] = "sr-me";
    AttachToLangPrimaryRequestVNextLanguageEnum["SrRs"] = "sr-rs";
    AttachToLangPrimaryRequestVNextLanguageEnum["SrXk"] = "sr-xk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Su"] = "su";
    AttachToLangPrimaryRequestVNextLanguageEnum["SuId"] = "su-id";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sv"] = "sv";
    AttachToLangPrimaryRequestVNextLanguageEnum["SvAx"] = "sv-ax";
    AttachToLangPrimaryRequestVNextLanguageEnum["SvFi"] = "sv-fi";
    AttachToLangPrimaryRequestVNextLanguageEnum["SvSe"] = "sv-se";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sw"] = "sw";
    AttachToLangPrimaryRequestVNextLanguageEnum["SwCd"] = "sw-cd";
    AttachToLangPrimaryRequestVNextLanguageEnum["SwKe"] = "sw-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["SwTz"] = "sw-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["SwUg"] = "sw-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["Sy"] = "sy";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ta"] = "ta";
    AttachToLangPrimaryRequestVNextLanguageEnum["TaIn"] = "ta-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["TaLk"] = "ta-lk";
    AttachToLangPrimaryRequestVNextLanguageEnum["TaMy"] = "ta-my";
    AttachToLangPrimaryRequestVNextLanguageEnum["TaSg"] = "ta-sg";
    AttachToLangPrimaryRequestVNextLanguageEnum["Te"] = "te";
    AttachToLangPrimaryRequestVNextLanguageEnum["TeIn"] = "te-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["Teo"] = "teo";
    AttachToLangPrimaryRequestVNextLanguageEnum["TeoKe"] = "teo-ke";
    AttachToLangPrimaryRequestVNextLanguageEnum["TeoUg"] = "teo-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["Tg"] = "tg";
    AttachToLangPrimaryRequestVNextLanguageEnum["TgTj"] = "tg-tj";
    AttachToLangPrimaryRequestVNextLanguageEnum["Th"] = "th";
    AttachToLangPrimaryRequestVNextLanguageEnum["ThTh"] = "th-th";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ti"] = "ti";
    AttachToLangPrimaryRequestVNextLanguageEnum["TiEr"] = "ti-er";
    AttachToLangPrimaryRequestVNextLanguageEnum["TiEt"] = "ti-et";
    AttachToLangPrimaryRequestVNextLanguageEnum["Tk"] = "tk";
    AttachToLangPrimaryRequestVNextLanguageEnum["TkTm"] = "tk-tm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Tl"] = "tl";
    AttachToLangPrimaryRequestVNextLanguageEnum["To"] = "to";
    AttachToLangPrimaryRequestVNextLanguageEnum["ToTo"] = "to-to";
    AttachToLangPrimaryRequestVNextLanguageEnum["Tr"] = "tr";
    AttachToLangPrimaryRequestVNextLanguageEnum["TrCy"] = "tr-cy";
    AttachToLangPrimaryRequestVNextLanguageEnum["TrTr"] = "tr-tr";
    AttachToLangPrimaryRequestVNextLanguageEnum["Tt"] = "tt";
    AttachToLangPrimaryRequestVNextLanguageEnum["TtRu"] = "tt-ru";
    AttachToLangPrimaryRequestVNextLanguageEnum["Twq"] = "twq";
    AttachToLangPrimaryRequestVNextLanguageEnum["TwqNe"] = "twq-ne";
    AttachToLangPrimaryRequestVNextLanguageEnum["Tzm"] = "tzm";
    AttachToLangPrimaryRequestVNextLanguageEnum["TzmMa"] = "tzm-ma";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ug"] = "ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["UgCn"] = "ug-cn";
    AttachToLangPrimaryRequestVNextLanguageEnum["Uk"] = "uk";
    AttachToLangPrimaryRequestVNextLanguageEnum["UkUa"] = "uk-ua";
    AttachToLangPrimaryRequestVNextLanguageEnum["Ur"] = "ur";
    AttachToLangPrimaryRequestVNextLanguageEnum["UrIn"] = "ur-in";
    AttachToLangPrimaryRequestVNextLanguageEnum["UrPk"] = "ur-pk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Uz"] = "uz";
    AttachToLangPrimaryRequestVNextLanguageEnum["UzAf"] = "uz-af";
    AttachToLangPrimaryRequestVNextLanguageEnum["UzUz"] = "uz-uz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Vai"] = "vai";
    AttachToLangPrimaryRequestVNextLanguageEnum["VaiLr"] = "vai-lr";
    AttachToLangPrimaryRequestVNextLanguageEnum["Vi"] = "vi";
    AttachToLangPrimaryRequestVNextLanguageEnum["ViVn"] = "vi-vn";
    AttachToLangPrimaryRequestVNextLanguageEnum["Vo"] = "vo";
    AttachToLangPrimaryRequestVNextLanguageEnum["Vo001"] = "vo-001";
    AttachToLangPrimaryRequestVNextLanguageEnum["Vun"] = "vun";
    AttachToLangPrimaryRequestVNextLanguageEnum["VunTz"] = "vun-tz";
    AttachToLangPrimaryRequestVNextLanguageEnum["Wae"] = "wae";
    AttachToLangPrimaryRequestVNextLanguageEnum["WaeCh"] = "wae-ch";
    AttachToLangPrimaryRequestVNextLanguageEnum["Wo"] = "wo";
    AttachToLangPrimaryRequestVNextLanguageEnum["WoSn"] = "wo-sn";
    AttachToLangPrimaryRequestVNextLanguageEnum["Xh"] = "xh";
    AttachToLangPrimaryRequestVNextLanguageEnum["XhZa"] = "xh-za";
    AttachToLangPrimaryRequestVNextLanguageEnum["Xog"] = "xog";
    AttachToLangPrimaryRequestVNextLanguageEnum["XogUg"] = "xog-ug";
    AttachToLangPrimaryRequestVNextLanguageEnum["Yav"] = "yav";
    AttachToLangPrimaryRequestVNextLanguageEnum["YavCm"] = "yav-cm";
    AttachToLangPrimaryRequestVNextLanguageEnum["Yo"] = "yo";
    AttachToLangPrimaryRequestVNextLanguageEnum["YoBj"] = "yo-bj";
    AttachToLangPrimaryRequestVNextLanguageEnum["YoNg"] = "yo-ng";
    AttachToLangPrimaryRequestVNextLanguageEnum["Yue"] = "yue";
    AttachToLangPrimaryRequestVNextLanguageEnum["YueCn"] = "yue-cn";
    AttachToLangPrimaryRequestVNextLanguageEnum["YueHk"] = "yue-hk";
    AttachToLangPrimaryRequestVNextLanguageEnum["Zgh"] = "zgh";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZghMa"] = "zgh-ma";
    AttachToLangPrimaryRequestVNextLanguageEnum["Zh"] = "zh";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhCn"] = "zh-cn";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhHk"] = "zh-hk";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhMo"] = "zh-mo";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhSg"] = "zh-sg";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhTw"] = "zh-tw";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhHans"] = "zh-hans";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZhHant"] = "zh-hant";
    AttachToLangPrimaryRequestVNextLanguageEnum["Zu"] = "zu";
    AttachToLangPrimaryRequestVNextLanguageEnum["ZuZa"] = "zu-za";
})(AttachToLangPrimaryRequestVNextLanguageEnum = exports.AttachToLangPrimaryRequestVNextLanguageEnum || (exports.AttachToLangPrimaryRequestVNextLanguageEnum = {}));
