// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/models/BatchResponseBlogPostWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseBlogPostWithErrorsStatusEnum = exports.BatchResponseBlogPostWithErrors = void 0;
class BatchResponseBlogPostWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseBlogPostWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseBlogPostWithErrors = BatchResponseBlogPostWithErrors;
BatchResponseBlogPostWithErrors.discriminator = undefined;
BatchResponseBlogPostWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<BlogPost>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseBlogPostWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseBlogPostWithErrorsStatusEnum;
(function (BatchResponseBlogPostWithErrorsStatusEnum) {
    BatchResponseBlogPostWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseBlogPostWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseBlogPostWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseBlogPostWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseBlogPostWithErrorsStatusEnum = exports.BatchResponseBlogPostWithErrorsStatusEnum || (exports.BatchResponseBlogPostWithErrorsStatusEnum = {}));
