// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/models/BatchResponseBlogPost.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseBlogPostStatusEnum = exports.BatchResponseBlogPost = void 0;
class BatchResponseBlogPost {
    static getAttributeTypeMap() {
        return BatchResponseBlogPost.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseBlogPost = BatchResponseBlogPost;
BatchResponseBlogPost.discriminator = undefined;
BatchResponseBlogPost.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<BlogPost>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseBlogPostStatusEnum",
        "format": ""
    }
];
var BatchResponseBlogPostStatusEnum;
(function (BatchResponseBlogPostStatusEnum) {
    BatchResponseBlogPostStatusEnum["Pending"] = "PENDING";
    BatchResponseBlogPostStatusEnum["Processing"] = "PROCESSING";
    BatchResponseBlogPostStatusEnum["Canceled"] = "CANCELED";
    BatchResponseBlogPostStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseBlogPostStatusEnum = exports.BatchResponseBlogPostStatusEnum || (exports.BatchResponseBlogPostStatusEnum = {}));
