// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/models/BlogPost.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlogPostCurrentStateEnum = exports.BlogPostAbStatusEnum = exports.BlogPostContentTypeCategoryEnum = exports.BlogPostLanguageEnum = exports.BlogPost = void 0;
class BlogPost {
    static getAttributeTypeMap() {
        return BlogPost.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BlogPost = BlogPost;
BlogPost.discriminator = undefined;
BlogPost.attributeTypeMap = [
    {
        "name": "publishDate",
        "baseName": "publishDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "BlogPostLanguageEnum",
        "format": ""
    },
    {
        "name": "enableLayoutStylesheets",
        "baseName": "enableLayoutStylesheets",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "metaDescription",
        "baseName": "metaDescription",
        "type": "string",
        "format": ""
    },
    {
        "name": "attachedStylesheets",
        "baseName": "attachedStylesheets",
        "type": "Array<{ [key: string]: any; }>",
        "format": ""
    },
    {
        "name": "password",
        "baseName": "password",
        "type": "string",
        "format": ""
    },
    {
        "name": "htmlTitle",
        "baseName": "htmlTitle",
        "type": "string",
        "format": ""
    },
    {
        "name": "publishImmediately",
        "baseName": "publishImmediately",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "translations",
        "baseName": "translations",
        "type": "{ [key: string]: ContentLanguageVariation; }",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "state",
        "baseName": "state",
        "type": "string",
        "format": ""
    },
    {
        "name": "slug",
        "baseName": "slug",
        "type": "string",
        "format": ""
    },
    {
        "name": "createdById",
        "baseName": "createdById",
        "type": "string",
        "format": ""
    },
    {
        "name": "rssBody",
        "baseName": "rssBody",
        "type": "string",
        "format": ""
    },
    {
        "name": "currentlyPublished",
        "baseName": "currentlyPublished",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "archivedInDashboard",
        "baseName": "archivedInDashboard",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "contentTypeCategory",
        "baseName": "contentTypeCategory",
        "type": "BlogPostContentTypeCategoryEnum",
        "format": ""
    },
    {
        "name": "mabExperimentId",
        "baseName": "mabExperimentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "updatedById",
        "baseName": "updatedById",
        "type": "string",
        "format": ""
    },
    {
        "name": "translatedFromId",
        "baseName": "translatedFromId",
        "type": "string",
        "format": ""
    },
    {
        "name": "folderId",
        "baseName": "folderId",
        "type": "string",
        "format": ""
    },
    {
        "name": "widgetContainers",
        "baseName": "widgetContainers",
        "type": "{ [key: string]: any; }",
        "format": ""
    },
    {
        "name": "pageExpiryRedirectId",
        "baseName": "pageExpiryRedirectId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "dynamicPageDataSourceType",
        "baseName": "dynamicPageDataSourceType",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "featuredImage",
        "baseName": "featuredImage",
        "type": "string",
        "format": ""
    },
    {
        "name": "authorName",
        "baseName": "authorName",
        "type": "string",
        "format": ""
    },
    {
        "name": "domain",
        "baseName": "domain",
        "type": "string",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "dynamicPageHubDbTableId",
        "baseName": "dynamicPageHubDbTableId",
        "type": "string",
        "format": ""
    },
    {
        "name": "campaign",
        "baseName": "campaign",
        "type": "string",
        "format": ""
    },
    {
        "name": "dynamicPageDataSourceId",
        "baseName": "dynamicPageDataSourceId",
        "type": "string",
        "format": ""
    },
    {
        "name": "enableDomainStylesheets",
        "baseName": "enableDomainStylesheets",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "includeDefaultCustomCss",
        "baseName": "includeDefaultCustomCss",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "layoutSections",
        "baseName": "layoutSections",
        "type": "{ [key: string]: LayoutSection; }",
        "format": ""
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "footerHtml",
        "baseName": "footerHtml",
        "type": "string",
        "format": ""
    },
    {
        "name": "tagIds",
        "baseName": "tagIds",
        "type": "Array<number>",
        "format": "int64"
    },
    {
        "name": "widgets",
        "baseName": "widgets",
        "type": "{ [key: string]: any; }",
        "format": ""
    },
    {
        "name": "postSummary",
        "baseName": "postSummary",
        "type": "string",
        "format": ""
    },
    {
        "name": "headHtml",
        "baseName": "headHtml",
        "type": "string",
        "format": ""
    },
    {
        "name": "pageExpiryRedirectUrl",
        "baseName": "pageExpiryRedirectUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "abStatus",
        "baseName": "abStatus",
        "type": "BlogPostAbStatusEnum",
        "format": ""
    },
    {
        "name": "useFeaturedImage",
        "baseName": "useFeaturedImage",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "abTestId",
        "baseName": "abTestId",
        "type": "string",
        "format": ""
    },
    {
        "name": "featuredImageAltText",
        "baseName": "featuredImageAltText",
        "type": "string",
        "format": ""
    },
    {
        "name": "blogAuthorId",
        "baseName": "blogAuthorId",
        "type": "string",
        "format": ""
    },
    {
        "name": "contentGroupId",
        "baseName": "contentGroupId",
        "type": "string",
        "format": ""
    },
    {
        "name": "rssSummary",
        "baseName": "rssSummary",
        "type": "string",
        "format": ""
    },
    {
        "name": "pageExpiryEnabled",
        "baseName": "pageExpiryEnabled",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    },
    {
        "name": "enableGoogleAmpOutputOverride",
        "baseName": "enableGoogleAmpOutputOverride",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "publicAccessRules",
        "baseName": "publicAccessRules",
        "type": "Array<any>",
        "format": ""
    },
    {
        "name": "archivedAt",
        "baseName": "archivedAt",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "postBody",
        "baseName": "postBody",
        "type": "string",
        "format": ""
    },
    {
        "name": "themeSettingsValues",
        "baseName": "themeSettingsValues",
        "type": "{ [key: string]: any; }",
        "format": ""
    },
    {
        "name": "pageExpiryDate",
        "baseName": "pageExpiryDate",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "publicAccessRulesEnabled",
        "baseName": "publicAccessRulesEnabled",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "currentState",
        "baseName": "currentState",
        "type": "BlogPostCurrentStateEnum",
        "format": ""
    },
    {
        "name": "categoryId",
        "baseName": "categoryId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "linkRelCanonicalUrl",
        "baseName": "linkRelCanonicalUrl",
        "type": "string",
        "format": ""
    }
];
var BlogPostLanguageEnum;
(function (BlogPostLanguageEnum) {
    BlogPostLanguageEnum["Af"] = "af";
    BlogPostLanguageEnum["AfNa"] = "af-na";
    BlogPostLanguageEnum["AfZa"] = "af-za";
    BlogPostLanguageEnum["Agq"] = "agq";
    BlogPostLanguageEnum["AgqCm"] = "agq-cm";
    BlogPostLanguageEnum["Ak"] = "ak";
    BlogPostLanguageEnum["AkGh"] = "ak-gh";
    BlogPostLanguageEnum["Am"] = "am";
    BlogPostLanguageEnum["AmEt"] = "am-et";
    BlogPostLanguageEnum["Ar"] = "ar";
    BlogPostLanguageEnum["Ar001"] = "ar-001";
    BlogPostLanguageEnum["ArAe"] = "ar-ae";
    BlogPostLanguageEnum["ArBh"] = "ar-bh";
    BlogPostLanguageEnum["ArDj"] = "ar-dj";
    BlogPostLanguageEnum["ArDz"] = "ar-dz";
    BlogPostLanguageEnum["ArEg"] = "ar-eg";
    BlogPostLanguageEnum["ArEh"] = "ar-eh";
    BlogPostLanguageEnum["ArEr"] = "ar-er";
    BlogPostLanguageEnum["ArIl"] = "ar-il";
    BlogPostLanguageEnum["ArIq"] = "ar-iq";
    BlogPostLanguageEnum["ArJo"] = "ar-jo";
    BlogPostLanguageEnum["ArKm"] = "ar-km";
    BlogPostLanguageEnum["ArKw"] = "ar-kw";
    BlogPostLanguageEnum["ArLb"] = "ar-lb";
    BlogPostLanguageEnum["ArLy"] = "ar-ly";
    BlogPostLanguageEnum["ArMa"] = "ar-ma";
    BlogPostLanguageEnum["ArMr"] = "ar-mr";
    BlogPostLanguageEnum["ArOm"] = "ar-om";
    BlogPostLanguageEnum["ArPs"] = "ar-ps";
    BlogPostLanguageEnum["ArQa"] = "ar-qa";
    BlogPostLanguageEnum["ArSa"] = "ar-sa";
    BlogPostLanguageEnum["ArSd"] = "ar-sd";
    BlogPostLanguageEnum["ArSo"] = "ar-so";
    BlogPostLanguageEnum["ArSs"] = "ar-ss";
    BlogPostLanguageEnum["ArSy"] = "ar-sy";
    BlogPostLanguageEnum["ArTd"] = "ar-td";
    BlogPostLanguageEnum["ArTn"] = "ar-tn";
    BlogPostLanguageEnum["ArYe"] = "ar-ye";
    BlogPostLanguageEnum["As"] = "as";
    BlogPostLanguageEnum["AsIn"] = "as-in";
    BlogPostLanguageEnum["Asa"] = "asa";
    BlogPostLanguageEnum["AsaTz"] = "asa-tz";
    BlogPostLanguageEnum["Ast"] = "ast";
    BlogPostLanguageEnum["AstEs"] = "ast-es";
    BlogPostLanguageEnum["Az"] = "az";
    BlogPostLanguageEnum["AzAz"] = "az-az";
    BlogPostLanguageEnum["Bas"] = "bas";
    BlogPostLanguageEnum["BasCm"] = "bas-cm";
    BlogPostLanguageEnum["Be"] = "be";
    BlogPostLanguageEnum["BeBy"] = "be-by";
    BlogPostLanguageEnum["Bem"] = "bem";
    BlogPostLanguageEnum["BemZm"] = "bem-zm";
    BlogPostLanguageEnum["Bez"] = "bez";
    BlogPostLanguageEnum["BezTz"] = "bez-tz";
    BlogPostLanguageEnum["Bg"] = "bg";
    BlogPostLanguageEnum["BgBg"] = "bg-bg";
    BlogPostLanguageEnum["Bm"] = "bm";
    BlogPostLanguageEnum["BmMl"] = "bm-ml";
    BlogPostLanguageEnum["Bn"] = "bn";
    BlogPostLanguageEnum["BnBd"] = "bn-bd";
    BlogPostLanguageEnum["BnIn"] = "bn-in";
    BlogPostLanguageEnum["Bo"] = "bo";
    BlogPostLanguageEnum["BoCn"] = "bo-cn";
    BlogPostLanguageEnum["BoIn"] = "bo-in";
    BlogPostLanguageEnum["Br"] = "br";
    BlogPostLanguageEnum["BrFr"] = "br-fr";
    BlogPostLanguageEnum["Brx"] = "brx";
    BlogPostLanguageEnum["BrxIn"] = "brx-in";
    BlogPostLanguageEnum["Bs"] = "bs";
    BlogPostLanguageEnum["BsBa"] = "bs-ba";
    BlogPostLanguageEnum["Ca"] = "ca";
    BlogPostLanguageEnum["CaAd"] = "ca-ad";
    BlogPostLanguageEnum["CaEs"] = "ca-es";
    BlogPostLanguageEnum["CaFr"] = "ca-fr";
    BlogPostLanguageEnum["CaIt"] = "ca-it";
    BlogPostLanguageEnum["Ccp"] = "ccp";
    BlogPostLanguageEnum["CcpBd"] = "ccp-bd";
    BlogPostLanguageEnum["CcpIn"] = "ccp-in";
    BlogPostLanguageEnum["Ce"] = "ce";
    BlogPostLanguageEnum["CeRu"] = "ce-ru";
    BlogPostLanguageEnum["Ceb"] = "ceb";
    BlogPostLanguageEnum["CebPh"] = "ceb-ph";
    BlogPostLanguageEnum["Cgg"] = "cgg";
    BlogPostLanguageEnum["CggUg"] = "cgg-ug";
    BlogPostLanguageEnum["Chr"] = "chr";
    BlogPostLanguageEnum["ChrUs"] = "chr-us";
    BlogPostLanguageEnum["Ckb"] = "ckb";
    BlogPostLanguageEnum["CkbIq"] = "ckb-iq";
    BlogPostLanguageEnum["CkbIr"] = "ckb-ir";
    BlogPostLanguageEnum["Cs"] = "cs";
    BlogPostLanguageEnum["CsCz"] = "cs-cz";
    BlogPostLanguageEnum["Cu"] = "cu";
    BlogPostLanguageEnum["CuRu"] = "cu-ru";
    BlogPostLanguageEnum["Cy"] = "cy";
    BlogPostLanguageEnum["CyGb"] = "cy-gb";
    BlogPostLanguageEnum["Da"] = "da";
    BlogPostLanguageEnum["DaDk"] = "da-dk";
    BlogPostLanguageEnum["DaGl"] = "da-gl";
    BlogPostLanguageEnum["Dav"] = "dav";
    BlogPostLanguageEnum["DavKe"] = "dav-ke";
    BlogPostLanguageEnum["De"] = "de";
    BlogPostLanguageEnum["DeAt"] = "de-at";
    BlogPostLanguageEnum["DeBe"] = "de-be";
    BlogPostLanguageEnum["DeCh"] = "de-ch";
    BlogPostLanguageEnum["DeDe"] = "de-de";
    BlogPostLanguageEnum["DeGr"] = "de-gr";
    BlogPostLanguageEnum["DeIt"] = "de-it";
    BlogPostLanguageEnum["DeLi"] = "de-li";
    BlogPostLanguageEnum["DeLu"] = "de-lu";
    BlogPostLanguageEnum["Dje"] = "dje";
    BlogPostLanguageEnum["DjeNe"] = "dje-ne";
    BlogPostLanguageEnum["Doi"] = "doi";
    BlogPostLanguageEnum["DoiIn"] = "doi-in";
    BlogPostLanguageEnum["Dsb"] = "dsb";
    BlogPostLanguageEnum["DsbDe"] = "dsb-de";
    BlogPostLanguageEnum["Dua"] = "dua";
    BlogPostLanguageEnum["DuaCm"] = "dua-cm";
    BlogPostLanguageEnum["Dyo"] = "dyo";
    BlogPostLanguageEnum["DyoSn"] = "dyo-sn";
    BlogPostLanguageEnum["Dz"] = "dz";
    BlogPostLanguageEnum["DzBt"] = "dz-bt";
    BlogPostLanguageEnum["Ebu"] = "ebu";
    BlogPostLanguageEnum["EbuKe"] = "ebu-ke";
    BlogPostLanguageEnum["Ee"] = "ee";
    BlogPostLanguageEnum["EeGh"] = "ee-gh";
    BlogPostLanguageEnum["EeTg"] = "ee-tg";
    BlogPostLanguageEnum["El"] = "el";
    BlogPostLanguageEnum["ElCy"] = "el-cy";
    BlogPostLanguageEnum["ElGr"] = "el-gr";
    BlogPostLanguageEnum["En"] = "en";
    BlogPostLanguageEnum["En001"] = "en-001";
    BlogPostLanguageEnum["En150"] = "en-150";
    BlogPostLanguageEnum["EnAe"] = "en-ae";
    BlogPostLanguageEnum["EnAg"] = "en-ag";
    BlogPostLanguageEnum["EnAi"] = "en-ai";
    BlogPostLanguageEnum["EnAs"] = "en-as";
    BlogPostLanguageEnum["EnAt"] = "en-at";
    BlogPostLanguageEnum["EnAu"] = "en-au";
    BlogPostLanguageEnum["EnBb"] = "en-bb";
    BlogPostLanguageEnum["EnBe"] = "en-be";
    BlogPostLanguageEnum["EnBi"] = "en-bi";
    BlogPostLanguageEnum["EnBm"] = "en-bm";
    BlogPostLanguageEnum["EnBs"] = "en-bs";
    BlogPostLanguageEnum["EnBw"] = "en-bw";
    BlogPostLanguageEnum["EnBz"] = "en-bz";
    BlogPostLanguageEnum["EnCa"] = "en-ca";
    BlogPostLanguageEnum["EnCc"] = "en-cc";
    BlogPostLanguageEnum["EnCh"] = "en-ch";
    BlogPostLanguageEnum["EnCk"] = "en-ck";
    BlogPostLanguageEnum["EnCm"] = "en-cm";
    BlogPostLanguageEnum["EnCn"] = "en-cn";
    BlogPostLanguageEnum["EnCx"] = "en-cx";
    BlogPostLanguageEnum["EnCy"] = "en-cy";
    BlogPostLanguageEnum["EnDe"] = "en-de";
    BlogPostLanguageEnum["EnDg"] = "en-dg";
    BlogPostLanguageEnum["EnDk"] = "en-dk";
    BlogPostLanguageEnum["EnDm"] = "en-dm";
    BlogPostLanguageEnum["EnEe"] = "en-ee";
    BlogPostLanguageEnum["EnEr"] = "en-er";
    BlogPostLanguageEnum["EnFr"] = "en-fr";
    BlogPostLanguageEnum["EnFi"] = "en-fi";
    BlogPostLanguageEnum["EnFj"] = "en-fj";
    BlogPostLanguageEnum["EnFk"] = "en-fk";
    BlogPostLanguageEnum["EnFm"] = "en-fm";
    BlogPostLanguageEnum["EnGb"] = "en-gb";
    BlogPostLanguageEnum["EnGd"] = "en-gd";
    BlogPostLanguageEnum["EnGg"] = "en-gg";
    BlogPostLanguageEnum["EnGh"] = "en-gh";
    BlogPostLanguageEnum["EnGi"] = "en-gi";
    BlogPostLanguageEnum["EnGm"] = "en-gm";
    BlogPostLanguageEnum["EnGu"] = "en-gu";
    BlogPostLanguageEnum["EnGy"] = "en-gy";
    BlogPostLanguageEnum["EnHk"] = "en-hk";
    BlogPostLanguageEnum["EnIe"] = "en-ie";
    BlogPostLanguageEnum["EnIl"] = "en-il";
    BlogPostLanguageEnum["EnIm"] = "en-im";
    BlogPostLanguageEnum["EnIn"] = "en-in";
    BlogPostLanguageEnum["EnIo"] = "en-io";
    BlogPostLanguageEnum["EnJe"] = "en-je";
    BlogPostLanguageEnum["EnJm"] = "en-jm";
    BlogPostLanguageEnum["EnKe"] = "en-ke";
    BlogPostLanguageEnum["EnKi"] = "en-ki";
    BlogPostLanguageEnum["EnKn"] = "en-kn";
    BlogPostLanguageEnum["EnKy"] = "en-ky";
    BlogPostLanguageEnum["EnLc"] = "en-lc";
    BlogPostLanguageEnum["EnLr"] = "en-lr";
    BlogPostLanguageEnum["EnLs"] = "en-ls";
    BlogPostLanguageEnum["EnLu"] = "en-lu";
    BlogPostLanguageEnum["EnMg"] = "en-mg";
    BlogPostLanguageEnum["EnMh"] = "en-mh";
    BlogPostLanguageEnum["EnMo"] = "en-mo";
    BlogPostLanguageEnum["EnMp"] = "en-mp";
    BlogPostLanguageEnum["EnMs"] = "en-ms";
    BlogPostLanguageEnum["EnMt"] = "en-mt";
    BlogPostLanguageEnum["EnMu"] = "en-mu";
    BlogPostLanguageEnum["EnMw"] = "en-mw";
    BlogPostLanguageEnum["EnMx"] = "en-mx";
    BlogPostLanguageEnum["EnMy"] = "en-my";
    BlogPostLanguageEnum["EnNa"] = "en-na";
    BlogPostLanguageEnum["EnNf"] = "en-nf";
    BlogPostLanguageEnum["EnNg"] = "en-ng";
    BlogPostLanguageEnum["EnNl"] = "en-nl";
    BlogPostLanguageEnum["EnNr"] = "en-nr";
    BlogPostLanguageEnum["EnNu"] = "en-nu";
    BlogPostLanguageEnum["EnNz"] = "en-nz";
    BlogPostLanguageEnum["EnPg"] = "en-pg";
    BlogPostLanguageEnum["EnPh"] = "en-ph";
    BlogPostLanguageEnum["EnPk"] = "en-pk";
    BlogPostLanguageEnum["EnPn"] = "en-pn";
    BlogPostLanguageEnum["EnPr"] = "en-pr";
    BlogPostLanguageEnum["EnPw"] = "en-pw";
    BlogPostLanguageEnum["EnRw"] = "en-rw";
    BlogPostLanguageEnum["EnSb"] = "en-sb";
    BlogPostLanguageEnum["EnSc"] = "en-sc";
    BlogPostLanguageEnum["EnSd"] = "en-sd";
    BlogPostLanguageEnum["EnSe"] = "en-se";
    BlogPostLanguageEnum["EnSg"] = "en-sg";
    BlogPostLanguageEnum["EnSh"] = "en-sh";
    BlogPostLanguageEnum["EnSi"] = "en-si";
    BlogPostLanguageEnum["EnSl"] = "en-sl";
    BlogPostLanguageEnum["EnSs"] = "en-ss";
    BlogPostLanguageEnum["EnSx"] = "en-sx";
    BlogPostLanguageEnum["EnSz"] = "en-sz";
    BlogPostLanguageEnum["EnTc"] = "en-tc";
    BlogPostLanguageEnum["EnTk"] = "en-tk";
    BlogPostLanguageEnum["EnTo"] = "en-to";
    BlogPostLanguageEnum["EnTt"] = "en-tt";
    BlogPostLanguageEnum["EnTv"] = "en-tv";
    BlogPostLanguageEnum["EnTz"] = "en-tz";
    BlogPostLanguageEnum["EnUg"] = "en-ug";
    BlogPostLanguageEnum["EnUm"] = "en-um";
    BlogPostLanguageEnum["EnUs"] = "en-us";
    BlogPostLanguageEnum["EnVc"] = "en-vc";
    BlogPostLanguageEnum["EnVg"] = "en-vg";
    BlogPostLanguageEnum["EnVi"] = "en-vi";
    BlogPostLanguageEnum["EnVu"] = "en-vu";
    BlogPostLanguageEnum["EnWs"] = "en-ws";
    BlogPostLanguageEnum["EnZa"] = "en-za";
    BlogPostLanguageEnum["EnZm"] = "en-zm";
    BlogPostLanguageEnum["EnZw"] = "en-zw";
    BlogPostLanguageEnum["Eo"] = "eo";
    BlogPostLanguageEnum["Eo001"] = "eo-001";
    BlogPostLanguageEnum["Es"] = "es";
    BlogPostLanguageEnum["Es419"] = "es-419";
    BlogPostLanguageEnum["EsAr"] = "es-ar";
    BlogPostLanguageEnum["EsBo"] = "es-bo";
    BlogPostLanguageEnum["EsBr"] = "es-br";
    BlogPostLanguageEnum["EsBz"] = "es-bz";
    BlogPostLanguageEnum["EsCl"] = "es-cl";
    BlogPostLanguageEnum["EsCo"] = "es-co";
    BlogPostLanguageEnum["EsCr"] = "es-cr";
    BlogPostLanguageEnum["EsCu"] = "es-cu";
    BlogPostLanguageEnum["EsDo"] = "es-do";
    BlogPostLanguageEnum["EsEa"] = "es-ea";
    BlogPostLanguageEnum["EsEc"] = "es-ec";
    BlogPostLanguageEnum["EsEs"] = "es-es";
    BlogPostLanguageEnum["EsGq"] = "es-gq";
    BlogPostLanguageEnum["EsGt"] = "es-gt";
    BlogPostLanguageEnum["EsHn"] = "es-hn";
    BlogPostLanguageEnum["EsIc"] = "es-ic";
    BlogPostLanguageEnum["EsMx"] = "es-mx";
    BlogPostLanguageEnum["EsNi"] = "es-ni";
    BlogPostLanguageEnum["EsPa"] = "es-pa";
    BlogPostLanguageEnum["EsPe"] = "es-pe";
    BlogPostLanguageEnum["EsPh"] = "es-ph";
    BlogPostLanguageEnum["EsPr"] = "es-pr";
    BlogPostLanguageEnum["EsPy"] = "es-py";
    BlogPostLanguageEnum["EsSv"] = "es-sv";
    BlogPostLanguageEnum["EsUs"] = "es-us";
    BlogPostLanguageEnum["EsUy"] = "es-uy";
    BlogPostLanguageEnum["EsVe"] = "es-ve";
    BlogPostLanguageEnum["Et"] = "et";
    BlogPostLanguageEnum["EtEe"] = "et-ee";
    BlogPostLanguageEnum["Eu"] = "eu";
    BlogPostLanguageEnum["EuEs"] = "eu-es";
    BlogPostLanguageEnum["Ewo"] = "ewo";
    BlogPostLanguageEnum["EwoCm"] = "ewo-cm";
    BlogPostLanguageEnum["Fa"] = "fa";
    BlogPostLanguageEnum["FaAf"] = "fa-af";
    BlogPostLanguageEnum["FaIr"] = "fa-ir";
    BlogPostLanguageEnum["Ff"] = "ff";
    BlogPostLanguageEnum["FfBf"] = "ff-bf";
    BlogPostLanguageEnum["FfCm"] = "ff-cm";
    BlogPostLanguageEnum["FfGh"] = "ff-gh";
    BlogPostLanguageEnum["FfGm"] = "ff-gm";
    BlogPostLanguageEnum["FfGn"] = "ff-gn";
    BlogPostLanguageEnum["FfGw"] = "ff-gw";
    BlogPostLanguageEnum["FfLr"] = "ff-lr";
    BlogPostLanguageEnum["FfMr"] = "ff-mr";
    BlogPostLanguageEnum["FfNe"] = "ff-ne";
    BlogPostLanguageEnum["FfNg"] = "ff-ng";
    BlogPostLanguageEnum["FfSl"] = "ff-sl";
    BlogPostLanguageEnum["FfSn"] = "ff-sn";
    BlogPostLanguageEnum["Fi"] = "fi";
    BlogPostLanguageEnum["FiFi"] = "fi-fi";
    BlogPostLanguageEnum["Fil"] = "fil";
    BlogPostLanguageEnum["FilPh"] = "fil-ph";
    BlogPostLanguageEnum["Fo"] = "fo";
    BlogPostLanguageEnum["FoDk"] = "fo-dk";
    BlogPostLanguageEnum["FoFo"] = "fo-fo";
    BlogPostLanguageEnum["Fr"] = "fr";
    BlogPostLanguageEnum["FrBe"] = "fr-be";
    BlogPostLanguageEnum["FrBf"] = "fr-bf";
    BlogPostLanguageEnum["FrBi"] = "fr-bi";
    BlogPostLanguageEnum["FrBj"] = "fr-bj";
    BlogPostLanguageEnum["FrBl"] = "fr-bl";
    BlogPostLanguageEnum["FrCa"] = "fr-ca";
    BlogPostLanguageEnum["FrCd"] = "fr-cd";
    BlogPostLanguageEnum["FrCf"] = "fr-cf";
    BlogPostLanguageEnum["FrCg"] = "fr-cg";
    BlogPostLanguageEnum["FrCh"] = "fr-ch";
    BlogPostLanguageEnum["FrCi"] = "fr-ci";
    BlogPostLanguageEnum["FrCm"] = "fr-cm";
    BlogPostLanguageEnum["FrDj"] = "fr-dj";
    BlogPostLanguageEnum["FrDz"] = "fr-dz";
    BlogPostLanguageEnum["FrFr"] = "fr-fr";
    BlogPostLanguageEnum["FrGa"] = "fr-ga";
    BlogPostLanguageEnum["FrGf"] = "fr-gf";
    BlogPostLanguageEnum["FrGn"] = "fr-gn";
    BlogPostLanguageEnum["FrGp"] = "fr-gp";
    BlogPostLanguageEnum["FrGq"] = "fr-gq";
    BlogPostLanguageEnum["FrHt"] = "fr-ht";
    BlogPostLanguageEnum["FrKm"] = "fr-km";
    BlogPostLanguageEnum["FrLu"] = "fr-lu";
    BlogPostLanguageEnum["FrMa"] = "fr-ma";
    BlogPostLanguageEnum["FrMc"] = "fr-mc";
    BlogPostLanguageEnum["FrMf"] = "fr-mf";
    BlogPostLanguageEnum["FrMg"] = "fr-mg";
    BlogPostLanguageEnum["FrMl"] = "fr-ml";
    BlogPostLanguageEnum["FrMq"] = "fr-mq";
    BlogPostLanguageEnum["FrMr"] = "fr-mr";
    BlogPostLanguageEnum["FrMu"] = "fr-mu";
    BlogPostLanguageEnum["FrNc"] = "fr-nc";
    BlogPostLanguageEnum["FrNe"] = "fr-ne";
    BlogPostLanguageEnum["FrPf"] = "fr-pf";
    BlogPostLanguageEnum["FrPm"] = "fr-pm";
    BlogPostLanguageEnum["FrRe"] = "fr-re";
    BlogPostLanguageEnum["FrRw"] = "fr-rw";
    BlogPostLanguageEnum["FrSc"] = "fr-sc";
    BlogPostLanguageEnum["FrSn"] = "fr-sn";
    BlogPostLanguageEnum["FrSy"] = "fr-sy";
    BlogPostLanguageEnum["FrTd"] = "fr-td";
    BlogPostLanguageEnum["FrTg"] = "fr-tg";
    BlogPostLanguageEnum["FrTn"] = "fr-tn";
    BlogPostLanguageEnum["FrVu"] = "fr-vu";
    BlogPostLanguageEnum["FrWf"] = "fr-wf";
    BlogPostLanguageEnum["FrYt"] = "fr-yt";
    BlogPostLanguageEnum["Fur"] = "fur";
    BlogPostLanguageEnum["FurIt"] = "fur-it";
    BlogPostLanguageEnum["Fy"] = "fy";
    BlogPostLanguageEnum["FyNl"] = "fy-nl";
    BlogPostLanguageEnum["Ga"] = "ga";
    BlogPostLanguageEnum["GaGb"] = "ga-gb";
    BlogPostLanguageEnum["GaIe"] = "ga-ie";
    BlogPostLanguageEnum["Gd"] = "gd";
    BlogPostLanguageEnum["GdGb"] = "gd-gb";
    BlogPostLanguageEnum["Gl"] = "gl";
    BlogPostLanguageEnum["GlEs"] = "gl-es";
    BlogPostLanguageEnum["Gsw"] = "gsw";
    BlogPostLanguageEnum["GswCh"] = "gsw-ch";
    BlogPostLanguageEnum["GswFr"] = "gsw-fr";
    BlogPostLanguageEnum["GswLi"] = "gsw-li";
    BlogPostLanguageEnum["Gu"] = "gu";
    BlogPostLanguageEnum["GuIn"] = "gu-in";
    BlogPostLanguageEnum["Guz"] = "guz";
    BlogPostLanguageEnum["GuzKe"] = "guz-ke";
    BlogPostLanguageEnum["Gv"] = "gv";
    BlogPostLanguageEnum["GvIm"] = "gv-im";
    BlogPostLanguageEnum["Ha"] = "ha";
    BlogPostLanguageEnum["HaGh"] = "ha-gh";
    BlogPostLanguageEnum["HaNe"] = "ha-ne";
    BlogPostLanguageEnum["HaNg"] = "ha-ng";
    BlogPostLanguageEnum["Haw"] = "haw";
    BlogPostLanguageEnum["HawUs"] = "haw-us";
    BlogPostLanguageEnum["He"] = "he";
    BlogPostLanguageEnum["Hi"] = "hi";
    BlogPostLanguageEnum["HiIn"] = "hi-in";
    BlogPostLanguageEnum["Hr"] = "hr";
    BlogPostLanguageEnum["HrBa"] = "hr-ba";
    BlogPostLanguageEnum["HrHr"] = "hr-hr";
    BlogPostLanguageEnum["Hsb"] = "hsb";
    BlogPostLanguageEnum["HsbDe"] = "hsb-de";
    BlogPostLanguageEnum["Hu"] = "hu";
    BlogPostLanguageEnum["HuHu"] = "hu-hu";
    BlogPostLanguageEnum["Hy"] = "hy";
    BlogPostLanguageEnum["HyAm"] = "hy-am";
    BlogPostLanguageEnum["Ia"] = "ia";
    BlogPostLanguageEnum["Ia001"] = "ia-001";
    BlogPostLanguageEnum["Id"] = "id";
    BlogPostLanguageEnum["Ig"] = "ig";
    BlogPostLanguageEnum["IgNg"] = "ig-ng";
    BlogPostLanguageEnum["Ii"] = "ii";
    BlogPostLanguageEnum["IiCn"] = "ii-cn";
    BlogPostLanguageEnum["IdId"] = "id-id";
    BlogPostLanguageEnum["Is"] = "is";
    BlogPostLanguageEnum["IsIs"] = "is-is";
    BlogPostLanguageEnum["It"] = "it";
    BlogPostLanguageEnum["ItCh"] = "it-ch";
    BlogPostLanguageEnum["ItIt"] = "it-it";
    BlogPostLanguageEnum["ItSm"] = "it-sm";
    BlogPostLanguageEnum["ItVa"] = "it-va";
    BlogPostLanguageEnum["HeIl"] = "he-il";
    BlogPostLanguageEnum["Ja"] = "ja";
    BlogPostLanguageEnum["JaJp"] = "ja-jp";
    BlogPostLanguageEnum["Jgo"] = "jgo";
    BlogPostLanguageEnum["JgoCm"] = "jgo-cm";
    BlogPostLanguageEnum["Yi"] = "yi";
    BlogPostLanguageEnum["Yi001"] = "yi-001";
    BlogPostLanguageEnum["Jmc"] = "jmc";
    BlogPostLanguageEnum["JmcTz"] = "jmc-tz";
    BlogPostLanguageEnum["Jv"] = "jv";
    BlogPostLanguageEnum["JvId"] = "jv-id";
    BlogPostLanguageEnum["Ka"] = "ka";
    BlogPostLanguageEnum["KaGe"] = "ka-ge";
    BlogPostLanguageEnum["Kab"] = "kab";
    BlogPostLanguageEnum["KabDz"] = "kab-dz";
    BlogPostLanguageEnum["Kam"] = "kam";
    BlogPostLanguageEnum["KamKe"] = "kam-ke";
    BlogPostLanguageEnum["Kde"] = "kde";
    BlogPostLanguageEnum["KdeTz"] = "kde-tz";
    BlogPostLanguageEnum["Kea"] = "kea";
    BlogPostLanguageEnum["KeaCv"] = "kea-cv";
    BlogPostLanguageEnum["Khq"] = "khq";
    BlogPostLanguageEnum["KhqMl"] = "khq-ml";
    BlogPostLanguageEnum["Ki"] = "ki";
    BlogPostLanguageEnum["KiKe"] = "ki-ke";
    BlogPostLanguageEnum["Kk"] = "kk";
    BlogPostLanguageEnum["KkKz"] = "kk-kz";
    BlogPostLanguageEnum["Kkj"] = "kkj";
    BlogPostLanguageEnum["KkjCm"] = "kkj-cm";
    BlogPostLanguageEnum["Kl"] = "kl";
    BlogPostLanguageEnum["KlGl"] = "kl-gl";
    BlogPostLanguageEnum["Kln"] = "kln";
    BlogPostLanguageEnum["KlnKe"] = "kln-ke";
    BlogPostLanguageEnum["Km"] = "km";
    BlogPostLanguageEnum["KmKh"] = "km-kh";
    BlogPostLanguageEnum["Kn"] = "kn";
    BlogPostLanguageEnum["KnIn"] = "kn-in";
    BlogPostLanguageEnum["Ko"] = "ko";
    BlogPostLanguageEnum["KoKp"] = "ko-kp";
    BlogPostLanguageEnum["KoKr"] = "ko-kr";
    BlogPostLanguageEnum["Kok"] = "kok";
    BlogPostLanguageEnum["KokIn"] = "kok-in";
    BlogPostLanguageEnum["Ks"] = "ks";
    BlogPostLanguageEnum["KsIn"] = "ks-in";
    BlogPostLanguageEnum["Ksb"] = "ksb";
    BlogPostLanguageEnum["KsbTz"] = "ksb-tz";
    BlogPostLanguageEnum["Ksf"] = "ksf";
    BlogPostLanguageEnum["KsfCm"] = "ksf-cm";
    BlogPostLanguageEnum["Ksh"] = "ksh";
    BlogPostLanguageEnum["KshDe"] = "ksh-de";
    BlogPostLanguageEnum["Kw"] = "kw";
    BlogPostLanguageEnum["KwGb"] = "kw-gb";
    BlogPostLanguageEnum["Ku"] = "ku";
    BlogPostLanguageEnum["KuTr"] = "ku-tr";
    BlogPostLanguageEnum["Ky"] = "ky";
    BlogPostLanguageEnum["KyKg"] = "ky-kg";
    BlogPostLanguageEnum["Lag"] = "lag";
    BlogPostLanguageEnum["LagTz"] = "lag-tz";
    BlogPostLanguageEnum["Lb"] = "lb";
    BlogPostLanguageEnum["LbLu"] = "lb-lu";
    BlogPostLanguageEnum["Lg"] = "lg";
    BlogPostLanguageEnum["LgUg"] = "lg-ug";
    BlogPostLanguageEnum["Lkt"] = "lkt";
    BlogPostLanguageEnum["LktUs"] = "lkt-us";
    BlogPostLanguageEnum["Ln"] = "ln";
    BlogPostLanguageEnum["LnAo"] = "ln-ao";
    BlogPostLanguageEnum["LnCd"] = "ln-cd";
    BlogPostLanguageEnum["LnCf"] = "ln-cf";
    BlogPostLanguageEnum["LnCg"] = "ln-cg";
    BlogPostLanguageEnum["Lo"] = "lo";
    BlogPostLanguageEnum["LoLa"] = "lo-la";
    BlogPostLanguageEnum["Lrc"] = "lrc";
    BlogPostLanguageEnum["LrcIq"] = "lrc-iq";
    BlogPostLanguageEnum["LrcIr"] = "lrc-ir";
    BlogPostLanguageEnum["Lt"] = "lt";
    BlogPostLanguageEnum["LtLt"] = "lt-lt";
    BlogPostLanguageEnum["Lu"] = "lu";
    BlogPostLanguageEnum["LuCd"] = "lu-cd";
    BlogPostLanguageEnum["Luo"] = "luo";
    BlogPostLanguageEnum["LuoKe"] = "luo-ke";
    BlogPostLanguageEnum["Luy"] = "luy";
    BlogPostLanguageEnum["LuyKe"] = "luy-ke";
    BlogPostLanguageEnum["Lv"] = "lv";
    BlogPostLanguageEnum["LvLv"] = "lv-lv";
    BlogPostLanguageEnum["Mai"] = "mai";
    BlogPostLanguageEnum["MaiIn"] = "mai-in";
    BlogPostLanguageEnum["Mas"] = "mas";
    BlogPostLanguageEnum["MasKe"] = "mas-ke";
    BlogPostLanguageEnum["MasTz"] = "mas-tz";
    BlogPostLanguageEnum["Mer"] = "mer";
    BlogPostLanguageEnum["MerKe"] = "mer-ke";
    BlogPostLanguageEnum["Mfe"] = "mfe";
    BlogPostLanguageEnum["MfeMu"] = "mfe-mu";
    BlogPostLanguageEnum["Mg"] = "mg";
    BlogPostLanguageEnum["MgMg"] = "mg-mg";
    BlogPostLanguageEnum["Mgh"] = "mgh";
    BlogPostLanguageEnum["MghMz"] = "mgh-mz";
    BlogPostLanguageEnum["Mgo"] = "mgo";
    BlogPostLanguageEnum["MgoCm"] = "mgo-cm";
    BlogPostLanguageEnum["Mi"] = "mi";
    BlogPostLanguageEnum["MiNz"] = "mi-nz";
    BlogPostLanguageEnum["Mk"] = "mk";
    BlogPostLanguageEnum["MkMk"] = "mk-mk";
    BlogPostLanguageEnum["Ml"] = "ml";
    BlogPostLanguageEnum["MlIn"] = "ml-in";
    BlogPostLanguageEnum["Mn"] = "mn";
    BlogPostLanguageEnum["MnMn"] = "mn-mn";
    BlogPostLanguageEnum["Mni"] = "mni";
    BlogPostLanguageEnum["MniIn"] = "mni-in";
    BlogPostLanguageEnum["Mr"] = "mr";
    BlogPostLanguageEnum["MrIn"] = "mr-in";
    BlogPostLanguageEnum["Ms"] = "ms";
    BlogPostLanguageEnum["MsBn"] = "ms-bn";
    BlogPostLanguageEnum["MsId"] = "ms-id";
    BlogPostLanguageEnum["MsMy"] = "ms-my";
    BlogPostLanguageEnum["MsSg"] = "ms-sg";
    BlogPostLanguageEnum["Mt"] = "mt";
    BlogPostLanguageEnum["MtMt"] = "mt-mt";
    BlogPostLanguageEnum["Mua"] = "mua";
    BlogPostLanguageEnum["MuaCm"] = "mua-cm";
    BlogPostLanguageEnum["My"] = "my";
    BlogPostLanguageEnum["MyMm"] = "my-mm";
    BlogPostLanguageEnum["Mzn"] = "mzn";
    BlogPostLanguageEnum["MznIr"] = "mzn-ir";
    BlogPostLanguageEnum["Naq"] = "naq";
    BlogPostLanguageEnum["NaqNa"] = "naq-na";
    BlogPostLanguageEnum["Nb"] = "nb";
    BlogPostLanguageEnum["NbNo"] = "nb-no";
    BlogPostLanguageEnum["NbSj"] = "nb-sj";
    BlogPostLanguageEnum["Nd"] = "nd";
    BlogPostLanguageEnum["NdZw"] = "nd-zw";
    BlogPostLanguageEnum["Nds"] = "nds";
    BlogPostLanguageEnum["NdsDe"] = "nds-de";
    BlogPostLanguageEnum["NdsNl"] = "nds-nl";
    BlogPostLanguageEnum["Ne"] = "ne";
    BlogPostLanguageEnum["NeIn"] = "ne-in";
    BlogPostLanguageEnum["NeNp"] = "ne-np";
    BlogPostLanguageEnum["Nl"] = "nl";
    BlogPostLanguageEnum["NlAw"] = "nl-aw";
    BlogPostLanguageEnum["NlBe"] = "nl-be";
    BlogPostLanguageEnum["NlCh"] = "nl-ch";
    BlogPostLanguageEnum["NlBq"] = "nl-bq";
    BlogPostLanguageEnum["NlCw"] = "nl-cw";
    BlogPostLanguageEnum["NlLu"] = "nl-lu";
    BlogPostLanguageEnum["NlNl"] = "nl-nl";
    BlogPostLanguageEnum["NlSr"] = "nl-sr";
    BlogPostLanguageEnum["NlSx"] = "nl-sx";
    BlogPostLanguageEnum["Nmg"] = "nmg";
    BlogPostLanguageEnum["NmgCm"] = "nmg-cm";
    BlogPostLanguageEnum["Nn"] = "nn";
    BlogPostLanguageEnum["NnNo"] = "nn-no";
    BlogPostLanguageEnum["Nnh"] = "nnh";
    BlogPostLanguageEnum["NnhCm"] = "nnh-cm";
    BlogPostLanguageEnum["No"] = "no";
    BlogPostLanguageEnum["NoNo"] = "no-no";
    BlogPostLanguageEnum["Nus"] = "nus";
    BlogPostLanguageEnum["NusSs"] = "nus-ss";
    BlogPostLanguageEnum["Nyn"] = "nyn";
    BlogPostLanguageEnum["NynUg"] = "nyn-ug";
    BlogPostLanguageEnum["Om"] = "om";
    BlogPostLanguageEnum["OmEt"] = "om-et";
    BlogPostLanguageEnum["OmKe"] = "om-ke";
    BlogPostLanguageEnum["Or"] = "or";
    BlogPostLanguageEnum["OrIn"] = "or-in";
    BlogPostLanguageEnum["Os"] = "os";
    BlogPostLanguageEnum["OsGe"] = "os-ge";
    BlogPostLanguageEnum["OsRu"] = "os-ru";
    BlogPostLanguageEnum["Pa"] = "pa";
    BlogPostLanguageEnum["PaIn"] = "pa-in";
    BlogPostLanguageEnum["PaPk"] = "pa-pk";
    BlogPostLanguageEnum["Pcm"] = "pcm";
    BlogPostLanguageEnum["PcmNg"] = "pcm-ng";
    BlogPostLanguageEnum["Pl"] = "pl";
    BlogPostLanguageEnum["PlPl"] = "pl-pl";
    BlogPostLanguageEnum["Prg"] = "prg";
    BlogPostLanguageEnum["Prg001"] = "prg-001";
    BlogPostLanguageEnum["Ps"] = "ps";
    BlogPostLanguageEnum["PsAf"] = "ps-af";
    BlogPostLanguageEnum["PsPk"] = "ps-pk";
    BlogPostLanguageEnum["Pt"] = "pt";
    BlogPostLanguageEnum["PtAo"] = "pt-ao";
    BlogPostLanguageEnum["PtBr"] = "pt-br";
    BlogPostLanguageEnum["PtCh"] = "pt-ch";
    BlogPostLanguageEnum["PtCv"] = "pt-cv";
    BlogPostLanguageEnum["PtGq"] = "pt-gq";
    BlogPostLanguageEnum["PtGw"] = "pt-gw";
    BlogPostLanguageEnum["PtLu"] = "pt-lu";
    BlogPostLanguageEnum["PtMo"] = "pt-mo";
    BlogPostLanguageEnum["PtMz"] = "pt-mz";
    BlogPostLanguageEnum["PtPt"] = "pt-pt";
    BlogPostLanguageEnum["PtSt"] = "pt-st";
    BlogPostLanguageEnum["PtTl"] = "pt-tl";
    BlogPostLanguageEnum["Qu"] = "qu";
    BlogPostLanguageEnum["QuBo"] = "qu-bo";
    BlogPostLanguageEnum["QuEc"] = "qu-ec";
    BlogPostLanguageEnum["QuPe"] = "qu-pe";
    BlogPostLanguageEnum["Rm"] = "rm";
    BlogPostLanguageEnum["RmCh"] = "rm-ch";
    BlogPostLanguageEnum["Rn"] = "rn";
    BlogPostLanguageEnum["RnBi"] = "rn-bi";
    BlogPostLanguageEnum["Ro"] = "ro";
    BlogPostLanguageEnum["RoMd"] = "ro-md";
    BlogPostLanguageEnum["RoRo"] = "ro-ro";
    BlogPostLanguageEnum["Rof"] = "rof";
    BlogPostLanguageEnum["RofTz"] = "rof-tz";
    BlogPostLanguageEnum["Ru"] = "ru";
    BlogPostLanguageEnum["RuBy"] = "ru-by";
    BlogPostLanguageEnum["RuKg"] = "ru-kg";
    BlogPostLanguageEnum["RuKz"] = "ru-kz";
    BlogPostLanguageEnum["RuMd"] = "ru-md";
    BlogPostLanguageEnum["RuRu"] = "ru-ru";
    BlogPostLanguageEnum["RuUa"] = "ru-ua";
    BlogPostLanguageEnum["Rw"] = "rw";
    BlogPostLanguageEnum["RwRw"] = "rw-rw";
    BlogPostLanguageEnum["Rwk"] = "rwk";
    BlogPostLanguageEnum["RwkTz"] = "rwk-tz";
    BlogPostLanguageEnum["Sa"] = "sa";
    BlogPostLanguageEnum["SaIn"] = "sa-in";
    BlogPostLanguageEnum["Sah"] = "sah";
    BlogPostLanguageEnum["SahRu"] = "sah-ru";
    BlogPostLanguageEnum["Saq"] = "saq";
    BlogPostLanguageEnum["SaqKe"] = "saq-ke";
    BlogPostLanguageEnum["Sat"] = "sat";
    BlogPostLanguageEnum["SatIn"] = "sat-in";
    BlogPostLanguageEnum["Sbp"] = "sbp";
    BlogPostLanguageEnum["SbpTz"] = "sbp-tz";
    BlogPostLanguageEnum["Sd"] = "sd";
    BlogPostLanguageEnum["SdIn"] = "sd-in";
    BlogPostLanguageEnum["SdPk"] = "sd-pk";
    BlogPostLanguageEnum["Se"] = "se";
    BlogPostLanguageEnum["SeFi"] = "se-fi";
    BlogPostLanguageEnum["SeNo"] = "se-no";
    BlogPostLanguageEnum["SeSe"] = "se-se";
    BlogPostLanguageEnum["Seh"] = "seh";
    BlogPostLanguageEnum["SehMz"] = "seh-mz";
    BlogPostLanguageEnum["Ses"] = "ses";
    BlogPostLanguageEnum["SesMl"] = "ses-ml";
    BlogPostLanguageEnum["Sg"] = "sg";
    BlogPostLanguageEnum["SgCf"] = "sg-cf";
    BlogPostLanguageEnum["Shi"] = "shi";
    BlogPostLanguageEnum["ShiMa"] = "shi-ma";
    BlogPostLanguageEnum["Si"] = "si";
    BlogPostLanguageEnum["SiLk"] = "si-lk";
    BlogPostLanguageEnum["Sk"] = "sk";
    BlogPostLanguageEnum["SkSk"] = "sk-sk";
    BlogPostLanguageEnum["Sl"] = "sl";
    BlogPostLanguageEnum["SlSi"] = "sl-si";
    BlogPostLanguageEnum["Smn"] = "smn";
    BlogPostLanguageEnum["SmnFi"] = "smn-fi";
    BlogPostLanguageEnum["Sn"] = "sn";
    BlogPostLanguageEnum["SnZw"] = "sn-zw";
    BlogPostLanguageEnum["So"] = "so";
    BlogPostLanguageEnum["SoDj"] = "so-dj";
    BlogPostLanguageEnum["SoEt"] = "so-et";
    BlogPostLanguageEnum["SoKe"] = "so-ke";
    BlogPostLanguageEnum["SoSo"] = "so-so";
    BlogPostLanguageEnum["Sq"] = "sq";
    BlogPostLanguageEnum["SqAl"] = "sq-al";
    BlogPostLanguageEnum["SqMk"] = "sq-mk";
    BlogPostLanguageEnum["SqXk"] = "sq-xk";
    BlogPostLanguageEnum["Sr"] = "sr";
    BlogPostLanguageEnum["SrBa"] = "sr-ba";
    BlogPostLanguageEnum["SrCs"] = "sr-cs";
    BlogPostLanguageEnum["SrMe"] = "sr-me";
    BlogPostLanguageEnum["SrRs"] = "sr-rs";
    BlogPostLanguageEnum["SrXk"] = "sr-xk";
    BlogPostLanguageEnum["Su"] = "su";
    BlogPostLanguageEnum["SuId"] = "su-id";
    BlogPostLanguageEnum["Sv"] = "sv";
    BlogPostLanguageEnum["SvAx"] = "sv-ax";
    BlogPostLanguageEnum["SvFi"] = "sv-fi";
    BlogPostLanguageEnum["SvSe"] = "sv-se";
    BlogPostLanguageEnum["Sw"] = "sw";
    BlogPostLanguageEnum["SwCd"] = "sw-cd";
    BlogPostLanguageEnum["SwKe"] = "sw-ke";
    BlogPostLanguageEnum["SwTz"] = "sw-tz";
    BlogPostLanguageEnum["SwUg"] = "sw-ug";
    BlogPostLanguageEnum["Sy"] = "sy";
    BlogPostLanguageEnum["Ta"] = "ta";
    BlogPostLanguageEnum["TaIn"] = "ta-in";
    BlogPostLanguageEnum["TaLk"] = "ta-lk";
    BlogPostLanguageEnum["TaMy"] = "ta-my";
    BlogPostLanguageEnum["TaSg"] = "ta-sg";
    BlogPostLanguageEnum["Te"] = "te";
    BlogPostLanguageEnum["TeIn"] = "te-in";
    BlogPostLanguageEnum["Teo"] = "teo";
    BlogPostLanguageEnum["TeoKe"] = "teo-ke";
    BlogPostLanguageEnum["TeoUg"] = "teo-ug";
    BlogPostLanguageEnum["Tg"] = "tg";
    BlogPostLanguageEnum["TgTj"] = "tg-tj";
    BlogPostLanguageEnum["Th"] = "th";
    BlogPostLanguageEnum["ThTh"] = "th-th";
    BlogPostLanguageEnum["Ti"] = "ti";
    BlogPostLanguageEnum["TiEr"] = "ti-er";
    BlogPostLanguageEnum["TiEt"] = "ti-et";
    BlogPostLanguageEnum["Tk"] = "tk";
    BlogPostLanguageEnum["TkTm"] = "tk-tm";
    BlogPostLanguageEnum["Tl"] = "tl";
    BlogPostLanguageEnum["To"] = "to";
    BlogPostLanguageEnum["ToTo"] = "to-to";
    BlogPostLanguageEnum["Tr"] = "tr";
    BlogPostLanguageEnum["TrCy"] = "tr-cy";
    BlogPostLanguageEnum["TrTr"] = "tr-tr";
    BlogPostLanguageEnum["Tt"] = "tt";
    BlogPostLanguageEnum["TtRu"] = "tt-ru";
    BlogPostLanguageEnum["Twq"] = "twq";
    BlogPostLanguageEnum["TwqNe"] = "twq-ne";
    BlogPostLanguageEnum["Tzm"] = "tzm";
    BlogPostLanguageEnum["TzmMa"] = "tzm-ma";
    BlogPostLanguageEnum["Ug"] = "ug";
    BlogPostLanguageEnum["UgCn"] = "ug-cn";
    BlogPostLanguageEnum["Uk"] = "uk";
    BlogPostLanguageEnum["UkUa"] = "uk-ua";
    BlogPostLanguageEnum["Ur"] = "ur";
    BlogPostLanguageEnum["UrIn"] = "ur-in";
    BlogPostLanguageEnum["UrPk"] = "ur-pk";
    BlogPostLanguageEnum["Uz"] = "uz";
    BlogPostLanguageEnum["UzAf"] = "uz-af";
    BlogPostLanguageEnum["UzUz"] = "uz-uz";
    BlogPostLanguageEnum["Vai"] = "vai";
    BlogPostLanguageEnum["VaiLr"] = "vai-lr";
    BlogPostLanguageEnum["Vi"] = "vi";
    BlogPostLanguageEnum["ViVn"] = "vi-vn";
    BlogPostLanguageEnum["Vo"] = "vo";
    BlogPostLanguageEnum["Vo001"] = "vo-001";
    BlogPostLanguageEnum["Vun"] = "vun";
    BlogPostLanguageEnum["VunTz"] = "vun-tz";
    BlogPostLanguageEnum["Wae"] = "wae";
    BlogPostLanguageEnum["WaeCh"] = "wae-ch";
    BlogPostLanguageEnum["Wo"] = "wo";
    BlogPostLanguageEnum["WoSn"] = "wo-sn";
    BlogPostLanguageEnum["Xh"] = "xh";
    BlogPostLanguageEnum["XhZa"] = "xh-za";
    BlogPostLanguageEnum["Xog"] = "xog";
    BlogPostLanguageEnum["XogUg"] = "xog-ug";
    BlogPostLanguageEnum["Yav"] = "yav";
    BlogPostLanguageEnum["YavCm"] = "yav-cm";
    BlogPostLanguageEnum["Yo"] = "yo";
    BlogPostLanguageEnum["YoBj"] = "yo-bj";
    BlogPostLanguageEnum["YoNg"] = "yo-ng";
    BlogPostLanguageEnum["Yue"] = "yue";
    BlogPostLanguageEnum["YueCn"] = "yue-cn";
    BlogPostLanguageEnum["YueHk"] = "yue-hk";
    BlogPostLanguageEnum["Zgh"] = "zgh";
    BlogPostLanguageEnum["ZghMa"] = "zgh-ma";
    BlogPostLanguageEnum["Zh"] = "zh";
    BlogPostLanguageEnum["ZhCn"] = "zh-cn";
    BlogPostLanguageEnum["ZhHk"] = "zh-hk";
    BlogPostLanguageEnum["ZhMo"] = "zh-mo";
    BlogPostLanguageEnum["ZhSg"] = "zh-sg";
    BlogPostLanguageEnum["ZhTw"] = "zh-tw";
    BlogPostLanguageEnum["ZhHans"] = "zh-hans";
    BlogPostLanguageEnum["ZhHant"] = "zh-hant";
    BlogPostLanguageEnum["Zu"] = "zu";
    BlogPostLanguageEnum["ZuZa"] = "zu-za";
})(BlogPostLanguageEnum = exports.BlogPostLanguageEnum || (exports.BlogPostLanguageEnum = {}));
var BlogPostContentTypeCategoryEnum;
(function (BlogPostContentTypeCategoryEnum) {
    BlogPostContentTypeCategoryEnum["_0"] = "0";
    BlogPostContentTypeCategoryEnum["_1"] = "1";
    BlogPostContentTypeCategoryEnum["_2"] = "2";
    BlogPostContentTypeCategoryEnum["_3"] = "3";
    BlogPostContentTypeCategoryEnum["_4"] = "4";
    BlogPostContentTypeCategoryEnum["_5"] = "5";
    BlogPostContentTypeCategoryEnum["_6"] = "6";
    BlogPostContentTypeCategoryEnum["_7"] = "7";
    BlogPostContentTypeCategoryEnum["_8"] = "8";
    BlogPostContentTypeCategoryEnum["_9"] = "9";
    BlogPostContentTypeCategoryEnum["_10"] = "10";
    BlogPostContentTypeCategoryEnum["_11"] = "11";
    BlogPostContentTypeCategoryEnum["_12"] = "12";
    BlogPostContentTypeCategoryEnum["_13"] = "13";
    BlogPostContentTypeCategoryEnum["_14"] = "14";
    BlogPostContentTypeCategoryEnum["_15"] = "15";
})(BlogPostContentTypeCategoryEnum = exports.BlogPostContentTypeCategoryEnum || (exports.BlogPostContentTypeCategoryEnum = {}));
var BlogPostAbStatusEnum;
(function (BlogPostAbStatusEnum) {
    BlogPostAbStatusEnum["Master"] = "master";
    BlogPostAbStatusEnum["Variant"] = "variant";
    BlogPostAbStatusEnum["LoserVariant"] = "loser_variant";
    BlogPostAbStatusEnum["MabMaster"] = "mab_master";
    BlogPostAbStatusEnum["MabVariant"] = "mab_variant";
    BlogPostAbStatusEnum["AutomatedMaster"] = "automated_master";
    BlogPostAbStatusEnum["AutomatedVariant"] = "automated_variant";
    BlogPostAbStatusEnum["AutomatedLoserVariant"] = "automated_loser_variant";
})(BlogPostAbStatusEnum = exports.BlogPostAbStatusEnum || (exports.BlogPostAbStatusEnum = {}));
var BlogPostCurrentStateEnum;
(function (BlogPostCurrentStateEnum) {
    BlogPostCurrentStateEnum["Automated"] = "AUTOMATED";
    BlogPostCurrentStateEnum["AutomatedDraft"] = "AUTOMATED_DRAFT";
    BlogPostCurrentStateEnum["AutomatedSending"] = "AUTOMATED_SENDING";
    BlogPostCurrentStateEnum["AutomatedForForm"] = "AUTOMATED_FOR_FORM";
    BlogPostCurrentStateEnum["AutomatedForFormBuffer"] = "AUTOMATED_FOR_FORM_BUFFER";
    BlogPostCurrentStateEnum["AutomatedForFormDraft"] = "AUTOMATED_FOR_FORM_DRAFT";
    BlogPostCurrentStateEnum["AutomatedForFormLegacy"] = "AUTOMATED_FOR_FORM_LEGACY";
    BlogPostCurrentStateEnum["BlogEmailDraft"] = "BLOG_EMAIL_DRAFT";
    BlogPostCurrentStateEnum["BlogEmailPublished"] = "BLOG_EMAIL_PUBLISHED";
    BlogPostCurrentStateEnum["Draft"] = "DRAFT";
    BlogPostCurrentStateEnum["DraftAb"] = "DRAFT_AB";
    BlogPostCurrentStateEnum["DraftAbVariant"] = "DRAFT_AB_VARIANT";
    BlogPostCurrentStateEnum["Error"] = "ERROR";
    BlogPostCurrentStateEnum["LoserAbVariant"] = "LOSER_AB_VARIANT";
    BlogPostCurrentStateEnum["PageStub"] = "PAGE_STUB";
    BlogPostCurrentStateEnum["PreProcessing"] = "PRE_PROCESSING";
    BlogPostCurrentStateEnum["Processing"] = "PROCESSING";
    BlogPostCurrentStateEnum["Published"] = "PUBLISHED";
    BlogPostCurrentStateEnum["PublishedAb"] = "PUBLISHED_AB";
    BlogPostCurrentStateEnum["PublishedAbVariant"] = "PUBLISHED_AB_VARIANT";
    BlogPostCurrentStateEnum["PublishedOrScheduled"] = "PUBLISHED_OR_SCHEDULED";
    BlogPostCurrentStateEnum["RssToEmailDraft"] = "RSS_TO_EMAIL_DRAFT";
    BlogPostCurrentStateEnum["RssToEmailPublished"] = "RSS_TO_EMAIL_PUBLISHED";
    BlogPostCurrentStateEnum["Scheduled"] = "SCHEDULED";
    BlogPostCurrentStateEnum["ScheduledAb"] = "SCHEDULED_AB";
    BlogPostCurrentStateEnum["ScheduledOrPublished"] = "SCHEDULED_OR_PUBLISHED";
    BlogPostCurrentStateEnum["AutomatedAb"] = "AUTOMATED_AB";
    BlogPostCurrentStateEnum["AutomatedAbVariant"] = "AUTOMATED_AB_VARIANT";
    BlogPostCurrentStateEnum["AutomatedDraftAb"] = "AUTOMATED_DRAFT_AB";
    BlogPostCurrentStateEnum["AutomatedDraftAbvariant"] = "AUTOMATED_DRAFT_ABVARIANT";
    BlogPostCurrentStateEnum["AutomatedLoserAbvariant"] = "AUTOMATED_LOSER_ABVARIANT";
})(BlogPostCurrentStateEnum = exports.BlogPostCurrentStateEnum || (exports.BlogPostCurrentStateEnum = {}));
