// @hubspot/ui-extensions/dist/types.js
export class ExtensionEvent {
    type;
    bubbles;
    timeStamp;
    id;
    constructor(event) {
        this.bubbles = event.bubbles;
        this.type = event.type;
        this.timeStamp = event.timeStamp;
        this.id = crypto.randomUUID();
    }
}
export class FormSubmitExtensionEvent extends ExtensionEvent {
    targetValue;
    constructor(value, event) {
        super(event);
        this.targetValue = value;
    }
}
export const chartFieldTypes = {
    datetime: 'datetime',
    linear: 'linear',
    category: 'category',
};
export const iconNames = {
    success: 'success',
    remove: 'remove',
    add: 'add',
    attach: 'attach',
    date: 'date',
    copy: 'duplicate',
    delete: 'delete',
    edit: 'edit',
    email: 'email',
    exclamation: 'exclamation',
    question: 'question',
    home: 'home',
    location: 'location',
    upCarat: 'upCarat',
    downCarat: 'downCarat',
    warning: 'warning',
    shoppingCart: 'cart',
    clock: 'time',
    comment: 'comments',
    contact: 'contacts',
    star: 'favorite',
    file: 'file',
    reports: 'reports',
    video: 'video',
    robot: 'simpleBot',
    refresh: 'refresh',
    faceHappy: 'emoji',
    faceNeutral: 'emojiLineNeutral',
    faceSad: 'emojiLineSad',
    upload: 'upload',
    download: 'download',
    left: 'left',
    right: 'right',
    dataSync: 'dataSync',
    imageGallery: 'imageGallery',
    search: 'search',
    save: 'saveEditableView',
    notification: 'notification',
    bulb: 'bulb',
    settings: 'settings',
};
/**
 * @category Serverless
 */
export var ServerlessExecutionStatus;
(function (ServerlessExecutionStatus) {
    ServerlessExecutionStatus["Success"] = "SUCCESS";
    ServerlessExecutionStatus["Error"] = "ERROR";
})(ServerlessExecutionStatus || (ServerlessExecutionStatus = {}));
export const EXTENSION_POINT_LOCATIONS = [
    'crm.preview',
    'crm.record.sidebar',
    'crm.record.tab',
    'helpdesk.sidebar',
    'uie.playground.middle',
    'settings',
];
/** @deprecated use ExtensionEvent/FormSubmitExtensionEvent instead */
export class RemoteEvent {
    type;
    bubbles;
    timeStamp;
    targetValue;
    constructor(value, event) {
        this.bubbles = event.bubbles;
        this.type = event.type;
        this.timeStamp = event.timeStamp;
        this.targetValue = value;
    }
}
