// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectBlogPostsApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectBlogPostsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableBlogPostsApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(param, options) {
        return this.api.archiveWithHttpInfo(param.objectId, param.archived, options).toPromise();
    }
    archive(param, options) {
        return this.api.archive(param.objectId, param.archived, options).toPromise();
    }
    archiveBatchWithHttpInfo(param, options) {
        return this.api.archiveBatchWithHttpInfo(param.batchInputString, options).toPromise();
    }
    archiveBatch(param, options) {
        return this.api.archiveBatch(param.batchInputString, options).toPromise();
    }
    attachToLangGroupWithHttpInfo(param, options) {
        return this.api.attachToLangGroupWithHttpInfo(param.attachToLangPrimaryRequestVNext, options).toPromise();
    }
    attachToLangGroup(param, options) {
        return this.api.attachToLangGroup(param.attachToLangPrimaryRequestVNext, options).toPromise();
    }
    cloneWithHttpInfo(param, options) {
        return this.api.cloneWithHttpInfo(param.contentCloneRequestVNext, options).toPromise();
    }
    clone(param, options) {
        return this.api.clone(param.contentCloneRequestVNext, options).toPromise();
    }
    createWithHttpInfo(param, options) {
        return this.api.createWithHttpInfo(param.blogPost, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.blogPost, options).toPromise();
    }
    createBatchWithHttpInfo(param, options) {
        return this.api.createBatchWithHttpInfo(param.batchInputBlogPost, options).toPromise();
    }
    createBatch(param, options) {
        return this.api.createBatch(param.batchInputBlogPost, options).toPromise();
    }
    createLangVariationWithHttpInfo(param, options) {
        return this.api.createLangVariationWithHttpInfo(param.blogPostLanguageCloneRequestVNext, options).toPromise();
    }
    createLangVariation(param, options) {
        return this.api.createLangVariation(param.blogPostLanguageCloneRequestVNext, options).toPromise();
    }
    detachFromLangGroupWithHttpInfo(param, options) {
        return this.api.detachFromLangGroupWithHttpInfo(param.detachFromLangGroupRequestVNext, options).toPromise();
    }
    detachFromLangGroup(param, options) {
        return this.api.detachFromLangGroup(param.detachFromLangGroupRequestVNext, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.objectId, param.archived, param.property, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.objectId, param.archived, param.property, options).toPromise();
    }
    getDraftByIdWithHttpInfo(param, options) {
        return this.api.getDraftByIdWithHttpInfo(param.objectId, options).toPromise();
    }
    getDraftById(param, options) {
        return this.api.getDraftById(param.objectId, options).toPromise();
    }
    getPageWithHttpInfo(param = {}, options) {
        return this.api.getPageWithHttpInfo(param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.sort, param.after, param.limit, param.archived, param.property, options).toPromise();
    }
    getPage(param = {}, options) {
        return this.api.getPage(param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.sort, param.after, param.limit, param.archived, param.property, options).toPromise();
    }
    getPreviousVersionWithHttpInfo(param, options) {
        return this.api.getPreviousVersionWithHttpInfo(param.objectId, param.revisionId, options).toPromise();
    }
    getPreviousVersion(param, options) {
        return this.api.getPreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    getPreviousVersionsWithHttpInfo(param, options) {
        return this.api.getPreviousVersionsWithHttpInfo(param.objectId, param.after, param.before, param.limit, options).toPromise();
    }
    getPreviousVersions(param, options) {
        return this.api.getPreviousVersions(param.objectId, param.after, param.before, param.limit, options).toPromise();
    }
    pushLiveWithHttpInfo(param, options) {
        return this.api.pushLiveWithHttpInfo(param.objectId, options).toPromise();
    }
    pushLive(param, options) {
        return this.api.pushLive(param.objectId, options).toPromise();
    }
    readBatchWithHttpInfo(param, options) {
        return this.api.readBatchWithHttpInfo(param.batchInputString, param.archived, options).toPromise();
    }
    readBatch(param, options) {
        return this.api.readBatch(param.batchInputString, param.archived, options).toPromise();
    }
    resetDraftWithHttpInfo(param, options) {
        return this.api.resetDraftWithHttpInfo(param.objectId, options).toPromise();
    }
    resetDraft(param, options) {
        return this.api.resetDraft(param.objectId, options).toPromise();
    }
    restorePreviousVersionWithHttpInfo(param, options) {
        return this.api.restorePreviousVersionWithHttpInfo(param.objectId, param.revisionId, options).toPromise();
    }
    restorePreviousVersion(param, options) {
        return this.api.restorePreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    restorePreviousVersionToDraftWithHttpInfo(param, options) {
        return this.api.restorePreviousVersionToDraftWithHttpInfo(param.objectId, param.revisionId, options).toPromise();
    }
    restorePreviousVersionToDraft(param, options) {
        return this.api.restorePreviousVersionToDraft(param.objectId, param.revisionId, options).toPromise();
    }
    scheduleWithHttpInfo(param, options) {
        return this.api.scheduleWithHttpInfo(param.contentScheduleRequestVNext, options).toPromise();
    }
    schedule(param, options) {
        return this.api.schedule(param.contentScheduleRequestVNext, options).toPromise();
    }
    setLangPrimaryWithHttpInfo(param, options) {
        return this.api.setLangPrimaryWithHttpInfo(param.setNewLanguagePrimaryRequestVNext, options).toPromise();
    }
    setLangPrimary(param, options) {
        return this.api.setLangPrimary(param.setNewLanguagePrimaryRequestVNext, options).toPromise();
    }
    updateWithHttpInfo(param, options) {
        return this.api.updateWithHttpInfo(param.objectId, param.blogPost, param.archived, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.objectId, param.blogPost, param.archived, options).toPromise();
    }
    updateBatchWithHttpInfo(param, options) {
        return this.api.updateBatchWithHttpInfo(param.batchInputJsonNode, param.archived, options).toPromise();
    }
    updateBatch(param, options) {
        return this.api.updateBatch(param.batchInputJsonNode, param.archived, options).toPromise();
    }
    updateDraftWithHttpInfo(param, options) {
        return this.api.updateDraftWithHttpInfo(param.objectId, param.blogPost, options).toPromise();
    }
    updateDraft(param, options) {
        return this.api.updateDraft(param.objectId, param.blogPost, options).toPromise();
    }
    updateLangsWithHttpInfo(param, options) {
        return this.api.updateLangsWithHttpInfo(param.updateLanguagesRequestVNext, options).toPromise();
    }
    updateLangs(param, options) {
        return this.api.updateLangs(param.updateLanguagesRequestVNext, options).toPromise();
    }
}
exports.ObjectBlogPostsApi = ObjectBlogPostsApi;
