// @hubspot/api-client/lib/codegen/crm/properties/models/BatchResponseProperty.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePropertyStatusEnum = exports.BatchResponseProperty = void 0;
class BatchResponseProperty {
    static getAttributeTypeMap() {
        return BatchResponseProperty.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseProperty = BatchResponseProperty;
BatchResponseProperty.discriminator = undefined;
BatchResponseProperty.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<Property>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePropertyStatusEnum",
        "format": ""
    }
];
var BatchResponsePropertyStatusEnum;
(function (BatchResponsePropertyStatusEnum) {
    BatchResponsePropertyStatusEnum["Pending"] = "PENDING";
    BatchResponsePropertyStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePropertyStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePropertyStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePropertyStatusEnum = exports.BatchResponsePropertyStatusEnum || (exports.BatchResponsePropertyStatusEnum = {}));
