// @hubspot/api-client/lib/codegen/crm/properties/models/BatchResponsePropertyWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePropertyWithErrorsStatusEnum = exports.BatchResponsePropertyWithErrors = void 0;
class BatchResponsePropertyWithErrors {
    static getAttributeTypeMap() {
        return BatchResponsePropertyWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePropertyWithErrors = BatchResponsePropertyWithErrors;
BatchResponsePropertyWithErrors.discriminator = undefined;
BatchResponsePropertyWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<Property>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePropertyWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponsePropertyWithErrorsStatusEnum;
(function (BatchResponsePropertyWithErrorsStatusEnum) {
    BatchResponsePropertyWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponsePropertyWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePropertyWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePropertyWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePropertyWithErrorsStatusEnum = exports.BatchResponsePropertyWithErrorsStatusEnum || (exports.BatchResponsePropertyWithErrorsStatusEnum = {}));
