// @hubspot/api-client/lib/codegen/crm/properties/models/PropertyCreate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyCreateFieldTypeEnum = exports.PropertyCreateTypeEnum = exports.PropertyCreate = void 0;
class PropertyCreate {
    static getAttributeTypeMap() {
        return PropertyCreate.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PropertyCreate = PropertyCreate;
PropertyCreate.discriminator = undefined;
PropertyCreate.attributeTypeMap = [
    {
        "name": "hidden",
        "baseName": "hidden",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "displayOrder",
        "baseName": "displayOrder",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PropertyCreateTypeEnum",
        "format": ""
    },
    {
        "name": "formField",
        "baseName": "formField",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "groupName",
        "baseName": "groupName",
        "type": "string",
        "format": ""
    },
    {
        "name": "referencedObjectType",
        "baseName": "referencedObjectType",
        "type": "string",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<OptionInput>",
        "format": ""
    },
    {
        "name": "calculationFormula",
        "baseName": "calculationFormula",
        "type": "string",
        "format": ""
    },
    {
        "name": "hasUniqueValue",
        "baseName": "hasUniqueValue",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "PropertyCreateFieldTypeEnum",
        "format": ""
    },
    {
        "name": "externalOptions",
        "baseName": "externalOptions",
        "type": "boolean",
        "format": ""
    }
];
var PropertyCreateTypeEnum;
(function (PropertyCreateTypeEnum) {
    PropertyCreateTypeEnum["String"] = "string";
    PropertyCreateTypeEnum["Number"] = "number";
    PropertyCreateTypeEnum["Date"] = "date";
    PropertyCreateTypeEnum["Datetime"] = "datetime";
    PropertyCreateTypeEnum["Enumeration"] = "enumeration";
    PropertyCreateTypeEnum["Bool"] = "bool";
})(PropertyCreateTypeEnum = exports.PropertyCreateTypeEnum || (exports.PropertyCreateTypeEnum = {}));
var PropertyCreateFieldTypeEnum;
(function (PropertyCreateFieldTypeEnum) {
    PropertyCreateFieldTypeEnum["Textarea"] = "textarea";
    PropertyCreateFieldTypeEnum["Text"] = "text";
    PropertyCreateFieldTypeEnum["Date"] = "date";
    PropertyCreateFieldTypeEnum["File"] = "file";
    PropertyCreateFieldTypeEnum["Number"] = "number";
    PropertyCreateFieldTypeEnum["Select"] = "select";
    PropertyCreateFieldTypeEnum["Radio"] = "radio";
    PropertyCreateFieldTypeEnum["Checkbox"] = "checkbox";
    PropertyCreateFieldTypeEnum["Booleancheckbox"] = "booleancheckbox";
    PropertyCreateFieldTypeEnum["CalculationEquation"] = "calculation_equation";
})(PropertyCreateFieldTypeEnum = exports.PropertyCreateFieldTypeEnum || (exports.PropertyCreateFieldTypeEnum = {}));
