// @hubspot/api-client/lib/codegen/crm/properties/models/PropertyUpdate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyUpdateFieldTypeEnum = exports.PropertyUpdateTypeEnum = exports.PropertyUpdate = void 0;
class PropertyUpdate {
    static getAttributeTypeMap() {
        return PropertyUpdate.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PropertyUpdate = PropertyUpdate;
PropertyUpdate.discriminator = undefined;
PropertyUpdate.attributeTypeMap = [
    {
        "name": "groupName",
        "baseName": "groupName",
        "type": "string",
        "format": ""
    },
    {
        "name": "hidden",
        "baseName": "hidden",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<OptionInput>",
        "format": ""
    },
    {
        "name": "displayOrder",
        "baseName": "displayOrder",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "calculationFormula",
        "baseName": "calculationFormula",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PropertyUpdateTypeEnum",
        "format": ""
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "PropertyUpdateFieldTypeEnum",
        "format": ""
    },
    {
        "name": "formField",
        "baseName": "formField",
        "type": "boolean",
        "format": ""
    }
];
var PropertyUpdateTypeEnum;
(function (PropertyUpdateTypeEnum) {
    PropertyUpdateTypeEnum["String"] = "string";
    PropertyUpdateTypeEnum["Number"] = "number";
    PropertyUpdateTypeEnum["Date"] = "date";
    PropertyUpdateTypeEnum["Datetime"] = "datetime";
    PropertyUpdateTypeEnum["Enumeration"] = "enumeration";
    PropertyUpdateTypeEnum["Bool"] = "bool";
})(PropertyUpdateTypeEnum = exports.PropertyUpdateTypeEnum || (exports.PropertyUpdateTypeEnum = {}));
var PropertyUpdateFieldTypeEnum;
(function (PropertyUpdateFieldTypeEnum) {
    PropertyUpdateFieldTypeEnum["Textarea"] = "textarea";
    PropertyUpdateFieldTypeEnum["Text"] = "text";
    PropertyUpdateFieldTypeEnum["Date"] = "date";
    PropertyUpdateFieldTypeEnum["File"] = "file";
    PropertyUpdateFieldTypeEnum["Number"] = "number";
    PropertyUpdateFieldTypeEnum["Select"] = "select";
    PropertyUpdateFieldTypeEnum["Radio"] = "radio";
    PropertyUpdateFieldTypeEnum["Checkbox"] = "checkbox";
    PropertyUpdateFieldTypeEnum["Booleancheckbox"] = "booleancheckbox";
    PropertyUpdateFieldTypeEnum["CalculationEquation"] = "calculation_equation";
})(PropertyUpdateFieldTypeEnum = exports.PropertyUpdateFieldTypeEnum || (exports.PropertyUpdateFieldTypeEnum = {}));
