// @hubspot/api-client/lib/codegen/cms/hubdb/models/BatchResponseHubDbTableRowV3.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseHubDbTableRowV3StatusEnum = exports.BatchResponseHubDbTableRowV3 = void 0;
class BatchResponseHubDbTableRowV3 {
    static getAttributeTypeMap() {
        return BatchResponseHubDbTableRowV3.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseHubDbTableRowV3 = BatchResponseHubDbTableRowV3;
BatchResponseHubDbTableRowV3.discriminator = undefined;
BatchResponseHubDbTableRowV3.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<HubDbTableRowV3>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseHubDbTableRowV3StatusEnum",
        "format": ""
    }
];
var BatchResponseHubDbTableRowV3StatusEnum;
(function (BatchResponseHubDbTableRowV3StatusEnum) {
    BatchResponseHubDbTableRowV3StatusEnum["Pending"] = "PENDING";
    BatchResponseHubDbTableRowV3StatusEnum["Processing"] = "PROCESSING";
    BatchResponseHubDbTableRowV3StatusEnum["Canceled"] = "CANCELED";
    BatchResponseHubDbTableRowV3StatusEnum["Complete"] = "COMPLETE";
})(BatchResponseHubDbTableRowV3StatusEnum = exports.BatchResponseHubDbTableRowV3StatusEnum || (exports.BatchResponseHubDbTableRowV3StatusEnum = {}));
