// @hubspot/api-client/lib/codegen/cms/hubdb/models/BatchResponseHubDbTableRowV3WithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseHubDbTableRowV3WithErrorsStatusEnum = exports.BatchResponseHubDbTableRowV3WithErrors = void 0;
class BatchResponseHubDbTableRowV3WithErrors {
    static getAttributeTypeMap() {
        return BatchResponseHubDbTableRowV3WithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseHubDbTableRowV3WithErrors = BatchResponseHubDbTableRowV3WithErrors;
BatchResponseHubDbTableRowV3WithErrors.discriminator = undefined;
BatchResponseHubDbTableRowV3WithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<HubDbTableRowV3>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseHubDbTableRowV3WithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseHubDbTableRowV3WithErrorsStatusEnum;
(function (BatchResponseHubDbTableRowV3WithErrorsStatusEnum) {
    BatchResponseHubDbTableRowV3WithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseHubDbTableRowV3WithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseHubDbTableRowV3WithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseHubDbTableRowV3WithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseHubDbTableRowV3WithErrorsStatusEnum = exports.BatchResponseHubDbTableRowV3WithErrorsStatusEnum || (exports.BatchResponseHubDbTableRowV3WithErrorsStatusEnum = {}));
