// @hubspot/api-client/lib/codegen/cms/hubdb/models/ColumnRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ColumnRequestTypeEnum = exports.ColumnRequest = void 0;
class ColumnRequest {
    static getAttributeTypeMap() {
        return ColumnRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ColumnRequest = ColumnRequest;
ColumnRequest.discriminator = undefined;
ColumnRequest.attributeTypeMap = [
    {
        "name": "foreignTableId",
        "baseName": "foreignTableId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<Option>",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ColumnRequestTypeEnum",
        "format": ""
    },
    {
        "name": "foreignColumnId",
        "baseName": "foreignColumnId",
        "type": "number",
        "format": "int32"
    }
];
var ColumnRequestTypeEnum;
(function (ColumnRequestTypeEnum) {
    ColumnRequestTypeEnum["Null"] = "NULL";
    ColumnRequestTypeEnum["Text"] = "TEXT";
    ColumnRequestTypeEnum["Number"] = "NUMBER";
    ColumnRequestTypeEnum["Url"] = "URL";
    ColumnRequestTypeEnum["Image"] = "IMAGE";
    ColumnRequestTypeEnum["Select"] = "SELECT";
    ColumnRequestTypeEnum["Multiselect"] = "MULTISELECT";
    ColumnRequestTypeEnum["Boolean"] = "BOOLEAN";
    ColumnRequestTypeEnum["Location"] = "LOCATION";
    ColumnRequestTypeEnum["Date"] = "DATE";
    ColumnRequestTypeEnum["Datetime"] = "DATETIME";
    ColumnRequestTypeEnum["Currency"] = "CURRENCY";
    ColumnRequestTypeEnum["Richtext"] = "RICHTEXT";
    ColumnRequestTypeEnum["ForeignId"] = "FOREIGN_ID";
    ColumnRequestTypeEnum["Video"] = "VIDEO";
    ColumnRequestTypeEnum["Cta"] = "CTA";
    ColumnRequestTypeEnum["File"] = "FILE";
    ColumnRequestTypeEnum["Json"] = "JSON";
    ColumnRequestTypeEnum["Composite"] = "COMPOSITE";
    ColumnRequestTypeEnum["Code"] = "CODE";
})(ColumnRequestTypeEnum = exports.ColumnRequestTypeEnum || (exports.ColumnRequestTypeEnum = {}));
