// @hubspot/api-client/lib/codegen/communication_preferences/models/PublicSubscriptionStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicSubscriptionStatusSourceOfStatusEnum = exports.PublicSubscriptionStatusStatusEnum = exports.PublicSubscriptionStatusLegalBasisEnum = exports.PublicSubscriptionStatus = void 0;
class PublicSubscriptionStatus {
    static getAttributeTypeMap() {
        return PublicSubscriptionStatus.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicSubscriptionStatus = PublicSubscriptionStatus;
PublicSubscriptionStatus.discriminator = undefined;
PublicSubscriptionStatus.attributeTypeMap = [
    {
        "name": "brandId",
        "baseName": "brandId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalBasis",
        "baseName": "legalBasis",
        "type": "PublicSubscriptionStatusLegalBasisEnum",
        "format": ""
    },
    {
        "name": "preferenceGroupName",
        "baseName": "preferenceGroupName",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalBasisExplanation",
        "baseName": "legalBasisExplanation",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PublicSubscriptionStatusStatusEnum",
        "format": ""
    },
    {
        "name": "sourceOfStatus",
        "baseName": "sourceOfStatus",
        "type": "PublicSubscriptionStatusSourceOfStatusEnum",
        "format": ""
    }
];
var PublicSubscriptionStatusLegalBasisEnum;
(function (PublicSubscriptionStatusLegalBasisEnum) {
    PublicSubscriptionStatusLegalBasisEnum["LegitimateInterestPql"] = "LEGITIMATE_INTEREST_PQL";
    PublicSubscriptionStatusLegalBasisEnum["LegitimateInterestClient"] = "LEGITIMATE_INTEREST_CLIENT";
    PublicSubscriptionStatusLegalBasisEnum["PerformanceOfContract"] = "PERFORMANCE_OF_CONTRACT";
    PublicSubscriptionStatusLegalBasisEnum["ConsentWithNotice"] = "CONSENT_WITH_NOTICE";
    PublicSubscriptionStatusLegalBasisEnum["NonGdpr"] = "NON_GDPR";
    PublicSubscriptionStatusLegalBasisEnum["ProcessAndStore"] = "PROCESS_AND_STORE";
    PublicSubscriptionStatusLegalBasisEnum["LegitimateInterestOther"] = "LEGITIMATE_INTEREST_OTHER";
})(PublicSubscriptionStatusLegalBasisEnum = exports.PublicSubscriptionStatusLegalBasisEnum || (exports.PublicSubscriptionStatusLegalBasisEnum = {}));
var PublicSubscriptionStatusStatusEnum;
(function (PublicSubscriptionStatusStatusEnum) {
    PublicSubscriptionStatusStatusEnum["Subscribed"] = "SUBSCRIBED";
    PublicSubscriptionStatusStatusEnum["NotSubscribed"] = "NOT_SUBSCRIBED";
})(PublicSubscriptionStatusStatusEnum = exports.PublicSubscriptionStatusStatusEnum || (exports.PublicSubscriptionStatusStatusEnum = {}));
var PublicSubscriptionStatusSourceOfStatusEnum;
(function (PublicSubscriptionStatusSourceOfStatusEnum) {
    PublicSubscriptionStatusSourceOfStatusEnum["PortalWideStatus"] = "PORTAL_WIDE_STATUS";
    PublicSubscriptionStatusSourceOfStatusEnum["BrandWideStatus"] = "BRAND_WIDE_STATUS";
    PublicSubscriptionStatusSourceOfStatusEnum["SubscriptionStatus"] = "SUBSCRIPTION_STATUS";
})(PublicSubscriptionStatusSourceOfStatusEnum = exports.PublicSubscriptionStatusSourceOfStatusEnum || (exports.PublicSubscriptionStatusSourceOfStatusEnum = {}));
