// @hubspot/api-client/lib/codegen/communication_preferences/models/PublicUpdateSubscriptionStatusRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicUpdateSubscriptionStatusRequestLegalBasisEnum = exports.PublicUpdateSubscriptionStatusRequest = void 0;
class PublicUpdateSubscriptionStatusRequest {
    static getAttributeTypeMap() {
        return PublicUpdateSubscriptionStatusRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicUpdateSubscriptionStatusRequest = PublicUpdateSubscriptionStatusRequest;
PublicUpdateSubscriptionStatusRequest.discriminator = undefined;
PublicUpdateSubscriptionStatusRequest.attributeTypeMap = [
    {
        "name": "emailAddress",
        "baseName": "emailAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalBasis",
        "baseName": "legalBasis",
        "type": "PublicUpdateSubscriptionStatusRequestLegalBasisEnum",
        "format": ""
    },
    {
        "name": "subscriptionId",
        "baseName": "subscriptionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalBasisExplanation",
        "baseName": "legalBasisExplanation",
        "type": "string",
        "format": ""
    }
];
var PublicUpdateSubscriptionStatusRequestLegalBasisEnum;
(function (PublicUpdateSubscriptionStatusRequestLegalBasisEnum) {
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["LegitimateInterestPql"] = "LEGITIMATE_INTEREST_PQL";
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["LegitimateInterestClient"] = "LEGITIMATE_INTEREST_CLIENT";
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["PerformanceOfContract"] = "PERFORMANCE_OF_CONTRACT";
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["ConsentWithNotice"] = "CONSENT_WITH_NOTICE";
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["NonGdpr"] = "NON_GDPR";
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["ProcessAndStore"] = "PROCESS_AND_STORE";
    PublicUpdateSubscriptionStatusRequestLegalBasisEnum["LegitimateInterestOther"] = "LEGITIMATE_INTEREST_OTHER";
})(PublicUpdateSubscriptionStatusRequestLegalBasisEnum = exports.PublicUpdateSubscriptionStatusRequestLegalBasisEnum || (exports.PublicUpdateSubscriptionStatusRequestLegalBasisEnum = {}));
