// @hubspot/api-client/lib/codegen/crm/associations/models/BatchResponsePublicAssociationWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationWithErrorsStatusEnum = exports.BatchResponsePublicAssociationWithErrors = void 0;
class BatchResponsePublicAssociationWithErrors {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationWithErrors = BatchResponsePublicAssociationWithErrors;
BatchResponsePublicAssociationWithErrors.discriminator = undefined;
BatchResponsePublicAssociationWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociation>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationWithErrorsStatusEnum;
(function (BatchResponsePublicAssociationWithErrorsStatusEnum) {
    BatchResponsePublicAssociationWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationWithErrorsStatusEnum = exports.BatchResponsePublicAssociationWithErrorsStatusEnum || (exports.BatchResponsePublicAssociationWithErrorsStatusEnum = {}));
