// @hubspot/api-client/lib/codegen/crm/associations/models/BatchResponsePublicAssociationMultiWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationMultiWithErrorsStatusEnum = exports.BatchResponsePublicAssociationMultiWithErrors = void 0;
class BatchResponsePublicAssociationMultiWithErrors {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationMultiWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationMultiWithErrors = BatchResponsePublicAssociationMultiWithErrors;
BatchResponsePublicAssociationMultiWithErrors.discriminator = undefined;
BatchResponsePublicAssociationMultiWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationMulti>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationMultiWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationMultiWithErrorsStatusEnum;
(function (BatchResponsePublicAssociationMultiWithErrorsStatusEnum) {
    BatchResponsePublicAssociationMultiWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationMultiWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationMultiWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationMultiWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationMultiWithErrorsStatusEnum = exports.BatchResponsePublicAssociationMultiWithErrorsStatusEnum || (exports.BatchResponsePublicAssociationMultiWithErrorsStatusEnum = {}));
