// @hubspot/api-client/lib/codegen/crm/associations/models/BatchResponsePublicAssociation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationStatusEnum = exports.BatchResponsePublicAssociation = void 0;
class BatchResponsePublicAssociation {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociation.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociation = BatchResponsePublicAssociation;
BatchResponsePublicAssociation.discriminator = undefined;
BatchResponsePublicAssociation.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociation>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationStatusEnum;
(function (BatchResponsePublicAssociationStatusEnum) {
    BatchResponsePublicAssociationStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationStatusEnum = exports.BatchResponsePublicAssociationStatusEnum || (exports.BatchResponsePublicAssociationStatusEnum = {}));
