// @hubspot/api-client/lib/codegen/crm/associations/models/BatchResponsePublicAssociationMulti.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationMultiStatusEnum = exports.BatchResponsePublicAssociationMulti = void 0;
class BatchResponsePublicAssociationMulti {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationMulti.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationMulti = BatchResponsePublicAssociationMulti;
BatchResponsePublicAssociationMulti.discriminator = undefined;
BatchResponsePublicAssociationMulti.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationMulti>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationMultiStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationMultiStatusEnum;
(function (BatchResponsePublicAssociationMultiStatusEnum) {
    BatchResponsePublicAssociationMultiStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationMultiStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationMultiStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationMultiStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationMultiStatusEnum = exports.BatchResponsePublicAssociationMultiStatusEnum || (exports.BatchResponsePublicAssociationMultiStatusEnum = {}));
