// @hubspot/api-client/lib/codegen/marketing/transactional/models/EmailSendStatusView.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailSendStatusViewStatusEnum = exports.EmailSendStatusViewSendResultEnum = exports.EmailSendStatusView = void 0;
class EmailSendStatusView {
    static getAttributeTypeMap() {
        return EmailSendStatusView.attributeTypeMap;
    }
    constructor() {
    }
}
exports.EmailSendStatusView = EmailSendStatusView;
EmailSendStatusView.discriminator = undefined;
EmailSendStatusView.attributeTypeMap = [
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "EventIdView",
        "format": ""
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "statusId",
        "baseName": "statusId",
        "type": "string",
        "format": ""
    },
    {
        "name": "sendResult",
        "baseName": "sendResult",
        "type": "EmailSendStatusViewSendResultEnum",
        "format": ""
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "EmailSendStatusViewStatusEnum",
        "format": ""
    }
];
var EmailSendStatusViewSendResultEnum;
(function (EmailSendStatusViewSendResultEnum) {
    EmailSendStatusViewSendResultEnum["Sent"] = "SENT";
    EmailSendStatusViewSendResultEnum["IdempotentIgnore"] = "IDEMPOTENT_IGNORE";
    EmailSendStatusViewSendResultEnum["Queued"] = "QUEUED";
    EmailSendStatusViewSendResultEnum["IdempotentFail"] = "IDEMPOTENT_FAIL";
    EmailSendStatusViewSendResultEnum["Throttled"] = "THROTTLED";
    EmailSendStatusViewSendResultEnum["EmailDisabled"] = "EMAIL_DISABLED";
    EmailSendStatusViewSendResultEnum["PortalSuspended"] = "PORTAL_SUSPENDED";
    EmailSendStatusViewSendResultEnum["InvalidToAddress"] = "INVALID_TO_ADDRESS";
    EmailSendStatusViewSendResultEnum["BlockedDomain"] = "BLOCKED_DOMAIN";
    EmailSendStatusViewSendResultEnum["PreviouslyBounced"] = "PREVIOUSLY_BOUNCED";
    EmailSendStatusViewSendResultEnum["EmailUnconfirmed"] = "EMAIL_UNCONFIRMED";
    EmailSendStatusViewSendResultEnum["PreviousSpam"] = "PREVIOUS_SPAM";
    EmailSendStatusViewSendResultEnum["PreviouslyUnsubscribedMessage"] = "PREVIOUSLY_UNSUBSCRIBED_MESSAGE";
    EmailSendStatusViewSendResultEnum["PreviouslyUnsubscribedPortal"] = "PREVIOUSLY_UNSUBSCRIBED_PORTAL";
    EmailSendStatusViewSendResultEnum["InvalidFromAddress"] = "INVALID_FROM_ADDRESS";
    EmailSendStatusViewSendResultEnum["CampaignCancelled"] = "CAMPAIGN_CANCELLED";
    EmailSendStatusViewSendResultEnum["ValidationFailed"] = "VALIDATION_FAILED";
    EmailSendStatusViewSendResultEnum["MtaIgnore"] = "MTA_IGNORE";
    EmailSendStatusViewSendResultEnum["BlockedAddress"] = "BLOCKED_ADDRESS";
    EmailSendStatusViewSendResultEnum["PortalOverLimit"] = "PORTAL_OVER_LIMIT";
    EmailSendStatusViewSendResultEnum["PortalExpired"] = "PORTAL_EXPIRED";
    EmailSendStatusViewSendResultEnum["PortalMissingMarketingScope"] = "PORTAL_MISSING_MARKETING_SCOPE";
    EmailSendStatusViewSendResultEnum["MissingTemplateProperties"] = "MISSING_TEMPLATE_PROPERTIES";
    EmailSendStatusViewSendResultEnum["MissingRequiredParameter"] = "MISSING_REQUIRED_PARAMETER";
    EmailSendStatusViewSendResultEnum["PortalAuthenticationFailure"] = "PORTAL_AUTHENTICATION_FAILURE";
    EmailSendStatusViewSendResultEnum["MissingContent"] = "MISSING_CONTENT";
    EmailSendStatusViewSendResultEnum["CorruptInput"] = "CORRUPT_INPUT";
    EmailSendStatusViewSendResultEnum["TemplateRenderException"] = "TEMPLATE_RENDER_EXCEPTION";
    EmailSendStatusViewSendResultEnum["GraymailSuppressed"] = "GRAYMAIL_SUPPRESSED";
    EmailSendStatusViewSendResultEnum["UnconfiguredSendingDomain"] = "UNCONFIGURED_SENDING_DOMAIN";
    EmailSendStatusViewSendResultEnum["Undeliverable"] = "UNDELIVERABLE";
    EmailSendStatusViewSendResultEnum["CancelledAbuse"] = "CANCELLED_ABUSE";
    EmailSendStatusViewSendResultEnum["QuarantinedAddress"] = "QUARANTINED_ADDRESS";
    EmailSendStatusViewSendResultEnum["AddressOnlyAcceptedOnProd"] = "ADDRESS_ONLY_ACCEPTED_ON_PROD";
    EmailSendStatusViewSendResultEnum["PortalNotAuthorizedForApplication"] = "PORTAL_NOT_AUTHORIZED_FOR_APPLICATION";
    EmailSendStatusViewSendResultEnum["AddressListBombed"] = "ADDRESS_LIST_BOMBED";
    EmailSendStatusViewSendResultEnum["AddressOptedOut"] = "ADDRESS_OPTED_OUT";
    EmailSendStatusViewSendResultEnum["RecipientFatigueSuppressed"] = "RECIPIENT_FATIGUE_SUPPRESSED";
    EmailSendStatusViewSendResultEnum["TooManyRecipients"] = "TOO_MANY_RECIPIENTS";
    EmailSendStatusViewSendResultEnum["PreviouslyUnsubscribedBrand"] = "PREVIOUSLY_UNSUBSCRIBED_BRAND";
    EmailSendStatusViewSendResultEnum["NonMarketableContact"] = "NON_MARKETABLE_CONTACT";
    EmailSendStatusViewSendResultEnum["PreviouslyUnsubscribedBusinessUnit"] = "PREVIOUSLY_UNSUBSCRIBED_BUSINESS_UNIT";
    EmailSendStatusViewSendResultEnum["GdprDoiEnabled"] = "GDPR_DOI_ENABLED";
})(EmailSendStatusViewSendResultEnum = exports.EmailSendStatusViewSendResultEnum || (exports.EmailSendStatusViewSendResultEnum = {}));
var EmailSendStatusViewStatusEnum;
(function (EmailSendStatusViewStatusEnum) {
    EmailSendStatusViewStatusEnum["Pending"] = "PENDING";
    EmailSendStatusViewStatusEnum["Processing"] = "PROCESSING";
    EmailSendStatusViewStatusEnum["Canceled"] = "CANCELED";
    EmailSendStatusViewStatusEnum["Complete"] = "COMPLETE";
})(EmailSendStatusViewStatusEnum = exports.EmailSendStatusViewStatusEnum || (exports.EmailSendStatusViewStatusEnum = {}));
