// @hubspot/api-client/lib/codegen/cms/pages/models/BatchResponsePageWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePageWithErrorsStatusEnum = exports.BatchResponsePageWithErrors = void 0;
class BatchResponsePageWithErrors {
    static getAttributeTypeMap() {
        return BatchResponsePageWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePageWithErrors = BatchResponsePageWithErrors;
BatchResponsePageWithErrors.discriminator = undefined;
BatchResponsePageWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<Page>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePageWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponsePageWithErrorsStatusEnum;
(function (BatchResponsePageWithErrorsStatusEnum) {
    BatchResponsePageWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponsePageWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePageWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePageWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePageWithErrorsStatusEnum = exports.BatchResponsePageWithErrorsStatusEnum || (exports.BatchResponsePageWithErrorsStatusEnum = {}));
