// @hubspot/api-client/lib/codegen/files/models/FileActionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileActionResponseStatusEnum = exports.FileActionResponse = void 0;
class FileActionResponse {
    static getAttributeTypeMap() {
        return FileActionResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FileActionResponse = FileActionResponse;
FileActionResponse.discriminator = undefined;
FileActionResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "result",
        "type": "any",
        "format": ""
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "taskId",
        "baseName": "taskId",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "FileActionResponseStatusEnum",
        "format": ""
    }
];
var FileActionResponseStatusEnum;
(function (FileActionResponseStatusEnum) {
    FileActionResponseStatusEnum["Pending"] = "PENDING";
    FileActionResponseStatusEnum["Processing"] = "PROCESSING";
    FileActionResponseStatusEnum["Canceled"] = "CANCELED";
    FileActionResponseStatusEnum["Complete"] = "COMPLETE";
})(FileActionResponseStatusEnum = exports.FileActionResponseStatusEnum || (exports.FileActionResponseStatusEnum = {}));
