// @hubspot/api-client/lib/codegen/files/models/FolderActionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FolderActionResponseStatusEnum = exports.FolderActionResponse = void 0;
class FolderActionResponse {
    static getAttributeTypeMap() {
        return FolderActionResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FolderActionResponse = FolderActionResponse;
FolderActionResponse.discriminator = undefined;
FolderActionResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "result",
        "type": "Folder",
        "format": ""
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "taskId",
        "baseName": "taskId",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "FolderActionResponseStatusEnum",
        "format": ""
    }
];
var FolderActionResponseStatusEnum;
(function (FolderActionResponseStatusEnum) {
    FolderActionResponseStatusEnum["Pending"] = "PENDING";
    FolderActionResponseStatusEnum["Processing"] = "PROCESSING";
    FolderActionResponseStatusEnum["Canceled"] = "CANCELED";
    FolderActionResponseStatusEnum["Complete"] = "COMPLETE";
})(FolderActionResponseStatusEnum = exports.FolderActionResponseStatusEnum || (exports.FolderActionResponseStatusEnum = {}));
