// @hubspot/api-client/lib/codegen/files/models/ImportFromUrlInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportFromUrlInputDuplicateValidationStrategyEnum = exports.ImportFromUrlInputDuplicateValidationScopeEnum = exports.ImportFromUrlInputAccessEnum = exports.ImportFromUrlInput = void 0;
class ImportFromUrlInput {
    static getAttributeTypeMap() {
        return ImportFromUrlInput.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ImportFromUrlInput = ImportFromUrlInput;
ImportFromUrlInput.discriminator = undefined;
ImportFromUrlInput.attributeTypeMap = [
    {
        "name": "folderPath",
        "baseName": "folderPath",
        "type": "string",
        "format": ""
    },
    {
        "name": "access",
        "baseName": "access",
        "type": "ImportFromUrlInputAccessEnum",
        "format": ""
    },
    {
        "name": "duplicateValidationScope",
        "baseName": "duplicateValidationScope",
        "type": "ImportFromUrlInputDuplicateValidationScopeEnum",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "duplicateValidationStrategy",
        "baseName": "duplicateValidationStrategy",
        "type": "ImportFromUrlInputDuplicateValidationStrategyEnum",
        "format": ""
    },
    {
        "name": "ttl",
        "baseName": "ttl",
        "type": "string",
        "format": ""
    },
    {
        "name": "overwrite",
        "baseName": "overwrite",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    },
    {
        "name": "folderId",
        "baseName": "folderId",
        "type": "string",
        "format": ""
    }
];
var ImportFromUrlInputAccessEnum;
(function (ImportFromUrlInputAccessEnum) {
    ImportFromUrlInputAccessEnum["PublicIndexable"] = "PUBLIC_INDEXABLE";
    ImportFromUrlInputAccessEnum["PublicNotIndexable"] = "PUBLIC_NOT_INDEXABLE";
    ImportFromUrlInputAccessEnum["HiddenIndexable"] = "HIDDEN_INDEXABLE";
    ImportFromUrlInputAccessEnum["HiddenNotIndexable"] = "HIDDEN_NOT_INDEXABLE";
    ImportFromUrlInputAccessEnum["HiddenPrivate"] = "HIDDEN_PRIVATE";
    ImportFromUrlInputAccessEnum["Private"] = "PRIVATE";
})(ImportFromUrlInputAccessEnum = exports.ImportFromUrlInputAccessEnum || (exports.ImportFromUrlInputAccessEnum = {}));
var ImportFromUrlInputDuplicateValidationScopeEnum;
(function (ImportFromUrlInputDuplicateValidationScopeEnum) {
    ImportFromUrlInputDuplicateValidationScopeEnum["EntirePortal"] = "ENTIRE_PORTAL";
    ImportFromUrlInputDuplicateValidationScopeEnum["ExactFolder"] = "EXACT_FOLDER";
})(ImportFromUrlInputDuplicateValidationScopeEnum = exports.ImportFromUrlInputDuplicateValidationScopeEnum || (exports.ImportFromUrlInputDuplicateValidationScopeEnum = {}));
var ImportFromUrlInputDuplicateValidationStrategyEnum;
(function (ImportFromUrlInputDuplicateValidationStrategyEnum) {
    ImportFromUrlInputDuplicateValidationStrategyEnum["None"] = "NONE";
    ImportFromUrlInputDuplicateValidationStrategyEnum["Reject"] = "REJECT";
    ImportFromUrlInputDuplicateValidationStrategyEnum["ReturnExisting"] = "RETURN_EXISTING";
})(ImportFromUrlInputDuplicateValidationStrategyEnum = exports.ImportFromUrlInputDuplicateValidationStrategyEnum || (exports.ImportFromUrlInputDuplicateValidationStrategyEnum = {}));
