// @hubspot/api-client/lib/codegen/files/models/ModelFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelFileAccessEnum = exports.ModelFile = void 0;
class ModelFile {
    static getAttributeTypeMap() {
        return ModelFile.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ModelFile = ModelFile;
ModelFile.discriminator = undefined;
ModelFile.attributeTypeMap = [
    {
        "name": "extension",
        "baseName": "extension",
        "type": "string",
        "format": ""
    },
    {
        "name": "access",
        "baseName": "access",
        "type": "ModelFileAccessEnum",
        "format": ""
    },
    {
        "name": "parentFolderId",
        "baseName": "parentFolderId",
        "type": "string",
        "format": ""
    },
    {
        "name": "encoding",
        "baseName": "encoding",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string",
        "format": ""
    },
    {
        "name": "isUsableInContent",
        "baseName": "isUsableInContent",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiresAt",
        "baseName": "expiresAt",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "archivedAt",
        "baseName": "archivedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "path",
        "baseName": "path",
        "type": "string",
        "format": ""
    },
    {
        "name": "size",
        "baseName": "size",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "width",
        "baseName": "width",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "defaultHostingUrl",
        "baseName": "defaultHostingUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "height",
        "baseName": "height",
        "type": "number",
        "format": "int32"
    }
];
var ModelFileAccessEnum;
(function (ModelFileAccessEnum) {
    ModelFileAccessEnum["PublicIndexable"] = "PUBLIC_INDEXABLE";
    ModelFileAccessEnum["PublicNotIndexable"] = "PUBLIC_NOT_INDEXABLE";
    ModelFileAccessEnum["HiddenIndexable"] = "HIDDEN_INDEXABLE";
    ModelFileAccessEnum["HiddenNotIndexable"] = "HIDDEN_NOT_INDEXABLE";
    ModelFileAccessEnum["HiddenPrivate"] = "HIDDEN_PRIVATE";
    ModelFileAccessEnum["Private"] = "PRIVATE";
})(ModelFileAccessEnum = exports.ModelFileAccessEnum || (exports.ModelFileAccessEnum = {}));
