// @hubspot/api-client/lib/codegen/automation/actions/models/FieldTypeDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldTypeDefinitionFieldTypeEnum = exports.FieldTypeDefinitionTypeEnum = exports.FieldTypeDefinitionReferencedObjectTypeEnum = exports.FieldTypeDefinition = void 0;
class FieldTypeDefinition {
    static getAttributeTypeMap() {
        return FieldTypeDefinition.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FieldTypeDefinition = FieldTypeDefinition;
FieldTypeDefinition.discriminator = undefined;
FieldTypeDefinition.attributeTypeMap = [
    {
        "name": "helpText",
        "baseName": "helpText",
        "type": "string",
        "format": ""
    },
    {
        "name": "referencedObjectType",
        "baseName": "referencedObjectType",
        "type": "FieldTypeDefinitionReferencedObjectTypeEnum",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<Option>",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "externalOptionsReferenceType",
        "baseName": "externalOptionsReferenceType",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "FieldTypeDefinitionTypeEnum",
        "format": ""
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "FieldTypeDefinitionFieldTypeEnum",
        "format": ""
    },
    {
        "name": "optionsUrl",
        "baseName": "optionsUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "externalOptions",
        "baseName": "externalOptions",
        "type": "boolean",
        "format": ""
    }
];
var FieldTypeDefinitionReferencedObjectTypeEnum;
(function (FieldTypeDefinitionReferencedObjectTypeEnum) {
    FieldTypeDefinitionReferencedObjectTypeEnum["Contact"] = "CONTACT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Company"] = "COMPANY";
    FieldTypeDefinitionReferencedObjectTypeEnum["Deal"] = "DEAL";
    FieldTypeDefinitionReferencedObjectTypeEnum["Engagement"] = "ENGAGEMENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Ticket"] = "TICKET";
    FieldTypeDefinitionReferencedObjectTypeEnum["Owner"] = "OWNER";
    FieldTypeDefinitionReferencedObjectTypeEnum["Product"] = "PRODUCT";
    FieldTypeDefinitionReferencedObjectTypeEnum["LineItem"] = "LINE_ITEM";
    FieldTypeDefinitionReferencedObjectTypeEnum["BetDeliverableService"] = "BET_DELIVERABLE_SERVICE";
    FieldTypeDefinitionReferencedObjectTypeEnum["Content"] = "CONTENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Conversation"] = "CONVERSATION";
    FieldTypeDefinitionReferencedObjectTypeEnum["BetAlert"] = "BET_ALERT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Portal"] = "PORTAL";
    FieldTypeDefinitionReferencedObjectTypeEnum["Quote"] = "QUOTE";
    FieldTypeDefinitionReferencedObjectTypeEnum["FormSubmissionInbounddb"] = "FORM_SUBMISSION_INBOUNDDB";
    FieldTypeDefinitionReferencedObjectTypeEnum["Quota"] = "QUOTA";
    FieldTypeDefinitionReferencedObjectTypeEnum["Unsubscribe"] = "UNSUBSCRIBE";
    FieldTypeDefinitionReferencedObjectTypeEnum["Communication"] = "COMMUNICATION";
    FieldTypeDefinitionReferencedObjectTypeEnum["FeedbackSubmission"] = "FEEDBACK_SUBMISSION";
    FieldTypeDefinitionReferencedObjectTypeEnum["Attribution"] = "ATTRIBUTION";
    FieldTypeDefinitionReferencedObjectTypeEnum["SalesforceSyncError"] = "SALESFORCE_SYNC_ERROR";
    FieldTypeDefinitionReferencedObjectTypeEnum["RestorableCrmObject"] = "RESTORABLE_CRM_OBJECT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Hub"] = "HUB";
    FieldTypeDefinitionReferencedObjectTypeEnum["LandingPage"] = "LANDING_PAGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["ProductOrFolder"] = "PRODUCT_OR_FOLDER";
    FieldTypeDefinitionReferencedObjectTypeEnum["Task"] = "TASK";
    FieldTypeDefinitionReferencedObjectTypeEnum["Form"] = "FORM";
    FieldTypeDefinitionReferencedObjectTypeEnum["MarketingEmail"] = "MARKETING_EMAIL";
    FieldTypeDefinitionReferencedObjectTypeEnum["AdAccount"] = "AD_ACCOUNT";
    FieldTypeDefinitionReferencedObjectTypeEnum["AdCampaign"] = "AD_CAMPAIGN";
    FieldTypeDefinitionReferencedObjectTypeEnum["AdGroup"] = "AD_GROUP";
    FieldTypeDefinitionReferencedObjectTypeEnum["Ad"] = "AD";
    FieldTypeDefinitionReferencedObjectTypeEnum["Keyword"] = "KEYWORD";
    FieldTypeDefinitionReferencedObjectTypeEnum["Campaign"] = "CAMPAIGN";
    FieldTypeDefinitionReferencedObjectTypeEnum["SocialChannel"] = "SOCIAL_CHANNEL";
    FieldTypeDefinitionReferencedObjectTypeEnum["SocialPost"] = "SOCIAL_POST";
    FieldTypeDefinitionReferencedObjectTypeEnum["SitePage"] = "SITE_PAGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["BlogPost"] = "BLOG_POST";
    FieldTypeDefinitionReferencedObjectTypeEnum["Import"] = "IMPORT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Export"] = "EXPORT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Cta"] = "CTA";
    FieldTypeDefinitionReferencedObjectTypeEnum["TaskTemplate"] = "TASK_TEMPLATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["AutomationPlatformFlow"] = "AUTOMATION_PLATFORM_FLOW";
    FieldTypeDefinitionReferencedObjectTypeEnum["ObjectList"] = "OBJECT_LIST";
    FieldTypeDefinitionReferencedObjectTypeEnum["Note"] = "NOTE";
    FieldTypeDefinitionReferencedObjectTypeEnum["MeetingEvent"] = "MEETING_EVENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Call"] = "CALL";
    FieldTypeDefinitionReferencedObjectTypeEnum["Email"] = "EMAIL";
    FieldTypeDefinitionReferencedObjectTypeEnum["PublishingTask"] = "PUBLISHING_TASK";
    FieldTypeDefinitionReferencedObjectTypeEnum["ConversationSession"] = "CONVERSATION_SESSION";
    FieldTypeDefinitionReferencedObjectTypeEnum["ContactCreateAttribution"] = "CONTACT_CREATE_ATTRIBUTION";
    FieldTypeDefinitionReferencedObjectTypeEnum["Invoice"] = "INVOICE";
    FieldTypeDefinitionReferencedObjectTypeEnum["MarketingEvent"] = "MARKETING_EVENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["ConversationInbox"] = "CONVERSATION_INBOX";
    FieldTypeDefinitionReferencedObjectTypeEnum["Chatflow"] = "CHATFLOW";
    FieldTypeDefinitionReferencedObjectTypeEnum["MediaBridge"] = "MEDIA_BRIDGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["Sequence"] = "SEQUENCE";
    FieldTypeDefinitionReferencedObjectTypeEnum["SequenceStep"] = "SEQUENCE_STEP";
    FieldTypeDefinitionReferencedObjectTypeEnum["Forecast"] = "FORECAST";
    FieldTypeDefinitionReferencedObjectTypeEnum["Snippet"] = "SNIPPET";
    FieldTypeDefinitionReferencedObjectTypeEnum["Template"] = "TEMPLATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["DealCreateAttribution"] = "DEAL_CREATE_ATTRIBUTION";
    FieldTypeDefinitionReferencedObjectTypeEnum["QuoteTemplate"] = "QUOTE_TEMPLATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["QuoteModule"] = "QUOTE_MODULE";
    FieldTypeDefinitionReferencedObjectTypeEnum["QuoteModuleField"] = "QUOTE_MODULE_FIELD";
    FieldTypeDefinitionReferencedObjectTypeEnum["QuoteField"] = "QUOTE_FIELD";
    FieldTypeDefinitionReferencedObjectTypeEnum["SequenceEnrollment"] = "SEQUENCE_ENROLLMENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Subscription"] = "SUBSCRIPTION";
    FieldTypeDefinitionReferencedObjectTypeEnum["AcceptanceTest"] = "ACCEPTANCE_TEST";
    FieldTypeDefinitionReferencedObjectTypeEnum["SocialBroadcast"] = "SOCIAL_BROADCAST";
    FieldTypeDefinitionReferencedObjectTypeEnum["DealSplit"] = "DEAL_SPLIT";
    FieldTypeDefinitionReferencedObjectTypeEnum["DealRegistration"] = "DEAL_REGISTRATION";
    FieldTypeDefinitionReferencedObjectTypeEnum["GoalTarget"] = "GOAL_TARGET";
    FieldTypeDefinitionReferencedObjectTypeEnum["GoalTargetGroup"] = "GOAL_TARGET_GROUP";
    FieldTypeDefinitionReferencedObjectTypeEnum["PortalObjectSyncMessage"] = "PORTAL_OBJECT_SYNC_MESSAGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["FileManagerFile"] = "FILE_MANAGER_FILE";
    FieldTypeDefinitionReferencedObjectTypeEnum["FileManagerFolder"] = "FILE_MANAGER_FOLDER";
    FieldTypeDefinitionReferencedObjectTypeEnum["SequenceStepEnrollment"] = "SEQUENCE_STEP_ENROLLMENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Approval"] = "APPROVAL";
    FieldTypeDefinitionReferencedObjectTypeEnum["ApprovalStep"] = "APPROVAL_STEP";
    FieldTypeDefinitionReferencedObjectTypeEnum["CtaVariant"] = "CTA_VARIANT";
    FieldTypeDefinitionReferencedObjectTypeEnum["SalesDocument"] = "SALES_DOCUMENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Discount"] = "DISCOUNT";
    FieldTypeDefinitionReferencedObjectTypeEnum["Fee"] = "FEE";
    FieldTypeDefinitionReferencedObjectTypeEnum["Tax"] = "TAX";
    FieldTypeDefinitionReferencedObjectTypeEnum["MarketingCalendar"] = "MARKETING_CALENDAR";
    FieldTypeDefinitionReferencedObjectTypeEnum["PermissionsTesting"] = "PERMISSIONS_TESTING";
    FieldTypeDefinitionReferencedObjectTypeEnum["PrivacyScannerCookie"] = "PRIVACY_SCANNER_COOKIE";
    FieldTypeDefinitionReferencedObjectTypeEnum["DataSyncState"] = "DATA_SYNC_STATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["WebInteractive"] = "WEB_INTERACTIVE";
    FieldTypeDefinitionReferencedObjectTypeEnum["Playbook"] = "PLAYBOOK";
    FieldTypeDefinitionReferencedObjectTypeEnum["Folder"] = "FOLDER";
    FieldTypeDefinitionReferencedObjectTypeEnum["PlaybookQuestion"] = "PLAYBOOK_QUESTION";
    FieldTypeDefinitionReferencedObjectTypeEnum["PlaybookSubmission"] = "PLAYBOOK_SUBMISSION";
    FieldTypeDefinitionReferencedObjectTypeEnum["PlaybookSubmissionAnswer"] = "PLAYBOOK_SUBMISSION_ANSWER";
    FieldTypeDefinitionReferencedObjectTypeEnum["CommercePayment"] = "COMMERCE_PAYMENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["GscProperty"] = "GSC_PROPERTY";
    FieldTypeDefinitionReferencedObjectTypeEnum["SoxProtectedDummyType"] = "SOX_PROTECTED_DUMMY_TYPE";
    FieldTypeDefinitionReferencedObjectTypeEnum["BlogListingPage"] = "BLOG_LISTING_PAGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["QuarantinedSubmission"] = "QUARANTINED_SUBMISSION";
    FieldTypeDefinitionReferencedObjectTypeEnum["PaymentSchedule"] = "PAYMENT_SCHEDULE";
    FieldTypeDefinitionReferencedObjectTypeEnum["PaymentScheduleInstallment"] = "PAYMENT_SCHEDULE_INSTALLMENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["MarketingCampaignUtm"] = "MARKETING_CAMPAIGN_UTM";
    FieldTypeDefinitionReferencedObjectTypeEnum["DiscountTemplate"] = "DISCOUNT_TEMPLATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["DiscountCode"] = "DISCOUNT_CODE";
    FieldTypeDefinitionReferencedObjectTypeEnum["FeedbackSurvey"] = "FEEDBACK_SURVEY";
    FieldTypeDefinitionReferencedObjectTypeEnum["CmsUrl"] = "CMS_URL";
    FieldTypeDefinitionReferencedObjectTypeEnum["SalesTask"] = "SALES_TASK";
    FieldTypeDefinitionReferencedObjectTypeEnum["SalesWorkload"] = "SALES_WORKLOAD";
    FieldTypeDefinitionReferencedObjectTypeEnum["User"] = "USER";
    FieldTypeDefinitionReferencedObjectTypeEnum["PostalMail"] = "POSTAL_MAIL";
    FieldTypeDefinitionReferencedObjectTypeEnum["SchemasBackendTest"] = "SCHEMAS_BACKEND_TEST";
    FieldTypeDefinitionReferencedObjectTypeEnum["PaymentLink"] = "PAYMENT_LINK";
    FieldTypeDefinitionReferencedObjectTypeEnum["SubmissionTag"] = "SUBMISSION_TAG";
    FieldTypeDefinitionReferencedObjectTypeEnum["CampaignStep"] = "CAMPAIGN_STEP";
    FieldTypeDefinitionReferencedObjectTypeEnum["SchedulingPage"] = "SCHEDULING_PAGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["SoxProtectedTestType"] = "SOX_PROTECTED_TEST_TYPE";
    FieldTypeDefinitionReferencedObjectTypeEnum["Order"] = "ORDER";
    FieldTypeDefinitionReferencedObjectTypeEnum["MarketingSms"] = "MARKETING_SMS";
    FieldTypeDefinitionReferencedObjectTypeEnum["PartnerAccount"] = "PARTNER_ACCOUNT";
    FieldTypeDefinitionReferencedObjectTypeEnum["CampaignTemplate"] = "CAMPAIGN_TEMPLATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["CampaignTemplateStep"] = "CAMPAIGN_TEMPLATE_STEP";
    FieldTypeDefinitionReferencedObjectTypeEnum["Playlist"] = "PLAYLIST";
    FieldTypeDefinitionReferencedObjectTypeEnum["Clip"] = "CLIP";
    FieldTypeDefinitionReferencedObjectTypeEnum["CampaignBudgetItem"] = "CAMPAIGN_BUDGET_ITEM";
    FieldTypeDefinitionReferencedObjectTypeEnum["CampaignSpendItem"] = "CAMPAIGN_SPEND_ITEM";
    FieldTypeDefinitionReferencedObjectTypeEnum["Mic"] = "MIC";
    FieldTypeDefinitionReferencedObjectTypeEnum["ContentAudit"] = "CONTENT_AUDIT";
    FieldTypeDefinitionReferencedObjectTypeEnum["ContentAuditPage"] = "CONTENT_AUDIT_PAGE";
    FieldTypeDefinitionReferencedObjectTypeEnum["PlaylistFolder"] = "PLAYLIST_FOLDER";
    FieldTypeDefinitionReferencedObjectTypeEnum["Lead"] = "LEAD";
    FieldTypeDefinitionReferencedObjectTypeEnum["AbandonedCart"] = "ABANDONED_CART";
    FieldTypeDefinitionReferencedObjectTypeEnum["ExternalWebUrl"] = "EXTERNAL_WEB_URL";
    FieldTypeDefinitionReferencedObjectTypeEnum["View"] = "VIEW";
    FieldTypeDefinitionReferencedObjectTypeEnum["ViewBlock"] = "VIEW_BLOCK";
    FieldTypeDefinitionReferencedObjectTypeEnum["Roster"] = "ROSTER";
    FieldTypeDefinitionReferencedObjectTypeEnum["Cart"] = "CART";
    FieldTypeDefinitionReferencedObjectTypeEnum["AutomationPlatformFlowAction"] = "AUTOMATION_PLATFORM_FLOW_ACTION";
    FieldTypeDefinitionReferencedObjectTypeEnum["SocialProfile"] = "SOCIAL_PROFILE";
    FieldTypeDefinitionReferencedObjectTypeEnum["PartnerClient"] = "PARTNER_CLIENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["RosterMember"] = "ROSTER_MEMBER";
    FieldTypeDefinitionReferencedObjectTypeEnum["MarketingEventAttendance"] = "MARKETING_EVENT_ATTENDANCE";
    FieldTypeDefinitionReferencedObjectTypeEnum["AllPages"] = "ALL_PAGES";
    FieldTypeDefinitionReferencedObjectTypeEnum["AiForecast"] = "AI_FORECAST";
    FieldTypeDefinitionReferencedObjectTypeEnum["CrmPipelinesDummyType"] = "CRM_PIPELINES_DUMMY_TYPE";
    FieldTypeDefinitionReferencedObjectTypeEnum["KnowledgeArticle"] = "KNOWLEDGE_ARTICLE";
    FieldTypeDefinitionReferencedObjectTypeEnum["PropertyInfo"] = "PROPERTY_INFO";
    FieldTypeDefinitionReferencedObjectTypeEnum["DataPrivacyConsent"] = "DATA_PRIVACY_CONSENT";
    FieldTypeDefinitionReferencedObjectTypeEnum["GoalTemplate"] = "GOAL_TEMPLATE";
    FieldTypeDefinitionReferencedObjectTypeEnum["ScoreConfiguration"] = "SCORE_CONFIGURATION";
    FieldTypeDefinitionReferencedObjectTypeEnum["Audience"] = "AUDIENCE";
    FieldTypeDefinitionReferencedObjectTypeEnum["PartnerClientRevenue"] = "PARTNER_CLIENT_REVENUE";
    FieldTypeDefinitionReferencedObjectTypeEnum["AutomationJourney"] = "AUTOMATION_JOURNEY";
    FieldTypeDefinitionReferencedObjectTypeEnum["Unknown"] = "UNKNOWN";
})(FieldTypeDefinitionReferencedObjectTypeEnum = exports.FieldTypeDefinitionReferencedObjectTypeEnum || (exports.FieldTypeDefinitionReferencedObjectTypeEnum = {}));
var FieldTypeDefinitionTypeEnum;
(function (FieldTypeDefinitionTypeEnum) {
    FieldTypeDefinitionTypeEnum["String"] = "string";
    FieldTypeDefinitionTypeEnum["Number"] = "number";
    FieldTypeDefinitionTypeEnum["Bool"] = "bool";
    FieldTypeDefinitionTypeEnum["Datetime"] = "datetime";
    FieldTypeDefinitionTypeEnum["Enumeration"] = "enumeration";
    FieldTypeDefinitionTypeEnum["Date"] = "date";
    FieldTypeDefinitionTypeEnum["PhoneNumber"] = "phone_number";
    FieldTypeDefinitionTypeEnum["CurrencyNumber"] = "currency_number";
    FieldTypeDefinitionTypeEnum["Json"] = "json";
    FieldTypeDefinitionTypeEnum["ObjectCoordinates"] = "object_coordinates";
})(FieldTypeDefinitionTypeEnum = exports.FieldTypeDefinitionTypeEnum || (exports.FieldTypeDefinitionTypeEnum = {}));
var FieldTypeDefinitionFieldTypeEnum;
(function (FieldTypeDefinitionFieldTypeEnum) {
    FieldTypeDefinitionFieldTypeEnum["Booleancheckbox"] = "booleancheckbox";
    FieldTypeDefinitionFieldTypeEnum["Checkbox"] = "checkbox";
    FieldTypeDefinitionFieldTypeEnum["Date"] = "date";
    FieldTypeDefinitionFieldTypeEnum["File"] = "file";
    FieldTypeDefinitionFieldTypeEnum["Number"] = "number";
    FieldTypeDefinitionFieldTypeEnum["Phonenumber"] = "phonenumber";
    FieldTypeDefinitionFieldTypeEnum["Radio"] = "radio";
    FieldTypeDefinitionFieldTypeEnum["Select"] = "select";
    FieldTypeDefinitionFieldTypeEnum["Text"] = "text";
    FieldTypeDefinitionFieldTypeEnum["Textarea"] = "textarea";
    FieldTypeDefinitionFieldTypeEnum["CalculationEquation"] = "calculation_equation";
    FieldTypeDefinitionFieldTypeEnum["CalculationRollup"] = "calculation_rollup";
    FieldTypeDefinitionFieldTypeEnum["CalculationScore"] = "calculation_score";
    FieldTypeDefinitionFieldTypeEnum["CalculationReadTime"] = "calculation_read_time";
    FieldTypeDefinitionFieldTypeEnum["Unknown"] = "unknown";
    FieldTypeDefinitionFieldTypeEnum["Html"] = "html";
})(FieldTypeDefinitionFieldTypeEnum = exports.FieldTypeDefinitionFieldTypeEnum || (exports.FieldTypeDefinitionFieldTypeEnum = {}));
