// @hubspot/api-client/lib/codegen/automation/actions/models/InputFieldDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InputFieldDefinitionSupportedValueTypesEnum = exports.InputFieldDefinition = void 0;
class InputFieldDefinition {
    static getAttributeTypeMap() {
        return InputFieldDefinition.attributeTypeMap;
    }
    constructor() {
    }
}
exports.InputFieldDefinition = InputFieldDefinition;
InputFieldDefinition.discriminator = undefined;
InputFieldDefinition.attributeTypeMap = [
    {
        "name": "isRequired",
        "baseName": "isRequired",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "automationFieldType",
        "baseName": "automationFieldType",
        "type": "string",
        "format": ""
    },
    {
        "name": "typeDefinition",
        "baseName": "typeDefinition",
        "type": "FieldTypeDefinition",
        "format": ""
    },
    {
        "name": "supportedValueTypes",
        "baseName": "supportedValueTypes",
        "type": "Array<InputFieldDefinitionSupportedValueTypesEnum>",
        "format": ""
    }
];
var InputFieldDefinitionSupportedValueTypesEnum;
(function (InputFieldDefinitionSupportedValueTypesEnum) {
    InputFieldDefinitionSupportedValueTypesEnum["StaticValue"] = "STATIC_VALUE";
    InputFieldDefinitionSupportedValueTypesEnum["ObjectProperty"] = "OBJECT_PROPERTY";
    InputFieldDefinitionSupportedValueTypesEnum["FieldData"] = "FIELD_DATA";
    InputFieldDefinitionSupportedValueTypesEnum["FetchedObjectProperty"] = "FETCHED_OBJECT_PROPERTY";
    InputFieldDefinitionSupportedValueTypesEnum["EnrollmentEventProperty"] = "ENROLLMENT_EVENT_PROPERTY";
})(InputFieldDefinitionSupportedValueTypesEnum = exports.InputFieldDefinitionSupportedValueTypesEnum || (exports.InputFieldDefinitionSupportedValueTypesEnum = {}));
