// @hubspot/api-client/lib/codegen/crm/imports/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectPublicImportsApi = exports.ObjectCoreApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectCoreApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableCoreApi(configuration, requestFactory, responseProcessor);
    }
    cancelWithHttpInfo(param, options) {
        return this.api.cancelWithHttpInfo(param.importId, options).toPromise();
    }
    cancel(param, options) {
        return this.api.cancel(param.importId, options).toPromise();
    }
    createWithHttpInfo(param = {}, options) {
        return this.api.createWithHttpInfo(param.files, param.importRequest, options).toPromise();
    }
    create(param = {}, options) {
        return this.api.create(param.files, param.importRequest, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.importId, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.importId, options).toPromise();
    }
    getPageWithHttpInfo(param = {}, options) {
        return this.api.getPageWithHttpInfo(param.after, param.before, param.limit, options).toPromise();
    }
    getPage(param = {}, options) {
        return this.api.getPage(param.after, param.before, param.limit, options).toPromise();
    }
}
exports.ObjectCoreApi = ObjectCoreApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectPublicImportsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservablePublicImportsApi(configuration, requestFactory, responseProcessor);
    }
    getErrorsWithHttpInfo(param, options) {
        return this.api.getErrorsWithHttpInfo(param.importId, param.after, param.limit, options).toPromise();
    }
    getErrors(param, options) {
        return this.api.getErrors(param.importId, param.after, param.limit, options).toPromise();
    }
}
exports.ObjectPublicImportsApi = ObjectPublicImportsApi;
