// @hubspot/api-client/lib/codegen/crm/objects/feedback_submissions/models/BatchResponseSimplePublicObjectWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSimplePublicObjectWithErrorsStatusEnum = exports.BatchResponseSimplePublicObjectWithErrors = void 0;
class BatchResponseSimplePublicObjectWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseSimplePublicObjectWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseSimplePublicObjectWithErrors = BatchResponseSimplePublicObjectWithErrors;
BatchResponseSimplePublicObjectWithErrors.discriminator = undefined;
BatchResponseSimplePublicObjectWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SimplePublicObject>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSimplePublicObjectWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseSimplePublicObjectWithErrorsStatusEnum;
(function (BatchResponseSimplePublicObjectWithErrorsStatusEnum) {
    BatchResponseSimplePublicObjectWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseSimplePublicObjectWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseSimplePublicObjectWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseSimplePublicObjectWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseSimplePublicObjectWithErrorsStatusEnum = exports.BatchResponseSimplePublicObjectWithErrorsStatusEnum || (exports.BatchResponseSimplePublicObjectWithErrorsStatusEnum = {}));
