// @hubspot/api-client/lib/codegen/marketing/forms/models/DependentFieldFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DependentFieldFilterOperatorEnum = exports.DependentFieldFilter = void 0;
class DependentFieldFilter {
    static getAttributeTypeMap() {
        return DependentFieldFilter.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DependentFieldFilter = DependentFieldFilter;
DependentFieldFilter.discriminator = undefined;
DependentFieldFilter.attributeTypeMap = [
    {
        "name": "rangeStart",
        "baseName": "rangeStart",
        "type": "string",
        "format": ""
    },
    {
        "name": "values",
        "baseName": "values",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "DependentFieldFilterOperatorEnum",
        "format": ""
    },
    {
        "name": "rangeEnd",
        "baseName": "rangeEnd",
        "type": "string",
        "format": ""
    }
];
var DependentFieldFilterOperatorEnum;
(function (DependentFieldFilterOperatorEnum) {
    DependentFieldFilterOperatorEnum["Eq"] = "eq";
    DependentFieldFilterOperatorEnum["Neq"] = "neq";
    DependentFieldFilterOperatorEnum["Contains"] = "contains";
    DependentFieldFilterOperatorEnum["DoesntContain"] = "doesnt_contain";
    DependentFieldFilterOperatorEnum["StrStartsWith"] = "str_starts_with";
    DependentFieldFilterOperatorEnum["StrEndsWith"] = "str_ends_with";
    DependentFieldFilterOperatorEnum["Lt"] = "lt";
    DependentFieldFilterOperatorEnum["Lte"] = "lte";
    DependentFieldFilterOperatorEnum["Gt"] = "gt";
    DependentFieldFilterOperatorEnum["Gte"] = "gte";
    DependentFieldFilterOperatorEnum["Between"] = "between";
    DependentFieldFilterOperatorEnum["NotBetween"] = "not_between";
    DependentFieldFilterOperatorEnum["WithinTimeReverse"] = "within_time_reverse";
    DependentFieldFilterOperatorEnum["WithinTime"] = "within_time";
    DependentFieldFilterOperatorEnum["SetAny"] = "set_any";
    DependentFieldFilterOperatorEnum["SetNotAny"] = "set_not_any";
    DependentFieldFilterOperatorEnum["SetAll"] = "set_all";
    DependentFieldFilterOperatorEnum["SetNotAll"] = "set_not_all";
    DependentFieldFilterOperatorEnum["SetEq"] = "set_eq";
    DependentFieldFilterOperatorEnum["SetNeq"] = "set_neq";
    DependentFieldFilterOperatorEnum["IsNotEmpty"] = "is_not_empty";
})(DependentFieldFilterOperatorEnum = exports.DependentFieldFilterOperatorEnum || (exports.DependentFieldFilterOperatorEnum = {}));
