// @hubspot/api-client/lib/codegen/marketing/forms/models/FieldGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldGroupRichTextTypeEnum = exports.FieldGroupGroupTypeEnum = exports.FieldGroup = void 0;
class FieldGroup {
    static getAttributeTypeMap() {
        return FieldGroup.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FieldGroup = FieldGroup;
FieldGroup.discriminator = undefined;
FieldGroup.attributeTypeMap = [
    {
        "name": "groupType",
        "baseName": "groupType",
        "type": "FieldGroupGroupTypeEnum",
        "format": ""
    },
    {
        "name": "richTextType",
        "baseName": "richTextType",
        "type": "FieldGroupRichTextTypeEnum",
        "format": ""
    },
    {
        "name": "richText",
        "baseName": "richText",
        "type": "string",
        "format": ""
    },
    {
        "name": "fields",
        "baseName": "fields",
        "type": "Array<DependentFieldDependentField>",
        "format": ""
    }
];
var FieldGroupGroupTypeEnum;
(function (FieldGroupGroupTypeEnum) {
    FieldGroupGroupTypeEnum["DefaultGroup"] = "default_group";
    FieldGroupGroupTypeEnum["Progressive"] = "progressive";
    FieldGroupGroupTypeEnum["Queued"] = "queued";
})(FieldGroupGroupTypeEnum = exports.FieldGroupGroupTypeEnum || (exports.FieldGroupGroupTypeEnum = {}));
var FieldGroupRichTextTypeEnum;
(function (FieldGroupRichTextTypeEnum) {
    FieldGroupRichTextTypeEnum["Text"] = "text";
    FieldGroupRichTextTypeEnum["Image"] = "image";
})(FieldGroupRichTextTypeEnum = exports.FieldGroupRichTextTypeEnum || (exports.FieldGroupRichTextTypeEnum = {}));
