// @hubspot/api-client/lib/codegen/marketing/forms/models/FormDisplayOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormDisplayOptionsThemeEnum = exports.FormDisplayOptions = void 0;
class FormDisplayOptions {
    static getAttributeTypeMap() {
        return FormDisplayOptions.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FormDisplayOptions = FormDisplayOptions;
FormDisplayOptions.discriminator = undefined;
FormDisplayOptions.attributeTypeMap = [
    {
        "name": "renderRawHtml",
        "baseName": "renderRawHtml",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "cssClass",
        "baseName": "cssClass",
        "type": "string",
        "format": ""
    },
    {
        "name": "theme",
        "baseName": "theme",
        "type": "FormDisplayOptionsThemeEnum",
        "format": ""
    },
    {
        "name": "submitButtonText",
        "baseName": "submitButtonText",
        "type": "string",
        "format": ""
    },
    {
        "name": "style",
        "baseName": "style",
        "type": "FormStyle",
        "format": ""
    }
];
var FormDisplayOptionsThemeEnum;
(function (FormDisplayOptionsThemeEnum) {
    FormDisplayOptionsThemeEnum["DefaultStyle"] = "default_style";
    FormDisplayOptionsThemeEnum["Canvas"] = "canvas";
    FormDisplayOptionsThemeEnum["Linear"] = "linear";
    FormDisplayOptionsThemeEnum["Round"] = "round";
    FormDisplayOptionsThemeEnum["Sharp"] = "sharp";
    FormDisplayOptionsThemeEnum["Legacy"] = "legacy";
})(FormDisplayOptionsThemeEnum = exports.FormDisplayOptionsThemeEnum || (exports.FormDisplayOptionsThemeEnum = {}));
