// @hubspot/api-client/lib/codegen/marketing/forms/models/HubSpotFormConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HubSpotFormConfigurationLanguageEnum = exports.HubSpotFormConfiguration = void 0;
class HubSpotFormConfiguration {
    static getAttributeTypeMap() {
        return HubSpotFormConfiguration.attributeTypeMap;
    }
    constructor() {
    }
}
exports.HubSpotFormConfiguration = HubSpotFormConfiguration;
HubSpotFormConfiguration.discriminator = undefined;
HubSpotFormConfiguration.attributeTypeMap = [
    {
        "name": "createNewContactForNewEmail",
        "baseName": "createNewContactForNewEmail",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "editable",
        "baseName": "editable",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "allowLinkToResetKnownValues",
        "baseName": "allowLinkToResetKnownValues",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "lifecycleStages",
        "baseName": "lifecycleStages",
        "type": "Array<LifecycleStage>",
        "format": ""
    },
    {
        "name": "postSubmitAction",
        "baseName": "postSubmitAction",
        "type": "FormPostSubmitAction",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "HubSpotFormConfigurationLanguageEnum",
        "format": ""
    },
    {
        "name": "prePopulateKnownValues",
        "baseName": "prePopulateKnownValues",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "cloneable",
        "baseName": "cloneable",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "notifyContactOwner",
        "baseName": "notifyContactOwner",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "recaptchaEnabled",
        "baseName": "recaptchaEnabled",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "archivable",
        "baseName": "archivable",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "notifyRecipients",
        "baseName": "notifyRecipients",
        "type": "Array<string>",
        "format": ""
    }
];
var HubSpotFormConfigurationLanguageEnum;
(function (HubSpotFormConfigurationLanguageEnum) {
    HubSpotFormConfigurationLanguageEnum["Af"] = "af";
    HubSpotFormConfigurationLanguageEnum["ArEg"] = "ar-eg";
    HubSpotFormConfigurationLanguageEnum["Bg"] = "bg";
    HubSpotFormConfigurationLanguageEnum["Bn"] = "bn";
    HubSpotFormConfigurationLanguageEnum["CaEs"] = "ca-es";
    HubSpotFormConfigurationLanguageEnum["Cs"] = "cs";
    HubSpotFormConfigurationLanguageEnum["Da"] = "da";
    HubSpotFormConfigurationLanguageEnum["De"] = "de";
    HubSpotFormConfigurationLanguageEnum["El"] = "el";
    HubSpotFormConfigurationLanguageEnum["En"] = "en";
    HubSpotFormConfigurationLanguageEnum["Es"] = "es";
    HubSpotFormConfigurationLanguageEnum["EsMx"] = "es-mx";
    HubSpotFormConfigurationLanguageEnum["Fi"] = "fi";
    HubSpotFormConfigurationLanguageEnum["Fr"] = "fr";
    HubSpotFormConfigurationLanguageEnum["FrCa"] = "fr-ca";
    HubSpotFormConfigurationLanguageEnum["HeIl"] = "he-il";
    HubSpotFormConfigurationLanguageEnum["Hr"] = "hr";
    HubSpotFormConfigurationLanguageEnum["Hu"] = "hu";
    HubSpotFormConfigurationLanguageEnum["Id"] = "id";
    HubSpotFormConfigurationLanguageEnum["It"] = "it";
    HubSpotFormConfigurationLanguageEnum["Ja"] = "ja";
    HubSpotFormConfigurationLanguageEnum["Ko"] = "ko";
    HubSpotFormConfigurationLanguageEnum["Lt"] = "lt";
    HubSpotFormConfigurationLanguageEnum["Ms"] = "ms";
    HubSpotFormConfigurationLanguageEnum["Nl"] = "nl";
    HubSpotFormConfigurationLanguageEnum["NoNo"] = "no-no";
    HubSpotFormConfigurationLanguageEnum["Pl"] = "pl";
    HubSpotFormConfigurationLanguageEnum["Pt"] = "pt";
    HubSpotFormConfigurationLanguageEnum["PtBr"] = "pt-br";
    HubSpotFormConfigurationLanguageEnum["Ro"] = "ro";
    HubSpotFormConfigurationLanguageEnum["Ru"] = "ru";
    HubSpotFormConfigurationLanguageEnum["Sk"] = "sk";
    HubSpotFormConfigurationLanguageEnum["Sl"] = "sl";
    HubSpotFormConfigurationLanguageEnum["Sv"] = "sv";
    HubSpotFormConfigurationLanguageEnum["Th"] = "th";
    HubSpotFormConfigurationLanguageEnum["Tl"] = "tl";
    HubSpotFormConfigurationLanguageEnum["Tr"] = "tr";
    HubSpotFormConfigurationLanguageEnum["Uk"] = "uk";
    HubSpotFormConfigurationLanguageEnum["Vi"] = "vi";
    HubSpotFormConfigurationLanguageEnum["ZhCn"] = "zh-cn";
    HubSpotFormConfigurationLanguageEnum["ZhHk"] = "zh-hk";
    HubSpotFormConfigurationLanguageEnum["ZhTw"] = "zh-tw";
})(HubSpotFormConfigurationLanguageEnum = exports.HubSpotFormConfigurationLanguageEnum || (exports.HubSpotFormConfigurationLanguageEnum = {}));
