// @hubspot/api-client/lib/codegen/crm/objects/leads/models/BatchResponseSimplePublicUpsertObjectWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum = exports.BatchResponseSimplePublicUpsertObjectWithErrors = void 0;
class BatchResponseSimplePublicUpsertObjectWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseSimplePublicUpsertObjectWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseSimplePublicUpsertObjectWithErrors = BatchResponseSimplePublicUpsertObjectWithErrors;
BatchResponseSimplePublicUpsertObjectWithErrors.discriminator = undefined;
BatchResponseSimplePublicUpsertObjectWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SimplePublicUpsertObject>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum;
(function (BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum) {
    BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum = exports.BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum || (exports.BatchResponseSimplePublicUpsertObjectWithErrorsStatusEnum = {}));
