// @hubspot/api-client/lib/codegen/crm/schemas/models/ObjectTypePropertyCreate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectTypePropertyCreateNumberDisplayHintEnum = exports.ObjectTypePropertyCreateTextDisplayHintEnum = exports.ObjectTypePropertyCreateTypeEnum = exports.ObjectTypePropertyCreateOptionSortStrategyEnum = exports.ObjectTypePropertyCreate = void 0;
class ObjectTypePropertyCreate {
    static getAttributeTypeMap() {
        return ObjectTypePropertyCreate.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ObjectTypePropertyCreate = ObjectTypePropertyCreate;
ObjectTypePropertyCreate.discriminator = undefined;
ObjectTypePropertyCreate.attributeTypeMap = [
    {
        "name": "hidden",
        "baseName": "hidden",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "optionSortStrategy",
        "baseName": "optionSortStrategy",
        "type": "ObjectTypePropertyCreateOptionSortStrategyEnum",
        "format": ""
    },
    {
        "name": "displayOrder",
        "baseName": "displayOrder",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "showCurrencySymbol",
        "baseName": "showCurrencySymbol",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ObjectTypePropertyCreateTypeEnum",
        "format": ""
    },
    {
        "name": "formField",
        "baseName": "formField",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "groupName",
        "baseName": "groupName",
        "type": "string",
        "format": ""
    },
    {
        "name": "referencedObjectType",
        "baseName": "referencedObjectType",
        "type": "string",
        "format": ""
    },
    {
        "name": "textDisplayHint",
        "baseName": "textDisplayHint",
        "type": "ObjectTypePropertyCreateTextDisplayHintEnum",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<OptionInput>",
        "format": ""
    },
    {
        "name": "searchableInGlobalSearch",
        "baseName": "searchableInGlobalSearch",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "numberDisplayHint",
        "baseName": "numberDisplayHint",
        "type": "ObjectTypePropertyCreateNumberDisplayHintEnum",
        "format": ""
    },
    {
        "name": "hasUniqueValue",
        "baseName": "hasUniqueValue",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "string",
        "format": ""
    }
];
var ObjectTypePropertyCreateOptionSortStrategyEnum;
(function (ObjectTypePropertyCreateOptionSortStrategyEnum) {
    ObjectTypePropertyCreateOptionSortStrategyEnum["DisplayOrder"] = "DISPLAY_ORDER";
    ObjectTypePropertyCreateOptionSortStrategyEnum["Alphabetical"] = "ALPHABETICAL";
})(ObjectTypePropertyCreateOptionSortStrategyEnum = exports.ObjectTypePropertyCreateOptionSortStrategyEnum || (exports.ObjectTypePropertyCreateOptionSortStrategyEnum = {}));
var ObjectTypePropertyCreateTypeEnum;
(function (ObjectTypePropertyCreateTypeEnum) {
    ObjectTypePropertyCreateTypeEnum["String"] = "string";
    ObjectTypePropertyCreateTypeEnum["Number"] = "number";
    ObjectTypePropertyCreateTypeEnum["Date"] = "date";
    ObjectTypePropertyCreateTypeEnum["Datetime"] = "datetime";
    ObjectTypePropertyCreateTypeEnum["Enumeration"] = "enumeration";
    ObjectTypePropertyCreateTypeEnum["Bool"] = "bool";
})(ObjectTypePropertyCreateTypeEnum = exports.ObjectTypePropertyCreateTypeEnum || (exports.ObjectTypePropertyCreateTypeEnum = {}));
var ObjectTypePropertyCreateTextDisplayHintEnum;
(function (ObjectTypePropertyCreateTextDisplayHintEnum) {
    ObjectTypePropertyCreateTextDisplayHintEnum["UnformattedSingleLine"] = "unformatted_single_line";
    ObjectTypePropertyCreateTextDisplayHintEnum["MultiLine"] = "multi_line";
    ObjectTypePropertyCreateTextDisplayHintEnum["Email"] = "email";
    ObjectTypePropertyCreateTextDisplayHintEnum["PhoneNumber"] = "phone_number";
    ObjectTypePropertyCreateTextDisplayHintEnum["DomainName"] = "domain_name";
    ObjectTypePropertyCreateTextDisplayHintEnum["IpAddress"] = "ip_address";
    ObjectTypePropertyCreateTextDisplayHintEnum["PhysicalAddress"] = "physical_address";
    ObjectTypePropertyCreateTextDisplayHintEnum["PostalCode"] = "postal_code";
})(ObjectTypePropertyCreateTextDisplayHintEnum = exports.ObjectTypePropertyCreateTextDisplayHintEnum || (exports.ObjectTypePropertyCreateTextDisplayHintEnum = {}));
var ObjectTypePropertyCreateNumberDisplayHintEnum;
(function (ObjectTypePropertyCreateNumberDisplayHintEnum) {
    ObjectTypePropertyCreateNumberDisplayHintEnum["Unformatted"] = "unformatted";
    ObjectTypePropertyCreateNumberDisplayHintEnum["Formatted"] = "formatted";
    ObjectTypePropertyCreateNumberDisplayHintEnum["Currency"] = "currency";
    ObjectTypePropertyCreateNumberDisplayHintEnum["Percentage"] = "percentage";
    ObjectTypePropertyCreateNumberDisplayHintEnum["Duration"] = "duration";
    ObjectTypePropertyCreateNumberDisplayHintEnum["Probability"] = "probability";
})(ObjectTypePropertyCreateNumberDisplayHintEnum = exports.ObjectTypePropertyCreateNumberDisplayHintEnum || (exports.ObjectTypePropertyCreateNumberDisplayHintEnum = {}));
