// @hubspot/api-client/lib/codegen/crm/objects/meetings/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSearchApi = exports.ObjectBatchApi = exports.ObjectBasicApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectBasicApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableBasicApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(param, options) {
        return this.api.archiveWithHttpInfo(param.meetingId, options).toPromise();
    }
    archive(param, options) {
        return this.api.archive(param.meetingId, options).toPromise();
    }
    createWithHttpInfo(param, options) {
        return this.api.createWithHttpInfo(param.simplePublicObjectInputForCreate, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.simplePublicObjectInputForCreate, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.meetingId, param.properties, param.propertiesWithHistory, param.associations, param.archived, param.idProperty, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.meetingId, param.properties, param.propertiesWithHistory, param.associations, param.archived, param.idProperty, options).toPromise();
    }
    getPageWithHttpInfo(param = {}, options) {
        return this.api.getPageWithHttpInfo(param.limit, param.after, param.properties, param.propertiesWithHistory, param.associations, param.archived, options).toPromise();
    }
    getPage(param = {}, options) {
        return this.api.getPage(param.limit, param.after, param.properties, param.propertiesWithHistory, param.associations, param.archived, options).toPromise();
    }
    updateWithHttpInfo(param, options) {
        return this.api.updateWithHttpInfo(param.meetingId, param.simplePublicObjectInput, param.idProperty, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.meetingId, param.simplePublicObjectInput, param.idProperty, options).toPromise();
    }
}
exports.ObjectBasicApi = ObjectBasicApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectBatchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableBatchApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(param, options) {
        return this.api.archiveWithHttpInfo(param.batchInputSimplePublicObjectId, options).toPromise();
    }
    archive(param, options) {
        return this.api.archive(param.batchInputSimplePublicObjectId, options).toPromise();
    }
    createWithHttpInfo(param, options) {
        return this.api.createWithHttpInfo(param.batchInputSimplePublicObjectInputForCreate, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.batchInputSimplePublicObjectInputForCreate, options).toPromise();
    }
    readWithHttpInfo(param, options) {
        return this.api.readWithHttpInfo(param.batchReadInputSimplePublicObjectId, param.archived, options).toPromise();
    }
    read(param, options) {
        return this.api.read(param.batchReadInputSimplePublicObjectId, param.archived, options).toPromise();
    }
    updateWithHttpInfo(param, options) {
        return this.api.updateWithHttpInfo(param.batchInputSimplePublicObjectBatchInput, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.batchInputSimplePublicObjectBatchInput, options).toPromise();
    }
    upsertWithHttpInfo(param, options) {
        return this.api.upsertWithHttpInfo(param.batchInputSimplePublicObjectBatchInputUpsert, options).toPromise();
    }
    upsert(param, options) {
        return this.api.upsert(param.batchInputSimplePublicObjectBatchInputUpsert, options).toPromise();
    }
}
exports.ObjectBatchApi = ObjectBatchApi;
const ObservableAPI_3 = require("./ObservableAPI");
class ObjectSearchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableSearchApi(configuration, requestFactory, responseProcessor);
    }
    doSearchWithHttpInfo(param, options) {
        return this.api.doSearchWithHttpInfo(param.publicObjectSearchRequest, options).toPromise();
    }
    doSearch(param, options) {
        return this.api.doSearch(param.publicObjectSearchRequest, options).toPromise();
    }
}
exports.ObjectSearchApi = ObjectSearchApi;
