// @hubspot/api-client/lib/codegen/crm/timeline/models/BatchResponseTimelineEventResponseWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseTimelineEventResponseWithErrorsStatusEnum = exports.BatchResponseTimelineEventResponseWithErrors = void 0;
class BatchResponseTimelineEventResponseWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseTimelineEventResponseWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseTimelineEventResponseWithErrors = BatchResponseTimelineEventResponseWithErrors;
BatchResponseTimelineEventResponseWithErrors.discriminator = undefined;
BatchResponseTimelineEventResponseWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<TimelineEventResponse>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseTimelineEventResponseWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseTimelineEventResponseWithErrorsStatusEnum;
(function (BatchResponseTimelineEventResponseWithErrorsStatusEnum) {
    BatchResponseTimelineEventResponseWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseTimelineEventResponseWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseTimelineEventResponseWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseTimelineEventResponseWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseTimelineEventResponseWithErrorsStatusEnum = exports.BatchResponseTimelineEventResponseWithErrorsStatusEnum || (exports.BatchResponseTimelineEventResponseWithErrorsStatusEnum = {}));
