// @hubspot/api-client/lib/codegen/crm/timeline/models/BatchResponseTimelineEventResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseTimelineEventResponseStatusEnum = exports.BatchResponseTimelineEventResponse = void 0;
class BatchResponseTimelineEventResponse {
    static getAttributeTypeMap() {
        return BatchResponseTimelineEventResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseTimelineEventResponse = BatchResponseTimelineEventResponse;
BatchResponseTimelineEventResponse.discriminator = undefined;
BatchResponseTimelineEventResponse.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<TimelineEventResponse>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseTimelineEventResponseStatusEnum",
        "format": ""
    }
];
var BatchResponseTimelineEventResponseStatusEnum;
(function (BatchResponseTimelineEventResponseStatusEnum) {
    BatchResponseTimelineEventResponseStatusEnum["Pending"] = "PENDING";
    BatchResponseTimelineEventResponseStatusEnum["Processing"] = "PROCESSING";
    BatchResponseTimelineEventResponseStatusEnum["Canceled"] = "CANCELED";
    BatchResponseTimelineEventResponseStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseTimelineEventResponseStatusEnum = exports.BatchResponseTimelineEventResponseStatusEnum || (exports.BatchResponseTimelineEventResponseStatusEnum = {}));
