// @hubspot/api-client/lib/codegen/crm/timeline/models/TimelineEventTemplateToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineEventTemplateTokenTypeEnum = exports.TimelineEventTemplateToken = void 0;
class TimelineEventTemplateToken {
    static getAttributeTypeMap() {
        return TimelineEventTemplateToken.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TimelineEventTemplateToken = TimelineEventTemplateToken;
TimelineEventTemplateToken.discriminator = undefined;
TimelineEventTemplateToken.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<TimelineEventTemplateTokenOption>",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectPropertyName",
        "baseName": "objectPropertyName",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TimelineEventTemplateTokenTypeEnum",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    }
];
var TimelineEventTemplateTokenTypeEnum;
(function (TimelineEventTemplateTokenTypeEnum) {
    TimelineEventTemplateTokenTypeEnum["Date"] = "date";
    TimelineEventTemplateTokenTypeEnum["Enumeration"] = "enumeration";
    TimelineEventTemplateTokenTypeEnum["Number"] = "number";
    TimelineEventTemplateTokenTypeEnum["String"] = "string";
})(TimelineEventTemplateTokenTypeEnum = exports.TimelineEventTemplateTokenTypeEnum || (exports.TimelineEventTemplateTokenTypeEnum = {}));
