// @hubspot/api-client/lib/codegen/crm/lists/models/PublicUnifiedEventsFilterBranch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicUnifiedEventsFilterBranchOperatorEnum = exports.PublicUnifiedEventsFilterBranchFilterBranchTypeEnum = exports.PublicUnifiedEventsFilterBranch = void 0;
class PublicUnifiedEventsFilterBranch {
    static getAttributeTypeMap() {
        return PublicUnifiedEventsFilterBranch.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicUnifiedEventsFilterBranch = PublicUnifiedEventsFilterBranch;
PublicUnifiedEventsFilterBranch.discriminator = undefined;
PublicUnifiedEventsFilterBranch.attributeTypeMap = [
    {
        "name": "filterBranchType",
        "baseName": "filterBranchType",
        "type": "PublicUnifiedEventsFilterBranchFilterBranchTypeEnum",
        "format": ""
    },
    {
        "name": "filterBranches",
        "baseName": "filterBranches",
        "type": "Array<PublicPropertyAssociationFilterBranchFilterBranchesInner>",
        "format": ""
    },
    {
        "name": "eventTypeId",
        "baseName": "eventTypeId",
        "type": "string",
        "format": ""
    },
    {
        "name": "coalescingRefineBy",
        "baseName": "coalescingRefineBy",
        "type": "PublicFormSubmissionFilterCoalescingRefineBy",
        "format": ""
    },
    {
        "name": "filterBranchOperator",
        "baseName": "filterBranchOperator",
        "type": "string",
        "format": ""
    },
    {
        "name": "filters",
        "baseName": "filters",
        "type": "Array<PublicPropertyAssociationFilterBranchFiltersInner>",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "PublicUnifiedEventsFilterBranchOperatorEnum",
        "format": ""
    }
];
var PublicUnifiedEventsFilterBranchFilterBranchTypeEnum;
(function (PublicUnifiedEventsFilterBranchFilterBranchTypeEnum) {
    PublicUnifiedEventsFilterBranchFilterBranchTypeEnum["UnifiedEvents"] = "UNIFIED_EVENTS";
})(PublicUnifiedEventsFilterBranchFilterBranchTypeEnum = exports.PublicUnifiedEventsFilterBranchFilterBranchTypeEnum || (exports.PublicUnifiedEventsFilterBranchFilterBranchTypeEnum = {}));
var PublicUnifiedEventsFilterBranchOperatorEnum;
(function (PublicUnifiedEventsFilterBranchOperatorEnum) {
    PublicUnifiedEventsFilterBranchOperatorEnum["Completed"] = "HAS_COMPLETED";
    PublicUnifiedEventsFilterBranchOperatorEnum["NotCompleted"] = "HAS_NOT_COMPLETED";
})(PublicUnifiedEventsFilterBranchOperatorEnum = exports.PublicUnifiedEventsFilterBranchOperatorEnum || (exports.PublicUnifiedEventsFilterBranchOperatorEnum = {}));
