// @hubspot/api-client/lib/codegen/crm/lists/models/PublicEmailEventFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicEmailEventFilterOperatorEnum = exports.PublicEmailEventFilterFilterTypeEnum = exports.PublicEmailEventFilter = void 0;
class PublicEmailEventFilter {
    static getAttributeTypeMap() {
        return PublicEmailEventFilter.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicEmailEventFilter = PublicEmailEventFilter;
PublicEmailEventFilter.discriminator = undefined;
PublicEmailEventFilter.attributeTypeMap = [
    {
        "name": "clickUrl",
        "baseName": "clickUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "level",
        "baseName": "level",
        "type": "string",
        "format": ""
    },
    {
        "name": "pruningRefineBy",
        "baseName": "pruningRefineBy",
        "type": "PublicFormSubmissionFilterCoalescingRefineBy",
        "format": ""
    },
    {
        "name": "appId",
        "baseName": "appId",
        "type": "string",
        "format": ""
    },
    {
        "name": "emailId",
        "baseName": "emailId",
        "type": "string",
        "format": ""
    },
    {
        "name": "filterType",
        "baseName": "filterType",
        "type": "PublicEmailEventFilterFilterTypeEnum",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "PublicEmailEventFilterOperatorEnum",
        "format": ""
    }
];
var PublicEmailEventFilterFilterTypeEnum;
(function (PublicEmailEventFilterFilterTypeEnum) {
    PublicEmailEventFilterFilterTypeEnum["EmailEvent"] = "EMAIL_EVENT";
})(PublicEmailEventFilterFilterTypeEnum = exports.PublicEmailEventFilterFilterTypeEnum || (exports.PublicEmailEventFilterFilterTypeEnum = {}));
var PublicEmailEventFilterOperatorEnum;
(function (PublicEmailEventFilterOperatorEnum) {
    PublicEmailEventFilterOperatorEnum["LinkClicked"] = "LINK_CLICKED";
    PublicEmailEventFilterOperatorEnum["MarkedSpam"] = "MARKED_SPAM";
    PublicEmailEventFilterOperatorEnum["Opened"] = "OPENED";
    PublicEmailEventFilterOperatorEnum["OpenedButLinkNotClicked"] = "OPENED_BUT_LINK_NOT_CLICKED";
    PublicEmailEventFilterOperatorEnum["OpenedButNotReplied"] = "OPENED_BUT_NOT_REPLIED";
    PublicEmailEventFilterOperatorEnum["Replied"] = "REPLIED";
    PublicEmailEventFilterOperatorEnum["Unsubscribed"] = "UNSUBSCRIBED";
    PublicEmailEventFilterOperatorEnum["Bounced"] = "BOUNCED";
    PublicEmailEventFilterOperatorEnum["Received"] = "RECEIVED";
    PublicEmailEventFilterOperatorEnum["ReceivedButNotOpened"] = "RECEIVED_BUT_NOT_OPENED";
    PublicEmailEventFilterOperatorEnum["Sent"] = "SENT";
    PublicEmailEventFilterOperatorEnum["SentButLinkNotClicked"] = "SENT_BUT_LINK_NOT_CLICKED";
    PublicEmailEventFilterOperatorEnum["SentButNotReceived"] = "SENT_BUT_NOT_RECEIVED";
})(PublicEmailEventFilterOperatorEnum = exports.PublicEmailEventFilterOperatorEnum || (exports.PublicEmailEventFilterOperatorEnum = {}));
