// @hubspot/api-client/lib/codegen/crm/lists/models/PublicFormSubmissionOnPageFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicFormSubmissionOnPageFilterOperatorEnum = exports.PublicFormSubmissionOnPageFilterFilterTypeEnum = exports.PublicFormSubmissionOnPageFilter = void 0;
class PublicFormSubmissionOnPageFilter {
    static getAttributeTypeMap() {
        return PublicFormSubmissionOnPageFilter.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicFormSubmissionOnPageFilter = PublicFormSubmissionOnPageFilter;
PublicFormSubmissionOnPageFilter.discriminator = undefined;
PublicFormSubmissionOnPageFilter.attributeTypeMap = [
    {
        "name": "formId",
        "baseName": "formId",
        "type": "string",
        "format": ""
    },
    {
        "name": "coalescingRefineBy",
        "baseName": "coalescingRefineBy",
        "type": "PublicFormSubmissionFilterCoalescingRefineBy",
        "format": ""
    },
    {
        "name": "pruningRefineBy",
        "baseName": "pruningRefineBy",
        "type": "PublicFormSubmissionFilterCoalescingRefineBy",
        "format": ""
    },
    {
        "name": "filterType",
        "baseName": "filterType",
        "type": "PublicFormSubmissionOnPageFilterFilterTypeEnum",
        "format": ""
    },
    {
        "name": "pageId",
        "baseName": "pageId",
        "type": "string",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "PublicFormSubmissionOnPageFilterOperatorEnum",
        "format": ""
    }
];
var PublicFormSubmissionOnPageFilterFilterTypeEnum;
(function (PublicFormSubmissionOnPageFilterFilterTypeEnum) {
    PublicFormSubmissionOnPageFilterFilterTypeEnum["FormSubmissionOnPage"] = "FORM_SUBMISSION_ON_PAGE";
})(PublicFormSubmissionOnPageFilterFilterTypeEnum = exports.PublicFormSubmissionOnPageFilterFilterTypeEnum || (exports.PublicFormSubmissionOnPageFilterFilterTypeEnum = {}));
var PublicFormSubmissionOnPageFilterOperatorEnum;
(function (PublicFormSubmissionOnPageFilterOperatorEnum) {
    PublicFormSubmissionOnPageFilterOperatorEnum["FilledOut"] = "FILLED_OUT";
    PublicFormSubmissionOnPageFilterOperatorEnum["NotFilledOut"] = "NOT_FILLED_OUT";
})(PublicFormSubmissionOnPageFilterOperatorEnum = exports.PublicFormSubmissionOnPageFilterOperatorEnum || (exports.PublicFormSubmissionOnPageFilterOperatorEnum = {}));
