// @hubspot/api-client/lib/codegen/crm/lists/models/PublicFormSubmissionFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicFormSubmissionFilterOperatorEnum = exports.PublicFormSubmissionFilterFilterTypeEnum = exports.PublicFormSubmissionFilter = void 0;
class PublicFormSubmissionFilter {
    static getAttributeTypeMap() {
        return PublicFormSubmissionFilter.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicFormSubmissionFilter = PublicFormSubmissionFilter;
PublicFormSubmissionFilter.discriminator = undefined;
PublicFormSubmissionFilter.attributeTypeMap = [
    {
        "name": "formId",
        "baseName": "formId",
        "type": "string",
        "format": ""
    },
    {
        "name": "coalescingRefineBy",
        "baseName": "coalescingRefineBy",
        "type": "PublicFormSubmissionFilterCoalescingRefineBy",
        "format": ""
    },
    {
        "name": "pruningRefineBy",
        "baseName": "pruningRefineBy",
        "type": "PublicFormSubmissionFilterCoalescingRefineBy",
        "format": ""
    },
    {
        "name": "filterType",
        "baseName": "filterType",
        "type": "PublicFormSubmissionFilterFilterTypeEnum",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "PublicFormSubmissionFilterOperatorEnum",
        "format": ""
    }
];
var PublicFormSubmissionFilterFilterTypeEnum;
(function (PublicFormSubmissionFilterFilterTypeEnum) {
    PublicFormSubmissionFilterFilterTypeEnum["FormSubmission"] = "FORM_SUBMISSION";
})(PublicFormSubmissionFilterFilterTypeEnum = exports.PublicFormSubmissionFilterFilterTypeEnum || (exports.PublicFormSubmissionFilterFilterTypeEnum = {}));
var PublicFormSubmissionFilterOperatorEnum;
(function (PublicFormSubmissionFilterOperatorEnum) {
    PublicFormSubmissionFilterOperatorEnum["FilledOut"] = "FILLED_OUT";
    PublicFormSubmissionFilterOperatorEnum["NotFilledOut"] = "NOT_FILLED_OUT";
})(PublicFormSubmissionFilterOperatorEnum = exports.PublicFormSubmissionFilterOperatorEnum || (exports.PublicFormSubmissionFilterOperatorEnum = {}));
