// @hubspot/api-client/lib/codegen/crm/lists/models/PublicIndexedTimePointIndexReference.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicIndexedTimePointIndexReferenceDayOfWeekEnum = exports.PublicIndexedTimePointIndexReferenceReferenceTypeEnum = exports.PublicIndexedTimePointIndexReference = void 0;
class PublicIndexedTimePointIndexReference {
    static getAttributeTypeMap() {
        return PublicIndexedTimePointIndexReference.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicIndexedTimePointIndexReference = PublicIndexedTimePointIndexReference;
PublicIndexedTimePointIndexReference.discriminator = undefined;
PublicIndexedTimePointIndexReference.attributeTypeMap = [
    {
        "name": "hour",
        "baseName": "hour",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "millisecond",
        "baseName": "millisecond",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "referenceType",
        "baseName": "referenceType",
        "type": "PublicIndexedTimePointIndexReferenceReferenceTypeEnum",
        "format": ""
    },
    {
        "name": "minute",
        "baseName": "minute",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "second",
        "baseName": "second",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "dayOfWeek",
        "baseName": "dayOfWeek",
        "type": "PublicIndexedTimePointIndexReferenceDayOfWeekEnum",
        "format": ""
    },
    {
        "name": "month",
        "baseName": "month",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "day",
        "baseName": "day",
        "type": "number",
        "format": "int32"
    }
];
var PublicIndexedTimePointIndexReferenceReferenceTypeEnum;
(function (PublicIndexedTimePointIndexReferenceReferenceTypeEnum) {
    PublicIndexedTimePointIndexReferenceReferenceTypeEnum["Month"] = "MONTH";
})(PublicIndexedTimePointIndexReferenceReferenceTypeEnum = exports.PublicIndexedTimePointIndexReferenceReferenceTypeEnum || (exports.PublicIndexedTimePointIndexReferenceReferenceTypeEnum = {}));
var PublicIndexedTimePointIndexReferenceDayOfWeekEnum;
(function (PublicIndexedTimePointIndexReferenceDayOfWeekEnum) {
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Monday"] = "MONDAY";
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Tuesday"] = "TUESDAY";
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Wednesday"] = "WEDNESDAY";
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Thursday"] = "THURSDAY";
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Friday"] = "FRIDAY";
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Saturday"] = "SATURDAY";
    PublicIndexedTimePointIndexReferenceDayOfWeekEnum["Sunday"] = "SUNDAY";
})(PublicIndexedTimePointIndexReferenceDayOfWeekEnum = exports.PublicIndexedTimePointIndexReferenceDayOfWeekEnum || (exports.PublicIndexedTimePointIndexReferenceDayOfWeekEnum = {}));
